<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 Statistics for Categorical Data | JABSTB: Statistical Design and Analysis of Experiments with R</title>
  <meta name="description" content="Experimental biostatistics using R." />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 Statistics for Categorical Data | JABSTB: Statistical Design and Analysis of Experiments with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Experimental biostatistics using R." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 Statistics for Categorical Data | JABSTB: Statistical Design and Analysis of Experiments with R" />
  
  <meta name="twitter:description" content="Experimental biostatistics using R." />
  

<meta name="author" content="TJ Murphy PhD, Department of Pharmacology and Chemical Biology, School of Medicine, Emory University, Atlanta, GA biostats538@gmail.com" />


<meta name="date" content="2020-02-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="normal.html"/>
<link rel="next" href="chisquare.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">JABSTB</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="author.html"><a href="author.html"><i class="fa fa-check"></i><b>1</b> About the author and book</a></li>
<li class="chapter" data-level="2" data-path="history.html"><a href="history.html"><i class="fa fa-check"></i><b>2</b> A Brief History of Experimental Design</a></li>
<li class="chapter" data-level="3" data-path="software.html"><a href="software.html"><i class="fa fa-check"></i><b>3</b> Installing and Understanding the software</a><ul>
<li class="chapter" data-level="3.1" data-path="software.html"><a href="software.html#spring-2020-term-versions"><i class="fa fa-check"></i><b>3.1</b> Spring 2020 term versions</a><ul>
<li class="chapter" data-level="3.1.1" data-path="software.html"><a href="software.html#what-are-these"><i class="fa fa-check"></i><b>3.1.1</b> What are these?</a></li>
<li class="chapter" data-level="3.1.2" data-path="software.html"><a href="software.html#rule-1-install-r-first-install-rstudio-second"><i class="fa fa-check"></i><b>3.1.2</b> Rule 1: Install R first, install RStudio second</a></li>
<li class="chapter" data-level="3.1.3" data-path="software.html"><a href="software.html#rule-2-dont-pay"><i class="fa fa-check"></i><b>3.1.3</b> Rule 2: Don’t pay!</a></li>
<li class="chapter" data-level="3.1.4" data-path="software.html"><a href="software.html#rule-3-dont-compile"><i class="fa fa-check"></i><b>3.1.4</b> Rule 3: Don’t compile!</a></li>
<li class="chapter" data-level="3.1.5" data-path="software.html"><a href="software.html#rule-4-install-the-right-software-for-your-machine"><i class="fa fa-check"></i><b>3.1.5</b> Rule 4: Install the right software for your machine</a></li>
<li class="chapter" data-level="3.1.6" data-path="software.html"><a href="software.html#rule-5-use-this-installation-guide"><i class="fa fa-check"></i><b>3.1.6</b> Rule 5: Use this installation guide</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="software.html"><a href="software.html#after-installation"><i class="fa fa-check"></i><b>3.2</b> After installation</a><ul>
<li class="chapter" data-level="3.2.1" data-path="software.html"><a href="software.html#what-is-r"><i class="fa fa-check"></i><b>3.2.1</b> What is R?</a></li>
<li class="chapter" data-level="3.2.2" data-path="software.html"><a href="software.html#r-packages"><i class="fa fa-check"></i><b>3.2.2</b> R packages</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="software.html"><a href="software.html#important-avoid-compiling"><i class="fa fa-check"></i><b>3.3</b> Important: Avoid compiling</a><ul>
<li class="chapter" data-level="3.3.1" data-path="software.html"><a href="software.html#reasons-not-to-compile"><i class="fa fa-check"></i><b>3.3.1</b> Reasons not to compile</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>4</b> Start Learning R</a><ul>
<li class="chapter" data-level="4.1" data-path="getting-started.html"><a href="getting-started.html#digging-in-with-rrstudio"><i class="fa fa-check"></i><b>4.1</b> Digging in with R/RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started.html"><a href="getting-started.html#the-tidyverse-package"><i class="fa fa-check"></i><b>4.2</b> The tidyverse package</a><ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started.html"><a href="getting-started.html#do-swirlstats"><i class="fa fa-check"></i><b>4.2.1</b> Do swirlstats</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started.html"><a href="getting-started.html#other-resources"><i class="fa fa-check"></i><b>4.3</b> Other resources</a><ul>
<li class="chapter" data-level="4.3.1" data-path="getting-started.html"><a href="getting-started.html#internet-search"><i class="fa fa-check"></i><b>4.3.1</b> Internet search</a></li>
<li class="chapter" data-level="4.3.2" data-path="getting-started.html"><a href="getting-started.html#help-documents"><i class="fa fa-check"></i><b>4.3.2</b> Help Documents</a></li>
<li class="chapter" data-level="4.3.3" data-path="getting-started.html"><a href="getting-started.html#package-vignettes"><i class="fa fa-check"></i><b>4.3.3</b> Package Vignettes</a></li>
<li class="chapter" data-level="4.3.4" data-path="getting-started.html"><a href="getting-started.html#people"><i class="fa fa-check"></i><b>4.3.4</b> People</a></li>
<li class="chapter" data-level="4.3.5" data-path="getting-started.html"><a href="getting-started.html#typos"><i class="fa fa-check"></i><b>4.3.5</b> typos</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="philos.html"><a href="philos.html"><i class="fa fa-check"></i><b>5</b> What is your philosophy?</a><ul>
<li class="chapter" data-level="5.1" data-path="philos.html"><a href="philos.html#exploration-and-uncertainty"><i class="fa fa-check"></i><b>5.1</b> Exploration and uncertainty</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bigpic.html"><a href="bigpic.html"><i class="fa fa-check"></i><b>6</b> Stats: The Big Picture</a><ul>
<li class="chapter" data-level="6.1" data-path="bigpic.html"><a href="bigpic.html#what-are-experimental-statistics"><i class="fa fa-check"></i><b>6.1</b> What are experimental statistics?</a><ul>
<li class="chapter" data-level="6.1.1" data-path="bigpic.html"><a href="bigpic.html#descriptive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Descriptive modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="bigpic.html"><a href="bigpic.html#statistical-inference"><i class="fa fa-check"></i><b>6.1.2</b> Statistical inference</a></li>
<li class="chapter" data-level="6.1.3" data-path="bigpic.html"><a href="bigpic.html#experimental-design"><i class="fa fa-check"></i><b>6.1.3</b> Experimental design</a></li>
<li class="chapter" data-level="6.1.4" data-path="bigpic.html"><a href="bigpic.html#statistics-as-an-anti-bias-framework"><i class="fa fa-check"></i><b>6.1.4</b> Statistics as an anti-bias framework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>7</b> Statistical Sampling</a><ul>
<li class="chapter" data-level="7.1" data-path="sampling.html"><a href="sampling.html#experimental-units"><i class="fa fa-check"></i><b>7.1</b> Experimental units</a><ul>
<li class="chapter" data-level="7.1.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-to-define-the-experimental-unit"><i class="fa fa-check"></i><b>7.1.1</b> A simple test to define the experimental unit</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="sampling.html"><a href="sampling.html#independent-replicates"><i class="fa fa-check"></i><b>7.2</b> Independent Replicates</a><ul>
<li class="chapter" data-level="7.2.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-for-independence"><i class="fa fa-check"></i><b>7.2.1</b> A simple test for independence</a></li>
<li class="chapter" data-level="7.2.2" data-path="sampling.html"><a href="sampling.html#some-replication-examples"><i class="fa fa-check"></i><b>7.2.2</b> Some replication examples</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="sampling.html"><a href="sampling.html#random-process"><i class="fa fa-check"></i><b>7.3</b> Random process</a></li>
<li class="chapter" data-level="7.4" data-path="sampling.html"><a href="sampling.html#statistically-valid-samples"><i class="fa fa-check"></i><b>7.4</b> Statistically valid samples</a><ul>
<li class="chapter" data-level="7.4.1" data-path="sampling.html"><a href="sampling.html#select-random-subjects"><i class="fa fa-check"></i><b>7.4.1</b> Select random subjects</a></li>
<li class="chapter" data-level="7.4.2" data-path="sampling.html"><a href="sampling.html#randomize-to-sequence"><i class="fa fa-check"></i><b>7.4.2</b> Randomize to sequence</a></li>
<li class="chapter" data-level="7.4.3" data-path="sampling.html"><a href="sampling.html#randomize-to-location"><i class="fa fa-check"></i><b>7.4.3</b> Randomize to location</a></li>
<li class="chapter" data-level="7.4.4" data-path="sampling.html"><a href="sampling.html#randomize-to-block"><i class="fa fa-check"></i><b>7.4.4</b> Randomize to block</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="sampling.html"><a href="sampling.html#independence-of-replicates"><i class="fa fa-check"></i><b>7.5</b> Independence of replicates</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>8</b> Data Classification</a><ul>
<li class="chapter" data-level="8.1" data-path="data.html"><a href="data.html#dependent-and-independent-variables"><i class="fa fa-check"></i><b>8.1</b> Dependent and independent variables</a><ul>
<li class="chapter" data-level="8.1.1" data-path="data.html"><a href="data.html#statistical-notation"><i class="fa fa-check"></i><b>8.1.1</b> Statistical notation</a></li>
<li class="chapter" data-level="8.1.2" data-path="data.html"><a href="data.html#when-there-is-no-independent-variable"><i class="fa fa-check"></i><b>8.1.2</b> When there is no independent variable</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="data.html"><a href="data.html#discrete-or-continuous-variables"><i class="fa fa-check"></i><b>8.2</b> Discrete or continuous variables</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data.html"><a href="data.html#measured-variables"><i class="fa fa-check"></i><b>8.2.1</b> Measured variables</a></li>
<li class="chapter" data-level="8.2.2" data-path="data.html"><a href="data.html#discrete-categorical-and-ordinal-variables"><i class="fa fa-check"></i><b>8.2.2</b> Discrete categorical and ordinal variables</a></li>
<li class="chapter" data-level="8.2.3" data-path="data.html"><a href="data.html#rescaling-variables"><i class="fa fa-check"></i><b>8.2.3</b> Rescaling variables</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data.html"><a href="data.html#working-with-data"><i class="fa fa-check"></i><b>8.3</b> Working with data</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data.html"><a href="data.html#entering-data"><i class="fa fa-check"></i><b>8.3.1</b> Entering data</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data.html"><a href="data.html#summary"><i class="fa fa-check"></i><b>8.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dispersion.html"><a href="dispersion.html"><i class="fa fa-check"></i><b>9</b> Variability, Accuracy and Precision</a><ul>
<li class="chapter" data-level="9.0.1" data-path="dispersion.html"><a href="dispersion.html#data-are-messy"><i class="fa fa-check"></i><b>9.0.1</b> Data are messy</a></li>
<li class="chapter" data-level="9.1" data-path="dispersion.html"><a href="dispersion.html#illustration-of-a-model-with-error"><i class="fa fa-check"></i><b>9.1</b> Illustration of a model with error</a></li>
<li class="chapter" data-level="9.2" data-path="dispersion.html"><a href="dispersion.html#variance-quantifying-variation-by-least-squares"><i class="fa fa-check"></i><b>9.2</b> Variance: Quantifying variation by least squares</a><ul>
<li class="chapter" data-level="9.2.1" data-path="dispersion.html"><a href="dispersion.html#definition-of-variance"><i class="fa fa-check"></i><b>9.2.1</b> Definition of variance</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="dispersion.html"><a href="dispersion.html#standard-deviation"><i class="fa fa-check"></i><b>9.3</b> Standard deviation</a><ul>
<li class="chapter" data-level="9.3.1" data-path="dispersion.html"><a href="dispersion.html#what-does-the-standard-deviation-tell-us"><i class="fa fa-check"></i><b>9.3.1</b> What does the standard deviation tell us</a></li>
<li class="chapter" data-level="9.3.2" data-path="dispersion.html"><a href="dispersion.html#ways-of-describing-sample-variability"><i class="fa fa-check"></i><b>9.3.2</b> Ways of describing sample variability</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="dispersion.html"><a href="dispersion.html#precision-and-accuracy"><i class="fa fa-check"></i><b>9.4</b> Precision and Accuracy</a></li>
<li class="chapter" data-level="9.5" data-path="dispersion.html"><a href="dispersion.html#standard-error"><i class="fa fa-check"></i><b>9.5</b> Standard error</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dispersion.html"><a href="dispersion.html#what-exactly-does-the-standard-error-represent"><i class="fa fa-check"></i><b>9.5.1</b> What exactly does the standard error represent?</a></li>
<li class="chapter" data-level="9.5.2" data-path="dispersion.html"><a href="dispersion.html#should-i-use-sd-or-sem"><i class="fa fa-check"></i><b>9.5.2</b> “Should I use SD or SEM?”</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dispersion.html"><a href="dispersion.html#confidence-intervals"><i class="fa fa-check"></i><b>9.6</b> Confidence intervals</a><ul>
<li class="chapter" data-level="9.6.1" data-path="dispersion.html"><a href="dispersion.html#ci-definition"><i class="fa fa-check"></i><b>9.6.1</b> CI Definition</a></li>
<li class="chapter" data-level="9.6.2" data-path="dispersion.html"><a href="dispersion.html#evaluating-confidence-intervals"><i class="fa fa-check"></i><b>9.6.2</b> Evaluating confidence intervals</a></li>
<li class="chapter" data-level="9.6.3" data-path="dispersion.html"><a href="dispersion.html#ci-simulation-tools"><i class="fa fa-check"></i><b>9.6.3</b> CI Simulation tools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="dispersion.html"><a href="dispersion.html#key-take-aways"><i class="fa fa-check"></i><b>9.7</b> Key take aways</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="hypotheses.html"><a href="hypotheses.html"><i class="fa fa-check"></i><b>10</b> Framing statistical hypotheses</a><ul>
<li class="chapter" data-level="10.1" data-path="hypotheses.html"><a href="hypotheses.html#the-decision-process"><i class="fa fa-check"></i><b>10.1</b> The decision process</a></li>
<li class="chapter" data-level="10.2" data-path="hypotheses.html"><a href="hypotheses.html#popper-and-falsification"><i class="fa fa-check"></i><b>10.2</b> Popper and falsification</a></li>
<li class="chapter" data-level="10.3" data-path="hypotheses.html"><a href="hypotheses.html#statistical-hypothesis-rubric"><i class="fa fa-check"></i><b>10.3</b> Statistical hypothesis rubric</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="error.html"><a href="error.html"><i class="fa fa-check"></i><b>11</b> Error</a><ul>
<li class="chapter" data-level="11.1" data-path="error.html"><a href="error.html#setting-type-1-and-type-2-error-thresholds"><i class="fa fa-check"></i><b>11.1</b> Setting type 1 and type 2 error thresholds</a><ul>
<li class="chapter" data-level="11.1.1" data-path="error.html"><a href="error.html#setting-alpha-the-type-1-error"><i class="fa fa-check"></i><b>11.1.1</b> Setting alpha-the type 1 error</a></li>
<li class="chapter" data-level="11.1.2" data-path="error.html"><a href="error.html#power-setting-beta-the-type-2-error"><i class="fa fa-check"></i><b>11.1.2</b> Power: Setting beta-the type 2 error</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="error.html"><a href="error.html#striking-the-right-balance"><i class="fa fa-check"></i><b>11.2</b> Striking the right balance</a></li>
<li class="chapter" data-level="11.3" data-path="error.html"><a href="error.html#false-discovery-rate"><i class="fa fa-check"></i><b>11.3</b> False discovery rate</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="pvalues.html"><a href="pvalues.html"><i class="fa fa-check"></i><b>12</b> P Values</a><ul>
<li class="chapter" data-level="12.1" data-path="pvalues.html"><a href="pvalues.html#definition"><i class="fa fa-check"></i><b>12.1</b> Definition</a></li>
<li class="chapter" data-level="12.2" data-path="pvalues.html"><a href="pvalues.html#test-statistics-come-from-null-distributions"><i class="fa fa-check"></i><b>12.2</b> Test statistics come from null distributions</a><ul>
<li class="chapter" data-level="12.2.1" data-path="pvalues.html"><a href="pvalues.html#p-value-behavior"><i class="fa fa-check"></i><b>12.2.1</b> P-value behavior</a></li>
<li class="chapter" data-level="12.2.2" data-path="pvalues.html"><a href="pvalues.html#p-values-and-the-decision-process"><i class="fa fa-check"></i><b>12.2.2</b> P-values and the decision process</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="pvalues.html"><a href="pvalues.html#a-blood-glucose-p-value"><i class="fa fa-check"></i><b>12.3</b> A blood glucose p-value</a></li>
<li class="chapter" data-level="12.4" data-path="pvalues.html"><a href="pvalues.html#how-p-values-should-be-interpreted"><i class="fa fa-check"></i><b>12.4</b> How p-values should be interpreted</a></li>
<li class="chapter" data-level="12.5" data-path="pvalues.html"><a href="pvalues.html#interpretation"><i class="fa fa-check"></i><b>12.5</b> Interpretation</a></li>
<li class="chapter" data-level="12.6" data-path="pvalues.html"><a href="pvalues.html#criticisms-of-p-values"><i class="fa fa-check"></i><b>12.6</b> Criticisms of p-values</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="jaxwest7.html"><a href="jaxwest7.html"><i class="fa fa-check"></i><b>13</b> Reproducible Data Munging in R</a><ul>
<li class="chapter" data-level="13.1" data-path="jaxwest7.html"><a href="jaxwest7.html#jaxwest7-glucose-data"><i class="fa fa-check"></i><b>13.1</b> Jaxwest7 glucose data</a><ul>
<li class="chapter" data-level="13.1.1" data-path="jaxwest7.html"><a href="jaxwest7.html#inspect-the-jaxwest7-data"><i class="fa fa-check"></i><b>13.1.1</b> Inspect the Jaxwest7 data</a></li>
<li class="chapter" data-level="13.1.2" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-glucose-concentration-data-into-r"><i class="fa fa-check"></i><b>13.1.2</b> Munge the glucose concentration data into R</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="jaxwest7.html"><a href="jaxwest7.html#explore-the-jaxwest7-data"><i class="fa fa-check"></i><b>13.2</b> Explore the Jaxwest7 data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="binomial.html"><a href="binomial.html"><i class="fa fa-check"></i><b>14</b> The Binomial Distribution</a><ul>
<li class="chapter" data-level="14.1" data-path="binomial.html"><a href="binomial.html#dbinom"><i class="fa fa-check"></i><b>14.1</b> dbinom</a></li>
<li class="chapter" data-level="14.2" data-path="binomial.html"><a href="binomial.html#pbinom"><i class="fa fa-check"></i><b>14.2</b> pbinom</a></li>
<li class="chapter" data-level="14.3" data-path="binomial.html"><a href="binomial.html#qbinom"><i class="fa fa-check"></i><b>14.3</b> qbinom</a></li>
<li class="chapter" data-level="14.4" data-path="binomial.html"><a href="binomial.html#rbinom"><i class="fa fa-check"></i><b>14.4</b> rbinom</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="poisson.html"><a href="poisson.html"><i class="fa fa-check"></i><b>15</b> The Poisson Distribution</a><ul>
<li class="chapter" data-level="15.1" data-path="poisson.html"><a href="poisson.html#poisson-events"><i class="fa fa-check"></i><b>15.1</b> Poisson Events</a></li>
<li class="chapter" data-level="15.2" data-path="poisson.html"><a href="poisson.html#dpois"><i class="fa fa-check"></i><b>15.2</b> dpois</a></li>
<li class="chapter" data-level="15.3" data-path="poisson.html"><a href="poisson.html#ppois"><i class="fa fa-check"></i><b>15.3</b> ppois</a></li>
<li class="chapter" data-level="15.4" data-path="poisson.html"><a href="poisson.html#rpois"><i class="fa fa-check"></i><b>15.4</b> rpois</a></li>
<li class="chapter" data-level="15.5" data-path="poisson.html"><a href="poisson.html#overdispersion"><i class="fa fa-check"></i><b>15.5</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="normal.html"><a href="normal.html"><i class="fa fa-check"></i><b>16</b> The Normal Distribution</a><ul>
<li class="chapter" data-level="16.0.1" data-path="normal.html"><a href="normal.html#the-standard-normal"><i class="fa fa-check"></i><b>16.0.1</b> The Standard Normal</a></li>
<li class="chapter" data-level="16.1" data-path="normal.html"><a href="normal.html#dnorm"><i class="fa fa-check"></i><b>16.1</b> dnorm</a></li>
<li class="chapter" data-level="16.2" data-path="normal.html"><a href="normal.html#pnorm"><i class="fa fa-check"></i><b>16.2</b> pnorm</a><ul>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-p-values-using-pnorm"><i class="fa fa-check"></i>Calculating “p-values”" using pnorm</a></li>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-percentiles-using-pnorm"><i class="fa fa-check"></i>Calculating percentiles using pnorm</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="normal.html"><a href="normal.html#qnorm"><i class="fa fa-check"></i><b>16.3</b> qnorm</a></li>
<li class="chapter" data-level="16.4" data-path="normal.html"><a href="normal.html#rnorm"><i class="fa fa-check"></i><b>16.4</b> rnorm</a><ul>
<li class="chapter" data-level="16.4.1" data-path="normal.html"><a href="normal.html#plotting-histograms-of-some-rnorm-samples"><i class="fa fa-check"></i><b>16.4.1</b> Plotting histograms of some rnorm samples</a></li>
<li class="chapter" data-level="16.4.2" data-path="normal.html"><a href="normal.html#bins-and-binwidth"><i class="fa fa-check"></i><b>16.4.2</b> Bins and Binwidth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="categorical.html"><a href="categorical.html"><i class="fa fa-check"></i><b>17</b> Statistics for Categorical Data</a><ul>
<li class="chapter" data-level="17.1" data-path="categorical.html"><a href="categorical.html#types-of-categorical-data"><i class="fa fa-check"></i><b>17.1</b> Types of categorical data</a><ul>
<li class="chapter" data-level="17.1.1" data-path="categorical.html"><a href="categorical.html#proportions"><i class="fa fa-check"></i><b>17.1.1</b> Proportions</a></li>
<li class="chapter" data-level="17.1.2" data-path="categorical.html"><a href="categorical.html#frequencies"><i class="fa fa-check"></i><b>17.1.2</b> Frequencies</a></li>
<li class="chapter" data-level="17.1.3" data-path="categorical.html"><a href="categorical.html#associations"><i class="fa fa-check"></i><b>17.1.3</b> Associations</a></li>
<li class="chapter" data-level="17.1.4" data-path="categorical.html"><a href="categorical.html#statistics-covered-here"><i class="fa fa-check"></i><b>17.1.4</b> Statistics Covered Here</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="categorical.html"><a href="categorical.html#exact-v-asymptotic-calculations-of-p-values"><i class="fa fa-check"></i><b>17.2</b> Exact v Asymptotic Calculations of p-values</a><ul>
<li class="chapter" data-level="17.2.1" data-path="categorical.html"><a href="categorical.html#choosing-exact-or-asymptotic"><i class="fa fa-check"></i><b>17.2.1</b> Choosing exact or asymptotic</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="categorical.html"><a href="categorical.html#overview-of-the-types-of-hypothesis-testing"><i class="fa fa-check"></i><b>17.3</b> Overview of the types of hypothesis testing</a><ul>
<li class="chapter" data-level="17.3.1" data-path="categorical.html"><a href="categorical.html#proportion-analysis"><i class="fa fa-check"></i><b>17.3.1</b> Proportion analysis</a></li>
<li class="chapter" data-level="17.3.2" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-testing"><i class="fa fa-check"></i><b>17.3.2</b> Goodness of fit testing</a></li>
<li class="chapter" data-level="17.3.3" data-path="categorical.html"><a href="categorical.html#contingency-analysis"><i class="fa fa-check"></i><b>17.3.3</b> Contingency Analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="categorical.html"><a href="categorical.html#comparing-proportions"><i class="fa fa-check"></i><b>17.4</b> Comparing proportions</a><ul>
<li class="chapter" data-level="17.4.1" data-path="categorical.html"><a href="categorical.html#a-mouse-t-cell-pilot-experiment-the-cytokine-inducible-antigen-gradstudin"><i class="fa fa-check"></i><b>17.4.1</b> A Mouse T Cell Pilot Experiment: The Cytokine-inducible antigen gradstudin</a></li>
<li class="chapter" data-level="17.4.2" data-path="categorical.html"><a href="categorical.html#calculating-proportions"><i class="fa fa-check"></i><b>17.4.2</b> Calculating Proportions</a></li>
<li class="chapter" data-level="17.4.3" data-path="categorical.html"><a href="categorical.html#what-a-proportion-estimates"><i class="fa fa-check"></i><b>17.4.3</b> What A Proportion Estimates</a></li>
<li class="chapter" data-level="17.4.4" data-path="categorical.html"><a href="categorical.html#confidence-intervals-of-proportions"><i class="fa fa-check"></i><b>17.4.4</b> Confidence Intervals of Proportions</a></li>
<li class="chapter" data-level="17.4.5" data-path="categorical.html"><a href="categorical.html#a-one-sample-proportion-test"><i class="fa fa-check"></i><b>17.4.5</b> A One-Sample Proportion Test</a></li>
<li class="chapter" data-level="17.4.6" data-path="categorical.html"><a href="categorical.html#an-exact-test-for-one-proportion"><i class="fa fa-check"></i><b>17.4.6</b> An exact test for one proportion</a></li>
<li class="chapter" data-level="17.4.7" data-path="categorical.html"><a href="categorical.html#comparing-two-proportions"><i class="fa fa-check"></i><b>17.4.7</b> Comparing Two Proportions</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="categorical.html"><a href="categorical.html#exact-tests-for-two-proportions"><i class="fa fa-check"></i><b>17.5</b> Exact tests for two proportions</a></li>
<li class="chapter" data-level="17.6" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-tests"><i class="fa fa-check"></i><b>17.6</b> Goodness of fit Tests</a><ul>
<li class="chapter" data-level="17.6.1" data-path="categorical.html"><a href="categorical.html#an-exact-goodness-of-fit-test"><i class="fa fa-check"></i><b>17.6.1</b> An Exact Goodness of Fit test</a></li>
<li class="chapter" data-level="17.6.2" data-path="categorical.html"><a href="categorical.html#an-approximate-goodness-of-fit-test"><i class="fa fa-check"></i><b>17.6.2</b> An Approximate Goodness of Fit test</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="categorical.html"><a href="categorical.html#contingency-testing"><i class="fa fa-check"></i><b>17.7</b> Contingency Testing</a><ul>
<li class="chapter" data-level="17.7.1" data-path="categorical.html"><a href="categorical.html#interpretation-of-contingency-results"><i class="fa fa-check"></i><b>17.7.1</b> Interpretation of Contingency Results</a></li>
<li class="chapter" data-level="17.7.2" data-path="categorical.html"><a href="categorical.html#write-up-4"><i class="fa fa-check"></i><b>17.7.2</b> Write Up</a></li>
<li class="chapter" data-level="17.7.3" data-path="categorical.html"><a href="categorical.html#interpretation-of-chi-square-output"><i class="fa fa-check"></i><b>17.7.3</b> Interpretation of chi-square output</a></li>
<li class="chapter" data-level="17.7.4" data-path="categorical.html"><a href="categorical.html#write-up-5"><i class="fa fa-check"></i><b>17.7.4</b> Write Up</a></li>
<li class="chapter" data-level="17.7.5" data-path="categorical.html"><a href="categorical.html#which-contingency-test-is-best"><i class="fa fa-check"></i><b>17.7.5</b> Which contingency test is best?</a></li>
<li class="chapter" data-level="17.7.6" data-path="categorical.html"><a href="categorical.html#higher-dimension-contingency-analysis"><i class="fa fa-check"></i><b>17.7.6</b> Higher dimension contingency analysis</a></li>
<li class="chapter" data-level="17.7.7" data-path="categorical.html"><a href="categorical.html#other-experimental-designs-involving-categorical-data"><i class="fa fa-check"></i><b>17.7.7</b> Other experimental designs involving categorical data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="categorical.html"><a href="categorical.html#power-analysis-for-proportion-tests"><i class="fa fa-check"></i><b>17.8</b> Power analysis for proportion tests</a></li>
<li class="chapter" data-level="17.9" data-path="categorical.html"><a href="categorical.html#power-analysis-functions-for-proportion-tests"><i class="fa fa-check"></i><b>17.9</b> Power analysis functions for proportion tests</a></li>
<li class="chapter" data-level="17.10" data-path="categorical.html"><a href="categorical.html#graphing-proportions"><i class="fa fa-check"></i><b>17.10</b> Graphing Proportions</a></li>
<li class="chapter" data-level="17.11" data-path="categorical.html"><a href="categorical.html#key-takeaways"><i class="fa fa-check"></i><b>17.11</b> Key Takeaways</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="chisquare.html"><a href="chisquare.html"><i class="fa fa-check"></i><b>18</b> The Chi-square Distribution</a><ul>
<li class="chapter" data-level="18.1" data-path="chisquare.html"><a href="chisquare.html#background"><i class="fa fa-check"></i><b>18.1</b> Background</a></li>
<li class="chapter" data-level="18.2" data-path="chisquare.html"><a href="chisquare.html#dchisq"><i class="fa fa-check"></i><b>18.2</b> dchisq</a></li>
<li class="chapter" data-level="18.3" data-path="chisquare.html"><a href="chisquare.html#pchisq"><i class="fa fa-check"></i><b>18.3</b> pchisq</a><ul>
<li class="chapter" data-level="18.3.1" data-path="chisquare.html"><a href="chisquare.html#calculating-p-values-from-pchisq"><i class="fa fa-check"></i><b>18.3.1</b> Calculating p-values from pchisq</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="chisquare.html"><a href="chisquare.html#qchisq"><i class="fa fa-check"></i><b>18.4</b> qchisq</a></li>
<li class="chapter" data-level="18.5" data-path="chisquare.html"><a href="chisquare.html#rchisq"><i class="fa fa-check"></i><b>18.5</b> rchisq</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="nonparametrics.html"><a href="nonparametrics.html"><i class="fa fa-check"></i><b>19</b> Nonparametric Statistical Tests</a><ul>
<li class="chapter" data-level="19.1" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-sampling-distributions"><i class="fa fa-check"></i><b>19.1</b> Nonparametric sampling distributions</a></li>
<li class="chapter" data-level="19.2" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-discrete-data"><i class="fa fa-check"></i><b>19.2</b> Experiments involving discrete data</a><ul>
<li class="chapter" data-level="19.2.1" data-path="nonparametrics.html"><a href="nonparametrics.html#ordered-data-1"><i class="fa fa-check"></i><b>19.2.1</b> Ordered data</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-deviant-data"><i class="fa fa-check"></i><b>19.3</b> Experiments involving deviant Data</a></li>
<li class="chapter" data-level="19.4" data-path="nonparametrics.html"><a href="nonparametrics.html#sign-test"><i class="fa fa-check"></i><b>19.4</b> Sign Test</a><ul>
<li class="chapter" data-level="19.4.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-1"><i class="fa fa-check"></i><b>19.4.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.4.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-6"><i class="fa fa-check"></i><b>19.4.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-one-group"><i class="fa fa-check"></i><b>19.5</b> Wilcoxon Sign Rank Test for One Group</a><ul>
<li class="chapter" data-level="19.5.1" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-experimental-designs"><i class="fa fa-check"></i><b>19.5.1</b> Wilcoxon Sign Rank Experimental Designs</a></li>
<li class="chapter" data-level="19.5.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-2"><i class="fa fa-check"></i><b>19.5.2</b> Interpretation</a></li>
<li class="chapter" data-level="19.5.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-7"><i class="fa fa-check"></i><b>19.5.3</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-mann-whitney-rank-sum-test-for-2-independent-groups"><i class="fa fa-check"></i><b>19.6</b> Wilcoxon Mann Whitney Rank Sum Test for 2 independent groups</a><ul>
<li class="chapter" data-level="19.6.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-3"><i class="fa fa-check"></i><b>19.6.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.6.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-8"><i class="fa fa-check"></i><b>19.6.2</b> Write Up</a></li>
<li class="chapter" data-level="19.6.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot"><i class="fa fa-check"></i><b>19.6.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-paired-groups"><i class="fa fa-check"></i><b>19.7</b> Wilcoxon Sign Rank Test for paired groups</a><ul>
<li class="chapter" data-level="19.7.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-4"><i class="fa fa-check"></i><b>19.7.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.7.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-9"><i class="fa fa-check"></i><b>19.7.2</b> Write up</a></li>
<li class="chapter" data-level="19.7.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot-1"><i class="fa fa-check"></i><b>19.7.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="nonparametrics.html"><a href="nonparametrics.html#kruskal-wallis"><i class="fa fa-check"></i><b>19.8</b> Kruskal-Wallis</a><ul>
<li class="chapter" data-level="19.8.1" data-path="nonparametrics.html"><a href="nonparametrics.html#the-experimental-design"><i class="fa fa-check"></i><b>19.8.1</b> The experimental design</a></li>
<li class="chapter" data-level="19.8.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-10"><i class="fa fa-check"></i><b>19.8.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.9" data-path="nonparametrics.html"><a href="nonparametrics.html#friedman-test"><i class="fa fa-check"></i><b>19.9</b> Friedman test</a><ul>
<li class="chapter" data-level="19.9.1" data-path="nonparametrics.html"><a href="nonparametrics.html#blocked-experimental-design"><i class="fa fa-check"></i><b>19.9.1</b> Blocked experimental design</a></li>
<li class="chapter" data-level="19.9.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-5"><i class="fa fa-check"></i><b>19.9.2</b> Interpretation</a></li>
<li class="chapter" data-level="19.9.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-11"><i class="fa fa-check"></i><b>19.9.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="19.10" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-power-calculations"><i class="fa fa-check"></i><b>19.10</b> Nonparametric power calculations</a><ul>
<li class="chapter" data-level="19.10.1" data-path="nonparametrics.html"><a href="nonparametrics.html#how-it-works"><i class="fa fa-check"></i><b>19.10.1</b> How it works</a></li>
<li class="chapter" data-level="19.10.2" data-path="nonparametrics.html"><a href="nonparametrics.html#simulating-response-data"><i class="fa fa-check"></i><b>19.10.2</b> Simulating response data</a></li>
<li class="chapter" data-level="19.10.3" data-path="nonparametrics.html"><a href="nonparametrics.html#an-example"><i class="fa fa-check"></i><b>19.10.3</b> An example</a></li>
<li class="chapter" data-level="19.10.4" data-path="nonparametrics.html"><a href="nonparametrics.html#nonpara.pwr"><i class="fa fa-check"></i><b>19.10.4</b> nonpara.pwr</a></li>
</ul></li>
<li class="chapter" data-level="19.11" data-path="nonparametrics.html"><a href="nonparametrics.html#summary-1"><i class="fa fa-check"></i><b>19.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="signrank.html"><a href="signrank.html"><i class="fa fa-check"></i><b>20</b> Signed Rank Distribution</a><ul>
<li class="chapter" data-level="20.1" data-path="signrank.html"><a href="signrank.html#transformation-of-data-into-sign-ranks"><i class="fa fa-check"></i><b>20.1</b> Transformation of data into sign ranks</a><ul>
<li class="chapter" data-level="20.1.1" data-path="signrank.html"><a href="signrank.html#for-a-one-group-sample"><i class="fa fa-check"></i><b>20.1.1</b> For a one group sample</a></li>
<li class="chapter" data-level="20.1.2" data-path="signrank.html"><a href="signrank.html#for-a-paired-sample"><i class="fa fa-check"></i><b>20.1.2</b> For a paired sample</a></li>
<li class="chapter" data-level="20.1.3" data-path="signrank.html"><a href="signrank.html#the-sign-rank-test-statistic-in-r"><i class="fa fa-check"></i><b>20.1.3</b> The sign rank test statistic in R</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="signrank.html"><a href="signrank.html#rs-four-sign-rank-distribution-functions"><i class="fa fa-check"></i><b>20.2</b> R’s Four Sign Rank Distribution Functions</a><ul>
<li class="chapter" data-level="20.2.1" data-path="signrank.html"><a href="signrank.html#dsignrank"><i class="fa fa-check"></i><b>20.2.1</b> dsignrank</a></li>
<li class="chapter" data-level="20.2.2" data-path="signrank.html"><a href="signrank.html#psignrank"><i class="fa fa-check"></i><b>20.2.2</b> psignrank</a></li>
<li class="chapter" data-level="20.2.3" data-path="signrank.html"><a href="signrank.html#qsignrank"><i class="fa fa-check"></i><b>20.2.3</b> qsignrank</a></li>
<li class="chapter" data-level="20.2.4" data-path="signrank.html"><a href="signrank.html#rsignrank"><i class="fa fa-check"></i><b>20.2.4</b> rsignrank</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="ranksum.html"><a href="ranksum.html"><i class="fa fa-check"></i><b>21</b> Rank Sum Distribution</a><ul>
<li class="chapter" data-level="21.0.1" data-path="ranksum.html"><a href="ranksum.html#transformation-of-data-into-rank-summs"><i class="fa fa-check"></i><b>21.0.1</b> Transformation of data into rank summs</a></li>
<li class="chapter" data-level="21.0.2" data-path="ranksum.html"><a href="ranksum.html#the-sign-rank-test-statistic-in-r-1"><i class="fa fa-check"></i><b>21.0.2</b> The sign rank test statistic in R</a></li>
<li class="chapter" data-level="21.1" data-path="ranksum.html"><a href="ranksum.html#rs-four-sign-rank-distribution-functions-1"><i class="fa fa-check"></i><b>21.1</b> R’s Four Sign Rank Distribution Functions</a><ul>
<li class="chapter" data-level="21.1.1" data-path="ranksum.html"><a href="ranksum.html#dwilcox"><i class="fa fa-check"></i><b>21.1.1</b> dwilcox</a></li>
<li class="chapter" data-level="21.1.2" data-path="ranksum.html"><a href="ranksum.html#pwilcox"><i class="fa fa-check"></i><b>21.1.2</b> pwilcox</a></li>
<li class="chapter" data-level="21.1.3" data-path="ranksum.html"><a href="ranksum.html#qwilcox"><i class="fa fa-check"></i><b>21.1.3</b> qwilcox</a></li>
<li class="chapter" data-level="21.1.4" data-path="ranksum.html"><a href="ranksum.html#rwilcox"><i class="fa fa-check"></i><b>21.1.4</b> rwilcox</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="ttests.html"><a href="ttests.html"><i class="fa fa-check"></i><b>22</b> The t-tests</a><ul>
<li class="chapter" data-level="22.1" data-path="ttests.html"><a href="ttests.html#data-assumptions-for-t-tests"><i class="fa fa-check"></i><b>22.1</b> Data assumptions for t-tests</a></li>
<li class="chapter" data-level="22.2" data-path="ttests.html"><a href="ttests.html#the-t-statistic"><i class="fa fa-check"></i><b>22.2</b> The t Statistic</a><ul>
<li class="chapter" data-level="22.2.1" data-path="ttests.html"><a href="ttests.html#one-sample-t-tests"><i class="fa fa-check"></i><b>22.2.1</b> One sample t tests</a></li>
<li class="chapter" data-level="22.2.2" data-path="ttests.html"><a href="ttests.html#unpaired-t-tests"><i class="fa fa-check"></i><b>22.2.2</b> Unpaired t tests</a></li>
<li class="chapter" data-level="22.2.3" data-path="ttests.html"><a href="ttests.html#paired-t-tests"><i class="fa fa-check"></i><b>22.2.3</b> Paired t tests</a></li>
</ul></li>
<li class="chapter" data-level="22.3" data-path="ttests.html"><a href="ttests.html#t-test-hypotheses"><i class="fa fa-check"></i><b>22.3</b> t Test Hypotheses</a><ul>
<li class="chapter" data-level="22.3.1" data-path="ttests.html"><a href="ttests.html#one-sided-or-two-sided"><i class="fa fa-check"></i><b>22.3.1</b> One-sided or two-sided??</a></li>
<li class="chapter" data-level="22.3.2" data-path="ttests.html"><a href="ttests.html#one-sample-hypotheses"><i class="fa fa-check"></i><b>22.3.2</b> One sample hypotheses</a></li>
<li class="chapter" data-level="22.3.3" data-path="ttests.html"><a href="ttests.html#unpaired-hypotheses"><i class="fa fa-check"></i><b>22.3.3</b> Unpaired hypotheses</a></li>
<li class="chapter" data-level="22.3.4" data-path="ttests.html"><a href="ttests.html#paired-hypotheses"><i class="fa fa-check"></i><b>22.3.4</b> Paired hypotheses</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="ttests.html"><a href="ttests.html#confidence-intervals-of-means"><i class="fa fa-check"></i><b>22.4</b> Confidence Intervals of Means</a></li>
<li class="chapter" data-level="22.5" data-path="ttests.html"><a href="ttests.html#parameters-small-samples-and-reliability"><i class="fa fa-check"></i><b>22.5</b> Parameters, small samples, and reliability</a><ul>
<li class="chapter" data-level="22.5.1" data-path="ttests.html"><a href="ttests.html#reporting-sd-or-sem"><i class="fa fa-check"></i><b>22.5.1</b> Reporting SD or SEM?</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="ttests.html"><a href="ttests.html#t-tests-running-the-analysis"><i class="fa fa-check"></i><b>22.6</b> t Tests: Running the analysis</a><ul>
<li class="chapter" data-level="22.6.1" data-path="ttests.html"><a href="ttests.html#one-sample-t-test"><i class="fa fa-check"></i><b>22.6.1</b> One sample t test</a></li>
<li class="chapter" data-level="22.6.2" data-path="ttests.html"><a href="ttests.html#unpaired-t-test"><i class="fa fa-check"></i><b>22.6.2</b> Unpaired t test</a></li>
<li class="chapter" data-level="22.6.3" data-path="ttests.html"><a href="ttests.html#paired-t-test"><i class="fa fa-check"></i><b>22.6.3</b> Paired t Test</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="ttests.html"><a href="ttests.html#plotting-t-tests"><i class="fa fa-check"></i><b>22.7</b> Plotting t Tests</a><ul>
<li class="chapter" data-level="22.7.1" data-path="ttests.html"><a href="ttests.html#one-sample"><i class="fa fa-check"></i><b>22.7.1</b> One-sample</a></li>
<li class="chapter" data-level="22.7.2" data-path="ttests.html"><a href="ttests.html#unpaired"><i class="fa fa-check"></i><b>22.7.2</b> Unpaired</a></li>
<li class="chapter" data-level="22.7.3" data-path="ttests.html"><a href="ttests.html#paired"><i class="fa fa-check"></i><b>22.7.3</b> Paired</a></li>
</ul></li>
<li class="chapter" data-level="22.8" data-path="ttests.html"><a href="ttests.html#t-test-power"><i class="fa fa-check"></i><b>22.8</b> t Test Power</a><ul>
<li class="chapter" data-level="22.8.1" data-path="ttests.html"><a href="ttests.html#pwr.t.test"><i class="fa fa-check"></i><b>22.8.1</b> pwr.t.test</a></li>
<li class="chapter" data-level="22.8.2" data-path="ttests.html"><a href="ttests.html#interpretation-9"><i class="fa fa-check"></i><b>22.8.2</b> Interpretation</a></li>
<li class="chapter" data-level="22.8.3" data-path="ttests.html"><a href="ttests.html#t-pwr-a-monte-carlo-method"><i class="fa fa-check"></i><b>22.8.3</b> t-pwr: A Monte Carlo method</a></li>
</ul></li>
<li class="chapter" data-level="22.9" data-path="ttests.html"><a href="ttests.html#summary-2"><i class="fa fa-check"></i><b>22.9</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="ttestmc.html"><a href="ttestmc.html"><i class="fa fa-check"></i><b>23</b> Statistical design of t-tests</a><ul>
<li class="chapter" data-level="23.1" data-path="ttestmc.html"><a href="ttestmc.html#about-this-chapter"><i class="fa fa-check"></i><b>23.1</b> About this chapter</a><ul>
<li class="chapter" data-level="23.1.1" data-path="ttestmc.html"><a href="ttestmc.html#overview-of-the-monte-carlo"><i class="fa fa-check"></i><b>23.1.1</b> Overview of the Monte Carlo</a></li>
<li class="chapter" data-level="23.1.2" data-path="ttestmc.html"><a href="ttestmc.html#tldr-on-data-simulation"><i class="fa fa-check"></i><b>23.1.2</b> tl;dr on data simulation</a></li>
<li class="chapter" data-level="23.1.3" data-path="ttestmc.html"><a href="ttestmc.html#tldr-on-the-test"><i class="fa fa-check"></i><b>23.1.3</b> tl;dr on the test</a></li>
<li class="chapter" data-level="23.1.4" data-path="ttestmc.html"><a href="ttestmc.html#changing-n"><i class="fa fa-check"></i><b>23.1.4</b> Changing n</a></li>
<li class="chapter" data-level="23.1.5" data-path="ttestmc.html"><a href="ttestmc.html#diabetes-drug-scenario"><i class="fa fa-check"></i><b>23.1.5</b> Diabetes drug scenario</a></li>
<li class="chapter" data-level="23.1.6" data-path="ttestmc.html"><a href="ttestmc.html#what-would-be-a-scientifically-meaningful-response"><i class="fa fa-check"></i><b>23.1.6</b> What would be a scientifically meaningful response?</a></li>
</ul></li>
<li class="chapter" data-level="23.2" data-path="ttestmc.html"><a href="ttestmc.html#one-sample-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.2</b> One sample t-test Monte Carlo</a><ul>
<li class="chapter" data-level="23.2.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1"><i class="fa fa-check"></i><b>23.2.1</b> Step 1</a></li>
<li class="chapter" data-level="23.2.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2"><i class="fa fa-check"></i><b>23.2.2</b> Step 2</a></li>
<li class="chapter" data-level="23.2.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3"><i class="fa fa-check"></i><b>23.2.3</b> Step 3</a></li>
<li class="chapter" data-level="23.2.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4"><i class="fa fa-check"></i><b>23.2.4</b> Step 4</a></li>
<li class="chapter" data-level="23.2.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5"><i class="fa fa-check"></i><b>23.2.5</b> Step 5</a></li>
<li class="chapter" data-level="23.2.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6"><i class="fa fa-check"></i><b>23.2.6</b> Step 6</a></li>
<li class="chapter" data-level="23.2.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7"><i class="fa fa-check"></i><b>23.2.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.3" data-path="ttestmc.html"><a href="ttestmc.html#unpaired-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.3</b> Unpaired t-test Monte Carlo</a><ul>
<li class="chapter" data-level="23.3.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-1"><i class="fa fa-check"></i><b>23.3.1</b> Step 1</a></li>
<li class="chapter" data-level="23.3.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-1"><i class="fa fa-check"></i><b>23.3.2</b> Step 2</a></li>
<li class="chapter" data-level="23.3.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-1"><i class="fa fa-check"></i><b>23.3.3</b> Step 3</a></li>
<li class="chapter" data-level="23.3.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-1"><i class="fa fa-check"></i><b>23.3.4</b> Step 4</a></li>
<li class="chapter" data-level="23.3.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-1"><i class="fa fa-check"></i><b>23.3.5</b> Step 5</a></li>
<li class="chapter" data-level="23.3.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-1"><i class="fa fa-check"></i><b>23.3.6</b> Step 6</a></li>
<li class="chapter" data-level="23.3.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-1"><i class="fa fa-check"></i><b>23.3.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.4" data-path="ttestmc.html"><a href="ttestmc.html#paired-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.4</b> Paired t-test Monte Carlo</a><ul>
<li class="chapter" data-level="23.4.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-2"><i class="fa fa-check"></i><b>23.4.1</b> Step 1</a></li>
<li class="chapter" data-level="23.4.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-2"><i class="fa fa-check"></i><b>23.4.2</b> Step 2</a></li>
<li class="chapter" data-level="23.4.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-2"><i class="fa fa-check"></i><b>23.4.3</b> Step 3</a></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="ttestmc.html"><a href="ttestmc.html#step-4-2"><i class="fa fa-check"></i><b>23.5</b> Step 4</a><ul>
<li class="chapter" data-level="23.5.1" data-path="ttestmc.html"><a href="ttestmc.html#step-5-2"><i class="fa fa-check"></i><b>23.5.1</b> Step 5</a></li>
<li class="chapter" data-level="23.5.2" data-path="ttestmc.html"><a href="ttestmc.html#step-6-2"><i class="fa fa-check"></i><b>23.5.2</b> Step 6</a></li>
<li class="chapter" data-level="23.5.3" data-path="ttestmc.html"><a href="ttestmc.html#step-7-2"><i class="fa fa-check"></i><b>23.5.3</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.6" data-path="ttestmc.html"><a href="ttestmc.html#comparison-to-pwr-tests"><i class="fa fa-check"></i><b>23.6</b> Comparison to pwr tests</a></li>
<li class="chapter" data-level="23.7" data-path="ttestmc.html"><a href="ttestmc.html#dt"><i class="fa fa-check"></i><b>23.7</b> dt</a></li>
<li class="chapter" data-level="23.8" data-path="ttestmc.html"><a href="ttestmc.html#pt"><i class="fa fa-check"></i><b>23.8</b> pt</a></li>
<li class="chapter" data-level="23.9" data-path="ttestmc.html"><a href="ttestmc.html#qt"><i class="fa fa-check"></i><b>23.9</b> qt</a></li>
<li class="chapter" data-level="23.10" data-path="ttestmc.html"><a href="ttestmc.html#rt"><i class="fa fa-check"></i><b>23.10</b> rt</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="simcorrelation.html"><a href="simcorrelation.html"><i class="fa fa-check"></i><b>24</b> Simulating correlated variables</a><ul>
<li class="chapter" data-level="24.1" data-path="simcorrelation.html"><a href="simcorrelation.html#estimating-correlation-between-two-variables"><i class="fa fa-check"></i><b>24.1</b> Estimating correlation between two variables</a></li>
<li class="chapter" data-level="24.2" data-path="simcorrelation.html"><a href="simcorrelation.html#simulating-correlated-variables"><i class="fa fa-check"></i><b>24.2</b> Simulating correlated variables</a></li>
<li class="chapter" data-level="24.3" data-path="simcorrelation.html"><a href="simcorrelation.html#monte-carlo-simulation"><i class="fa fa-check"></i><b>24.3</b> Monte Carlo simulation</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="introanova.html"><a href="introanova.html"><i class="fa fa-check"></i><b>25</b> Introduction to ANOVA</a><ul>
<li class="chapter" data-level="25.1" data-path="introanova.html"><a href="introanova.html#factors-and-levels"><i class="fa fa-check"></i><b>25.1</b> Factors and levels</a></li>
<li class="chapter" data-level="25.2" data-path="introanova.html"><a href="introanova.html#anova-models-one--two--and-three-way"><i class="fa fa-check"></i><b>25.2</b> ANOVA models: One-, Two-, and Three-way</a></li>
<li class="chapter" data-level="25.3" data-path="introanova.html"><a href="introanova.html#anova-inference-protocol"><i class="fa fa-check"></i><b>25.3</b> ANOVA inference protocol</a></li>
<li class="chapter" data-level="25.4" data-path="introanova.html"><a href="introanova.html#anova-calculations"><i class="fa fa-check"></i><b>25.4</b> ANOVA calculations</a><ul>
<li class="chapter" data-level="25.4.1" data-path="introanova.html"><a href="introanova.html#sums-of-squares-partitioning"><i class="fa fa-check"></i><b>25.4.1</b> Sums of Squares partitioning</a></li>
<li class="chapter" data-level="25.4.2" data-path="introanova.html"><a href="introanova.html#degrees-of-freedom"><i class="fa fa-check"></i><b>25.4.2</b> Degrees of freedom</a></li>
<li class="chapter" data-level="25.4.3" data-path="introanova.html"><a href="introanova.html#the-mean-squares"><i class="fa fa-check"></i><b>25.4.3</b> The mean squares</a></li>
<li class="chapter" data-level="25.4.4" data-path="introanova.html"><a href="introanova.html#the-anova-table"><i class="fa fa-check"></i><b>25.4.4</b> The ANOVA table</a></li>
<li class="chapter" data-level="25.4.5" data-path="introanova.html"><a href="introanova.html#the-f-test"><i class="fa fa-check"></i><b>25.4.5</b> The F-test</a></li>
<li class="chapter" data-level="25.4.6" data-path="introanova.html"><a href="introanova.html#post-hoc-group-comparisons"><i class="fa fa-check"></i><b>25.4.6</b> Post-hoc group comparisons</a></li>
</ul></li>
<li class="chapter" data-level="25.5" data-path="introanova.html"><a href="introanova.html#completely-randomized-or-related-measures"><i class="fa fa-check"></i><b>25.5</b> Completely randomized or related measures</a><ul>
<li class="chapter" data-level="25.5.1" data-path="introanova.html"><a href="introanova.html#the-problem-of-lost-data-in-related-measures-designs"><i class="fa fa-check"></i><b>25.5.1</b> The problem of lost data in related measures designs</a></li>
</ul></li>
<li class="chapter" data-level="25.6" data-path="introanova.html"><a href="introanova.html#two-way-anova"><i class="fa fa-check"></i><b>25.6</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="25.7" data-path="introanova.html"><a href="introanova.html#other-anova-models"><i class="fa fa-check"></i><b>25.7</b> Other ANOVA models</a><ul>
<li class="chapter" data-level="25.7.1" data-path="introanova.html"><a href="introanova.html#r-and-anova"><i class="fa fa-check"></i><b>25.7.1</b> R and ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="introanova.html"><a href="introanova.html#alternatives-to-anova"><i class="fa fa-check"></i><b>25.8</b> Alternatives to ANOVA</a><ul>
<li class="chapter" data-level="25.8.1" data-path="introanova.html"><a href="introanova.html#screw-anova-just-tell-me-how-to-t-test-everything"><i class="fa fa-check"></i><b>25.8.1</b> Screw ANOVA, Just Tell Me How to t-Test Everything</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="fdistr.html"><a href="fdistr.html"><i class="fa fa-check"></i><b>26</b> The F distribution</a><ul>
<li class="chapter" data-level="26.1" data-path="fdistr.html"><a href="fdistr.html#background-1"><i class="fa fa-check"></i><b>26.1</b> Background</a><ul>
<li class="chapter" data-level="26.1.1" data-path="fdistr.html"><a href="fdistr.html#sample-variance-and-fs-pdf"><i class="fa fa-check"></i><b>26.1.1</b> Sample Variance and F’s PDF</a></li>
</ul></li>
<li class="chapter" data-level="26.2" data-path="fdistr.html"><a href="fdistr.html#df"><i class="fa fa-check"></i><b>26.2</b> df</a></li>
<li class="chapter" data-level="26.3" data-path="fdistr.html"><a href="fdistr.html#pf"><i class="fa fa-check"></i><b>26.3</b> pf</a></li>
<li class="chapter" data-level="26.4" data-path="fdistr.html"><a href="fdistr.html#qf"><i class="fa fa-check"></i><b>26.4</b> qf</a></li>
<li class="chapter" data-level="26.5" data-path="fdistr.html"><a href="fdistr.html#rf"><i class="fa fa-check"></i><b>26.5</b> rf</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="onewayanova.html"><a href="onewayanova.html"><i class="fa fa-check"></i><b>27</b> One-way ANOVA Completely Randomized</a><ul>
<li class="chapter" data-level="27.1" data-path="onewayanova.html"><a href="onewayanova.html#using-ezanova"><i class="fa fa-check"></i><b>27.1</b> Using <code>ezANOVA</code></a></li>
<li class="chapter" data-level="27.2" data-path="onewayanova.html"><a href="onewayanova.html#the-chickwt-data-set"><i class="fa fa-check"></i><b>27.2</b> The chickwt data set</a><ul>
<li class="chapter" data-level="27.2.1" data-path="onewayanova.html"><a href="onewayanova.html#inspect-the-data"><i class="fa fa-check"></i><b>27.2.1</b> Inspect the data</a></li>
</ul></li>
<li class="chapter" data-level="27.3" data-path="onewayanova.html"><a href="onewayanova.html#run-the-anova"><i class="fa fa-check"></i><b>27.3</b> Run the ANOVA</a><ul>
<li class="chapter" data-level="27.3.1" data-path="onewayanova.html"><a href="onewayanova.html#run-the-chickwts-one-way-anova"><i class="fa fa-check"></i><b>27.3.1</b> Run the chickwts One Way ANOVA</a></li>
<li class="chapter" data-level="27.3.2" data-path="onewayanova.html"><a href="onewayanova.html#interpreting-the-one-way-cr-anova-output"><i class="fa fa-check"></i><b>27.3.2</b> Interpreting the One-Way CR ANOVA Output</a></li>
</ul></li>
<li class="chapter" data-level="27.4" data-path="onewayanova.html"><a href="onewayanova.html#posthoc"><i class="fa fa-check"></i><b>27.4</b> Post hoc pairwise comparisons</a><ul>
<li class="chapter" data-level="27.4.1" data-path="onewayanova.html"><a href="onewayanova.html#overview-of-options"><i class="fa fa-check"></i><b>27.4.1</b> Overview of options</a></li>
</ul></li>
<li class="chapter" data-level="27.5" data-path="onewayanova.html"><a href="onewayanova.html#reporting-the-result"><i class="fa fa-check"></i><b>27.5</b> Reporting the result</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="onewayRM.html"><a href="onewayRM.html"><i class="fa fa-check"></i><b>28</b> One-way ANOVA Related Measures</a><ul>
<li class="chapter" data-level="28.1" data-path="onewayRM.html"><a href="onewayRM.html#data-prep"><i class="fa fa-check"></i><b>28.1</b> Data prep</a></li>
<li class="chapter" data-level="28.2" data-path="onewayRM.html"><a href="onewayRM.html#run-the-anova-1"><i class="fa fa-check"></i><b>28.2</b> Run the ANOVA</a></li>
<li class="chapter" data-level="28.3" data-path="onewayRM.html"><a href="onewayRM.html#interpretation-10"><i class="fa fa-check"></i><b>28.3</b> Interpretation</a></li>
<li class="chapter" data-level="28.4" data-path="onewayRM.html"><a href="onewayRM.html#post-hoc-analysis"><i class="fa fa-check"></i><b>28.4</b> Post-hoc analysis</a></li>
<li class="chapter" data-level="28.5" data-path="onewayRM.html"><a href="onewayRM.html#write-up-12"><i class="fa fa-check"></i><b>28.5</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="twowayCR.html"><a href="twowayCR.html"><i class="fa fa-check"></i><b>29</b> Two-way ANOVA Completely Randomized</a><ul>
<li class="chapter" data-level="29.1" data-path="twowayCR.html"><a href="twowayCR.html#effect-of-strain-and-diet-on-liver"><i class="fa fa-check"></i><b>29.1</b> Effect of Strain and Diet on Liver</a></li>
<li class="chapter" data-level="29.2" data-path="twowayCR.html"><a href="twowayCR.html#the-test"><i class="fa fa-check"></i><b>29.2</b> The test</a></li>
<li class="chapter" data-level="29.3" data-path="twowayCR.html"><a href="twowayCR.html#interpretation-of-2-way-cr-anova-output"><i class="fa fa-check"></i><b>29.3</b> Interpretation of 2 Way CR ANOVA Output</a><ul>
<li class="chapter" data-level="29.3.1" data-path="twowayCR.html"><a href="twowayCR.html#levenes"><i class="fa fa-check"></i><b>29.3.1</b> Levene’s</a></li>
<li class="chapter" data-level="29.3.2" data-path="twowayCR.html"><a href="twowayCR.html#anova-table"><i class="fa fa-check"></i><b>29.3.2</b> ANOVA Table</a></li>
</ul></li>
<li class="chapter" data-level="29.4" data-path="twowayCR.html"><a href="twowayCR.html#post-hoc-multiple-comparisons"><i class="fa fa-check"></i><b>29.4</b> Post Hoc Multiple Comparisons</a><ul>
<li class="chapter" data-level="29.4.1" data-path="twowayCR.html"><a href="twowayCR.html#write-up-13"><i class="fa fa-check"></i><b>29.4.1</b> Write Up</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="twowayRM.html"><a href="twowayRM.html"><i class="fa fa-check"></i><b>30</b> Two-way ANOVA Related Measures</a><ul>
<li class="chapter" data-level="30.1" data-path="twowayRM.html"><a href="twowayRM.html#cell-culture"><i class="fa fa-check"></i><b>30.1</b> Cell culture</a></li>
<li class="chapter" data-level="30.2" data-path="twowayRM.html"><a href="twowayRM.html#the-test-1"><i class="fa fa-check"></i><b>30.2</b> The test</a></li>
<li class="chapter" data-level="30.3" data-path="twowayRM.html"><a href="twowayRM.html#interpretation-of-the-output"><i class="fa fa-check"></i><b>30.3</b> Interpretation of the output</a><ul>
<li class="chapter" data-level="30.3.1" data-path="twowayRM.html"><a href="twowayRM.html#anova-table-1"><i class="fa fa-check"></i><b>30.3.1</b> ANOVA Table</a></li>
<li class="chapter" data-level="30.3.2" data-path="twowayRM.html"><a href="twowayRM.html#mauchlys-sphericity-test"><i class="fa fa-check"></i><b>30.3.2</b> Mauchly’s Sphericity Test</a></li>
</ul></li>
<li class="chapter" data-level="30.4" data-path="twowayRM.html"><a href="twowayRM.html#post-hoc-multiple-comparisons-1"><i class="fa fa-check"></i><b>30.4</b> Post Hoc multiple comparisons</a></li>
<li class="chapter" data-level="30.5" data-path="twowayRM.html"><a href="twowayRM.html#write-up-14"><i class="fa fa-check"></i><b>30.5</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="twowaymixed.html"><a href="twowaymixed.html"><i class="fa fa-check"></i><b>31</b> Two-way ANOVA RM/CR</a><ul>
<li class="chapter" data-level="31.1" data-path="twowaymixed.html"><a href="twowaymixed.html#chickweight-dataset"><i class="fa fa-check"></i><b>31.1</b> ChickWeight Dataset</a></li>
<li class="chapter" data-level="31.2" data-path="twowaymixed.html"><a href="twowaymixed.html#munge-chickweight-data"><i class="fa fa-check"></i><b>31.2</b> Munge ChickWeight data</a></li>
<li class="chapter" data-level="31.3" data-path="twowaymixed.html"><a href="twowaymixed.html#the-test-2"><i class="fa fa-check"></i><b>31.3</b> The test</a></li>
<li class="chapter" data-level="31.4" data-path="twowaymixed.html"><a href="twowaymixed.html#interpreting-the-anova-output"><i class="fa fa-check"></i><b>31.4</b> Interpreting the ANOVA output</a><ul>
<li class="chapter" data-level="31.4.1" data-path="twowaymixed.html"><a href="twowaymixed.html#anova-the-anova-table-1"><i class="fa fa-check"></i><b>31.4.1</b> $ANOVA: The ANOVA table</a></li>
<li class="chapter" data-level="31.4.2" data-path="twowaymixed.html"><a href="twowaymixed.html#mauchlys-test-and-corrections"><i class="fa fa-check"></i><b>31.4.2</b> Mauchly’s Test and Corrections</a></li>
</ul></li>
<li class="chapter" data-level="31.5" data-path="twowaymixed.html"><a href="twowaymixed.html#post-hoc-pairwise-tests"><i class="fa fa-check"></i><b>31.5</b> Post hoc pairwise tests</a><ul>
<li class="chapter" data-level="31.5.1" data-path="twowaymixed.html"><a href="twowaymixed.html#heres-whats-been-discovered"><i class="fa fa-check"></i><b>31.5.1</b> Here’s what’s been discovered</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="jaxwest2.html"><a href="jaxwest2.html"><i class="fa fa-check"></i><b>32</b> Reproducible Data Munging Mostly with Tidyverse</a><ul>
<li class="chapter" data-level="32.1" data-path="jaxwest2.html"><a href="jaxwest2.html#look-at-the-original-data-carefully"><i class="fa fa-check"></i><b>32.1</b> Look at the original data carefully</a></li>
<li class="chapter" data-level="32.2" data-path="jaxwest2.html"><a href="jaxwest2.html#our-goal"><i class="fa fa-check"></i><b>32.2</b> Our goal</a></li>
<li class="chapter" data-level="32.3" data-path="jaxwest2.html"><a href="jaxwest2.html#read-the-data-into-r"><i class="fa fa-check"></i><b>32.3</b> Read the data into R</a></li>
<li class="chapter" data-level="32.4" data-path="jaxwest2.html"><a href="jaxwest2.html#select-the-variables"><i class="fa fa-check"></i><b>32.4</b> Select the variables</a></li>
<li class="chapter" data-level="32.5" data-path="jaxwest2.html"><a href="jaxwest2.html#trim-the-cases"><i class="fa fa-check"></i><b>32.5</b> Trim the cases</a></li>
<li class="chapter" data-level="32.6" data-path="jaxwest2.html"><a href="jaxwest2.html#go-long"><i class="fa fa-check"></i><b>32.6</b> Go long</a></li>
<li class="chapter" data-level="32.7" data-path="jaxwest2.html"><a href="jaxwest2.html#pull-out-the-values-for-the-day-variable"><i class="fa fa-check"></i><b>32.7</b> Pull out the values for the day variable</a></li>
<li class="chapter" data-level="32.8" data-path="jaxwest2.html"><a href="jaxwest2.html#convert-day-to-numeric"><i class="fa fa-check"></i><b>32.8</b> Convert day to numeric</a></li>
<li class="chapter" data-level="32.9" data-path="jaxwest2.html"><a href="jaxwest2.html#convert-tumor_vol-to-numeric"><i class="fa fa-check"></i><b>32.9</b> Convert tumor_vol to numeric</a></li>
<li class="chapter" data-level="32.10" data-path="jaxwest2.html"><a href="jaxwest2.html#deal-with-that-na"><i class="fa fa-check"></i><b>32.10</b> Deal with that NA</a></li>
<li class="chapter" data-level="32.11" data-path="jaxwest2.html"><a href="jaxwest2.html#convert-variables-to-factor"><i class="fa fa-check"></i><b>32.11</b> Convert variables to factor</a></li>
<li class="chapter" data-level="32.12" data-path="jaxwest2.html"><a href="jaxwest2.html#plot-the-data"><i class="fa fa-check"></i><b>32.12</b> Plot the data</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="anovamc.html"><a href="anovamc.html"><i class="fa fa-check"></i><b>33</b> ANOVA power using Monte Carlo</a><ul>
<li class="chapter" data-level="33.1" data-path="anovamc.html"><a href="anovamc.html#alternatives-to-monte-carlo"><i class="fa fa-check"></i><b>33.1</b> Alternatives to Monte Carlo</a></li>
<li class="chapter" data-level="33.2" data-path="anovamc.html"><a href="anovamc.html#what-is-monte-carlo"><i class="fa fa-check"></i><b>33.2</b> What is Monte Carlo</a></li>
<li class="chapter" data-level="33.3" data-path="anovamc.html"><a href="anovamc.html#one-way-completely-randomized-anova-monte-carlo"><i class="fa fa-check"></i><b>33.3</b> One-way completely randomized ANOVA Monte Carlo</a><ul>
<li class="chapter" data-level="33.3.1" data-path="anovamc.html"><a href="anovamc.html#directions"><i class="fa fa-check"></i><b>33.3.1</b> Directions</a></li>
<li class="chapter" data-level="33.3.2" data-path="anovamc.html"><a href="anovamc.html#step-1-create-initial-values"><i class="fa fa-check"></i><b>33.3.2</b> Step 1: Create initial values</a></li>
<li class="chapter" data-level="33.3.3" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-random-sample"><i class="fa fa-check"></i><b>33.3.3</b> Step 2: Visualize one random sample</a></li>
<li class="chapter" data-level="33.3.4" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator"><i class="fa fa-check"></i><b>33.3.4</b> Step 3: Run The Power Simulator</a></li>
<li class="chapter" data-level="33.3.5" data-path="anovamc.html"><a href="anovamc.html#step-4-optimize-for-suitable-power"><i class="fa fa-check"></i><b>33.3.5</b> Step 4: Optimize for suitable power</a></li>
<li class="chapter" data-level="33.3.6" data-path="anovamc.html"><a href="anovamc.html#notes-and-considerations"><i class="fa fa-check"></i><b>33.3.6</b> Notes And Considerations</a></li>
</ul></li>
<li class="chapter" data-level="33.4" data-path="anovamc.html"><a href="anovamc.html#one-way-related-measures-anova-monte-carlo"><i class="fa fa-check"></i><b>33.4</b> One-way related measures ANOVA Monte Carlo</a></li>
<li class="chapter" data-level="33.5" data-path="anovamc.html"><a href="anovamc.html#directions-1"><i class="fa fa-check"></i><b>33.5</b> Directions</a><ul>
<li class="chapter" data-level="33.5.1" data-path="anovamc.html"><a href="anovamc.html#step-1-initial-values"><i class="fa fa-check"></i><b>33.5.1</b> Step 1: Initial values</a></li>
<li class="chapter" data-level="33.5.2" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-sample"><i class="fa fa-check"></i><b>33.5.2</b> Step 2: Visualize one sample</a></li>
<li class="chapter" data-level="33.5.3" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator-1"><i class="fa fa-check"></i><b>33.5.3</b> Step 3: Run the power simulator</a></li>
<li class="chapter" data-level="33.5.4" data-path="anovamc.html"><a href="anovamc.html#step-4-should-anything-be-changed"><i class="fa fa-check"></i><b>33.5.4</b> Step 4: Should anything be changed?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="correl.html"><a href="correl.html"><i class="fa fa-check"></i><b>34</b> Correlation</a><ul>
<li class="chapter" data-level="34.1" data-path="correl.html"><a href="correl.html#correlation-causation"><i class="fa fa-check"></i><b>34.1</b> Correlation != Causation</a></li>
<li class="chapter" data-level="34.2" data-path="correl.html"><a href="correl.html#correlation-in-multivariate-outcomes-and-paired-designs"><i class="fa fa-check"></i><b>34.2</b> Correlation in Multivariate Outcomes and Paired Designs</a></li>
<li class="chapter" data-level="34.3" data-path="correl.html"><a href="correl.html#correlation-coefficients"><i class="fa fa-check"></i><b>34.3</b> Correlation coefficients</a><ul>
<li class="chapter" data-level="34.3.1" data-path="correl.html"><a href="correl.html#pearsons-correlation-coefficient"><i class="fa fa-check"></i><b>34.3.1</b> Pearson’s correlation coefficient</a></li>
<li class="chapter" data-level="34.3.2" data-path="correl.html"><a href="correl.html#spearmans-rank-correlation"><i class="fa fa-check"></i><b>34.3.2</b> Spearman’s rank correlation</a></li>
<li class="chapter" data-level="34.3.3" data-path="correl.html"><a href="correl.html#kendalls-tau"><i class="fa fa-check"></i><b>34.3.3</b> Kendall’s tau</a></li>
<li class="chapter" data-level="34.3.4" data-path="correl.html"><a href="correl.html#which-correlation-method-to-use"><i class="fa fa-check"></i><b>34.3.4</b> Which correlation method to use?</a></li>
<li class="chapter" data-level="34.3.5" data-path="correl.html"><a href="correl.html#r-correlation-analysis-functions"><i class="fa fa-check"></i><b>34.3.5</b> R correlation analysis functions</a></li>
<li class="chapter" data-level="34.3.6" data-path="correl.html"><a href="correl.html#plot-the-correlations"><i class="fa fa-check"></i><b>34.3.6</b> Plot the correlations</a></li>
<li class="chapter" data-level="34.3.7" data-path="correl.html"><a href="correl.html#calculate-a-correlation-coefficient-and-posthoc-test"><i class="fa fa-check"></i><b>34.3.7</b> Calculate a correlation coefficient and posthoc test</a></li>
<li class="chapter" data-level="34.3.8" data-path="correl.html"><a href="correl.html#interpretation-of-correlation-output"><i class="fa fa-check"></i><b>34.3.8</b> Interpretation of correlation output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="regress.html"><a href="regress.html"><i class="fa fa-check"></i><b>35</b> Linear Regression</a><ul>
<li class="chapter" data-level="35.1" data-path="regress.html"><a href="regress.html#the-linear-regression-model"><i class="fa fa-check"></i><b>35.1</b> The linear regression model</a></li>
<li class="chapter" data-level="35.2" data-path="regress.html"><a href="regress.html#least-squares-fitting"><i class="fa fa-check"></i><b>35.2</b> Least squares fitting</a></li>
<li class="chapter" data-level="35.3" data-path="regress.html"><a href="regress.html#the-practical-importance-of-linear-model-parameters"><i class="fa fa-check"></i><b>35.3</b> The practical importance of linear model parameters</a></li>
<li class="chapter" data-level="35.4" data-path="regress.html"><a href="regress.html#linear-model-standard-errors"><i class="fa fa-check"></i><b>35.4</b> Linear model standard errors</a></li>
<li class="chapter" data-level="35.5" data-path="regress.html"><a href="regress.html#linear-regression-in-r"><i class="fa fa-check"></i><b>35.5</b> Linear regression in R</a></li>
<li class="chapter" data-level="35.6" data-path="regress.html"><a href="regress.html#intepretation"><i class="fa fa-check"></i><b>35.6</b> Intepretation</a><ul>
<li class="chapter" data-level="35.6.1" data-path="regress.html"><a href="regress.html#residuals"><i class="fa fa-check"></i><b>35.6.1</b> Residuals</a></li>
<li class="chapter" data-level="35.6.2" data-path="regress.html"><a href="regress.html#coefficients"><i class="fa fa-check"></i><b>35.6.2</b> Coefficients</a></li>
<li class="chapter" data-level="35.6.3" data-path="regress.html"><a href="regress.html#degrees-of-freedom-1"><i class="fa fa-check"></i><b>35.6.3</b> Degrees of freedom</a></li>
<li class="chapter" data-level="35.6.4" data-path="regress.html"><a href="regress.html#r-squared"><i class="fa fa-check"></i><b>35.6.4</b> R-squared</a></li>
<li class="chapter" data-level="35.6.5" data-path="regress.html"><a href="regress.html#f-statistic"><i class="fa fa-check"></i><b>35.6.5</b> F-statistic</a></li>
<li class="chapter" data-level="35.6.6" data-path="regress.html"><a href="regress.html#plotting-regression-results"><i class="fa fa-check"></i><b>35.6.6</b> Plotting regression results</a></li>
<li class="chapter" data-level="35.6.7" data-path="regress.html"><a href="regress.html#visualizing-residuals"><i class="fa fa-check"></i><b>35.6.7</b> Visualizing residuals</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="nonlinearintro.html"><a href="nonlinearintro.html"><i class="fa fa-check"></i><b>36</b> Non-linear regression introduction</a><ul>
<li class="chapter" data-level="36.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#uses-for-nonlinear-regression"><i class="fa fa-check"></i><b>36.1</b> Uses for nonlinear regression</a></li>
<li class="chapter" data-level="36.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-and-parameters"><i class="fa fa-check"></i><b>36.2</b> Nonlinear models and parameters</a><ul>
<li class="chapter" data-level="36.2.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#hyperbolic-stimulus-response-functions"><i class="fa fa-check"></i><b>36.2.1</b> Hyperbolic stimulus response functions</a></li>
<li class="chapter" data-level="36.2.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#visualizing-nonlinear-data-and-log-scaling"><i class="fa fa-check"></i><b>36.2.2</b> Visualizing nonlinear data and log scaling</a></li>
</ul></li>
<li class="chapter" data-level="36.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#how-do-regression-fits-happen"><i class="fa fa-check"></i><b>36.3</b> How do regression fits happen?</a><ul>
<li class="chapter" data-level="36.3.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#selecting-the-right-model"><i class="fa fa-check"></i><b>36.3.1</b> Selecting the right model</a></li>
<li class="chapter" data-level="36.3.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#from-models-to-formulas"><i class="fa fa-check"></i><b>36.3.2</b> From models to formulas</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="nestedregress.html"><a href="nestedregress.html"><i class="fa fa-check"></i><b>37</b> Nested-model nonlinear regression</a><ul>
<li class="chapter" data-level="37.1" data-path="nestedregress.html"><a href="nestedregress.html#read-and-plot-the-data"><i class="fa fa-check"></i><b>37.1</b> Read and plot the data</a></li>
<li class="chapter" data-level="37.2" data-path="nestedregress.html"><a href="nestedregress.html#perform-the-nonlinear-regression"><i class="fa fa-check"></i><b>37.2</b> Perform the nonlinear regression</a><ul>
<li class="chapter" data-level="37.2.1" data-path="nestedregress.html"><a href="nestedregress.html#troubleshooting-the-regression"><i class="fa fa-check"></i><b>37.2.1</b> Troubleshooting the regression</a></li>
<li class="chapter" data-level="37.2.2" data-path="nestedregress.html"><a href="nestedregress.html#interpreting-the-parameters"><i class="fa fa-check"></i><b>37.2.2</b> Interpreting the parameters</a></li>
<li class="chapter" data-level="37.2.3" data-path="nestedregress.html"><a href="nestedregress.html#residual-plots-to-compare-fits"><i class="fa fa-check"></i><b>37.2.3</b> Residual plots to compare fits</a></li>
<li class="chapter" data-level="37.2.4" data-path="nestedregress.html"><a href="nestedregress.html#compare-aic"><i class="fa fa-check"></i><b>37.2.4</b> Compare AIC</a></li>
<li class="chapter" data-level="37.2.5" data-path="nestedregress.html"><a href="nestedregress.html#other-ways-to-compare-models"><i class="fa fa-check"></i><b>37.2.5</b> Other ways to compare models</a></li>
<li class="chapter" data-level="37.2.6" data-path="nestedregress.html"><a href="nestedregress.html#interpretation-of-this-one-replicate"><i class="fa fa-check"></i><b>37.2.6</b> Interpretation of this one replicate</a></li>
<li class="chapter" data-level="37.2.7" data-path="nestedregress.html"><a href="nestedregress.html#alternate-analysis"><i class="fa fa-check"></i><b>37.2.7</b> Alternate analysis</a></li>
</ul></li>
<li class="chapter" data-level="37.3" data-path="nestedregress.html"><a href="nestedregress.html#summary-3"><i class="fa fa-check"></i><b>37.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="38" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html"><i class="fa fa-check"></i><b>38</b> Nonlinear regression of independent replicates</a><ul>
<li class="chapter" data-level="38.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-dataset"><i class="fa fa-check"></i><b>38.1</b> The dataset</a></li>
<li class="chapter" data-level="38.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#munging-for-regression-analysis"><i class="fa fa-check"></i><b>38.2</b> Munging for regression analysis</a><ul>
<li class="chapter" data-level="38.2.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#average-the-technical-replicates"><i class="fa fa-check"></i><b>38.2.1</b> Average the technical replicates</a></li>
<li class="chapter" data-level="38.2.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#create-one-table"><i class="fa fa-check"></i><b>38.2.2</b> Create one table</a></li>
<li class="chapter" data-level="38.2.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#plot-the-data-1"><i class="fa fa-check"></i><b>38.2.3</b> Plot the data</a></li>
<li class="chapter" data-level="38.2.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#clean-up-regression-results"><i class="fa fa-check"></i><b>38.2.4</b> Clean up regression results</a></li>
</ul></li>
<li class="chapter" data-level="38.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#t-test-on-half-lives"><i class="fa fa-check"></i><b>38.3</b> T-test on half-lives</a></li>
<li class="chapter" data-level="38.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#conclusion"><i class="fa fa-check"></i><b>38.4</b> Conclusion</a></li>
<li class="chapter" data-level="38.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#summary-figure"><i class="fa fa-check"></i><b>38.5</b> Summary figure</a></li>
</ul></li>
<li class="chapter" data-level="39" data-path="multregress.html"><a href="multregress.html"><i class="fa fa-check"></i><b>39</b> Multiple regression</a><ul>
<li class="chapter" data-level="39.1" data-path="multregress.html"><a href="multregress.html#the-linear-regression-model-1"><i class="fa fa-check"></i><b>39.1</b> The linear regression model</a></li>
<li class="chapter" data-level="39.2" data-path="multregress.html"><a href="multregress.html#multiple-regression-models"><i class="fa fa-check"></i><b>39.2</b> Multiple regression models</a></li>
<li class="chapter" data-level="39.3" data-path="multregress.html"><a href="multregress.html#experimental-multiple-regression"><i class="fa fa-check"></i><b>39.3</b> Experimental multiple regression</a><ul>
<li class="chapter" data-level="39.3.1" data-path="multregress.html"><a href="multregress.html#a-one-factor-experiment-with-3-groups"><i class="fa fa-check"></i><b>39.3.1</b> A one factor experiment with 3 groups</a></li>
<li class="chapter" data-level="39.3.2" data-path="multregress.html"><a href="multregress.html#a-two-factor-experiment-with-6-groups"><i class="fa fa-check"></i><b>39.3.2</b> A two factor experiment with 6 groups</a></li>
</ul></li>
<li class="chapter" data-level="39.4" data-path="multregress.html"><a href="multregress.html#multiple-linear-mixed-models"><i class="fa fa-check"></i><b>39.4</b> Multiple linear mixed models</a></li>
<li class="chapter" data-level="39.5" data-path="multregress.html"><a href="multregress.html#multiple-regression-of-observational-data"><i class="fa fa-check"></i><b>39.5</b> Multiple regression of observational data</a></li>
</ul></li>
<li class="chapter" data-level="40" data-path="logregress.html"><a href="logregress.html"><i class="fa fa-check"></i><b>40</b> Logistic regression</a><ul>
<li class="chapter" data-level="40.1" data-path="logregress.html"><a href="logregress.html#uses-of-logistic-regression"><i class="fa fa-check"></i><b>40.1</b> Uses of logistic regression</a><ul>
<li class="chapter" data-level="40.1.1" data-path="logregress.html"><a href="logregress.html#sidebar-doing-logistic-regression-is-machine-learning"><i class="fa fa-check"></i><b>40.1.1</b> Sidebar: Doing logistic regression is machine learning</a></li>
</ul></li>
<li class="chapter" data-level="40.2" data-path="logregress.html"><a href="logregress.html#derivation-of-the-logistic-regression-model"><i class="fa fa-check"></i><b>40.2</b> Derivation of the logistic regression model</a><ul>
<li class="chapter" data-level="40.2.1" data-path="logregress.html"><a href="logregress.html#relationship-of-logit-to-odds-to-the-model-coefficients-and-probability"><i class="fa fa-check"></i><b>40.2.1</b> Relationship of logit to odds to the model coefficients and probability</a></li>
<li class="chapter" data-level="40.2.2" data-path="logregress.html"><a href="logregress.html#additional-types-of-logistic-regression-models"><i class="fa fa-check"></i><b>40.2.2</b> Additional types of logistic regression models</a></li>
</ul></li>
<li class="chapter" data-level="40.3" data-path="logregress.html"><a href="logregress.html#stress-and-survival"><i class="fa fa-check"></i><b>40.3</b> Stress and survival</a><ul>
<li class="chapter" data-level="40.3.1" data-path="logregress.html"><a href="logregress.html#interpretation-of-output"><i class="fa fa-check"></i><b>40.3.1</b> Interpretation of output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="41" data-path="mixedlogistic.html"><a href="mixedlogistic.html"><i class="fa fa-check"></i><b>41</b> Mixed model logistic regression</a><ul>
<li class="chapter" data-level="41.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#mixed-models-fixed-and-random-effects"><i class="fa fa-check"></i><b>41.1</b> Mixed models, fixed and random effects</a><ul>
<li class="chapter" data-level="41.1.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#nfat-localization-within-smooth-muscle-cells"><i class="fa fa-check"></i><b>41.1.1</b> NFAT localization within smooth muscle cells</a></li>
</ul></li>
<li class="chapter" data-level="41.2" data-path="mixedlogistic.html"><a href="mixedlogistic.html#data-1"><i class="fa fa-check"></i><b>41.2</b> Data</a><ul>
<li class="chapter" data-level="41.2.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#inference-1"><i class="fa fa-check"></i><b>41.2.1</b> Inference</a></li>
</ul></li>
<li class="chapter" data-level="41.3" data-path="mixedlogistic.html"><a href="mixedlogistic.html#alternative-analysis"><i class="fa fa-check"></i><b>41.3</b> Alternative analysis</a></li>
</ul></li>
<li class="chapter" data-level="42" data-path="lognormal.html"><a href="lognormal.html"><i class="fa fa-check"></i><b>42</b> The lognormal distribution</a><ul>
<li class="chapter" data-level="42.1" data-path="lognormal.html"><a href="lognormal.html#dlnorm"><i class="fa fa-check"></i><b>42.1</b> dlnorm</a></li>
<li class="chapter" data-level="42.2" data-path="lognormal.html"><a href="lognormal.html#plnorm"><i class="fa fa-check"></i><b>42.2</b> plnorm</a></li>
<li class="chapter" data-level="42.3" data-path="lognormal.html"><a href="lognormal.html#qlnorm"><i class="fa fa-check"></i><b>42.3</b> qlnorm</a></li>
<li class="chapter" data-level="42.4" data-path="lognormal.html"><a href="lognormal.html#rlnorm"><i class="fa fa-check"></i><b>42.4</b> rlnorm</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">JABSTB: Statistical Design and Analysis of Experiments with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="categorical" class="section level1">
<h1><span class="header-section-number">Chapter 17</span> Statistics for Categorical Data</h1>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb226-1" title="1"><span class="kw">library</span>(PropCIs)</a>
<a class="sourceLine" id="cb226-2" title="2"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb226-3" title="3"><span class="kw">library</span>(knitr)</a>
<a class="sourceLine" id="cb226-4" title="4"><span class="kw">library</span>(kableExtra)</a>
<a class="sourceLine" id="cb226-5" title="5"><span class="kw">library</span>(binom)</a>
<a class="sourceLine" id="cb226-6" title="6"><span class="kw">library</span>(pwr)</a>
<a class="sourceLine" id="cb226-7" title="7"><span class="kw">library</span>(statmod)</a>
<a class="sourceLine" id="cb226-8" title="8"><span class="kw">library</span>(EMT)</a></code></pre></div>
<p>Biomedical research is full of studies that count discrete events and sort them into nominal categories.</p>
<p>A common mistake made by many researchers is to use statistics designed for measured (continuous) variables on discrete count variables. For example, they transform count data into scalar measures (eg, percents, folds etc) and then apply statistics designed for continuous variables to events that are fundamentally discrete by nature. The problem with that is there are inherent differences in the behaviors of continuous and discrete variables.</p>
<p>Recognize that there are statistical tests that can be used to analyze discrete data directly, without resorting to these transformations.</p>
<div id="types-of-categorical-data" class="section level2">
<h2><span class="header-section-number">17.1</span> Types of categorical data</h2>
<ul>
<li>What proportion of cells express a specific antigen and does an experimental treatment cause that proportion to change?<br />
</li>
<li>What proportion of rats treated with an anxiolytic drug choose one chamber over others in a maze test?</li>
<li>How many people who express a certain biomarker go on to have cancer?</li>
</ul>
<p>In these three scenarios the primary data are counts. All of the study results have integer values. The counts are categorized with variable attributes, thus they are called categorical data.</p>
<p>In fact, the three scenarios above are very different experimental designs. The first represent experiments that compare simple proportions. The second compare frequencies, and the third is an association study. The analysis of these require using a common suite of statistical tools in slightly different ways.</p>
<p>Broadly, all of these tools boil down to dealing with proportions. A few types of proportions (eg, odds ratio, relative risk) that can be calculated from these datasets are sometimes used as effect size parameters. Other times we’d use a confidence interval as a way of conveying an effect size. Statistical tests are then used to evaluate whether these effect sizes, or frequencies, or simple proportions, are extreme enough relative to null counterparts so that we can conclude the experimental variable had some sort of effect.</p>
<div id="proportions" class="section level3">
<h3><span class="header-section-number">17.1.1</span> Proportions</h3>
<p>We might…</p>
<ul>
<li><p>Inactivate a gene hypothesized to function in the reproductive pathway. To test it, mating triangles would be set up to count the number of female mice that become pregnant or not.</p></li>
<li><p>Implant a tumor in mice, before counting the number of survivors and non-survivors at a given point later.</p></li>
<li><p>Mutate a protein that we hypothesize moves in and out of an intracellular compartment, before staining cells to count the number of cells where it is and is not located in a compartment of interest.</p></li>
</ul>
<p>Each of the examples above have <code>binomial</code> outcomes…pregnant vs not, dead vs alive, or inside vs outside.</p>
<p>In each case above, both the successful and the failed events are counted in the experiment. A proportion is a simple ratio of counts of success to counts of failures.</p>
</div>
<div id="frequencies" class="section level3">
<h3><span class="header-section-number">17.1.2</span> Frequencies</h3>
<p>Other kinds of counted data occur randomly in space and time. The examples below illustrate this. Note how only the number of events are recorded, rather than categorizing them as successes or failures. These counts therefore have the statistical property of frequency, which are counts of events in space or time.</p>
<p>We can…</p>
<ul>
<li><p>Expose a neuron to an excitatory neurotransmitter, then count the number of times it depolarizes.</p></li>
<li><p>In a model of asthma, count the number of immune cells that are washed out in a pulmonary lavage protocol after treatment with an immunosuppressive agent.</p></li>
</ul>
<p>The key difference for these compared to binomial events is that their non-event counterparts are meaningless. For example, it is not possible to measure the number of depolarizations that don’t occur, or know the number of immune cells that don’t wash out in the lavage.</p>
<p>Importantly, the dependent variable is simple discrete counts. If we count 555 events in a 5 minute window, the recorded value and units is 555 counts, not 111 counts per minute.</p>
</div>
<div id="associations" class="section level3">
<h3><span class="header-section-number">17.1.3</span> Associations</h3>
<p>Lastly, the examples below illustrate the design of association studies, which are based upon, according to the null hypothesis, independent predictors and outcomes.</p>
<p>Here are some examples of association study designs:</p>
<p>You might wish to</p>
<ul>
<li><p>identify causal alleles associated with a specific disease phenotype by counting the number of people with and without the disease, who have or don’t have a particular allelic variant.</p></li>
<li><p>determine if a history of exposure to certain carcinogens is associated with a higher risk of cancer by counting people with cancer who have been exposed.</p></li>
<li><p>know if a drug treatment causes a higher than expected frequency of a side-effect by counting the people on the drug with the side-effect.</p></li>
</ul>
<p>In the simplest (and most general) case, association studies are 2X2 in design: A predictor is either present or absent as the row factor, and an outcome was either a success for a failure as the column factor. Subjects are categorized into groups on the basis of where they fall in the 4 possible combinations that these 2X2’s allow for.</p>
<p>It should be noted that analysis of higher order association studies is also possible, which can be either symmetric (eg, 3x3) or non-symmetric (eg,) 9X2, 2x3, and so on.</p>
</div>
<div id="statistics-covered-here" class="section level3">
<h3><span class="header-section-number">17.1.4</span> Statistics Covered Here</h3>
<ul>
<li>Confidence intervals of proportions</li>
<li>One-sample proportions test</li>
<li>Two-sample proportions test</li>
<li>Goodness of fit tests</li>
<li>Tests of associations (aka contingency testing)</li>
<li>Power analysis of proportions (including Monte Carlo simulation)</li>
<li>Plotting proportions with ggplot2</li>
</ul>
</div>
</div>
<div id="exact-v-asymptotic-calculations-of-p-values" class="section level2">
<h2><span class="header-section-number">17.2</span> Exact v Asymptotic Calculations of p-values</h2>
<p>The statistical tests for hypotheses on categorical data fall into two broad categories: exact tests (<code>binom.test</code>, <code>fisher.test</code>, <code>multinomial.test</code>) and asymptotic tests (<code>prop.test</code>, <code>chisq.test</code>).</p>
<p>Exact tests calculate exact p-values. That’s made possible using factorial math. The <code>prop.test</code> and <code>chisq.test</code> generate asymptotic (aka, approximate) p-values. They calculate a <span class="math inline">\(\chi^2\)</span> test statistic from the data before mapping it to a <span class="math inline">\(\chi^2\)</span> probability density function. Because that function is continuous, the p-values it generates are asymptotically-estimated, rather than exactly calculated.</p>
<p>That should strike you as odd that a continuous sampling distribution is used for discrete data. The reasons are historical. In the early days, it was harder to calculate exact p-values than <span class="math inline">\(\chi^2\)</span> values. Tables existed (in text books) with critical values of the <span class="math inline">\(\chi^2\)</span> and associated p-values.</p>
<div id="choosing-exact-or-asymptotic" class="section level3">
<h3><span class="header-section-number">17.2.1</span> Choosing exact or asymptotic</h3>
<p>As a general rule, given the same datasets and arguments, exact and approximate hypothesis test functions will almost always give you p-values that differ, but only slightly.</p>
<p>That’s usually not a problem unless you’re near a threshold value.</p>
<p>Typically, an integrity crisis is evoked when that happens. For example, you don’t want to put yourself in a position to run the same data through two tests that generate p-values on either side of a threshold. To ask “Which test is”right???" is not the right question. Instead, if you find yourself in such a position ask yourself whether p-hacking is right (it is not).</p>
<p>You should use the test you said you’d use when you first designed the experiment. And if you didn’t pre-plan…or at least have some idea about where you are going…recognize that the exact tests are more accurate, even when they give you the answer you don’t want.</p>
<p>Another issue that arises is how well the tests perform with low count numbers. For example, as a rule of thumb, we’re advised to avoid using the <code>chisq.test</code> when the data have counts less than 5 in more than 20% of the cells because the accuracy of the <code>chisq.test</code> is less at these low cell counts. Use an exact test instead.</p>
</div>
</div>
<div id="overview-of-the-types-of-hypothesis-testing" class="section level2">
<h2><span class="header-section-number">17.3</span> Overview of the types of hypothesis testing</h2>
<p>We’ll go through each below in more detail, emphasizing practical experimental design and interpretation principles.</p>
<div id="proportion-analysis" class="section level3">
<h3><span class="header-section-number">17.3.1</span> Proportion analysis</h3>
<p>You can learn a lot about experimental statistics by thinking about proportions. Proportions are derived from events that can be classified as either successes or failures. Sometimes we want to compare simple proportions to decide if they are the same or not.</p>
</div>
<div id="goodness-of-fit-testing" class="section level3">
<h3><span class="header-section-number">17.3.2</span> Goodness of fit testing</h3>
<p>We do this when we want to compare the frequency distribution we observe in an experiment to the null expectation for that frequency distribution.</p>
</div>
<div id="contingency-analysis" class="section level3">
<h3><span class="header-section-number">17.3.3</span> Contingency Analysis</h3>
<p>Contingency analysis, otherwise known as tests of independence, are very different from goodness-of-fit test and simple proportion tests, in design and in purpose. They allow us to ask if two (or more) variables are associated with each other. Unlike a lot of the statistics we’ll deal with, there is a hint of a predictive element associated with these types of studies because the effect sizes we use to explain their results are related to odds and risk and likelihood. Which is not to say that we couldn’t use the same predictive concepts in proportions and goodness of fit testing.</p>
<p>Contingency tests are very common in epidemiology and in clinical science. You recognize them by their names as cohort studies, case control studies, and so forth.</p>
</div>
</div>
<div id="comparing-proportions" class="section level2">
<h2><span class="header-section-number">17.4</span> Comparing proportions</h2>
<p>In their simplest use, the tests here can be used to compare one proportion to another. Is the proportion of successes to failures that results from a treatment different from the proportion that results from control? We’ll dive into this further below.</p>
<div id="a-mouse-t-cell-pilot-experiment-the-cytokine-inducible-antigen-gradstudin" class="section level3">
<h3><span class="header-section-number">17.4.1</span> A Mouse T Cell Pilot Experiment: The Cytokine-inducible antigen gradstudin</h3>
<p>Let’s imagine a small pilot experiment to see how a cytokine affects T cells. This is a very crude thought experiment designed mostly to illustrate some principles.</p>
<p>A cytokine is injected into a single mouse. There is no control injection, just one mouse/one cytokine injection. A time later, blood is harvested from the mouse to measure an antigen on T cells. Let’s call the antigen gradstudin.</p>
<p>Assume a method exists to detect T cells that express gradstudin and that don’t express gradstudin.</p>
<p>That method implies some <code>nominal</code> criteria are established to <code>categorize</code> T cells as either expressing gradstudin or not. FACS machines are very useful for this. The machine typically produces <code>continuous</code> fluorescent data, where intensity is proportional to gradstudin levels. But we don’t care about the magnitude of the expression level, we just care whether it is there or not.</p>
<p>Based upon our scientific expertise, we establish cutoff gating criteria. A cell shows a level of fluorescence above which it is sorted as gradstudin positive.</p>
<p>The machine therefore returns simple counts of both gradstudin-positive and gradstudin-negative cells.</p>
</div>
<div id="calculating-proportions" class="section level3">
<h3><span class="header-section-number">17.4.2</span> Calculating Proportions</h3>
<p>Here’s the data, counts of cells expressing and not expressing gradstudin. It is a very simple dataset:</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb227-1" title="1">pos &lt;-<span class="st"> </span><span class="dv">5042</span></a>
<a class="sourceLine" id="cb227-2" title="2">neg &lt;-<span class="st"> </span><span class="dv">18492</span></a></code></pre></div>
<p><strong>A proportion is the count of a particular outcome relative to the total number of events.</strong></p>
<p>It’s customary to use the number of successes as the numerator. Whereas its customary to refer to the total number of events, <code>n</code>, as the trial number rather than as sample size, but they mean the same thing.</p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb228-1" title="1">n &lt;-<span class="st"> </span>pos<span class="op">+</span>neg <span class="co">#trial size</span></a>
<a class="sourceLine" id="cb228-2" title="2">prop &lt;-<span class="st"> </span>pos<span class="op">/</span>n</a>
<a class="sourceLine" id="cb228-3" title="3">prop</a></code></pre></div>
<pre><code>## [1] 0.2142432</code></pre>
</div>
<div id="what-a-proportion-estimates" class="section level3">
<h3><span class="header-section-number">17.4.3</span> What A Proportion Estimates</h3>
<p>This sample proportion is a descriptive statistic. It serves as a <code>point estimate</code> of the true proportion of the population we sampled. The only way to know the true proportion would be to count every T cell in every drop of the subjects blood, which is not going to happen.</p>
<p>This point estimate is statistically valid if our sample meets two conditions. First, that this is a random sample of the T cells in the subject’s blood. Second, if we consider every T cell in the sample as statistically independent of every other T cell.</p>
<p>We can safely assume those conditions are met. Strictly, as an estimate this proportion only infers the population of blood borne T cells IN THAT ONE subject. We really can’t generalize much further than that. Surely T cells are sequestered in other compartments (thymus, nodes, etc) that a blood draw cannot estimate.</p>
<p>Which is fine for our purposes now because we’re trying to keep this simple.</p>
</div>
<div id="confidence-intervals-of-proportions" class="section level3">
<h3><span class="header-section-number">17.4.4</span> Confidence Intervals of Proportions</h3>
<p>Confidence intervals (CI) have features of both descriptive and inferential statistics.</p>
<div id="definition-of-a-95-ci" class="section level4">
<h4><span class="header-section-number">17.4.4.1</span> Definition of a 95% CI</h4>
<p><strong>The range of proportions within which 95% of the time we would expect the true population proportion.</strong></p>
<p>There’s a lot going on there.</p>
<p>The value of the proportion we measured in the sample is a mathematical fact that is not in dispute. It is what it is. The question is, what does it represent?</p>
<p>Although there might be some error associated with measuring it, our single sample offers no real information about what that error might be.</p>
<p><strong>What is unknown is the true proportion of gradstudin+ T cells in the population we sampled.</strong></p>
<p>CI’s are designed to give us some insights into that unknown.</p>
<p>95% CI’s are a <code>range estimate</code> of what that true population proportion might be.
CI’s are calculated in part upon the quality of the point estimate. In the case of proportions, the quality of the point estimate is driven by the size of the sample, the number of counts that are involved in calculating the proportion.</p>
<p>As you might imagine intuitively, the more counts we have in the sample, the more confidence we should have that our proportion provides a good estimate of the population’s proportion.</p>
</div>
<div id="calculating-ci-with-r" class="section level4">
<h4><span class="header-section-number">17.4.4.2</span> Calculating CI with R</h4>
<p>Their are several ways to calculate the CI for a proportion. Because of that, when publishing CI’s of proportions it is important is to state which CI method is used.</p>
<p>Is one better than the other? Sometimes, yes. For now, let’s not worry about that. <a href="http://www.ucl.ac.uk/english-usage/staff/sean/resources/binomialpoisson.pdf">Wilson’s score interval with continuity correction</a>] is suggested as the most accurate for proportions.</p>
<p>Other methods are more commonly used than Wilson’s because they gained traction as being easier to compute by hand, and old habits die slowly.</p>
<p>Taking the data on cytokine induced gradstudin+ T cells, the chunk below illustrates how to use <code>PropCIs</code> to derive a Wilson score interval-based 95% CI:</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb230-1" title="1"><span class="kw">scoreci</span>(pos, n, <span class="dt">conf.level=</span><span class="fl">0.95</span>)</a></code></pre></div>
<pre><code>## 
## 
## 
## data:  
## 
## 95 percent confidence interval:
##  0.2090 0.2195</code></pre>
</div>
<div id="interpretation-of-a-ci" class="section level4">
<h4><span class="header-section-number">17.4.4.3</span> Interpretation of a CI</h4>
<p>The value of our sample proportion, 0.214, falls within this 95% CI. That’s not a big surprise, given the 95% CI was calculated from our proportion!</p>
<p>On the basis of the sample proportion, we can conclude from this CI that there is a 95% confidence the true proportion of gradstudin positive T cells falls within this very narrow range of 0.02090 to 0.2195.</p>
</div>
<div id="using-the-ci-as-a-quick-test-of-statistical-significance." class="section level4">
<h4><span class="header-section-number">17.4.4.4</span> Using the CI as a quick test of statistical significance.</h4>
<p>Let’s say, for example, that we have tremendous experience and great scientific reason to expect that only 15% of T cells would be gradstudin-positive. Does our sample proportion differ from that expectation?</p>
<p>Since a proportion of 0.15 is not within the 95% CI calculated above, we can conclude that the cytokine-induced sample proportion differs from this expectation at the 5% level of statistical significance.</p>
<p>We just did a statistical test, without running any software (sorta) or generating any p-values!! And it is perfectly valid inference.</p>
</div>
</div>
<div id="a-one-sample-proportion-test" class="section level3">
<h3><span class="header-section-number">17.4.5</span> A One-Sample Proportion Test</h3>
<p>Now we’ll use <code>prop.test</code> to run a test that generates a p-value to decide if the sample proportion we have above differs from 0.15. If you know something about a one sample t-test, then this should sound familiar to you.</p>
<div id="hypothesis-tested-in-a-one-sample-proportion-test" class="section level4">
<h4><span class="header-section-number">17.4.5.1</span> Hypothesis Tested in a One-Sample Proportion Test</h4>
<p>In this test the sample antigen-positive proportion is compared to a theoretical proportion. If the typical proportion of antigen-positive T cells within a blood sample is 15%, is the result after cytokine treatment different from this proportion?</p>
<p>Let’s say that our scientific hypothesis is that the cytokine induces the antigen on T cells. Since we are predicting an increase, for logical coherence we should establish a one-sided alternative (thus using <code>greater</code> as an argument in <code>prop.test</code> below) as our statistical hypothesis.</p>
<p>Our statistical hypothesis is the null. We’ll decide whether or not to reject the null on the basis of the test results. Philosophically, we’re using a falsification method.</p>
<p>The statistical alternate hypothesis: <span class="math inline">\(\pi&gt;15\%\)</span></p>
<p>The statistical null hypothesis: <span class="math inline">\(\pi\le15\%\)</span></p>
<p>Note how logically the two hypotheses are mutually exclusive and comprehensive.</p>
<p>We use Greek notation to represent the ‘true’ population proportion. This reminds us that a statistical hypothesis is an inferential test about the population proportion.</p>
<p>Again, there is no question that the sample proportion differs from a proportion of 15%. 21% does not equal 15%. That’s a simple numerical fact. Statistical tests are not necessary to make that assertion.</p>
<p>We use a test to draw inference on the composition of all of the T cells in the blood of the subject given this one sample. Thus, the sample p is only a point estimate of a true <span class="math inline">\(\pi\)</span> (which we notate using Greek letters).</p>
<p>The chunk below lays out these arguments using R’s <code>prop.test</code> function:</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb232-1" title="1"><span class="co">#pos and n in the test arguments are objects that were defined above!</span></a>
<a class="sourceLine" id="cb232-2" title="2"><span class="kw">prop.test</span>(</a>
<a class="sourceLine" id="cb232-3" title="3">  pos, n, <span class="dt">p=</span><span class="fl">0.15</span>, </a>
<a class="sourceLine" id="cb232-4" title="4">  <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span>, </a>
<a class="sourceLine" id="cb232-5" title="5">  <span class="dt">conf.level =</span> <span class="fl">0.95</span>, </a>
<a class="sourceLine" id="cb232-6" title="6">  <span class="dt">correct =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb232-7" title="7">  )</a></code></pre></div>
<pre><code>## 
##  1-sample proportions test with continuity correction
## 
## data:  pos out of n, null probability 0.15
## X-squared = 761.29, df = 1, p-value &lt; 2.2e-16
## alternative hypothesis: true p is greater than 0.15
## 95 percent confidence interval:
##  0.2098559 1.0000000
## sample estimates:
##         p 
## 0.2142432</code></pre>
</div>
<div id="interpreting-one-sample-prop.test-output" class="section level4">
<h4><span class="header-section-number">17.4.5.2</span> Interpreting one-sample prop.test output</h4>
<p>Like all statistical tests, this one is evaluated under the assumption that the null hypothesis is true. We use the test outcome to decide whether the null hypothesis should be rejected.</p>
<p>The <code>prop.test</code> conducts a chi-square analysis. The value of <span class="math inline">\(\chi^2\)</span> for this sample is very large, extreme. Read more about the chi-square sampling distribution in Chapter <a href="chisquare.html#chisquare">18</a> to get a feel for what these values mean.</p>
<p><strong>The p-value represents the probability of obtaining a <span class="math inline">\(\chi^2\)</span> value as larger or larger from this test, if the null is true.</strong></p>
<p>If the null hypothesis is true in this case, the probability of a <span class="math inline">\(\chi^2\)</span> value as large or larger than we obtained is 2.2e-16, which is very, very low.</p>
<p>The 95% CI is 0.2098 to 1.0. There is a 95% chance the population proportion is greater than 0.2098. The reason it differs from the Wilson’s CI calculated above is that we used <code>greater</code>as a one-sided hypothesis argument in the prop.test. These CI’s from one-sided hypothesis tests are not particularly useful when attempting to use the CI as an index of accuracy. But perhaps this oddity helps you understand how CI’s are tied to the hypothesis and thus inference.</p>
</div>
<div id="how-to-write-this-up" class="section level4">
<h4><span class="header-section-number">17.4.5.3</span> How to write this up</h4>
<p><em>The proportion of gradstudin positive T cells after cytokine treatment in the subject differs from an expected value of 0.15 (one-sided one-sample proportions test, p-value=2.2e-16, 95% CI = 0.209 to 1.0)</em></p>
</div>
</div>
<div id="an-exact-test-for-one-proportion" class="section level3">
<h3><span class="header-section-number">17.4.6</span> An exact test for one proportion</h3>
<p>R’s <code>binom.test</code> function is an exact test for whether a proportion differs from a theoretical expectation. It compares proportions using an entirely different procedure.</p>
<p>As a <strong>one-proportion test</strong> the <code>binom.test</code> gives an exact p-value derived from the binomial distribution, whereas the <code>prop.test</code> gives approximate p-values because it uses the chi-square distribution.</p>
<p>That distinction is hard to see with our examples here, but the differences will become more noticeable when analyzing samples with far fewer events.</p>
<p>Here’s the binomial test run on the sample we ran above in <code>prop.test</code>.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb234-1" title="1"><span class="kw">binom.test</span>(pos, n, <span class="dt">p=</span><span class="fl">0.15</span>, <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span>)</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  pos and n
## number of successes = 5042, number of trials = 23534, p-value &lt; 2.2e-16
## alternative hypothesis: true probability of success is greater than 0.15
## 95 percent confidence interval:
##  0.209849 1.000000
## sample estimates:
## probability of success 
##              0.2142432</code></pre>
<p>The results are about the same and the write up would be the same.</p>
</div>
<div id="comparing-two-proportions" class="section level3">
<h3><span class="header-section-number">17.4.7</span> Comparing Two Proportions</h3>
<p>We can stick with the T cell-cytokine-gradstudin scenario, but let’s change up the experiment a tad.</p>
<p>Let’s imagine we’ve withdrawn a sample of blood from a subject and enriched for T cells. Half of the sample is exposed in vitro to a cytokine for a few hours. The other half is exposed to vehicle as a control. We count the gradstudin-positive and gradstudin-negative T cells in both groups.</p>
<p>We now have a predictor group at two levels (treatment = vehicle or cytokine) and an outcome group at two levels (antigen = positive or negative)</p>
<div id="hypothesis-tested" class="section level4">
<h4><span class="header-section-number">17.4.7.1</span> Hypothesis Tested</h4>
<p>Let’s test the hypothesis that the proportion of positive T cells in the two samples differ. The choice is not to test whether one proportion is greater than the other. We just want to know if they differ.</p>
<p>The statistical hypotheses here differs from the one sample hypotheses in two ways.</p>
<p>First, notice how we’re comparing the population proportion of cytokine- to that for vehicle-treatment.</p>
<p>Second, we’re making this a two-tailed (<code>two.sided</code>) test instead of one-tailed (<code>greater</code>).</p>
<p>The statistical alternate hypothesis: <span class="math inline">\(\pi_c\ne\pi_v\)</span></p>
<p>The statistical null hypothesis: <span class="math inline">\(\pi_c=\pi_v\)</span></p>
<p>A second way of writing these hypotheses to say the same thing:</p>
<p>Alternate: <span class="math inline">\(\pi_c-\pi_v\ne0\)</span></p>
<p>Null: <span class="math inline">\(\pi_c-\pi_v=0\)</span></p>
</div>
<div id="running-the-test" class="section level4">
<h4><span class="header-section-number">17.4.7.2</span> Running the test</h4>
<p>Let’s say that here are the outcome data:</p>
<p>The data can be passed directly into <code>prop.test</code>.</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb236-1" title="1"><span class="co"># note x = successes, n = successes + failures</span></a>
<a class="sourceLine" id="cb236-2" title="2"></a>
<a class="sourceLine" id="cb236-3" title="3"><span class="kw">prop.test</span>(</a>
<a class="sourceLine" id="cb236-4" title="4">  <span class="dt">x=</span><span class="kw">c</span>(<span class="dv">567</span>, <span class="dv">412</span>),</a>
<a class="sourceLine" id="cb236-5" title="5">  <span class="dt">n=</span><span class="kw">c</span>(<span class="dv">1778</span>, <span class="dv">1897</span>),</a>
<a class="sourceLine" id="cb236-6" title="6">  <span class="dt">conf.level=</span><span class="fl">0.99</span></a>
<a class="sourceLine" id="cb236-7" title="7">  )</a></code></pre></div>
<pre><code>## 
##  2-sample test for equality of proportions with continuity correction
## 
## data:  c(567, 412) out of c(1778, 1897)
## X-squared = 48.066, df = 1, p-value = 4.121e-12
## alternative hypothesis: two.sided
## 99 percent confidence interval:
##  0.06368227 0.13974295
## sample estimates:
##    prop 1    prop 2 
## 0.3188976 0.2171850</code></pre>
</div>
<div id="interpretation-of-two-sample-proportions-test-output" class="section level4">
<h4><span class="header-section-number">17.4.7.3</span> Interpretation of Two-Sample proportions test output</h4>
<p>This test is evaluated under the assumption that the null hypothesis is true. The test results helps us decide whether the null hypothesis should be rejected. We’ll do that if the p-value is less than our type1 error threshold.</p>
<p>The <code>prop.test</code> conducts a chi-square analysis using the Yates continuity correction. This <span class="math inline">\(\chi^2\)</span> value is very large; extremely large. You can learn more about the chisquare test statistic in Chapter @(chisquare).</p>
<p>The p-value represents the probability of obtaining a <span class="math inline">\(\chi^2\)</span> value as larger or larger than that in the null distribution. If the null hypothesis is true in this case, the probability of that sized <span class="math inline">\(\chi^2\)</span> value is 4.12e-12, which is very low, and still much lower than our type1 error threshold.</p>
<p>We can reject the null.</p>
<p>If we subtract prop2 from prop1 we get a value of about 0.1017, as a point estimate for the difference between the proportion of gradstudin=positive cells in the vehicle and cytokine treated samples. The 95% CI represents a range of values that might include the difference between the two proportions.</p>
<p>We are 95% confident that the true difference between these proportions is from 0.0638 to 01397.</p>
<p>The 95% CI for the difference between two proportions does not include the value of 0. Since the 95% CI does not overlap with 0, we can conclude from it alone that there is a statistical difference between the two proportions.</p>
<p>We have to use scientific judgment to assess whether the lower and upper ranges of the CI represent values that are scientifically meaningful or relevant. It is possible for a CI to include values that are scientifically unremarkable, even if the value of zero is not included.</p>
</div>
<div id="write-up" class="section level4">
<h4><span class="header-section-number">17.4.7.4</span> Write Up</h4>
<p><em>The proportion of gradstudin positive T cells after cytokine differs from that in vehicle treated cells (two-sided two-sample proportions test, p-value=4.12e-12, 95% CI for proportion difference = 0.0638 to 01397)</em></p>
<p>Note how we don’t say “statistically significantly different”.</p>
</div>
</div>
</div>
<div id="exact-tests-for-two-proportions" class="section level2">
<h2><span class="header-section-number">17.5</span> Exact tests for two proportions</h2>
<p>An alternative to <code>prop.test</code> to <strong>compare two proportions</strong> is the <code>fisher.test</code>, which like the <code>binom.test</code> calculates exact p-values.</p>
<p>The <code>fisher.test</code> requires that data be input as a matrix or table of the successes and failures, so that involves a bit more munging.</p>
<p>In R a matrix differs from a dataframe in many ways. The cells within the matrix must all be of the same data type. In this case, numerical values. The matrix can be labeled with dimension names. Finally, it helps to understand that the matrix world operates in a row by column way. Thus, the vector of 4 values below fills the rows before filling the columns.</p>
<p>We need to make a matrix of the data first, then perform the <code>fisher.test</code> on that matrix:</p>
<p>Notice below how to enter the successes and the failures in the matrix. The matrix is also labeled with some dimension (row and column) names so it doesn’t get confusing.</p>
<p>The fisher test is run by passing the matrix into the <code>fisher.test</code> function:</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb238-1" title="1">M &lt;-<span class="st"> </span><span class="kw">matrix</span>(</a>
<a class="sourceLine" id="cb238-2" title="2">  <span class="kw">c</span>(<span class="dv">567</span>, <span class="dv">412</span>, <span class="dv">1211</span>, <span class="dv">1485</span>), </a>
<a class="sourceLine" id="cb238-3" title="3">  <span class="dt">nrow=</span><span class="dv">2</span>, </a>
<a class="sourceLine" id="cb238-4" title="4">  <span class="dt">dimnames =</span> <span class="kw">list</span>(</a>
<a class="sourceLine" id="cb238-5" title="5">    <span class="dt">Treat=</span><span class="kw">c</span>(<span class="st">&quot;Cytokine&quot;</span>, <span class="st">&quot;Vehicle&quot;</span>), </a>
<a class="sourceLine" id="cb238-6" title="6">    <span class="dt">Antigen=</span><span class="kw">c</span>(<span class="st">&quot;Positive&quot;</span>, <span class="st">&quot;Negative&quot;</span>)</a>
<a class="sourceLine" id="cb238-7" title="7">    )</a>
<a class="sourceLine" id="cb238-8" title="8">  )</a>
<a class="sourceLine" id="cb238-9" title="9">M</a></code></pre></div>
<pre><code>##           Antigen
## Treat      Positive Negative
##   Cytokine      567     1211
##   Vehicle       412     1485</code></pre>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb240-1" title="1"><span class="kw">fisher.test</span>(M)</a></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  M
## p-value = 3.433e-12
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.451760 1.962188
## sample estimates:
## odds ratio 
##   1.687312</code></pre>
<p>Notice the output differs from the prop.test. In addition to a p-value, the Fisher test produces an odds ratio and its confidence interval. The p-value leads to the same result and write-up as for the two proportion test.</p>
<p>As for the odds ratio…</p>
<p>Recall that the odds of an event that has a probability <span class="math inline">\(p\)</span> are <span class="math display">\[odds = \frac{p}{1-p}\]</span></p>
<p>Thus, the odds of gradstudin-positive cells are <span class="math display">\[odds = \frac{positive}{negative}\]</span></p>
<p>An odds ratio is the ratio of two odds. Here, it is the ratio of the odds of gradstudin-positive cells in response to cytokine to the odds of gradstudin-positive cells after vehicle.<span class="math display">\[OR = \frac{odds_{cyto}}{odds_{veh}}\]</span></p>
<p>It is probably best to leave the the OR as is. Report the ratio <em>per se</em> as the effect size as in the write up below.</p>
<p>However, one way to think of the OR is as a percent by multiplying the OR by 100. Thus, the odds of gradstudin positive cells in the cytokine group are 168% of their odds in the vehicle group.</p>
<div id="write-up-1" class="section level4">
<h4><span class="header-section-number">17.5.0.1</span> Write up</h4>
<p><em>The cytokine increases the odds of gradstudin-positive T cells by 1.687 compared to vehicle treatment (Fisher’s Exact Test for Count Data, p = 3.433e-12, OR 95% CI = 1.45 to 1.96)</em>.</p>
</div>
</div>
<div id="goodness-of-fit-tests" class="section level2">
<h2><span class="header-section-number">17.6</span> Goodness of fit Tests</h2>
<p>Goodness-of-fit tests are useful for testing hypotheses about patterns of counts in time or space. The tests questions whether the distribution of their observed frequencies differs from expectations of a null case. In other words, do the count patterns occur in a non-random pattern?</p>
<p>The shape of these datasets in their simplest form is either as 1 row or 1 column, where every data cell represents a time or space or nominal feature and the cell value is the the number of counts that occurred in that time or space or have that nominal feature.</p>
<p>Either the <code>multinomial.test</code> (for exact p-values) or the <code>chisq.test</code> (for approximate p-values) can be used for Goodness-of-fit testing. The latter is most commonly used, but may be less accurate when the number of counts in cells is low.</p>
<p>These designs compare the distribution of events to a hypothetical (or expected) model null distribution of those events.</p>
<p>These expected counts are entered in the test script as a <code>prop</code> or <code>p</code> argument.</p>
<p><em>This can be confusing. It’s important to recognize you should enter a vector of null probabilities in p or prop! Don’t enter the counts you hope to see if the test were positive!!! You always test the null.</em></p>
<p>Say we had a spatial memory test in which 28 independent subjects (rats) are placed into a maze for testing (one at a time) and we count which of 4 chambers they enter first. The results indicate they do so at the following frequencies: A=14, B=3, C=7, D=4.</p>
<p>Does this frequency distribution differ from the null expectation, A=7, B=7, C=7, D=7, where no chamber is more likely to be entered than another?</p>
<p>Failure is not an option in this design! Only successes are counted. Given enough time, a subject will always choose one of the 4 chambers. If a subject fails the task and never enters a chamber, it should be censured. If failure to enter a chamber is a fairly common outcome, then perhaps it should be the fifth nominal option.</p>
<p>Let’s test this at the 5% type1 error threshold:</p>
<div id="an-exact-goodness-of-fit-test" class="section level3">
<h3><span class="header-section-number">17.6.1</span> An Exact Goodness of Fit test</h3>
<p>NB:The <code>multinomial.test</code> function form the <code>EMT</code> package requires you to assert the null frequency distribution explicitly and as fractions whose sum is 1. This is unlike using the <code>chisq.test</code>, which coerces expected values for you unless you argue otherwise. Also note that in this example the probabilities are uniform. That need not be the case. The only requirement is that the sum of the probabilities equals one. The multinomial test shines brightest when you expect varied probs.</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb242-1" title="1">x &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A=</span><span class="dv">14</span>, <span class="dt">B=</span><span class="dv">3</span>, <span class="dt">C=</span><span class="dv">7</span>, <span class="dt">D=</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb242-2" title="2">prob &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A=</span>.<span class="dv">25</span>, <span class="dt">B=</span>.<span class="dv">25</span>, <span class="dt">C=</span>.<span class="dv">25</span>, <span class="dt">D=</span>.<span class="dv">25</span>)</a>
<a class="sourceLine" id="cb242-3" title="3"></a>
<a class="sourceLine" id="cb242-4" title="4"><span class="kw">multinomial.test</span>(</a>
<a class="sourceLine" id="cb242-5" title="5">  x, </a>
<a class="sourceLine" id="cb242-6" title="6">  <span class="dt">prob=</span>prob</a>
<a class="sourceLine" id="cb242-7" title="7">  )</a></code></pre></div>
<pre><code>## 
##  Exact Multinomial Test, distance measure: p
## 
##     Events    pObs    p.value
##       4495   1e-04     0.0235</code></pre>
<p>Note on the <code>multinomial.test</code>output: <a href="http://rinterested.github.io/statistics/multinomial_exact.html">Please see this site for further reading</a> for further information on what is represented by events. In this case the value for Events is <code>choose(28+3-1,4-1)</code>. pObs is a (rounded) multinomial probability for observing that exact permutation (<code>dmultinom(x=c(14,3,7,4), prob = c(0.25,0.25,0.25,0.25))</code>), which serves as a critical value. The p.value is the sum of all pObs for other permutations that are lower (more extreme) than this pObs.</p>
<p>We’re only interested in the p-value, since we’re using this function as an exact goodness-of-fit hypothesis test for the null hypothesis.</p>
<p>Why use an exact test rather than a <code>chisq.test</code>? Because we have two cells in the dataset with counts &lt; 5. An exact p-value test will be more accurate than <code>chisq.test</code>.</p>
<p>The test compares our sample frequency distribution to that in the null model. We actually wrote the latter explicitly in the function argument: null is a uniform distribution–the subjects are equally likely to enter each chamber.</p>
<p>H0: The probability of choice is equal for each chamber.
<span class="math inline">\(\pi_A=\pi_B=\pi_C=\pi_D\)</span>
H1: The probability of choice is not equal for each chamber.
<span class="math inline">\(\pi_A\ne\pi_B\ne\pi_C\ne\pi_D\)</span></p>
<p>Note: this is an omnibus test. It doesn’t explicitly tell us which chambers are preferred by the subjects.</p>
<p>Because the observed p-value is less than a pre-chosen threshold of 0.05, we reject the null hypothesis and conclude that the chamber choice is not equitable across the four options.</p>
<div id="write-up-2" class="section level4">
<h4><span class="header-section-number">17.6.1.1</span> Write up</h4>
<p><em>In a 4 chamber maze test, the subjects displayed a non-uniform chamber preference (Exact Multinomial Test, n=28, p=0.0235)</em></p>
<p>Note how this phrasing implies the null hypothesis.</p>
</div>
</div>
<div id="an-approximate-goodness-of-fit-test" class="section level3">
<h3><span class="header-section-number">17.6.2</span> An Approximate Goodness of Fit test</h3>
<p>The <span class="math inline">\(\chi^2\)</span> test of the same data is really simple to execute. We pass into the function the same vector of observed counts, <code>x</code>, that we configured above.</p>
<p>It offers the same conclusion, but note how the p-value is very different.</p>
<p>Note also that we didn’t enter the null frequency argument. The <code>chisq.test</code>function will coerce the null distribution if it is not entered as an argument explicitly, as you can see from the output for the second line. If for some reason to test against a non-uniform null distribution, you’ll need to write that in your argument explicitly (eg, p = c(A=0.5, B=0.25, C &amp; D = 0.125).</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb244-1" title="1"><span class="kw">chisq.test</span>(x)</a></code></pre></div>
<pre><code>## 
##  Chi-squared test for given probabilities
## 
## data:  x
## X-squared = 10.571, df = 3, p-value = 0.01428</code></pre>
<div id="write-up-3" class="section level4">
<h4><span class="header-section-number">17.6.2.1</span> Write up</h4>
<p>The interpretation is no different than for the exact test. The write up is:</p>
<p><em>In a 4 chamber maze test, the subjects displayed a nonuniform chamber preference (Chi square test for uniform probabilities, <span class="math inline">\(\chi^2\)</span>=10.571, df=3, p=0.01428)</em></p>
</div>
<div id="post-hoc-which-chamber-is-the-preferred" class="section level4">
<h4><span class="header-section-number">17.6.2.2</span> Post hoc: which chamber is the preferred?</h4>
<p>The result of the multinomial test can be unsatisfying in the sense that it only indicates the observed frequency pattern differs from that expected by chance. It doesn’t say which chamber preference explains the overall outcome.</p>
<p>This can be answered using a series of binomial tests. These would address whether a given choice differs from all the others.</p>
<p>The key here is that multiple comparisons will be made. Therefore an adjustment must be made to the type1 error threshold. Using a Bonferroni correction, if the experiment-wise type1 threshold is 5%, and 4 comparisons are to be made, then the threshold for any one comparison would be 5%/5, or 1.25%.</p>
<p>In later chapters (on ANOVA) we will talk about this adjustment in more detail.</p>
<p>Of the following four tests, you see that only the frequency of the A chamber differs from the null expectation. The test p-value of 0.004099 is lower than that for the reset type1 threshold of 0.0125.</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb246-1" title="1"><span class="co"># A chamber</span></a>
<a class="sourceLine" id="cb246-2" title="2"><span class="kw">binom.test</span>(<span class="dt">x=</span><span class="dv">14</span>, <span class="dt">n=</span><span class="dv">28</span>, <span class="dt">p=</span><span class="fl">0.25</span>)</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  14 and 28
## number of successes = 14, number of trials = 28, p-value = 0.004099
## alternative hypothesis: true probability of success is not equal to 0.25
## 95 percent confidence interval:
##  0.306471 0.693529
## sample estimates:
## probability of success 
##                    0.5</code></pre>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb248-1" title="1"><span class="co"># B chamber</span></a>
<a class="sourceLine" id="cb248-2" title="2"><span class="kw">binom.test</span>(<span class="dv">3</span>, <span class="dv">28</span>, <span class="dt">p=</span><span class="fl">0.25</span>)</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  3 and 28
## number of successes = 3, number of trials = 28, p-value = 0.123
## alternative hypothesis: true probability of success is not equal to 0.25
## 95 percent confidence interval:
##  0.02266509 0.28226440
## sample estimates:
## probability of success 
##              0.1071429</code></pre>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb250-1" title="1"><span class="co"># C chamber</span></a>
<a class="sourceLine" id="cb250-2" title="2"><span class="kw">binom.test</span>(<span class="dv">7</span>, <span class="dv">28</span>, <span class="dt">p=</span><span class="fl">0.25</span>)</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  7 and 28
## number of successes = 7, number of trials = 28, p-value = 1
## alternative hypothesis: true probability of success is not equal to 0.25
## 95 percent confidence interval:
##  0.1069080 0.4487155
## sample estimates:
## probability of success 
##                   0.25</code></pre>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb252-1" title="1"><span class="co"># D chamber</span></a>
<a class="sourceLine" id="cb252-2" title="2"><span class="kw">binom.test</span>(<span class="dv">4</span>, <span class="dv">28</span>, <span class="dt">p=</span><span class="fl">0.25</span>)</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  4 and 28
## number of successes = 4, number of trials = 28, p-value = 0.2738
## alternative hypothesis: true probability of success is not equal to 0.25
## 95 percent confidence interval:
##  0.04033563 0.32665267
## sample estimates:
## probability of success 
##              0.1428571</code></pre>
<p>Another way to report p-values from multiple comparisons is to state them as adjusted.</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb254-1" title="1"><span class="kw">p.adjust</span>(<span class="dt">p=</span><span class="fl">0.004099</span>, <span class="dt">method=</span><span class="st">&quot;bonferroni&quot;</span>, <span class="dt">n=</span><span class="dv">4</span>)</a></code></pre></div>
<pre><code>## [1] 0.016396</code></pre>
<p>Thus, a p-value of 0.004099 when adjusted for 4 multiple comparisons is equivalent to a p-value of 0.016396. This value yet remains below the pre-chosen threshold of 0.05.</p>
</div>
<div id="write-up-including-post-hoc-test" class="section level4">
<h4><span class="header-section-number">17.6.2.3</span> Write-up including post hoc test</h4>
<p><em>In a 4 chamber maze test, the subjects display a clear chamber preference (Chi square test for uniform probabilities, <span class="math inline">\(\chi^2\)</span>=10.571, df=3, p=0.01428). Multiple post hoc tests indicate only the chamber A frequencies differ from the null (Exact binomial test, adjusted p-value= 0.016396).</em></p>
</div>
</div>
</div>
<div id="contingency-testing" class="section level2">
<h2><span class="header-section-number">17.7</span> Contingency Testing</h2>
<p>Contingency testing is for deciding whether two or more variables are associated or not. These either explicitly (ie, when using <code>fisher.text</code>) or implicitly (ie, when using <code>chisq.test</code>) use ratio’s of proportions–the odds ratio, or relative risk, or the likelihood ratio, or sometimes other proportions–as parameters that express the magnitude of these associations.</p>
<p>In other words, the hypothesis test asks whether these ratio’s of proportions are more extreme than the null (which would be 1).</p>
<p>Let’s take the cancer marker data from the contingency analysis lecture. As you recall, a biomarker has been discovered that is hoped to be strongly associated with cancer. 100 people were tested for whether or not they have the marker, and whether or not they also go on to have cancer.</p>
<p>We’ll create a simple matrix then pass it through the <code>fisher.test</code> function to illustrate the procedure and interpretation.</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb256-1" title="1">x &lt;-<span class="st"> </span><span class="kw">matrix</span>(</a>
<a class="sourceLine" id="cb256-2" title="2">  <span class="kw">c</span>(<span class="dv">14</span>, <span class="dv">16</span>, <span class="dv">6</span>, <span class="dv">64</span>), </a>
<a class="sourceLine" id="cb256-3" title="3">  <span class="dt">ncol=</span><span class="dv">2</span>, </a>
<a class="sourceLine" id="cb256-4" title="4">  <span class="dt">dimnames =</span> <span class="kw">list</span>(</a>
<a class="sourceLine" id="cb256-5" title="5">    <span class="dt">Marker=</span><span class="kw">c</span>(<span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Absent&quot;</span>), </a>
<a class="sourceLine" id="cb256-6" title="6">    <span class="dt">Cancer =</span> <span class="kw">c</span>(<span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Absent&quot;</span>)</a>
<a class="sourceLine" id="cb256-7" title="7">    )</a>
<a class="sourceLine" id="cb256-8" title="8">  )</a>
<a class="sourceLine" id="cb256-9" title="9">x</a></code></pre></div>
<pre><code>##          Cancer
## Marker    Present Absent
##   Present      14      6
##   Absent       16     64</code></pre>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb258-1" title="1"><span class="kw">fisher.test</span>(x) </a></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  x
## p-value = 3.934e-05
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   2.762514 33.678765
## sample estimates:
## odds ratio 
##   9.061278</code></pre>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb260-1" title="1"><span class="co">#more argument customization than this is possible </span></a></code></pre></div>
<div id="interpretation-of-contingency-results" class="section level3">
<h3><span class="header-section-number">17.7.1</span> Interpretation of Contingency Results</h3>
<p>The odds of a person with the marker having cancer are 9.06 times greater than the cancer odds for those who don’t have the marker.</p>
<p>There is 95% confidence the true odds ratio in the population is between 2.76 and 33.68.</p>
<p>This is consistent with an association between the presence of this marker and the probability that cancer occurs.</p>
</div>
<div id="write-up-4" class="section level3">
<h3><span class="header-section-number">17.7.2</span> Write Up</h3>
<p><em>The large OR indicates the presence of this marker is strongly associated with cancer (n=100, OR = 9.06, 95% CI = 2.76 to 33.68, Fisher’s Exact Test for Count Data, p = 3.934e-05).</em></p>
<p>The word “strongly” is used to emphasize the effect size, which is OR, not the smallness of the p-value.</p>
<p>Here are the other tests you might use to conduct for a contingency analysis, to illustrate how they differ. First, a proportion test and a sort of proof that the prop test is just a chisquare test:</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb261-1" title="1">x &lt;-<span class="st"> </span><span class="kw">matrix</span>(</a>
<a class="sourceLine" id="cb261-2" title="2">  <span class="kw">c</span>(<span class="dv">14</span>, <span class="dv">16</span>, <span class="dv">6</span>, <span class="dv">64</span>), </a>
<a class="sourceLine" id="cb261-3" title="3">  <span class="dt">ncol=</span><span class="dv">2</span>, </a>
<a class="sourceLine" id="cb261-4" title="4">  <span class="dt">dimnames =</span> <span class="kw">list</span>(</a>
<a class="sourceLine" id="cb261-5" title="5">    <span class="dt">Marker=</span><span class="kw">c</span>(<span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Absent&quot;</span>), </a>
<a class="sourceLine" id="cb261-6" title="6">    <span class="dt">Cancer =</span> <span class="kw">c</span>(<span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Absent&quot;</span>)</a>
<a class="sourceLine" id="cb261-7" title="7">    )</a>
<a class="sourceLine" id="cb261-8" title="8">  )</a>
<a class="sourceLine" id="cb261-9" title="9">x</a></code></pre></div>
<pre><code>##          Cancer
## Marker    Present Absent
##   Present      14      6
##   Absent       16     64</code></pre>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb263-1" title="1"><span class="kw">prop.test</span>(x)</a></code></pre></div>
<pre><code>## 
##  2-sample test for equality of proportions with continuity correction
## 
## data:  x
## X-squared = 16.741, df = 1, p-value = 4.284e-05
## alternative hypothesis: two.sided
## 95 percent confidence interval:
##  0.2496194 0.7503806
## sample estimates:
## prop 1 prop 2 
##    0.7    0.2</code></pre>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb265-1" title="1"><span class="kw">chisq.test</span>(x)</a></code></pre></div>
<pre><code>## 
##  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
## 
## data:  x
## X-squared = 16.741, df = 1, p-value = 4.284e-05</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb267-1" title="1"><span class="kw">chisq.test</span>(x, <span class="dt">correct=</span>F)</a></code></pre></div>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  x
## X-squared = 19.048, df = 1, p-value = 1.275e-05</code></pre>
<p>First, note that the prop.test is just the chisq.test. You get the same <span class="math inline">\(\chi^2\)</span> value and p-value from both. They just differ in parameter output and input options. If you’d like a confidence interval use <code>prop.test</code>.</p>
<p>Second, note how turning off the Yates continuity correction changes the <span class="math inline">\(\chi^2\)</span> value and the p-value. That’s to be expected, it changes the calculation! Both the prop.test and chisq.test use Yates by default. The best way to think about Yate’s is that it acts as a smoothing function to take off some of the jigger in the calculation of the <span class="math inline">\(\chi^2\)</span> value.</p>
</div>
<div id="interpretation-of-chi-square-output" class="section level3">
<h3><span class="header-section-number">17.7.3</span> Interpretation of chi-square output</h3>
<p>There is an association between the presence of this marker and the probability that cancer occurs.</p>
<p>We <em>could</em> take the prop test’s calculation of the proportions and their difference, along with the 95% CI of their difference and make some hay out of that (the probability of getting cancer with the marker is 70%, and without the marker is 20%). But it’s more customary to use the odds ratio or relative risk rather than differences between probabilities to make effect size assertions in association studies.</p>
</div>
<div id="write-up-5" class="section level3">
<h3><span class="header-section-number">17.7.4</span> Write Up</h3>
<p>You would want to derive the odds ratio and its 95% CI, even though the <span class="math inline">\(\chi^2\)</span> test doesn’t produce it for you. The easiest way to do that is with <code>fisher.test</code>.</p>
<p>Having that:</p>
<p><em>The large OR indicates the presence of this marker is strongly associated with cancer (n=100, OR = 9.06, 95% CI = 2.76 to 33.68, Pearson’s Chi-square test with Yate’s continuity correction, p = 4.284e-05).</em></p>
<p>As before, the word “strongly” is used to emphasize the effect size, which is the OR, rather than the extremeness of the p-value.</p>
</div>
<div id="which-contingency-test-is-best" class="section level3">
<h3><span class="header-section-number">17.7.5</span> Which contingency test is best?</h3>
<p>With so many options, the question I’m sure you are asking is which is best to use for contingency analysis?</p>
<p>The answer is,
* be sure to make this decision in advance and use what you said you would use before you started the experiment.
* when it comes to p-values, are you an exactophile or an exactophobe? Use exact tests if the former.
* for datasets with low cell numbers (eg, counts less than 5 in a cell), exact tests tend to provide more accurate p-values.
* the fact that <code>fisher.test</code> generates the OR and its CI is very, very convenient.</p>
<p>I prefer the fisher.exact test. However, in R you’ll need to understand how to configure its arguments to get it to work on higher dimension contingency tables (eg, 2x3, 3x2, 3x3, etc).</p>
</div>
<div id="higher-dimension-contingency-analysis" class="section level3">
<h3><span class="header-section-number">17.7.6</span> Higher dimension contingency analysis</h3>
<p>Not infrequently we have studies with more than one level of a predictor variable and with two or more categorical outcomes (eg, 7x2, 2x3, 5x4, etc). In these instances we are interested in comparing a few to several combinations of outcomes.</p>
<p>A step-wise approach for these more complex analyses is to first run an omnibus <code>chisq.test</code> or <code>fisher.test</code> on the fully intact dimension. If the test is positive (the p-value is lower than a pre-chosen threshold), analyze 2x2 segments of the grid post hoc. That is accomplished by using one of the exact or the <code>chisq.test</code> on more discrete groupings. These tests can explain which of the proportions or associations explain the overall skew detected by the initial omnibus test. These post hoc analyses must include correction for multiple comparisons (eg, the Bonferroni correction) when drawing inference. To do this, collect a vector of p-values, on from each post hoc test, and pass it into R’s <code>p.adjust</code> function.</p>
<p>Here’s an example of a higher dimension array:</p>
<p>Gammaherpesviruses, such as EBV and KSHV, establish lifelong infection in B cells. At the peak of infection the cells have either germinal center B cell or plasma cell phenotypes. Very little is known about the immunoglobulins expressed in gammaherpesvirus infected cells. In a study using a mouse model of viral exposure to YFP-tagged gammherpesvirus, immunoglobulin isotypes (IG) were determined in randomly collected single cell clones that include both germinal center B cells and in plasma cells that were either infected or not (YFP+ or YFP-).</p>
<p>As a first step, the data are constructed as a matrix.</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb269-1" title="1">counts &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">10</span>,<span class="dv">9</span>,<span class="dv">15</span>,<span class="dv">29</span>,<span class="dv">17</span>,<span class="dv">11</span>,<span class="dv">40</span>,<span class="dv">50</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">13</span>,<span class="dv">25</span>,<span class="dv">59</span>,<span class="dv">61</span>,<span class="dv">43</span>),</a>
<a class="sourceLine" id="cb269-2" title="2">                 <span class="dt">nrow=</span><span class="dv">4</span>, <span class="dt">ncol=</span><span class="dv">5</span>,</a>
<a class="sourceLine" id="cb269-3" title="3"><span class="dt">dimnames=</span><span class="kw">list</span>(<span class="dt">Cell=</span><span class="kw">c</span>(<span class="st">&quot;YFP- GC B&quot;</span>, <span class="st">&quot;YFP+ GC B&quot;</span>, <span class="st">&quot;YFP- plasma&quot;</span>, <span class="st">&quot;YFP+ plasma&quot;</span>), <span class="dt">IG =</span><span class="kw">c</span>(<span class="st">&quot;G1&quot;</span>, <span class="st">&quot;G2b&quot;</span>, <span class="st">&quot;G2c&quot;</span>, <span class="st">&quot;G3&quot;</span>, <span class="st">&quot;M&quot;</span>)))</a>
<a class="sourceLine" id="cb269-4" title="4"></a>
<a class="sourceLine" id="cb269-5" title="5">counts</a></code></pre></div>
<pre><code>##              IG
## Cell          G1 G2b G2c G3  M
##   YFP- GC B    3  10  17  3 25
##   YFP+ GC B    0   9  11  2 59
##   YFP- plasma  0  15  40  2 61
##   YFP+ plasma  0  29  50 13 43</code></pre>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb271-1" title="1"><span class="kw">paste</span>(<span class="st">&quot;Total number of unique cells isotyped =&quot;</span>, <span class="kw">sum</span>(counts))</a></code></pre></div>
<pre><code>## [1] &quot;Total number of unique cells isotyped = 392&quot;</code></pre>
<p>Next, an omnibus Fisher test run at a type1 threshold of 5%. The combination calculations necessary to solve for an exact p-value in these higher order matrices won’t always converge to a solution. The <code>simulate.p.value</code> argument is often necessary in these cases. The p-value is not exact.</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb273-1" title="1"><span class="kw">fisher.test</span>(counts, <span class="dt">simulate.p.value =</span> T)</a></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data with simulated p-value (based on
##  2000 replicates)
## 
## data:  counts
## p-value = 0.0004998
## alternative hypothesis: two.sided</code></pre>
<div id="post-hoc-testing-of-higher-dimensions" class="section level4">
<h4><span class="header-section-number">17.7.6.1</span> Post hoc testing of higher dimensions</h4>
<p>A 4x5 matrix contains 20 groups, <span class="math inline">\(k\)</span>. The total possible comparisons is <span class="math inline">\(c=\frac{k(k-1)}{2}\)</span>, or 190 in this case. Many of these would be uninteresting.</p>
<p>Let scientific reasoning guide the posthoc testing. Thinking about the experiment two sets of comparisons seem good. The first set ask whether two rows differ in isotype frequencies. A second, and larger, set asks whether two rows differ in frequency for a specific isotype. In one sense, the first of these sets can serve as a sub-omnibus to guide the comparisons made in the second set.</p>
<p>Here are two examples for what the first set of comparisons might look like.</p>
<ul>
<li>Does viral infection affect immunoglobin isotype frequencies in germinal center cells?</li>
</ul>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb275-1" title="1"><span class="kw">fisher.test</span>(counts[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])</a></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  counts[1:2, 1:5]
## p-value = 0.002753
## alternative hypothesis: two.sided</code></pre>
<ul>
<li>Do isotype frequencies differ between virally-infected germinal center and plasma cells?</li>
</ul>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb277-1" title="1"><span class="kw">fisher.test</span>(counts[<span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>), <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])</a></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  counts[c(2, 4), 1:5]
## p-value = 9.09e-08
## alternative hypothesis: two.sided</code></pre>
<p>A few other questions could be asked.</p>
<p>Here’s an example of what the second set of questions would look like which focus on the frequencies of specific isotypes.</p>
<p>Although the p-value is unadjusted, the exact test result to the second question above suggests isotype frequencies do differ between virally-infected germinal center and plasma cells.</p>
<p>To focus on the isotype(s) that might explain this difference we would compare a series of two proportions, These would compare the counts for a given isotype to the sum of the counts for all other isotypes. Although the matrix-making code below looks a bit chunky, it clearly illustrates what ‘Other’ counts represent:</p>
<ul>
<li>Does viral infection cause a difference in M isotype expression between germinal center and plasma cells?</li>
</ul>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb279-1" title="1">m15 &lt;-<span class="st"> </span><span class="kw">matrix</span>(</a>
<a class="sourceLine" id="cb279-2" title="2">  <span class="kw">c</span>(counts[<span class="dv">2</span>,<span class="dv">5</span>],</a>
<a class="sourceLine" id="cb279-3" title="3">    counts[<span class="dv">4</span>,<span class="dv">5</span>],</a>
<a class="sourceLine" id="cb279-4" title="4">    <span class="kw">sum</span>(counts[<span class="dv">2</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])<span class="op">-</span>counts[<span class="dv">2</span>,<span class="dv">5</span>],</a>
<a class="sourceLine" id="cb279-5" title="5">    <span class="kw">sum</span>(counts[<span class="dv">4</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])<span class="op">-</span>counts[<span class="dv">4</span>,<span class="dv">5</span>]),</a>
<a class="sourceLine" id="cb279-6" title="6">  <span class="dt">nrow=</span><span class="dv">2</span>,<span class="dt">ncol=</span><span class="dv">2</span>,</a>
<a class="sourceLine" id="cb279-7" title="7">  <span class="dt">dimnames=</span><span class="kw">list</span>(<span class="dt">YFp=</span><span class="kw">c</span>(<span class="st">&quot;+ GC B&quot;</span>, <span class="st">&quot;+ plasma&quot;</span>), </a>
<a class="sourceLine" id="cb279-8" title="8">                <span class="dt">serotype=</span><span class="kw">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;Other&quot;</span>)</a>
<a class="sourceLine" id="cb279-9" title="9">                ))</a>
<a class="sourceLine" id="cb279-10" title="10">ftm15 &lt;-<span class="st"> </span><span class="kw">fisher.test</span>(m15)</a>
<a class="sourceLine" id="cb279-11" title="11">ftm15</a></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  m15
## p-value = 5.056e-09
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##   2.998242 11.094080
## sample estimates:
## odds ratio 
##   5.686205</code></pre>
<p>Although this p-value is also unadjusted, it certainly appears that gammaherpesvirus infection causes a difference in M isotype expression between germinal center and plasma cells.</p>
</div>
</div>
<div id="other-experimental-designs-involving-categorical-data" class="section level3">
<h3><span class="header-section-number">17.7.7</span> Other experimental designs involving categorical data</h3>
<p>Imagine an experiment to compare two or more conditions (eg, placebo v drug, wildtype vs mutant) and the outcome variable is discrete (eg, frequency counts or success and failure counts). The experiment involves several independent replications.</p>
<p>For example, cell depolarizations are counted both in the absence or presence of a condition on several different occasions.</p>
<table>
<caption>
<span id="tab:unnamed-chunk-163">Table 17.1: </span>Hypothetical replicates comparing depolarizations in mutant and wildtype cells should be analyzed using Poisson regression.
</caption>
<thead>
<tr>
<th style="text-align:left;">
replicate
</th>
<th style="text-align:left;">
predictor
</th>
<th style="text-align:right;">
counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
one
</td>
<td style="text-align:left;">
wildtype
</td>
<td style="text-align:right;">
110
</td>
</tr>
<tr>
<td style="text-align:left;">
two
</td>
<td style="text-align:left;">
wildtype
</td>
<td style="text-align:right;">
99
</td>
</tr>
<tr>
<td style="text-align:left;">
three
</td>
<td style="text-align:left;">
wildtype
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
one
</td>
<td style="text-align:left;">
mutant
</td>
<td style="text-align:right;">
165
</td>
</tr>
<tr>
<td style="text-align:left;">
two
</td>
<td style="text-align:left;">
mutant
</td>
<td style="text-align:right;">
138
</td>
</tr>
<tr>
<td style="text-align:left;">
three
</td>
<td style="text-align:left;">
mutant
</td>
<td style="text-align:right;">
139
</td>
</tr>
</tbody>
</table>
<p>Alternately, the fraction of cells in a culture dish that have died in the absence or presence of a condition is repeated a handful of times.</p>
<table>
<caption>
<span id="tab:unnamed-chunk-164">Table 17.2: </span>Hypothetical replicates comparing in mutant and wildtype cells should be analyzed using logistic regression.
</caption>
<thead>
<tr>
<th style="text-align:left;">
replicate
</th>
<th style="text-align:left;">
predictor
</th>
<th style="text-align:right;">
alive
</th>
<th style="text-align:right;">
dead
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
one
</td>
<td style="text-align:left;">
wildtype
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
71
</td>
</tr>
<tr>
<td style="text-align:left;">
two
</td>
<td style="text-align:left;">
wildtype
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
73
</td>
</tr>
<tr>
<td style="text-align:left;">
three
</td>
<td style="text-align:left;">
wildtype
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
69
</td>
</tr>
<tr>
<td style="text-align:left;">
one
</td>
<td style="text-align:left;">
mutant
</td>
<td style="text-align:right;">
64
</td>
<td style="text-align:right;">
40
</td>
</tr>
<tr>
<td style="text-align:left;">
two
</td>
<td style="text-align:left;">
mutant
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:right;">
38
</td>
</tr>
<tr>
<td style="text-align:left;">
three
</td>
<td style="text-align:left;">
mutant
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
39
</td>
</tr>
</tbody>
</table>
<p>The key distinction here, compared to what’s been discussed in this chapter up to this point, is that within each replicate, all of the events are intrinsically-linked. Through replication we’re establishing whether the frequencies of these counts are repeatable.</p>
<p>Logistic (for dichotomous data) or Poisson (for frequency data) regression are the appropriate analytical tools for these designs. These involve using the generalized linear model, conducted with the function <code>glm</code> or the function <code>glmer</code> (for so-called mixed models). These are discussed later in the logistic regression chapter.</p>
</div>
</div>
<div id="power-analysis-for-proportion-tests" class="section level2">
<h2><span class="header-section-number">17.8</span> Power analysis for proportion tests</h2>
<p>Power analysis should be done before starting an experiment. The purpose of a conducting power calculations a priori is to determine the number of trials, or subjects or sample size, to use for the study.</p>
<p>This is a two step process.</p>
<p>Step 1: What effect size will you test? Using scientific judgement, decide what is the value of a null proportion and an alternate that you think would be a scientifically meaningful proportion to observe. You need to have some insight into the system you’re studying to make these decisions. Often that comes from pilot studies or from previously published (or unpublished) work. What’s important is to establish an expectation of what a minimally scientifically significant outcome would look like.</p>
<p>Step 2: Calculate the number of subjects (or trials) you’ll need to study, given these proportions (and also given some type1 and type2 error tolerances).</p>
<p>There are several options in R for the second step.</p>
<p>In the examples below, we’re declaring a 5% difference between the null (0.15) and alternate (0.20) proportions would be a scientifically meaningful. We’re also using 5% for type1 error and 20% for type2 error (80% power) as error tolerance thresholds.</p>
</div>
<div id="power-analysis-functions-for-proportion-tests" class="section level2">
<h2><span class="header-section-number">17.9</span> Power analysis functions for proportion tests</h2>
<p>The function <code>pwr.p.test</code> is for one-sample proportion tests. The calculations below return a sample size n that should be used in the study, given a null and an alternate proportions, in addition to error rates.</p>
<p>NB: Since <code>pwr.p.test</code> takes a Cohen’s effect size as an argument, you 1st must calculate a Cohen effect size, h, given the alternate and null proportions you expect. The function <code>Es.h()</code> in the <code>pwr</code> package is useful for this. Then plug that effect size, <code>h</code>, into the proportion test power calculator.</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb281-1" title="1">h &lt;-<span class="st"> </span><span class="kw">ES.h</span>(<span class="dt">p1=</span><span class="fl">0.2</span>, <span class="dt">p2=</span><span class="fl">0.15</span>)</a>
<a class="sourceLine" id="cb281-2" title="2">h</a></code></pre></div>
<pre><code>## [1] 0.1318964</code></pre>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb283-1" title="1"><span class="kw">pwr.p.test</span>(</a>
<a class="sourceLine" id="cb283-2" title="2">  h, </a>
<a class="sourceLine" id="cb283-3" title="3">  <span class="dt">sig.level=</span><span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb283-4" title="4">  <span class="dt">power=</span><span class="fl">0.8</span>, </a>
<a class="sourceLine" id="cb283-5" title="5">  <span class="dt">alternative=</span><span class="st">&quot;two.sided&quot;</span></a>
<a class="sourceLine" id="cb283-6" title="6">  )</a></code></pre></div>
<pre><code>## 
##      proportion power calculation for binomial distribution (arcsine transformation) 
## 
##               h = 0.1318964
##               n = 451.1706
##       sig.level = 0.05
##           power = 0.8
##     alternative = two.sided</code></pre>
<p>The output indicates that a total of 450 random, independent events will be necessary for 80% power to detect a difference between a sample proportion of 0.2 to a hypothetical proportion of 0.15.</p>
<p><code>binom.power</code> is a function from the <code>binom</code> package. Instead of returning sample size, this function returns power, given sample size. As an a priori test, one iterates through (by hand) entering sample sizes until it returns an acceptable power. Then run the experiment at that sample size.</p>
<p>Note that it doesn’t give <em>exactly</em> the same result as <code>pwr.p.test</code>. The calculation differs, but the result is close.</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb285-1" title="1"><span class="kw">binom.power</span>(</a>
<a class="sourceLine" id="cb285-2" title="2">  <span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb285-3" title="3">  <span class="dt">n=</span><span class="dv">451</span>, </a>
<a class="sourceLine" id="cb285-4" title="4">  <span class="dt">p=</span><span class="fl">0.15</span>, </a>
<a class="sourceLine" id="cb285-5" title="5">  <span class="dt">alpha=</span><span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb285-6" title="6">  <span class="dt">alternative =</span> <span class="st">&quot;two.sided&quot;</span>, </a>
<a class="sourceLine" id="cb285-7" title="7">  <span class="dt">method=</span><span class="st">&quot;exact&quot;</span></a>
<a class="sourceLine" id="cb285-8" title="8">  )</a></code></pre></div>
<pre><code>## [1] 0.7908632</code></pre>
<p>To estimate sample size needed for a two-sample proportion test design, for example, to compare a negative control to a treatment, use the <code>power.prop.test</code> function.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb287-1" title="1"><span class="kw">power.prop.test</span>(</a>
<a class="sourceLine" id="cb287-2" title="2">  <span class="dt">p1=</span><span class="fl">0.15</span>, </a>
<a class="sourceLine" id="cb287-3" title="3">  <span class="dt">p2=</span><span class="fl">0.2</span>, </a>
<a class="sourceLine" id="cb287-4" title="4">  <span class="dt">sig.level =</span> <span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb287-5" title="5">  <span class="dt">power=</span><span class="fl">0.8</span></a>
<a class="sourceLine" id="cb287-6" title="6">  )</a></code></pre></div>
<pre><code>## 
##      Two-sample comparison of proportions power calculation 
## 
##               n = 905.3658
##              p1 = 0.15
##              p2 = 0.2
##       sig.level = 0.05
##           power = 0.8
##     alternative = two.sided
## 
## NOTE: n is number in *each* group</code></pre>
<p>Finally, the <code>statmod</code> package has the <code>power.fisher.test</code>, which returns the power for a Fisher’s exact test, given arguments of proportion, trial size and type1 error. Note how it is in close but not exact agreement with <code>power.prop.test</code>.</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb289-1" title="1"><span class="kw">power.fisher.test</span>(</a>
<a class="sourceLine" id="cb289-2" title="2">  <span class="fl">0.15</span>, <span class="fl">0.2</span>, <span class="dv">905</span>, <span class="dv">905</span>, <span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb289-3" title="3">  <span class="dt">alternative =</span> <span class="st">&quot;two.sided&quot;</span></a>
<a class="sourceLine" id="cb289-4" title="4">  )</a></code></pre></div>
<pre><code>## [1] 0.79</code></pre>
<div id="monte-carlo-power-simulations" class="section level4">
<h4><span class="header-section-number">17.9.0.1</span> Monte Carlo power simulations</h4>
<p>The Monte Carlo is a general technique in statistical computing to randomly repeat some calculation or process. Monte Carlo is the basis of all sorts of things, like permutation analyses, machine learning, fitting Bayesian models to datasets and more.</p>
<p>Monte Carlo’s are very simple.</p>
<p>Monte Carlo power simulations is the insane thing you would never do in real life, which is to repeat the same full experiment replete with random, independent replicates thousands of times. Just to get a long run average of how well it works.</p>
<p>The basic gist is to simulate and test a very large number of experiments. Each of these experiments is comprised of a random sample of some size, corresponding to your minimal effect size you define as scientifically meritorious.</p>
<p>Sample responses are randomly generated. These are run through the test of significance, to calculate a p-value. The fraction of simulations that are “hits”–for example, that have p-values &lt; 0.05, is the power of the experiment!</p>
<p>Simulations are re-run by adjusting the sample size until a desired power is achieved.</p>
<p>That’s the sample size you’d take for use in a real experiment!</p>
<p>The question the script addresses is this: What is the power of an experiment, given this trial size n, the null and alternate proportions evaluated, and the type1 error threshold?</p>
<p>If n is too low, the test will return a power below the threshold you set, meaning it is not adequately powered to test the difference between the null and alternate proportions.</p>
<p>Iterate through some a few sample sizes (n) until you arrive at an acceptable value for power.</p>
<div id="monte-carlo-simulation-for-prop.test-power" class="section level5">
<h5><span class="header-section-number">17.9.0.1.1</span> Monte Carlo simulation for prop.test power</h5>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb291-1" title="1"><span class="co">#these first few objects are the initializers</span></a>
<a class="sourceLine" id="cb291-2" title="2"></a>
<a class="sourceLine" id="cb291-3" title="3"><span class="co">#number of experiments to simulate, each of trial size n</span></a>
<a class="sourceLine" id="cb291-4" title="4">sims &lt;-<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb291-5" title="5"><span class="co">#expected null proportion</span></a>
<a class="sourceLine" id="cb291-6" title="6">null &lt;-<span class="st"> </span><span class="fl">0.15</span></a>
<a class="sourceLine" id="cb291-7" title="7"><span class="co">#expected minimal effect proportion</span></a>
<a class="sourceLine" id="cb291-8" title="8">alternate &lt;-<span class="st"> </span><span class="fl">0.20</span> </a>
<a class="sourceLine" id="cb291-9" title="9"></a>
<a class="sourceLine" id="cb291-10" title="10"><span class="co">#binomial trial size, just a guess, ha!</span></a>
<a class="sourceLine" id="cb291-11" title="11">n &lt;-<span class="st"> </span><span class="dv">450</span> </a>
<a class="sourceLine" id="cb291-12" title="12"><span class="co">#type 1 error threshold</span></a>
<a class="sourceLine" id="cb291-13" title="13">alpha &lt;-<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb291-14" title="14"></a>
<a class="sourceLine" id="cb291-15" title="15"><span class="co">#s1 is a random sample vector </span></a>
<a class="sourceLine" id="cb291-16" title="16"><span class="co">#each value is the number of </span></a>
<a class="sourceLine" id="cb291-17" title="17"><span class="co">#successes observed in a trial </span></a>
<a class="sourceLine" id="cb291-18" title="18"><span class="co">#of size n, given the alternate proportion.</span></a>
<a class="sourceLine" id="cb291-19" title="19"><span class="co">#it simulates the outcome of one experiment &quot;sims&quot; times</span></a>
<a class="sourceLine" id="cb291-20" title="20"></a>
<a class="sourceLine" id="cb291-21" title="21">s1 &lt;-<span class="st"> </span><span class="kw">rbinom</span>(sims, n, alternate)</a>
<a class="sourceLine" id="cb291-22" title="22"></a>
<a class="sourceLine" id="cb291-23" title="23"><span class="co">#t1 is a vector of p-values, </span></a>
<a class="sourceLine" id="cb291-24" title="24"><span class="co">#derived from a one sample proportion test </span></a>
<a class="sourceLine" id="cb291-25" title="25"><span class="co">#on each of the values in s1. </span></a>
<a class="sourceLine" id="cb291-26" title="26"><span class="co">#read from inside the function to see the logic</span></a>
<a class="sourceLine" id="cb291-27" title="27"></a>
<a class="sourceLine" id="cb291-28" title="28">t1 &lt;-<span class="st"> </span><span class="kw">unlist</span>(</a>
<a class="sourceLine" id="cb291-29" title="29">  <span class="kw">lapply</span>(</a>
<a class="sourceLine" id="cb291-30" title="30">    s1, </a>
<a class="sourceLine" id="cb291-31" title="31">    <span class="cf">function</span>(s1){</a>
<a class="sourceLine" id="cb291-32" title="32">      <span class="kw">prop.test</span>(</a>
<a class="sourceLine" id="cb291-33" title="33">        s1, n, null,</a>
<a class="sourceLine" id="cb291-34" title="34">        <span class="dt">alternative=</span><span class="st">&quot;two.sided&quot;</span>, </a>
<a class="sourceLine" id="cb291-35" title="35">        <span class="dt">conf.level=</span><span class="dv">1</span><span class="op">-</span>alpha, </a>
<a class="sourceLine" id="cb291-36" title="36">        <span class="dt">correct=</span>T)<span class="op">$</span>p.value</a>
<a class="sourceLine" id="cb291-37" title="37">      }</a>
<a class="sourceLine" id="cb291-38" title="38">    )</a>
<a class="sourceLine" id="cb291-39" title="39">  )</a>
<a class="sourceLine" id="cb291-40" title="40"></a>
<a class="sourceLine" id="cb291-41" title="41">power &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">which</span>(t1 <span class="op">&lt;</span><span class="st"> </span>alpha))<span class="op">/</span>sims</a>
<a class="sourceLine" id="cb291-42" title="42">power</a></code></pre></div>
<pre><code>## [1] 0.806</code></pre>
</div>
</div>
</div>
<div id="graphing-proportions" class="section level2">
<h2><span class="header-section-number">17.10</span> Graphing Proportions</h2>
<p>Here’s a few ggplot2-based ways of visualizing proportion data.</p>
<p>First thing is to create a dataframe of the proportion data since data fed into ggplot2 must be in dataframe format.</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb293-1" title="1">prop.df &lt;-<span class="st"> </span><span class="kw">tibble</span>(</a>
<a class="sourceLine" id="cb293-2" title="2">  <span class="dt">group=</span><span class="kw">c</span>(<span class="st">&quot;positive&quot;</span>, <span class="st">&quot;negative&quot;</span>), </a>
<a class="sourceLine" id="cb293-3" title="3">  <span class="dt">value=</span><span class="kw">c</span>(pos, neg)</a>
<a class="sourceLine" id="cb293-4" title="4">  )</a>
<a class="sourceLine" id="cb293-5" title="5">prop.df</a></code></pre></div>
<pre><code>## # A tibble: 2 x 2
##   group    value
##   &lt;chr&gt;    &lt;dbl&gt;
## 1 positive  5042
## 2 negative 18492</code></pre>
<div id="simple-stacked-bar-chart" class="section level5">
<h5><span class="header-section-number">17.10.0.0.1</span> Simple stacked bar chart</h5>
<p>These certainly emphasize proportion differences well, but make it difficult to visually estimate the differences between the two groups.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb295-1" title="1"><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb295-2" title="2">  prop.df,</a>
<a class="sourceLine" id="cb295-3" title="3">  (<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">&quot;&quot;</span>, </a>
<a class="sourceLine" id="cb295-4" title="4">       <span class="dt">y=</span>value, </a>
<a class="sourceLine" id="cb295-5" title="5">       <span class="dt">fill=</span>group)</a>
<a class="sourceLine" id="cb295-6" title="6">   )</a>
<a class="sourceLine" id="cb295-7" title="7">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb295-8" title="8"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">fill=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-171"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-171-1.png" alt="Stacked bar plots of raw counts can be difficult to interpret" width="672" />
<p class="caption">
Figure 17.1: Stacked bar plots of raw counts can be difficult to interpret
</p>
</div>
</div>
<div id="side-by-side-bar-chart" class="section level5">
<h5><span class="header-section-number">17.10.0.0.2</span> Side-by-side bar chart</h5>
<p>The gold standard for clear presentation of total counts and differences in counts between groups.</p>
<p>Note: When comparing <em>independent</em> counts there is no error to report. There’s no variation. Cells were classified as either having or not having the antigen. Every count is independent of all other counts.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb296-1" title="1"><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb296-2" title="2">  prop.df, </a>
<a class="sourceLine" id="cb296-3" title="3">  (<span class="kw">aes</span>(</a>
<a class="sourceLine" id="cb296-4" title="4">    <span class="dt">x=</span>group, </a>
<a class="sourceLine" id="cb296-5" title="5">    <span class="dt">y=</span>value, </a>
<a class="sourceLine" id="cb296-6" title="6">    <span class="dt">fill=</span>group)</a>
<a class="sourceLine" id="cb296-7" title="7">   )</a>
<a class="sourceLine" id="cb296-8" title="8">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb296-9" title="9"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">fill =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-172"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-172-1.png" alt="Side-by-side bar charts make it easier to see difference between raw counts." width="672" />
<p class="caption">
Figure 17.2: Side-by-side bar charts make it easier to see difference between raw counts.
</p>
</div>
</div>
<div id="plot-proportions-with-confidence-intervals" class="section level5">
<h5><span class="header-section-number">17.10.0.0.3</span> Plot proportions with confidence intervals</h5>
<p>However, adding confidence intervals to bar plots of proportions is a good way to visualize the uncertainty of counted data and comparing to other proportions.</p>
<table>
<caption>
<span id="tab:unnamed-chunk-173">Table 17.3: </span>Hypothetical counts of gradstudin-stained T cells, myCounts
</caption>
<thead>
<tr>
<th style="text-align:left;">
group
</th>
<th style="text-align:right;">
pos
</th>
<th style="text-align:right;">
neg
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
193
</td>
</tr>
<tr>
<td style="text-align:left;">
Cytokine1
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
186
</td>
</tr>
<tr>
<td style="text-align:left;">
Cytokine2
</td>
<td style="text-align:right;">
63
</td>
<td style="text-align:right;">
167
</td>
</tr>
</tbody>
</table>
<p>Munge the data directly within <code>ggplot</code> function to calculate values for the proportion and its upper and lower 95% confidence intervals. Proportions with overlapping CIs do not differ from each other. Note how the CIs are adjusted for the 3 possible comparisons using the Bonferroni correction (adjusted 95% confidence level = 1-0.05/3).</p>
<p>Note that the <code>rowwise</code> function is necessary for certain ‘legacy’ functions like <code>scoreci</code>.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb297-1" title="1"><span class="kw">ggplot</span>(myCounts <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rowwise</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb297-2" title="2"><span class="st">  </span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb297-3" title="3">    <span class="dt">prop=</span>pos<span class="op">/</span>(pos<span class="op">+</span>neg),</a>
<a class="sourceLine" id="cb297-4" title="4">    <span class="dt">lower=</span><span class="kw">scoreci</span>(pos, pos<span class="op">+</span>neg, <span class="fl">0.98333</span>)<span class="op">$</span>conf.int[<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb297-5" title="5">    <span class="dt">upper=</span><span class="kw">scoreci</span>(pos, pos<span class="op">+</span>neg, <span class="fl">0.98333</span>)<span class="op">$</span>conf.int[<span class="dv">2</span>]), <span class="kw">aes</span>(<span class="dt">x=</span>group))<span class="op">+</span></a>
<a class="sourceLine" id="cb297-6" title="6"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="kw">aes</span>(<span class="dt">y =</span> prop), <span class="dt">stat =</span><span class="st">&quot;identity&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;blue&quot;</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb297-7" title="7"><span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymin=</span>lower, <span class="dt">ymax=</span>upper), <span class="dt">width=</span><span class="fl">0.2</span>, <span class="dt">size=</span><span class="dv">2</span>)<span class="op">+</span></a>
<a class="sourceLine" id="cb297-8" title="8"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion of gradstudin positive cells&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-174"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-174-1.png" alt="You can infer two proportions differ from plots when their confidence intervals do not overlap. " width="672" />
<p class="caption">
Figure 17.3: You can infer two proportions differ from plots when their confidence intervals do not overlap.
</p>
</div>
</div>
<div id="pie-charts-and-mosaic-plots" class="section level5">
<h5><span class="header-section-number">17.10.0.0.4</span> Pie charts and mosaic plots</h5>
<p>I suggest you explore these options on Google. They can be visually appealing for showing relative differences in some cases, but they make it more difficult to easily see absolute counts and differentials.</p>
</div>
</div>
<div id="key-takeaways" class="section level2">
<h2><span class="header-section-number">17.11</span> Key Takeaways</h2>
<ul>
<li>Sorted variables have nominal values that are classified into categories and that are quantified as discrete integer counts.</li>
<li>Each count must be an independent replicate.</li>
<li>Sorted events can be modeled using <code>rbinom</code> or <code>rmultinom</code>. The latter is used when the event has more than two discrete outcomes.</li>
<li>The three basic types of experimental designs are questions about proportions, or about frequencies, or about associations.</li>
<li>The test functions mostly differ in whether they generate approximate or exact p-values. To avoid p-hacking, choose the test in advance.</li>
<li>When designing an experiment perform a power analysis to determine the number of independent events that will be necessary. When testing for small differences or effect sizes, these numbers needed for a stringent test can be surprisingly high.</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="normal.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chisquare.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["jabstb.pdf", "jabstb.epub"],
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
