<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Data Classification | JABSTB: Statistical Design and Analysis of Experiments with R</title>
  <meta name="description" content="Experimental biostatistics using R." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Data Classification | JABSTB: Statistical Design and Analysis of Experiments with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Experimental biostatistics using R." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Data Classification | JABSTB: Statistical Design and Analysis of Experiments with R" />
  
  <meta name="twitter:description" content="Experimental biostatistics using R." />
  

<meta name="author" content="TJ Murphy PhD, Department of Pharmacology and Chemical Biology, School of Medicine, Emory University, Atlanta, GA biostats538@gmail.com" />


<meta name="date" content="2022-01-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rlearn1.html"/>
<link rel="next" href="dispersion.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">JABSTB</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="author.html"><a href="author.html"><i class="fa fa-check"></i><b>1</b> About the author and book</a></li>
<li class="chapter" data-level="2" data-path="history.html"><a href="history.html"><i class="fa fa-check"></i><b>2</b> A Brief History of Experimental Design</a></li>
<li class="chapter" data-level="3" data-path="software.html"><a href="software.html"><i class="fa fa-check"></i><b>3</b> The software</a>
<ul>
<li class="chapter" data-level="3.1" data-path="software.html"><a href="software.html#you-do-not-need-to-install-anything-for-ibs538bios505"><i class="fa fa-check"></i><b>3.1</b> YOU DO NOT NEED TO INSTALL ANYTHING for IBS538/BIOS505</a></li>
<li class="chapter" data-level="3.2" data-path="software.html"><a href="software.html#tldr-if-you-wish-to-install"><i class="fa fa-check"></i><b>3.2</b> tl;dr if you wish to install</a></li>
<li class="chapter" data-level="3.3" data-path="software.html"><a href="software.html#spring-2022-term-versions"><i class="fa fa-check"></i><b>3.3</b> Spring 2022 term versions</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="software.html"><a href="software.html#what-are-these"><i class="fa fa-check"></i><b>3.3.1</b> What are these?</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="software.html"><a href="software.html#follow-these-rules"><i class="fa fa-check"></i><b>3.4</b> Follow these rules</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="software.html"><a href="software.html#rule-1-install-r-first-install-rstudio-next"><i class="fa fa-check"></i><b>3.4.1</b> Rule 1: Install R first, install RStudio next</a></li>
<li class="chapter" data-level="3.4.2" data-path="software.html"><a href="software.html#rule-2-dont-pay"><i class="fa fa-check"></i><b>3.4.2</b> Rule 2: Don’t pay!</a></li>
<li class="chapter" data-level="3.4.3" data-path="software.html"><a href="software.html#rule-3-dont-compile"><i class="fa fa-check"></i><b>3.4.3</b> Rule 3: Don’t compile!</a></li>
<li class="chapter" data-level="3.4.4" data-path="software.html"><a href="software.html#rule-4-install-the-right-r-software-for-your-machine"><i class="fa fa-check"></i><b>3.4.4</b> Rule 4: Install the right R software for your machine</a></li>
<li class="chapter" data-level="3.4.5" data-path="software.html"><a href="software.html#doubts"><i class="fa fa-check"></i><b>3.4.5</b> Doubts?</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="software.html"><a href="software.html#after-installation"><i class="fa fa-check"></i><b>3.5</b> After installation</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="software.html"><a href="software.html#problems"><i class="fa fa-check"></i><b>3.5.1</b> Problems?</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="software.html"><a href="software.html#what-is-r"><i class="fa fa-check"></i><b>3.6</b> What is R?</a></li>
<li class="chapter" data-level="3.7" data-path="software.html"><a href="software.html#what-are-r-packages"><i class="fa fa-check"></i><b>3.7</b> What are R packages?</a></li>
<li class="chapter" data-level="3.8" data-path="software.html"><a href="software.html#how-to-install-and-use-packages"><i class="fa fa-check"></i><b>3.8</b> How to install and use packages</a></li>
<li class="chapter" data-level="3.9" data-path="software.html"><a href="software.html#packages-well-need"><i class="fa fa-check"></i><b>3.9</b> Packages we’ll need</a></li>
<li class="chapter" data-level="3.10" data-path="software.html"><a href="software.html#super-important-always-avoid-compiling"><i class="fa fa-check"></i><b>3.10</b> Super important: Always avoid compiling</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="software.html"><a href="software.html#reasons-not-to-compile"><i class="fa fa-check"></i><b>3.10.1</b> Reasons not to compile</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="software.html"><a href="software.html#summary"><i class="fa fa-check"></i><b>3.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="getting_started.html"><a href="getting_started.html"><i class="fa fa-check"></i><b>4</b> Start Learning R</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting_started.html"><a href="getting_started.html#digging-in-with-rrstudio"><i class="fa fa-check"></i><b>4.1</b> Digging in with R/RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting_started.html"><a href="getting_started.html#the-tidyverse-package"><i class="fa fa-check"></i><b>4.2</b> The tidyverse package</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting_started.html"><a href="getting_started.html#do-swirlstats"><i class="fa fa-check"></i><b>4.2.1</b> Do swirlstats</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting_started.html"><a href="getting_started.html#other-resources"><i class="fa fa-check"></i><b>4.3</b> Other resources</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="getting_started.html"><a href="getting_started.html#internet-search"><i class="fa fa-check"></i><b>4.3.1</b> Internet search</a></li>
<li class="chapter" data-level="4.3.2" data-path="getting_started.html"><a href="getting_started.html#help-documents"><i class="fa fa-check"></i><b>4.3.2</b> Help Documents</a></li>
<li class="chapter" data-level="4.3.3" data-path="getting_started.html"><a href="getting_started.html#package-vignettes"><i class="fa fa-check"></i><b>4.3.3</b> Package Vignettes</a></li>
<li class="chapter" data-level="4.3.4" data-path="getting_started.html"><a href="getting_started.html#people"><i class="fa fa-check"></i><b>4.3.4</b> People</a></li>
<li class="chapter" data-level="4.3.5" data-path="getting_started.html"><a href="getting_started.html#typos"><i class="fa fa-check"></i><b>4.3.5</b> typos</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="philos.html"><a href="philos.html"><i class="fa fa-check"></i><b>5</b> What is your philosophy?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="philos.html"><a href="philos.html#our-philosophy-matters"><i class="fa fa-check"></i><b>5.1</b> Our philosophy matters</a></li>
<li class="chapter" data-level="5.2" data-path="philos.html"><a href="philos.html#exploration-and-uncertainty"><i class="fa fa-check"></i><b>5.2</b> Exploration and uncertainty</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bigpic.html"><a href="bigpic.html"><i class="fa fa-check"></i><b>6</b> Stats: The Big Picture</a>
<ul>
<li class="chapter" data-level="6.1" data-path="bigpic.html"><a href="bigpic.html#what-are-experimental-statistics"><i class="fa fa-check"></i><b>6.1</b> What are experimental statistics?</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="bigpic.html"><a href="bigpic.html#descriptive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Descriptive modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="bigpic.html"><a href="bigpic.html#statistical-inference"><i class="fa fa-check"></i><b>6.1.2</b> Statistical inference</a></li>
<li class="chapter" data-level="6.1.3" data-path="bigpic.html"><a href="bigpic.html#experimental-design"><i class="fa fa-check"></i><b>6.1.3</b> Experimental design</a></li>
<li class="chapter" data-level="6.1.4" data-path="bigpic.html"><a href="bigpic.html#statistics-as-an-anti-bias-framework"><i class="fa fa-check"></i><b>6.1.4</b> Statistics as an anti-bias framework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>7</b> Statistical Sampling</a>
<ul>
<li class="chapter" data-level="7.1" data-path="sampling.html"><a href="sampling.html#experimental-units"><i class="fa fa-check"></i><b>7.1</b> Experimental units</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-to-define-the-experimental-unit"><i class="fa fa-check"></i><b>7.1.1</b> A simple test to define the experimental unit</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="sampling.html"><a href="sampling.html#independent-replicates"><i class="fa fa-check"></i><b>7.2</b> Independent Replicates</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-for-independence"><i class="fa fa-check"></i><b>7.2.1</b> A simple test for independence</a></li>
<li class="chapter" data-level="7.2.2" data-path="sampling.html"><a href="sampling.html#some-replication-examples"><i class="fa fa-check"></i><b>7.2.2</b> Some replication examples</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="sampling.html"><a href="sampling.html#random-process"><i class="fa fa-check"></i><b>7.3</b> Random process</a></li>
<li class="chapter" data-level="7.4" data-path="sampling.html"><a href="sampling.html#statistically-valid-samples"><i class="fa fa-check"></i><b>7.4</b> Statistically valid samples</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="sampling.html"><a href="sampling.html#select-random-subjects"><i class="fa fa-check"></i><b>7.4.1</b> Select random subjects</a></li>
<li class="chapter" data-level="7.4.2" data-path="sampling.html"><a href="sampling.html#randomize-to-sequence"><i class="fa fa-check"></i><b>7.4.2</b> Randomize to sequence</a></li>
<li class="chapter" data-level="7.4.3" data-path="sampling.html"><a href="sampling.html#randomize-to-location"><i class="fa fa-check"></i><b>7.4.3</b> Randomize to location</a></li>
<li class="chapter" data-level="7.4.4" data-path="sampling.html"><a href="sampling.html#randomize-to-block"><i class="fa fa-check"></i><b>7.4.4</b> Randomize to block</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="sampling.html"><a href="sampling.html#independence-of-replicates"><i class="fa fa-check"></i><b>7.5</b> Independence of replicates</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="rlearn1.html"><a href="rlearn1.html"><i class="fa fa-check"></i><b>8</b> R Learnings for now</a>
<ul>
<li class="chapter" data-level="8.1" data-path="rlearn1.html"><a href="rlearn1.html#the-whole-r-thing"><i class="fa fa-check"></i><b>8.1</b> The whole R thing</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="rlearn1.html"><a href="rlearn1.html#words-matter"><i class="fa fa-check"></i><b>8.1.1</b> Words matter</a></li>
<li class="chapter" data-level="8.1.2" data-path="rlearn1.html"><a href="rlearn1.html#functions-arguments"><i class="fa fa-check"></i><b>8.1.2</b> Functions &amp; arguments</a></li>
<li class="chapter" data-level="8.1.3" data-path="rlearn1.html"><a href="rlearn1.html#functions-work-inside-other-functions"><i class="fa fa-check"></i><b>8.1.3</b> Functions work inside other functions</a></li>
<li class="chapter" data-level="8.1.4" data-path="rlearn1.html"><a href="rlearn1.html#divide-by-two"><i class="fa fa-check"></i><b>8.1.4</b> Divide by two</a></li>
<li class="chapter" data-level="8.1.5" data-path="rlearn1.html"><a href="rlearn1.html#steal-my-code"><i class="fa fa-check"></i><b>8.1.5</b> Steal my code</a></li>
<li class="chapter" data-level="8.1.6" data-path="rlearn1.html"><a href="rlearn1.html#mustknow"><i class="fa fa-check"></i><b>8.1.6</b> Nine MUST KNOW functions for now</a></li>
<li class="chapter" data-level="8.1.7" data-path="rlearn1.html"><a href="rlearn1.html#you-have-options"><i class="fa fa-check"></i><b>8.1.7</b> You have options</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="rlearn1.html"><a href="rlearn1.html#working-with-variables-in-r"><i class="fa fa-check"></i><b>8.2</b> Working with variables in R</a></li>
<li class="chapter" data-level="8.3" data-path="rlearn1.html"><a href="rlearn1.html#entering-data"><i class="fa fa-check"></i><b>8.3</b> Entering data</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="rlearn1.html"><a href="rlearn1.html#typing-in-variables-and-making-a-data-frame"><i class="fa fa-check"></i><b>8.3.1</b> Typing in variables and making a data frame</a></li>
<li class="chapter" data-level="8.3.2" data-path="rlearn1.html"><a href="rlearn1.html#using-datapasta"><i class="fa fa-check"></i><b>8.3.2</b> Using datapasta</a></li>
<li class="chapter" data-level="8.3.3" data-path="rlearn1.html"><a href="rlearn1.html#reading-a-file"><i class="fa fa-check"></i><b>8.3.3</b> Reading a file</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="rlearn1.html"><a href="rlearn1.html#subsetting-transforming-and-summarizing-data"><i class="fa fa-check"></i><b>8.4</b> Subsetting, transforming and summarizing data</a></li>
<li class="chapter" data-level="8.5" data-path="rlearn1.html"><a href="rlearn1.html#plotting-variables-with-ggplot2"><i class="fa fa-check"></i><b>8.5</b> Plotting variables with ggplot2</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="rlearn1.html"><a href="rlearn1.html#the-default-ggplot-function-creates-the-blank-canvas"><i class="fa fa-check"></i><b>8.5.1</b> The default ggplot function creates the blank canvas</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="rlearn1.html"><a href="rlearn1.html#summary-1"><i class="fa fa-check"></i><b>8.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>9</b> Data Classification</a>
<ul>
<li class="chapter" data-level="9.1" data-path="data.html"><a href="data.html#dependent-and-independent-variables"><i class="fa fa-check"></i><b>9.1</b> Dependent and independent variables</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="data.html"><a href="data.html#statistical-notation"><i class="fa fa-check"></i><b>9.1.1</b> Statistical notation</a></li>
<li class="chapter" data-level="9.1.2" data-path="data.html"><a href="data.html#when-there-is-no-independent-variable"><i class="fa fa-check"></i><b>9.1.2</b> When there is no independent variable</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="data.html"><a href="data.html#discrete-or-continuous-variables"><i class="fa fa-check"></i><b>9.2</b> Discrete or continuous variables</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="data.html"><a href="data.html#measured"><i class="fa fa-check"></i><b>9.2.1</b> Measured variables</a></li>
<li class="chapter" data-level="9.2.2" data-path="data.html"><a href="data.html#discrete-categorical-and-ordinal-variables"><i class="fa fa-check"></i><b>9.2.2</b> Discrete categorical and ordinal variables</a></li>
<li class="chapter" data-level="9.2.3" data-path="data.html"><a href="data.html#rescaling-variables"><i class="fa fa-check"></i><b>9.2.3</b> Rescaling variables</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="data.html"><a href="data.html#summary-2"><i class="fa fa-check"></i><b>9.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="dispersion.html"><a href="dispersion.html"><i class="fa fa-check"></i><b>10</b> Variability, Accuracy and Precision</a>
<ul>
<li class="chapter" data-level="10.1" data-path="dispersion.html"><a href="dispersion.html#data-are-messy"><i class="fa fa-check"></i><b>10.1</b> Data are messy</a></li>
<li class="chapter" data-level="10.2" data-path="dispersion.html"><a href="dispersion.html#illustration-of-a-model-with-error"><i class="fa fa-check"></i><b>10.2</b> Illustration of a model with error</a></li>
<li class="chapter" data-level="10.3" data-path="dispersion.html"><a href="dispersion.html#variance-quantifying-variation-by-least-squares"><i class="fa fa-check"></i><b>10.3</b> Variance: Quantifying variation by least squares</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="dispersion.html"><a href="dispersion.html#definition-of-variance"><i class="fa fa-check"></i><b>10.3.1</b> Definition of variance</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="dispersion.html"><a href="dispersion.html#standard-deviation"><i class="fa fa-check"></i><b>10.4</b> Standard deviation</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="dispersion.html"><a href="dispersion.html#what-does-the-standard-deviation-tell-us"><i class="fa fa-check"></i><b>10.4.1</b> What does the standard deviation tell us</a></li>
<li class="chapter" data-level="10.4.2" data-path="dispersion.html"><a href="dispersion.html#ways-of-describing-sample-variability"><i class="fa fa-check"></i><b>10.4.2</b> Ways of describing sample variability</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="dispersion.html"><a href="dispersion.html#precision-and-accuracy"><i class="fa fa-check"></i><b>10.5</b> Precision and Accuracy</a></li>
<li class="chapter" data-level="10.6" data-path="dispersion.html"><a href="dispersion.html#standard-error"><i class="fa fa-check"></i><b>10.6</b> Standard error</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="dispersion.html"><a href="dispersion.html#what-exactly-does-the-standard-error-represent"><i class="fa fa-check"></i><b>10.6.1</b> What exactly does the standard error represent?</a></li>
<li class="chapter" data-level="10.6.2" data-path="dispersion.html"><a href="dispersion.html#should-i-use-sd-or-sem"><i class="fa fa-check"></i><b>10.6.2</b> “Should I use SD or SEM?”</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="dispersion.html"><a href="dispersion.html#confidence-intervals"><i class="fa fa-check"></i><b>10.7</b> Confidence intervals</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="dispersion.html"><a href="dispersion.html#ci-definition"><i class="fa fa-check"></i><b>10.7.1</b> CI Definition</a></li>
<li class="chapter" data-level="10.7.2" data-path="dispersion.html"><a href="dispersion.html#evaluating-confidence-intervals"><i class="fa fa-check"></i><b>10.7.2</b> Evaluating confidence intervals</a></li>
<li class="chapter" data-level="10.7.3" data-path="dispersion.html"><a href="dispersion.html#ci-simulation-tools"><i class="fa fa-check"></i><b>10.7.3</b> CI Simulation tools</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="dispersion.html"><a href="dispersion.html#key-take-aways"><i class="fa fa-check"></i><b>10.8</b> Key take aways</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="hypotheses.html"><a href="hypotheses.html"><i class="fa fa-check"></i><b>11</b> Framing statistical hypotheses</a>
<ul>
<li class="chapter" data-level="11.1" data-path="hypotheses.html"><a href="hypotheses.html#the-decision-process"><i class="fa fa-check"></i><b>11.1</b> The decision process</a></li>
<li class="chapter" data-level="11.2" data-path="hypotheses.html"><a href="hypotheses.html#popper-and-falsification"><i class="fa fa-check"></i><b>11.2</b> Popper and falsification</a></li>
<li class="chapter" data-level="11.3" data-path="hypotheses.html"><a href="hypotheses.html#operational-decisions-made-in-advance-that-reduce-mistakes"><i class="fa fa-check"></i><b>11.3</b> Operational decisions made in advance that reduce mistakes</a></li>
<li class="chapter" data-level="11.4" data-path="hypotheses.html"><a href="hypotheses.html#statistical-hypothesis-rubric"><i class="fa fa-check"></i><b>11.4</b> Statistical hypothesis rubric</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="error.html"><a href="error.html"><i class="fa fa-check"></i><b>12</b> Error</a>
<ul>
<li class="chapter" data-level="12.1" data-path="error.html"><a href="error.html#setting-type-1-and-type-2-error-thresholds"><i class="fa fa-check"></i><b>12.1</b> Setting type 1 and type 2 error thresholds</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="error.html"><a href="error.html#setting-alpha-the-type-1-error"><i class="fa fa-check"></i><b>12.1.1</b> Setting alpha-the type 1 error</a></li>
<li class="chapter" data-level="12.1.2" data-path="error.html"><a href="error.html#power-setting-beta-the-type-2-error"><i class="fa fa-check"></i><b>12.1.2</b> Power: Setting beta-the type 2 error</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="error.html"><a href="error.html#striking-the-right-balance"><i class="fa fa-check"></i><b>12.2</b> Striking the right balance</a></li>
<li class="chapter" data-level="12.3" data-path="error.html"><a href="error.html#false-discovery-rate"><i class="fa fa-check"></i><b>12.3</b> False discovery rate</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="pvalues.html"><a href="pvalues.html"><i class="fa fa-check"></i><b>13</b> P Values</a>
<ul>
<li class="chapter" data-level="13.1" data-path="pvalues.html"><a href="pvalues.html#definition"><i class="fa fa-check"></i><b>13.1</b> Definition</a></li>
<li class="chapter" data-level="13.2" data-path="pvalues.html"><a href="pvalues.html#null-test-statistic-distributions"><i class="fa fa-check"></i><b>13.2</b> Null test statistic distributions</a></li>
<li class="chapter" data-level="13.3" data-path="pvalues.html"><a href="pvalues.html#computing-p-values"><i class="fa fa-check"></i><b>13.3</b> Computing p-values</a></li>
<li class="chapter" data-level="13.4" data-path="pvalues.html"><a href="pvalues.html#p-values-from-null-tests-are-uniform"><i class="fa fa-check"></i><b>13.4</b> P-values from null tests are uniform</a></li>
<li class="chapter" data-level="13.5" data-path="pvalues.html"><a href="pvalues.html#how-p-values-become-error-probabilities"><i class="fa fa-check"></i><b>13.5</b> How P-values become error probabilities</a></li>
<li class="chapter" data-level="13.6" data-path="pvalues.html"><a href="pvalues.html#p-values-as-dichotomous-switches"><i class="fa fa-check"></i><b>13.6</b> P-values as dichotomous switches</a></li>
<li class="chapter" data-level="13.7" data-path="pvalues.html"><a href="pvalues.html#p-values-in-true-effects"><i class="fa fa-check"></i><b>13.7</b> P-values in true effects</a></li>
<li class="chapter" data-level="13.8" data-path="pvalues.html"><a href="pvalues.html#criticisms-of-p-values"><i class="fa fa-check"></i><b>13.8</b> Criticisms of p-values</a></li>
<li class="chapter" data-level="13.9" data-path="pvalues.html"><a href="pvalues.html#summary-how-to-use-p-values"><i class="fa fa-check"></i><b>13.9</b> Summary: How to use p-values</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="jaxwest7.html"><a href="jaxwest7.html"><i class="fa fa-check"></i><b>14</b> Reproducible Data Munging in R</a>
<ul>
<li class="chapter" data-level="14.1" data-path="jaxwest7.html"><a href="jaxwest7.html#jaxwest7-data"><i class="fa fa-check"></i><b>14.1</b> Jaxwest7 data</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="jaxwest7.html"><a href="jaxwest7.html#inspect-the-source-data"><i class="fa fa-check"></i><b>14.1.1</b> Inspect the source data</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-glucose-data-into-r"><i class="fa fa-check"></i><b>14.2</b> Munge the glucose data into R</a></li>
<li class="chapter" data-level="14.3" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-bodyweight-data"><i class="fa fa-check"></i><b>14.3</b> Munge the bodyweight data</a></li>
<li class="chapter" data-level="14.4" data-path="jaxwest7.html"><a href="jaxwest7.html#merge-and-explore-the-glucose-and-bw-data"><i class="fa fa-check"></i><b>14.4</b> Merge and explore the glucose and bw data</a></li>
<li class="chapter" data-level="14.5" data-path="jaxwest7.html"><a href="jaxwest7.html#summary-3"><i class="fa fa-check"></i><b>14.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="binomial.html"><a href="binomial.html"><i class="fa fa-check"></i><b>15</b> The Binomial Distribution</a>
<ul>
<li class="chapter" data-level="15.1" data-path="binomial.html"><a href="binomial.html#dbinom"><i class="fa fa-check"></i><b>15.1</b> dbinom</a></li>
<li class="chapter" data-level="15.2" data-path="binomial.html"><a href="binomial.html#pbinom"><i class="fa fa-check"></i><b>15.2</b> pbinom</a></li>
<li class="chapter" data-level="15.3" data-path="binomial.html"><a href="binomial.html#qbinom"><i class="fa fa-check"></i><b>15.3</b> qbinom</a></li>
<li class="chapter" data-level="15.4" data-path="binomial.html"><a href="binomial.html#rbinom"><i class="fa fa-check"></i><b>15.4</b> rbinom</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="poisson.html"><a href="poisson.html"><i class="fa fa-check"></i><b>16</b> The Poisson Distribution</a>
<ul>
<li class="chapter" data-level="16.1" data-path="poisson.html"><a href="poisson.html#poisson-events"><i class="fa fa-check"></i><b>16.1</b> Poisson Events</a></li>
<li class="chapter" data-level="16.2" data-path="poisson.html"><a href="poisson.html#dpois"><i class="fa fa-check"></i><b>16.2</b> dpois</a></li>
<li class="chapter" data-level="16.3" data-path="poisson.html"><a href="poisson.html#ppois"><i class="fa fa-check"></i><b>16.3</b> ppois</a></li>
<li class="chapter" data-level="16.4" data-path="poisson.html"><a href="poisson.html#rpois"><i class="fa fa-check"></i><b>16.4</b> rpois</a></li>
<li class="chapter" data-level="16.5" data-path="poisson.html"><a href="poisson.html#overdispersion"><i class="fa fa-check"></i><b>16.5</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="normal.html"><a href="normal.html"><i class="fa fa-check"></i><b>17</b> The Normal Distribution</a>
<ul>
<li class="chapter" data-level="17.0.1" data-path="normal.html"><a href="normal.html#the-standard-normal"><i class="fa fa-check"></i><b>17.0.1</b> The Standard Normal</a></li>
<li class="chapter" data-level="17.1" data-path="normal.html"><a href="normal.html#dnorm"><i class="fa fa-check"></i><b>17.1</b> dnorm</a></li>
<li class="chapter" data-level="17.2" data-path="normal.html"><a href="normal.html#pnorm"><i class="fa fa-check"></i><b>17.2</b> pnorm</a>
<ul>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-p-values-using-pnorm"><i class="fa fa-check"></i>Calculating “p-values”" using pnorm</a></li>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-percentiles-using-pnorm"><i class="fa fa-check"></i>Calculating percentiles using pnorm</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="normal.html"><a href="normal.html#qnorm"><i class="fa fa-check"></i><b>17.3</b> qnorm</a></li>
<li class="chapter" data-level="17.4" data-path="normal.html"><a href="normal.html#rnorm"><i class="fa fa-check"></i><b>17.4</b> rnorm</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="normal.html"><a href="normal.html#plotting-histograms-of-some-rnorm-samples"><i class="fa fa-check"></i><b>17.4.1</b> Plotting histograms of some rnorm samples</a></li>
<li class="chapter" data-level="17.4.2" data-path="normal.html"><a href="normal.html#bins-and-binwidth"><i class="fa fa-check"></i><b>17.4.2</b> Bins and Binwidth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="lognormal.html"><a href="lognormal.html"><i class="fa fa-check"></i><b>18</b> The lognormal distribution</a>
<ul>
<li class="chapter" data-level="18.1" data-path="lognormal.html"><a href="lognormal.html#tldr"><i class="fa fa-check"></i><b>18.1</b> tl;dr</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="lognormal.html"><a href="lognormal.html#importance"><i class="fa fa-check"></i><b>18.1.1</b> Importance</a></li>
<li class="chapter" data-level="18.1.2" data-path="lognormal.html"><a href="lognormal.html#practical-importance-of-this-reciprocal-relationship"><i class="fa fa-check"></i><b>18.1.2</b> Practical importance of this reciprocal relationship</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="lognormal.html"><a href="lognormal.html#dlnorm"><i class="fa fa-check"></i><b>18.2</b> dlnorm</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="lognormal.html"><a href="lognormal.html#sidebar-arithmetic-and-geometric-means"><i class="fa fa-check"></i><b>18.2.1</b> Sidebar: Arithmetic and geometric means</a></li>
<li class="chapter" data-level="18.2.2" data-path="lognormal.html"><a href="lognormal.html#using-dlnorm"><i class="fa fa-check"></i><b>18.2.2</b> Using dlnorm</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="lognormal.html"><a href="lognormal.html#plnorm"><i class="fa fa-check"></i><b>18.3</b> plnorm</a></li>
<li class="chapter" data-level="18.4" data-path="lognormal.html"><a href="lognormal.html#qlnorm"><i class="fa fa-check"></i><b>18.4</b> qlnorm</a></li>
<li class="chapter" data-level="18.5" data-path="lognormal.html"><a href="lognormal.html#rlnorm"><i class="fa fa-check"></i><b>18.5</b> rlnorm</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="categorical.html"><a href="categorical.html"><i class="fa fa-check"></i><b>19</b> Statistics for Categorical Data</a>
<ul>
<li class="chapter" data-level="19.1" data-path="categorical.html"><a href="categorical.html#types-of-categorical-data"><i class="fa fa-check"></i><b>19.1</b> Types of categorical data</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="categorical.html"><a href="categorical.html#proportions"><i class="fa fa-check"></i><b>19.1.1</b> Proportions</a></li>
<li class="chapter" data-level="19.1.2" data-path="categorical.html"><a href="categorical.html#frequencies"><i class="fa fa-check"></i><b>19.1.2</b> Frequencies</a></li>
<li class="chapter" data-level="19.1.3" data-path="categorical.html"><a href="categorical.html#associations"><i class="fa fa-check"></i><b>19.1.3</b> Associations</a></li>
<li class="chapter" data-level="19.1.4" data-path="categorical.html"><a href="categorical.html#statistics-covered-here"><i class="fa fa-check"></i><b>19.1.4</b> Statistics Covered Here</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="categorical.html"><a href="categorical.html#exact-v-asymptotic-calculations-of-p-values"><i class="fa fa-check"></i><b>19.2</b> Exact v Asymptotic Calculations of p-values</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="categorical.html"><a href="categorical.html#choosing-exact-or-asymptotic"><i class="fa fa-check"></i><b>19.2.1</b> Choosing exact or asymptotic</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="categorical.html"><a href="categorical.html#overview-of-the-types-of-hypothesis-testing"><i class="fa fa-check"></i><b>19.3</b> Overview of the types of hypothesis testing</a>
<ul>
<li class="chapter" data-level="19.3.1" data-path="categorical.html"><a href="categorical.html#proportion-analysis"><i class="fa fa-check"></i><b>19.3.1</b> Proportion analysis</a></li>
<li class="chapter" data-level="19.3.2" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-testing"><i class="fa fa-check"></i><b>19.3.2</b> Goodness of fit testing</a></li>
<li class="chapter" data-level="19.3.3" data-path="categorical.html"><a href="categorical.html#contingency-analysis"><i class="fa fa-check"></i><b>19.3.3</b> Contingency Analysis</a></li>
</ul></li>
<li class="chapter" data-level="19.4" data-path="categorical.html"><a href="categorical.html#comparing-proportions"><i class="fa fa-check"></i><b>19.4</b> Comparing proportions</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="categorical.html"><a href="categorical.html#a-mouse-t-cell-pilot-experiment-the-cytokine-inducible-antigen-gradstudin"><i class="fa fa-check"></i><b>19.4.1</b> A Mouse T Cell Pilot Experiment: The Cytokine-inducible antigen gradstudin</a></li>
<li class="chapter" data-level="19.4.2" data-path="categorical.html"><a href="categorical.html#calculating-proportions"><i class="fa fa-check"></i><b>19.4.2</b> Calculating Proportions</a></li>
<li class="chapter" data-level="19.4.3" data-path="categorical.html"><a href="categorical.html#what-a-proportion-estimates"><i class="fa fa-check"></i><b>19.4.3</b> What A Proportion Estimates</a></li>
<li class="chapter" data-level="19.4.4" data-path="categorical.html"><a href="categorical.html#confidence-intervals-of-proportions"><i class="fa fa-check"></i><b>19.4.4</b> Confidence Intervals of Proportions</a></li>
<li class="chapter" data-level="19.4.5" data-path="categorical.html"><a href="categorical.html#a-one-sample-proportion-test"><i class="fa fa-check"></i><b>19.4.5</b> A One-Sample Proportion Test</a></li>
<li class="chapter" data-level="19.4.6" data-path="categorical.html"><a href="categorical.html#an-exact-test-for-one-proportion"><i class="fa fa-check"></i><b>19.4.6</b> An exact test for one proportion</a></li>
<li class="chapter" data-level="19.4.7" data-path="categorical.html"><a href="categorical.html#comparing-two-proportions"><i class="fa fa-check"></i><b>19.4.7</b> Comparing Two Proportions</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="categorical.html"><a href="categorical.html#exact-tests-for-two-proportions"><i class="fa fa-check"></i><b>19.5</b> Exact tests for two proportions</a></li>
<li class="chapter" data-level="19.6" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-tests"><i class="fa fa-check"></i><b>19.6</b> Goodness of fit Tests</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="categorical.html"><a href="categorical.html#an-exact-goodness-of-fit-test"><i class="fa fa-check"></i><b>19.6.1</b> An Exact Goodness of Fit test</a></li>
<li class="chapter" data-level="19.6.2" data-path="categorical.html"><a href="categorical.html#an-approximate-goodness-of-fit-test"><i class="fa fa-check"></i><b>19.6.2</b> An Approximate Goodness of Fit test</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="categorical.html"><a href="categorical.html#contingency-testing"><i class="fa fa-check"></i><b>19.7</b> Contingency Testing</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="categorical.html"><a href="categorical.html#interpretation-of-contingency-results"><i class="fa fa-check"></i><b>19.7.1</b> Interpretation of Contingency Results</a></li>
<li class="chapter" data-level="19.7.2" data-path="categorical.html"><a href="categorical.html#write-up-4"><i class="fa fa-check"></i><b>19.7.2</b> Write Up</a></li>
<li class="chapter" data-level="19.7.3" data-path="categorical.html"><a href="categorical.html#interpretation-of-chi-square-output"><i class="fa fa-check"></i><b>19.7.3</b> Interpretation of chi-square output</a></li>
<li class="chapter" data-level="19.7.4" data-path="categorical.html"><a href="categorical.html#write-up-5"><i class="fa fa-check"></i><b>19.7.4</b> Write Up</a></li>
<li class="chapter" data-level="19.7.5" data-path="categorical.html"><a href="categorical.html#which-contingency-test-is-best"><i class="fa fa-check"></i><b>19.7.5</b> Which contingency test is best?</a></li>
<li class="chapter" data-level="19.7.6" data-path="categorical.html"><a href="categorical.html#higher-dimension-contingency-analysis"><i class="fa fa-check"></i><b>19.7.6</b> Higher dimension contingency analysis</a></li>
<li class="chapter" data-level="19.7.7" data-path="categorical.html"><a href="categorical.html#other-experimental-designs-involving-categorical-data"><i class="fa fa-check"></i><b>19.7.7</b> Other experimental designs involving categorical data</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="categorical.html"><a href="categorical.html#power-analysis-for-proportion-tests"><i class="fa fa-check"></i><b>19.8</b> Power analysis for proportion tests</a></li>
<li class="chapter" data-level="19.9" data-path="categorical.html"><a href="categorical.html#power-analysis-functions-for-proportion-tests"><i class="fa fa-check"></i><b>19.9</b> Power analysis functions for proportion tests</a></li>
<li class="chapter" data-level="19.10" data-path="categorical.html"><a href="categorical.html#graphing-proportions"><i class="fa fa-check"></i><b>19.10</b> Graphing Proportions</a></li>
<li class="chapter" data-level="19.11" data-path="categorical.html"><a href="categorical.html#key-takeaways"><i class="fa fa-check"></i><b>19.11</b> Key Takeaways</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="chisquare.html"><a href="chisquare.html"><i class="fa fa-check"></i><b>20</b> The Chi-square Distribution</a>
<ul>
<li class="chapter" data-level="20.1" data-path="chisquare.html"><a href="chisquare.html#background"><i class="fa fa-check"></i><b>20.1</b> Background</a></li>
<li class="chapter" data-level="20.2" data-path="chisquare.html"><a href="chisquare.html#dchisq"><i class="fa fa-check"></i><b>20.2</b> dchisq</a></li>
<li class="chapter" data-level="20.3" data-path="chisquare.html"><a href="chisquare.html#pchisq"><i class="fa fa-check"></i><b>20.3</b> pchisq</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="chisquare.html"><a href="chisquare.html#calculating-p-values-from-pchisq"><i class="fa fa-check"></i><b>20.3.1</b> Calculating p-values from pchisq</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="chisquare.html"><a href="chisquare.html#qchisq"><i class="fa fa-check"></i><b>20.4</b> qchisq</a></li>
<li class="chapter" data-level="20.5" data-path="chisquare.html"><a href="chisquare.html#rchisq"><i class="fa fa-check"></i><b>20.5</b> rchisq</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="nonparametrics.html"><a href="nonparametrics.html"><i class="fa fa-check"></i><b>21</b> Nonparametric Statistical Tests</a>
<ul>
<li class="chapter" data-level="21.1" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-sampling-distributions"><i class="fa fa-check"></i><b>21.1</b> Nonparametric sampling distributions</a></li>
<li class="chapter" data-level="21.2" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-discrete-data"><i class="fa fa-check"></i><b>21.2</b> Experiments involving discrete data</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="nonparametrics.html"><a href="nonparametrics.html#ordered-data-1"><i class="fa fa-check"></i><b>21.2.1</b> Ordered data</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-deviant-data"><i class="fa fa-check"></i><b>21.3</b> Experiments involving deviant Data</a></li>
<li class="chapter" data-level="21.4" data-path="nonparametrics.html"><a href="nonparametrics.html#sign-test"><i class="fa fa-check"></i><b>21.4</b> Sign Test</a>
<ul>
<li class="chapter" data-level="21.4.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation"><i class="fa fa-check"></i><b>21.4.1</b> Interpretation</a></li>
<li class="chapter" data-level="21.4.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-6"><i class="fa fa-check"></i><b>21.4.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="21.5" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-one-group"><i class="fa fa-check"></i><b>21.5</b> Wilcoxon Sign Rank Test for One Group</a>
<ul>
<li class="chapter" data-level="21.5.1" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-experimental-designs"><i class="fa fa-check"></i><b>21.5.1</b> Wilcoxon Sign Rank Experimental Designs</a></li>
<li class="chapter" data-level="21.5.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-1"><i class="fa fa-check"></i><b>21.5.2</b> Interpretation</a></li>
<li class="chapter" data-level="21.5.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-7"><i class="fa fa-check"></i><b>21.5.3</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-mann-whitney-rank-sum-test-for-2-independent-groups"><i class="fa fa-check"></i><b>21.6</b> Wilcoxon Mann Whitney Rank Sum Test for 2 independent groups</a>
<ul>
<li class="chapter" data-level="21.6.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-2"><i class="fa fa-check"></i><b>21.6.1</b> Interpretation</a></li>
<li class="chapter" data-level="21.6.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-8"><i class="fa fa-check"></i><b>21.6.2</b> Write Up</a></li>
<li class="chapter" data-level="21.6.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot"><i class="fa fa-check"></i><b>21.6.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="21.7" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-paired-groups"><i class="fa fa-check"></i><b>21.7</b> Wilcoxon Sign Rank Test for paired groups</a>
<ul>
<li class="chapter" data-level="21.7.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-3"><i class="fa fa-check"></i><b>21.7.1</b> Interpretation</a></li>
<li class="chapter" data-level="21.7.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-9"><i class="fa fa-check"></i><b>21.7.2</b> Write up</a></li>
<li class="chapter" data-level="21.7.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot-1"><i class="fa fa-check"></i><b>21.7.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="21.8" data-path="nonparametrics.html"><a href="nonparametrics.html#kruskal-wallis"><i class="fa fa-check"></i><b>21.8</b> Kruskal-Wallis</a>
<ul>
<li class="chapter" data-level="21.8.1" data-path="nonparametrics.html"><a href="nonparametrics.html#the-experimental-design"><i class="fa fa-check"></i><b>21.8.1</b> The experimental design</a></li>
<li class="chapter" data-level="21.8.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-10"><i class="fa fa-check"></i><b>21.8.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="21.9" data-path="nonparametrics.html"><a href="nonparametrics.html#friedman-test"><i class="fa fa-check"></i><b>21.9</b> Friedman test</a>
<ul>
<li class="chapter" data-level="21.9.1" data-path="nonparametrics.html"><a href="nonparametrics.html#blocked-experimental-design"><i class="fa fa-check"></i><b>21.9.1</b> Blocked experimental design</a></li>
<li class="chapter" data-level="21.9.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-4"><i class="fa fa-check"></i><b>21.9.2</b> Interpretation</a></li>
<li class="chapter" data-level="21.9.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-11"><i class="fa fa-check"></i><b>21.9.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="21.10" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-power-calculations"><i class="fa fa-check"></i><b>21.10</b> Nonparametric power calculations</a>
<ul>
<li class="chapter" data-level="21.10.1" data-path="nonparametrics.html"><a href="nonparametrics.html#how-it-works"><i class="fa fa-check"></i><b>21.10.1</b> How it works</a></li>
<li class="chapter" data-level="21.10.2" data-path="nonparametrics.html"><a href="nonparametrics.html#simulating-response-data"><i class="fa fa-check"></i><b>21.10.2</b> Simulating response data</a></li>
<li class="chapter" data-level="21.10.3" data-path="nonparametrics.html"><a href="nonparametrics.html#an-example"><i class="fa fa-check"></i><b>21.10.3</b> An example</a></li>
<li class="chapter" data-level="21.10.4" data-path="nonparametrics.html"><a href="nonparametrics.html#nonpara.pwr"><i class="fa fa-check"></i><b>21.10.4</b> nonpara.pwr</a></li>
</ul></li>
<li class="chapter" data-level="21.11" data-path="nonparametrics.html"><a href="nonparametrics.html#summary-4"><i class="fa fa-check"></i><b>21.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="signrank.html"><a href="signrank.html"><i class="fa fa-check"></i><b>22</b> Signed Rank Distribution</a>
<ul>
<li class="chapter" data-level="22.1" data-path="signrank.html"><a href="signrank.html#transformation-of-data-into-sign-ranks"><i class="fa fa-check"></i><b>22.1</b> Transformation of data into sign ranks</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="signrank.html"><a href="signrank.html#for-a-one-group-sample"><i class="fa fa-check"></i><b>22.1.1</b> For a one group sample</a></li>
<li class="chapter" data-level="22.1.2" data-path="signrank.html"><a href="signrank.html#for-a-paired-sample"><i class="fa fa-check"></i><b>22.1.2</b> For a paired sample</a></li>
<li class="chapter" data-level="22.1.3" data-path="signrank.html"><a href="signrank.html#the-sign-rank-test-statistic-in-r"><i class="fa fa-check"></i><b>22.1.3</b> The sign rank test statistic in R</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="signrank.html"><a href="signrank.html#rs-four-sign-rank-distribution-functions"><i class="fa fa-check"></i><b>22.2</b> R’s Four Sign Rank Distribution Functions</a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="signrank.html"><a href="signrank.html#dsignrank"><i class="fa fa-check"></i><b>22.2.1</b> dsignrank</a></li>
<li class="chapter" data-level="22.2.2" data-path="signrank.html"><a href="signrank.html#psignrank"><i class="fa fa-check"></i><b>22.2.2</b> psignrank</a></li>
<li class="chapter" data-level="22.2.3" data-path="signrank.html"><a href="signrank.html#qsignrank"><i class="fa fa-check"></i><b>22.2.3</b> qsignrank</a></li>
<li class="chapter" data-level="22.2.4" data-path="signrank.html"><a href="signrank.html#rsignrank"><i class="fa fa-check"></i><b>22.2.4</b> rsignrank</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="ranksum.html"><a href="ranksum.html"><i class="fa fa-check"></i><b>23</b> Rank Sum Distribution</a>
<ul>
<li class="chapter" data-level="23.0.1" data-path="ranksum.html"><a href="ranksum.html#transformation-of-data-into-rank-summs"><i class="fa fa-check"></i><b>23.0.1</b> Transformation of data into rank summs</a></li>
<li class="chapter" data-level="23.0.2" data-path="ranksum.html"><a href="ranksum.html#the-sign-rank-test-statistic-in-r-1"><i class="fa fa-check"></i><b>23.0.2</b> The sign rank test statistic in R</a></li>
<li class="chapter" data-level="23.1" data-path="ranksum.html"><a href="ranksum.html#rs-four-sign-rank-distribution-functions-1"><i class="fa fa-check"></i><b>23.1</b> R’s Four Sign Rank Distribution Functions</a>
<ul>
<li class="chapter" data-level="23.1.1" data-path="ranksum.html"><a href="ranksum.html#dwilcox"><i class="fa fa-check"></i><b>23.1.1</b> dwilcox</a></li>
<li class="chapter" data-level="23.1.2" data-path="ranksum.html"><a href="ranksum.html#pwilcox"><i class="fa fa-check"></i><b>23.1.2</b> pwilcox</a></li>
<li class="chapter" data-level="23.1.3" data-path="ranksum.html"><a href="ranksum.html#qwilcox"><i class="fa fa-check"></i><b>23.1.3</b> qwilcox</a></li>
<li class="chapter" data-level="23.1.4" data-path="ranksum.html"><a href="ranksum.html#rwilcox"><i class="fa fa-check"></i><b>23.1.4</b> rwilcox</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="ttests.html"><a href="ttests.html"><i class="fa fa-check"></i><b>24</b> The t-tests</a>
<ul>
<li class="chapter" data-level="24.1" data-path="ttests.html"><a href="ttests.html#assumptions-for-t-tests"><i class="fa fa-check"></i><b>24.1</b> Assumptions for t-tests</a>
<ul>
<li class="chapter" data-level="24.1.1" data-path="ttests.html"><a href="ttests.html#strong-assumptions-t-tests-are-invalid-if-these-not-met"><i class="fa fa-check"></i><b>24.1.1</b> Strong assumptions, t-tests are invalid if these not met:</a></li>
<li class="chapter" data-level="24.1.2" data-path="ttests.html"><a href="ttests.html#weak-assumptions-t-tests-are-less-robust-when-these-are-not-met"><i class="fa fa-check"></i><b>24.1.2</b> Weak assumptions: t-tests are less robust when these are not met:</a></li>
<li class="chapter" data-level="24.1.3" data-path="ttests.html"><a href="ttests.html#on-random-residual-error"><i class="fa fa-check"></i><b>24.1.3</b> On random residual error</a></li>
<li class="chapter" data-level="24.1.4" data-path="ttests.html"><a href="ttests.html#on-the-normality-assumption"><i class="fa fa-check"></i><b>24.1.4</b> On the normality assumption</a></li>
</ul></li>
<li class="chapter" data-level="24.2" data-path="ttests.html"><a href="ttests.html#the-t-statistic"><i class="fa fa-check"></i><b>24.2</b> The t Statistic</a></li>
<li class="chapter" data-level="24.3" data-path="ttests.html"><a href="ttests.html#one-sample-t-tests"><i class="fa fa-check"></i><b>24.3</b> One sample t tests</a>
<ul>
<li class="chapter" data-level="24.3.1" data-path="ttests.html"><a href="ttests.html#example"><i class="fa fa-check"></i><b>24.3.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="24.4" data-path="ttests.html"><a href="ttests.html#unpaired-t-tests"><i class="fa fa-check"></i><b>24.4</b> Unpaired t tests</a>
<ul>
<li class="chapter" data-level="24.4.1" data-path="ttests.html"><a href="ttests.html#students-t-test"><i class="fa fa-check"></i><b>24.4.1</b> Student’s t-test</a></li>
<li class="chapter" data-level="24.4.2" data-path="ttests.html"><a href="ttests.html#welchs-t-test"><i class="fa fa-check"></i><b>24.4.2</b> Welch’s t-test</a></li>
<li class="chapter" data-level="24.4.3" data-path="ttests.html"><a href="ttests.html#example-1"><i class="fa fa-check"></i><b>24.4.3</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="24.5" data-path="ttests.html"><a href="ttests.html#paired-t-tests"><i class="fa fa-check"></i><b>24.5</b> Paired t tests</a>
<ul>
<li class="chapter" data-level="24.5.1" data-path="ttests.html"><a href="ttests.html#example-2"><i class="fa fa-check"></i><b>24.5.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="24.6" data-path="ttests.html"><a href="ttests.html#t-test-hypotheses"><i class="fa fa-check"></i><b>24.6</b> t Test Hypotheses</a></li>
<li class="chapter" data-level="24.7" data-path="ttests.html"><a href="ttests.html#one-sided-or-two-sided"><i class="fa fa-check"></i><b>24.7</b> One-sided or two-sided??</a>
<ul>
<li class="chapter" data-level="24.7.1" data-path="ttests.html"><a href="ttests.html#one-sample-hypotheses"><i class="fa fa-check"></i><b>24.7.1</b> One sample hypotheses</a></li>
<li class="chapter" data-level="24.7.2" data-path="ttests.html"><a href="ttests.html#unpaired-hypotheses"><i class="fa fa-check"></i><b>24.7.2</b> Unpaired hypotheses</a></li>
<li class="chapter" data-level="24.7.3" data-path="ttests.html"><a href="ttests.html#paired-hypotheses"><i class="fa fa-check"></i><b>24.7.3</b> Paired hypotheses</a></li>
</ul></li>
<li class="chapter" data-level="24.8" data-path="ttests.html"><a href="ttests.html#confidence-intervals-1"><i class="fa fa-check"></i><b>24.8</b> Confidence Intervals</a></li>
<li class="chapter" data-level="24.9" data-path="ttests.html"><a href="ttests.html#reliability-of-small-samples"><i class="fa fa-check"></i><b>24.9</b> Reliability of small samples</a>
<ul>
<li class="chapter" data-level="24.9.1" data-path="ttests.html"><a href="ttests.html#reporting-sd-or-sem"><i class="fa fa-check"></i><b>24.9.1</b> Reporting SD or SEM?</a></li>
</ul></li>
<li class="chapter" data-level="24.10" data-path="ttests.html"><a href="ttests.html#running-the-analysis-in-r"><i class="fa fa-check"></i><b>24.10</b> Running the analysis in R</a>
<ul>
<li class="chapter" data-level="24.10.1" data-path="ttests.html"><a href="ttests.html#one-sample-t-test"><i class="fa fa-check"></i><b>24.10.1</b> One sample t test</a></li>
<li class="chapter" data-level="24.10.2" data-path="ttests.html"><a href="ttests.html#unpaired-t-test"><i class="fa fa-check"></i><b>24.10.2</b> Unpaired t test</a></li>
<li class="chapter" data-level="24.10.3" data-path="ttests.html"><a href="ttests.html#paired-t-test"><i class="fa fa-check"></i><b>24.10.3</b> Paired t Test</a></li>
</ul></li>
<li class="chapter" data-level="24.11" data-path="ttests.html"><a href="ttests.html#plotting-t-tests"><i class="fa fa-check"></i><b>24.11</b> Plotting t Tests</a>
<ul>
<li class="chapter" data-level="24.11.1" data-path="ttests.html"><a href="ttests.html#one-sample"><i class="fa fa-check"></i><b>24.11.1</b> One-sample</a></li>
<li class="chapter" data-level="24.11.2" data-path="ttests.html"><a href="ttests.html#unpaired"><i class="fa fa-check"></i><b>24.11.2</b> Unpaired</a></li>
<li class="chapter" data-level="24.11.3" data-path="ttests.html"><a href="ttests.html#paired"><i class="fa fa-check"></i><b>24.11.3</b> Paired</a></li>
</ul></li>
<li class="chapter" data-level="24.12" data-path="ttests.html"><a href="ttests.html#t-test-power"><i class="fa fa-check"></i><b>24.12</b> t Test Power</a>
<ul>
<li class="chapter" data-level="24.12.1" data-path="ttests.html"><a href="ttests.html#pwr.t.test"><i class="fa fa-check"></i><b>24.12.1</b> pwr.t.test</a></li>
<li class="chapter" data-level="24.12.2" data-path="ttests.html"><a href="ttests.html#t-pwr-a-monte-carlo-method"><i class="fa fa-check"></i><b>24.12.2</b> t-pwr: A Monte Carlo method</a></li>
</ul></li>
<li class="chapter" data-level="24.13" data-path="ttests.html"><a href="ttests.html#summary-5"><i class="fa fa-check"></i><b>24.13</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="ttestmc.html"><a href="ttestmc.html"><i class="fa fa-check"></i><b>25</b> Statistical design of t-tests</a>
<ul>
<li class="chapter" data-level="25.1" data-path="ttestmc.html"><a href="ttestmc.html#about-this-chapter"><i class="fa fa-check"></i><b>25.1</b> About this chapter</a></li>
<li class="chapter" data-level="25.2" data-path="ttestmc.html"><a href="ttestmc.html#overview-of-the-monte-carlo"><i class="fa fa-check"></i><b>25.2</b> Overview of the Monte Carlo</a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="ttestmc.html"><a href="ttestmc.html#data-simulation"><i class="fa fa-check"></i><b>25.2.1</b> Data simulation</a></li>
<li class="chapter" data-level="25.2.2" data-path="ttestmc.html"><a href="ttestmc.html#the-test"><i class="fa fa-check"></i><b>25.2.2</b> The test</a></li>
</ul></li>
<li class="chapter" data-level="25.3" data-path="ttestmc.html"><a href="ttestmc.html#changing-sample-size"><i class="fa fa-check"></i><b>25.3</b> Changing Sample Size</a></li>
<li class="chapter" data-level="25.4" data-path="ttestmc.html"><a href="ttestmc.html#diabetes-drug-scenario"><i class="fa fa-check"></i><b>25.4</b> Diabetes drug scenario</a>
<ul>
<li class="chapter" data-level="25.4.1" data-path="ttestmc.html"><a href="ttestmc.html#what-would-be-a-scientifically-meaningful-response"><i class="fa fa-check"></i><b>25.4.1</b> What would be a scientifically meaningful response?</a></li>
</ul></li>
<li class="chapter" data-level="25.5" data-path="ttestmc.html"><a href="ttestmc.html#one-sample-t-test-monte-carlo"><i class="fa fa-check"></i><b>25.5</b> One sample t-test Monte Carlo</a>
<ul>
<li class="chapter" data-level="25.5.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-2"><i class="fa fa-check"></i><b>25.5.1</b> Step 1</a></li>
<li class="chapter" data-level="25.5.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-2"><i class="fa fa-check"></i><b>25.5.2</b> Step 2</a></li>
<li class="chapter" data-level="25.5.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-2"><i class="fa fa-check"></i><b>25.5.3</b> Step 3</a></li>
<li class="chapter" data-level="25.5.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-2"><i class="fa fa-check"></i><b>25.5.4</b> Step 4</a></li>
<li class="chapter" data-level="25.5.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-2"><i class="fa fa-check"></i><b>25.5.5</b> Step 5</a></li>
<li class="chapter" data-level="25.5.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-2"><i class="fa fa-check"></i><b>25.5.6</b> Step 6</a></li>
<li class="chapter" data-level="25.5.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-1"><i class="fa fa-check"></i><b>25.5.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="25.6" data-path="ttestmc.html"><a href="ttestmc.html#unpaired-t-test-monte-carlo"><i class="fa fa-check"></i><b>25.6</b> Unpaired t-test Monte Carlo</a>
<ul>
<li class="chapter" data-level="25.6.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-3"><i class="fa fa-check"></i><b>25.6.1</b> Step 1</a></li>
<li class="chapter" data-level="25.6.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-3"><i class="fa fa-check"></i><b>25.6.2</b> Step 2</a></li>
<li class="chapter" data-level="25.6.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-3"><i class="fa fa-check"></i><b>25.6.3</b> Step 3</a></li>
<li class="chapter" data-level="25.6.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-3"><i class="fa fa-check"></i><b>25.6.4</b> Step 4</a></li>
<li class="chapter" data-level="25.6.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-3"><i class="fa fa-check"></i><b>25.6.5</b> Step 5</a></li>
<li class="chapter" data-level="25.6.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-3"><i class="fa fa-check"></i><b>25.6.6</b> Step 6</a></li>
<li class="chapter" data-level="25.6.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-2"><i class="fa fa-check"></i><b>25.6.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="25.7" data-path="ttestmc.html"><a href="ttestmc.html#paired-t-test-monte-carlo"><i class="fa fa-check"></i><b>25.7</b> Paired t-test Monte Carlo</a>
<ul>
<li class="chapter" data-level="25.7.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-4"><i class="fa fa-check"></i><b>25.7.1</b> Step 1</a></li>
<li class="chapter" data-level="25.7.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-4"><i class="fa fa-check"></i><b>25.7.2</b> Step 2</a></li>
<li class="chapter" data-level="25.7.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-4"><i class="fa fa-check"></i><b>25.7.3</b> Step 3</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="ttestmc.html"><a href="ttestmc.html#step-4-4"><i class="fa fa-check"></i><b>25.8</b> Step 4</a>
<ul>
<li class="chapter" data-level="25.8.1" data-path="ttestmc.html"><a href="ttestmc.html#step-5-4"><i class="fa fa-check"></i><b>25.8.1</b> Step 5</a></li>
<li class="chapter" data-level="25.8.2" data-path="ttestmc.html"><a href="ttestmc.html#step-6-4"><i class="fa fa-check"></i><b>25.8.2</b> Step 6</a></li>
<li class="chapter" data-level="25.8.3" data-path="ttestmc.html"><a href="ttestmc.html#step-7-3"><i class="fa fa-check"></i><b>25.8.3</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="25.9" data-path="ttestmc.html"><a href="ttestmc.html#comparison-to-pwr-tests"><i class="fa fa-check"></i><b>25.9</b> Comparison to pwr tests</a></li>
<li class="chapter" data-level="25.10" data-path="ttestmc.html"><a href="ttestmc.html#summary-6"><i class="fa fa-check"></i><b>25.10</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="tdist.html"><a href="tdist.html"><i class="fa fa-check"></i><b>26</b> t Distributions</a>
<ul>
<li class="chapter" data-level="26.1" data-path="tdist.html"><a href="tdist.html#dt"><i class="fa fa-check"></i><b>26.1</b> dt</a></li>
<li class="chapter" data-level="26.2" data-path="tdist.html"><a href="tdist.html#pt"><i class="fa fa-check"></i><b>26.2</b> pt</a></li>
<li class="chapter" data-level="26.3" data-path="tdist.html"><a href="tdist.html#qt"><i class="fa fa-check"></i><b>26.3</b> qt</a></li>
<li class="chapter" data-level="26.4" data-path="tdist.html"><a href="tdist.html#rt"><i class="fa fa-check"></i><b>26.4</b> rt</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="simcorrelation.html"><a href="simcorrelation.html"><i class="fa fa-check"></i><b>27</b> Simulating correlated variables</a>
<ul>
<li class="chapter" data-level="27.1" data-path="simcorrelation.html"><a href="simcorrelation.html#estimating-correlation-between-two-variables"><i class="fa fa-check"></i><b>27.1</b> Estimating correlation between two variables</a></li>
<li class="chapter" data-level="27.2" data-path="simcorrelation.html"><a href="simcorrelation.html#simulating-correlated-variables"><i class="fa fa-check"></i><b>27.2</b> Simulating correlated variables</a></li>
<li class="chapter" data-level="27.3" data-path="simcorrelation.html"><a href="simcorrelation.html#monte-carlo-simulation"><i class="fa fa-check"></i><b>27.3</b> Monte Carlo simulation</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="introanova.html"><a href="introanova.html"><i class="fa fa-check"></i><b>28</b> Introduction to ANOVA</a>
<ul>
<li class="chapter" data-level="28.1" data-path="introanova.html"><a href="introanova.html#assumptions"><i class="fa fa-check"></i><b>28.1</b> Assumptions</a></li>
<li class="chapter" data-level="28.2" data-path="introanova.html"><a href="introanova.html#types-of-designs"><i class="fa fa-check"></i><b>28.2</b> Types of designs</a></li>
<li class="chapter" data-level="28.3" data-path="introanova.html"><a href="introanova.html#why-is-anova-so-common"><i class="fa fa-check"></i><b>28.3</b> Why is ANOVA so common?</a>
<ul>
<li class="chapter" data-level="28.3.1" data-path="introanova.html"><a href="introanova.html#first-anova-is-versatile."><i class="fa fa-check"></i><b>28.3.1</b> First, ANOVA is versatile.</a></li>
<li class="chapter" data-level="28.3.2" data-path="introanova.html"><a href="introanova.html#second-anova-is-efficient."><i class="fa fa-check"></i><b>28.3.2</b> Second, ANOVA is efficient.</a></li>
<li class="chapter" data-level="28.3.3" data-path="introanova.html"><a href="introanova.html#third-we-can-test-many-hypotheses-simultaneously-or-a-grand-hypothesis."><i class="fa fa-check"></i><b>28.3.3</b> Third, we can test many hypotheses simultaneously or a grand hypothesis.</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="introanova.html"><a href="introanova.html#jargon-factors-and-levels"><i class="fa fa-check"></i><b>28.4</b> Jargon: Factors and levels</a>
<ul>
<li class="chapter" data-level="28.4.1" data-path="introanova.html"><a href="introanova.html#working-with-factors-in-r"><i class="fa fa-check"></i><b>28.4.1</b> Working with factors in R</a></li>
</ul></li>
<li class="chapter" data-level="28.5" data-path="introanova.html"><a href="introanova.html#anova-models-one--two--and-three-way"><i class="fa fa-check"></i><b>28.5</b> ANOVA models: One-, Two-, and Three-way</a></li>
<li class="chapter" data-level="28.6" data-path="introanova.html"><a href="introanova.html#inference"><i class="fa fa-check"></i><b>28.6</b> Inference</a>
<ul>
<li class="chapter" data-level="28.6.1" data-path="introanova.html"><a href="introanova.html#a-quick-peek-at-posthoc-testing"><i class="fa fa-check"></i><b>28.6.1</b> A quick peek at posthoc testing</a></li>
</ul></li>
<li class="chapter" data-level="28.7" data-path="introanova.html"><a href="introanova.html#anova-calculations"><i class="fa fa-check"></i><b>28.7</b> ANOVA calculations</a>
<ul>
<li class="chapter" data-level="28.7.1" data-path="introanova.html"><a href="introanova.html#sums-of-squares-partitioning"><i class="fa fa-check"></i><b>28.7.1</b> Sums of Squares partitioning</a></li>
<li class="chapter" data-level="28.7.2" data-path="introanova.html"><a href="introanova.html#the-anova-table"><i class="fa fa-check"></i><b>28.7.2</b> The ANOVA table</a></li>
<li class="chapter" data-level="28.7.3" data-path="introanova.html"><a href="introanova.html#post-hoc-group-comparisons"><i class="fa fa-check"></i><b>28.7.3</b> Post-hoc group comparisons</a></li>
</ul></li>
<li class="chapter" data-level="28.8" data-path="introanova.html"><a href="introanova.html#completely-randomized-or-related-measures"><i class="fa fa-check"></i><b>28.8</b> Completely randomized or related measures</a>
<ul>
<li class="chapter" data-level="28.8.1" data-path="introanova.html"><a href="introanova.html#the-problem-of-lost-data-in-related-measures-designs"><i class="fa fa-check"></i><b>28.8.1</b> The problem of lost data in related measures designs</a></li>
</ul></li>
<li class="chapter" data-level="28.9" data-path="introanova.html"><a href="introanova.html#two-way-anova"><i class="fa fa-check"></i><b>28.9</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="28.10" data-path="introanova.html"><a href="introanova.html#other-anova-models"><i class="fa fa-check"></i><b>28.10</b> Other ANOVA models</a>
<ul>
<li class="chapter" data-level="28.10.1" data-path="introanova.html"><a href="introanova.html#r-and-anova"><i class="fa fa-check"></i><b>28.10.1</b> R and ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="28.11" data-path="introanova.html"><a href="introanova.html#alternatives-to-anova"><i class="fa fa-check"></i><b>28.11</b> Alternatives to ANOVA</a>
<ul>
<li class="chapter" data-level="28.11.1" data-path="introanova.html"><a href="introanova.html#screw-anova-just-tell-me-how-to-t-test-everything"><i class="fa fa-check"></i><b>28.11.1</b> Screw ANOVA, Just Tell Me How to t-Test Everything</a></li>
</ul></li>
<li class="chapter" data-level="28.12" data-path="introanova.html"><a href="introanova.html#summary-7"><i class="fa fa-check"></i><b>28.12</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="fdistr.html"><a href="fdistr.html"><i class="fa fa-check"></i><b>29</b> The F distribution</a>
<ul>
<li class="chapter" data-level="29.1" data-path="fdistr.html"><a href="fdistr.html#background-1"><i class="fa fa-check"></i><b>29.1</b> Background</a>
<ul>
<li class="chapter" data-level="29.1.1" data-path="fdistr.html"><a href="fdistr.html#sample-variance-and-fs-pdf"><i class="fa fa-check"></i><b>29.1.1</b> Sample Variance and F’s PDF</a></li>
</ul></li>
<li class="chapter" data-level="29.2" data-path="fdistr.html"><a href="fdistr.html#df"><i class="fa fa-check"></i><b>29.2</b> df</a></li>
<li class="chapter" data-level="29.3" data-path="fdistr.html"><a href="fdistr.html#pf"><i class="fa fa-check"></i><b>29.3</b> pf</a></li>
<li class="chapter" data-level="29.4" data-path="fdistr.html"><a href="fdistr.html#qf"><i class="fa fa-check"></i><b>29.4</b> qf</a></li>
<li class="chapter" data-level="29.5" data-path="fdistr.html"><a href="fdistr.html#rf"><i class="fa fa-check"></i><b>29.5</b> rf</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="onewayanova.html"><a href="onewayanova.html"><i class="fa fa-check"></i><b>30</b> One-way ANOVA Completely Randomized</a>
<ul>
<li class="chapter" data-level="30.1" data-path="onewayanova.html"><a href="onewayanova.html#using-ezanova"><i class="fa fa-check"></i><b>30.1</b> Using <code>ezANOVA</code></a></li>
<li class="chapter" data-level="30.2" data-path="onewayanova.html"><a href="onewayanova.html#the-chickwt-data-set"><i class="fa fa-check"></i><b>30.2</b> The chickwt data set</a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="onewayanova.html"><a href="onewayanova.html#inspect-the-data"><i class="fa fa-check"></i><b>30.2.1</b> Inspect the data</a></li>
</ul></li>
<li class="chapter" data-level="30.3" data-path="onewayanova.html"><a href="onewayanova.html#run-the-anova"><i class="fa fa-check"></i><b>30.3</b> Run the ANOVA</a>
<ul>
<li class="chapter" data-level="30.3.1" data-path="onewayanova.html"><a href="onewayanova.html#run-the-chickwts-one-way-anova"><i class="fa fa-check"></i><b>30.3.1</b> Run the chickwts One Way ANOVA</a></li>
<li class="chapter" data-level="30.3.2" data-path="onewayanova.html"><a href="onewayanova.html#interpreting-the-one-way-cr-anova-output"><i class="fa fa-check"></i><b>30.3.2</b> Interpreting the One-Way CR ANOVA Output</a></li>
</ul></li>
<li class="chapter" data-level="30.4" data-path="onewayanova.html"><a href="onewayanova.html#post-hoc-pairwise-comparisons"><i class="fa fa-check"></i><b>30.4</b> Post hoc pairwise comparisons</a>
<ul>
<li class="chapter" data-level="30.4.1" data-path="onewayanova.html"><a href="onewayanova.html#range-tests"><i class="fa fa-check"></i><b>30.4.1</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="30.5" data-path="onewayanova.html"><a href="onewayanova.html#summary-8"><i class="fa fa-check"></i><b>30.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="onewayRM.html"><a href="onewayRM.html"><i class="fa fa-check"></i><b>31</b> One-way ANOVA Related Measures</a>
<ul>
<li class="chapter" data-level="31.1" data-path="onewayRM.html"><a href="onewayRM.html#data-prep"><i class="fa fa-check"></i><b>31.1</b> Data prep</a></li>
<li class="chapter" data-level="31.2" data-path="onewayRM.html"><a href="onewayRM.html#data-visualization"><i class="fa fa-check"></i><b>31.2</b> Data visualization</a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="onewayRM.html"><a href="onewayRM.html#visualize-the-statistical-design"><i class="fa fa-check"></i><b>31.2.1</b> Visualize the statistical design</a></li>
</ul></li>
<li class="chapter" data-level="31.3" data-path="onewayRM.html"><a href="onewayRM.html#the-anova"><i class="fa fa-check"></i><b>31.3</b> The ANOVA</a>
<ul>
<li class="chapter" data-level="31.3.1" data-path="onewayRM.html"><a href="onewayRM.html#running-ezanova"><i class="fa fa-check"></i><b>31.3.1</b> Running ezANOVA</a></li>
</ul></li>
<li class="chapter" data-level="31.4" data-path="onewayRM.html"><a href="onewayRM.html#interpretation-5"><i class="fa fa-check"></i><b>31.4</b> Interpretation</a></li>
<li class="chapter" data-level="31.5" data-path="onewayRM.html"><a href="onewayRM.html#post-hoc-analysis"><i class="fa fa-check"></i><b>31.5</b> Post-hoc analysis</a></li>
<li class="chapter" data-level="31.6" data-path="onewayRM.html"><a href="onewayRM.html#write-up-12"><i class="fa fa-check"></i><b>31.6</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="twowayCR.html"><a href="twowayCR.html"><i class="fa fa-check"></i><b>32</b> Two-way ANOVA Completely Randomized</a>
<ul>
<li class="chapter" data-level="32.1" data-path="twowayCR.html"><a href="twowayCR.html#effect-of-strain-and-diet-on-liver"><i class="fa fa-check"></i><b>32.1</b> Effect of Strain and Diet on Liver</a></li>
<li class="chapter" data-level="32.2" data-path="twowayCR.html"><a href="twowayCR.html#the-test-1"><i class="fa fa-check"></i><b>32.2</b> The test</a></li>
<li class="chapter" data-level="32.3" data-path="twowayCR.html"><a href="twowayCR.html#interpretation-of-2-way-cr-anova-output"><i class="fa fa-check"></i><b>32.3</b> Interpretation of 2 Way CR ANOVA Output</a>
<ul>
<li class="chapter" data-level="32.3.1" data-path="twowayCR.html"><a href="twowayCR.html#levenes"><i class="fa fa-check"></i><b>32.3.1</b> Levene’s</a></li>
<li class="chapter" data-level="32.3.2" data-path="twowayCR.html"><a href="twowayCR.html#anova-table"><i class="fa fa-check"></i><b>32.3.2</b> ANOVA Table</a></li>
</ul></li>
<li class="chapter" data-level="32.4" data-path="twowayCR.html"><a href="twowayCR.html#post-hoc-multiple-comparisons"><i class="fa fa-check"></i><b>32.4</b> Post Hoc Multiple Comparisons</a>
<ul>
<li class="chapter" data-level="32.4.1" data-path="twowayCR.html"><a href="twowayCR.html#pairwise.t.tests"><i class="fa fa-check"></i><b>32.4.1</b> Pairwise.t.tests</a></li>
<li class="chapter" data-level="32.4.2" data-path="twowayCR.html"><a href="twowayCR.html#write-up-13"><i class="fa fa-check"></i><b>32.4.2</b> Write Up</a></li>
<li class="chapter" data-level="32.4.3" data-path="twowayCR.html"><a href="twowayCR.html#range-tests-1"><i class="fa fa-check"></i><b>32.4.3</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="32.5" data-path="twowayCR.html"><a href="twowayCR.html#summary-9"><i class="fa fa-check"></i><b>32.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="twowayRM.html"><a href="twowayRM.html"><i class="fa fa-check"></i><b>33</b> Two-way ANOVA Related Measures</a>
<ul>
<li class="chapter" data-level="33.1" data-path="twowayRM.html"><a href="twowayRM.html#cell-culture"><i class="fa fa-check"></i><b>33.1</b> Cell culture</a></li>
<li class="chapter" data-level="33.2" data-path="twowayRM.html"><a href="twowayRM.html#the-test-2"><i class="fa fa-check"></i><b>33.2</b> The test</a></li>
<li class="chapter" data-level="33.3" data-path="twowayRM.html"><a href="twowayRM.html#interpretation-of-the-output"><i class="fa fa-check"></i><b>33.3</b> Interpretation of the output</a>
<ul>
<li class="chapter" data-level="33.3.1" data-path="twowayRM.html"><a href="twowayRM.html#anova-table-1"><i class="fa fa-check"></i><b>33.3.1</b> ANOVA Table</a></li>
<li class="chapter" data-level="33.3.2" data-path="twowayRM.html"><a href="twowayRM.html#mauchlys-sphericity-test"><i class="fa fa-check"></i><b>33.3.2</b> Mauchly’s Sphericity Test</a></li>
</ul></li>
<li class="chapter" data-level="33.4" data-path="twowayRM.html"><a href="twowayRM.html#post-hoc-multiple-comparisons-1"><i class="fa fa-check"></i><b>33.4</b> Post Hoc multiple comparisons</a></li>
<li class="chapter" data-level="33.5" data-path="twowayRM.html"><a href="twowayRM.html#write-up-14"><i class="fa fa-check"></i><b>33.5</b> Write Up</a></li>
<li class="chapter" data-level="33.6" data-path="twowayRM.html"><a href="twowayRM.html#summary-10"><i class="fa fa-check"></i><b>33.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="34" data-path="twowaymixed.html"><a href="twowaymixed.html"><i class="fa fa-check"></i><b>34</b> Two-way ANOVA RM/CR</a>
<ul>
<li class="chapter" data-level="34.1" data-path="twowaymixed.html"><a href="twowaymixed.html#chickweight-data-set"><i class="fa fa-check"></i><b>34.1</b> ChickWeight Data set</a></li>
<li class="chapter" data-level="34.2" data-path="twowaymixed.html"><a href="twowaymixed.html#munge-chickweight-data"><i class="fa fa-check"></i><b>34.2</b> Munge ChickWeight data</a></li>
<li class="chapter" data-level="34.3" data-path="twowaymixed.html"><a href="twowaymixed.html#the-test-3"><i class="fa fa-check"></i><b>34.3</b> The test</a></li>
<li class="chapter" data-level="34.4" data-path="twowaymixed.html"><a href="twowaymixed.html#interpreting-the-anova-output"><i class="fa fa-check"></i><b>34.4</b> Interpreting the ANOVA output</a>
<ul>
<li class="chapter" data-level="34.4.1" data-path="twowaymixed.html"><a href="twowaymixed.html#anova-the-anova-table-1"><i class="fa fa-check"></i><b>34.4.1</b> $ANOVA: The ANOVA table</a></li>
<li class="chapter" data-level="34.4.2" data-path="twowaymixed.html"><a href="twowaymixed.html#mauchlys-test-and-corrections"><i class="fa fa-check"></i><b>34.4.2</b> Mauchly’s Test and Corrections</a></li>
</ul></li>
<li class="chapter" data-level="34.5" data-path="twowaymixed.html"><a href="twowaymixed.html#post-hoc"><i class="fa fa-check"></i><b>34.5</b> Post hoc</a>
<ul>
<li class="chapter" data-level="34.5.1" data-path="twowaymixed.html"><a href="twowaymixed.html#pairwise.t.tests-1"><i class="fa fa-check"></i><b>34.5.1</b> Pairwise.t.tests</a></li>
<li class="chapter" data-level="34.5.2" data-path="twowaymixed.html"><a href="twowaymixed.html#tukey-test-a-range-test"><i class="fa fa-check"></i><b>34.5.2</b> Tukey test: A range test</a></li>
<li class="chapter" data-level="34.5.3" data-path="twowaymixed.html"><a href="twowaymixed.html#heres-whats-been-discovered-by-the-tukeyhsd-test"><i class="fa fa-check"></i><b>34.5.3</b> Here’s what’s been discovered by the TukeyHSD test</a></li>
<li class="chapter" data-level="34.5.4" data-path="twowaymixed.html"><a href="twowaymixed.html#why-is-the-tukeyhsd-result-so-different-than-the-pairwise.t.test"><i class="fa fa-check"></i><b>34.5.4</b> Why is the TukeyHSD result so different than the pairwise.t.test??</a></li>
</ul></li>
<li class="chapter" data-level="34.6" data-path="twowaymixed.html"><a href="twowaymixed.html#summary-11"><i class="fa fa-check"></i><b>34.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="35" data-path="posthoc.html"><a href="posthoc.html"><i class="fa fa-check"></i><b>35</b> ANOVA Posthoc Comparisons</a>
<ul>
<li class="chapter" data-level="35.1" data-path="posthoc.html"><a href="posthoc.html#overview-of-options"><i class="fa fa-check"></i><b>35.1</b> Overview of options</a>
<ul>
<li class="chapter" data-level="35.1.1" data-path="posthoc.html"><a href="posthoc.html#pairwise.t.test-with-p-value-adjustments"><i class="fa fa-check"></i><b>35.1.1</b> Pairwise.t.test with p-value adjustments</a></li>
<li class="chapter" data-level="35.1.2" data-path="posthoc.html"><a href="posthoc.html#range-tests-2"><i class="fa fa-check"></i><b>35.1.2</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="35.2" data-path="posthoc.html"><a href="posthoc.html#reporting-the-result-1"><i class="fa fa-check"></i><b>35.2</b> Reporting the result</a></li>
<li class="chapter" data-level="35.3" data-path="posthoc.html"><a href="posthoc.html#summary-12"><i class="fa fa-check"></i><b>35.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="36" data-path="jaxwest2.html"><a href="jaxwest2.html"><i class="fa fa-check"></i><b>36</b> Reproducible Data Munging Mostly with Tidyverse</a>
<ul>
<li class="chapter" data-level="36.1" data-path="jaxwest2.html"><a href="jaxwest2.html#look-at-the-original-data-carefully"><i class="fa fa-check"></i><b>36.1</b> Look at the original data carefully</a></li>
<li class="chapter" data-level="36.2" data-path="jaxwest2.html"><a href="jaxwest2.html#our-goal"><i class="fa fa-check"></i><b>36.2</b> Our goal</a></li>
<li class="chapter" data-level="36.3" data-path="jaxwest2.html"><a href="jaxwest2.html#step-1-read-the-data-into-r"><i class="fa fa-check"></i><b>36.3</b> Step 1: Read the data into R</a></li>
<li class="chapter" data-level="36.4" data-path="jaxwest2.html"><a href="jaxwest2.html#step-2-select-the-variables"><i class="fa fa-check"></i><b>36.4</b> Step 2: Select the variables</a></li>
<li class="chapter" data-level="36.5" data-path="jaxwest2.html"><a href="jaxwest2.html#step-3-trim-the-cases"><i class="fa fa-check"></i><b>36.5</b> Step 3: Trim the cases</a></li>
<li class="chapter" data-level="36.6" data-path="jaxwest2.html"><a href="jaxwest2.html#step-4-we-have-a-variable-problem-to-fix"><i class="fa fa-check"></i><b>36.6</b> Step 4: We have a variable problem to fix</a></li>
<li class="chapter" data-level="36.7" data-path="jaxwest2.html"><a href="jaxwest2.html#step-5-impute-or-delete"><i class="fa fa-check"></i><b>36.7</b> Step 5: Impute or Delete</a></li>
<li class="chapter" data-level="36.8" data-path="jaxwest2.html"><a href="jaxwest2.html#step-6-go-long"><i class="fa fa-check"></i><b>36.8</b> Step 6: Go long</a></li>
<li class="chapter" data-level="36.9" data-path="jaxwest2.html"><a href="jaxwest2.html#step-7-convert-other-variables-to-factor"><i class="fa fa-check"></i><b>36.9</b> Step 7: Convert other variables to factor</a></li>
<li class="chapter" data-level="36.10" data-path="jaxwest2.html"><a href="jaxwest2.html#step-8-plot"><i class="fa fa-check"></i><b>36.10</b> Step 8: Plot</a></li>
<li class="chapter" data-level="36.11" data-path="jaxwest2.html"><a href="jaxwest2.html#step-9-run-the-anova"><i class="fa fa-check"></i><b>36.11</b> Step 9: Run the ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="37" data-path="anovamc.html"><a href="anovamc.html"><i class="fa fa-check"></i><b>37</b> ANOVA power using Monte Carlo</a>
<ul>
<li class="chapter" data-level="37.1" data-path="anovamc.html"><a href="anovamc.html#alternatives-to-monte-carlo"><i class="fa fa-check"></i><b>37.1</b> Alternatives to Monte Carlo</a></li>
<li class="chapter" data-level="37.2" data-path="anovamc.html"><a href="anovamc.html#what-is-monte-carlo"><i class="fa fa-check"></i><b>37.2</b> What is Monte Carlo</a></li>
<li class="chapter" data-level="37.3" data-path="anovamc.html"><a href="anovamc.html#one-way-completely-randomized-anova-monte-carlo"><i class="fa fa-check"></i><b>37.3</b> One-way completely randomized ANOVA Monte Carlo</a>
<ul>
<li class="chapter" data-level="37.3.1" data-path="anovamc.html"><a href="anovamc.html#directions"><i class="fa fa-check"></i><b>37.3.1</b> Directions</a></li>
<li class="chapter" data-level="37.3.2" data-path="anovamc.html"><a href="anovamc.html#step-1-create-initial-values"><i class="fa fa-check"></i><b>37.3.2</b> Step 1: Create initial values</a></li>
<li class="chapter" data-level="37.3.3" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-random-sample"><i class="fa fa-check"></i><b>37.3.3</b> Step 2: Visualize one random sample</a></li>
<li class="chapter" data-level="37.3.4" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator"><i class="fa fa-check"></i><b>37.3.4</b> Step 3: Run The Power Simulator</a></li>
<li class="chapter" data-level="37.3.5" data-path="anovamc.html"><a href="anovamc.html#step-4-optimize-for-suitable-power"><i class="fa fa-check"></i><b>37.3.5</b> Step 4: Optimize for suitable power</a></li>
<li class="chapter" data-level="37.3.6" data-path="anovamc.html"><a href="anovamc.html#notes-and-considerations"><i class="fa fa-check"></i><b>37.3.6</b> Notes And Considerations</a></li>
</ul></li>
<li class="chapter" data-level="37.4" data-path="anovamc.html"><a href="anovamc.html#one-way-related-measures-anova-monte-carlo"><i class="fa fa-check"></i><b>37.4</b> One-way related measures ANOVA Monte Carlo</a></li>
<li class="chapter" data-level="37.5" data-path="anovamc.html"><a href="anovamc.html#directions-1"><i class="fa fa-check"></i><b>37.5</b> Directions</a>
<ul>
<li class="chapter" data-level="37.5.1" data-path="anovamc.html"><a href="anovamc.html#step-1-initial-values"><i class="fa fa-check"></i><b>37.5.1</b> Step 1: Initial values</a></li>
<li class="chapter" data-level="37.5.2" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-sample"><i class="fa fa-check"></i><b>37.5.2</b> Step 2: Visualize one sample</a></li>
<li class="chapter" data-level="37.5.3" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator-1"><i class="fa fa-check"></i><b>37.5.3</b> Step 3: Run the power simulator</a></li>
<li class="chapter" data-level="37.5.4" data-path="anovamc.html"><a href="anovamc.html#step-4-should-anything-be-changed"><i class="fa fa-check"></i><b>37.5.4</b> Step 4: Should anything be changed?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="manova.html"><a href="manova.html"><i class="fa fa-check"></i><b>38</b> Multivariate Analysis of Variance (MANOVA)</a>
<ul>
<li class="chapter" data-level="38.1" data-path="manova.html"><a href="manova.html#what-does-manova-test"><i class="fa fa-check"></i><b>38.1</b> What does MANOVA test?</a></li>
<li class="chapter" data-level="38.2" data-path="manova.html"><a href="manova.html#types-of-experiments"><i class="fa fa-check"></i><b>38.2</b> Types of experiments</a></li>
<li class="chapter" data-level="38.3" data-path="manova.html"><a href="manova.html#manova-is-not-many-anovas"><i class="fa fa-check"></i><b>38.3</b> MANOVA is not many ANOVAs</a>
<ul>
<li class="chapter" data-level="38.3.1" data-path="manova.html"><a href="manova.html#why-manova"><i class="fa fa-check"></i><b>38.3.1</b> Why MANOVA?</a></li>
</ul></li>
<li class="chapter" data-level="38.4" data-path="manova.html"><a href="manova.html#assumptions-1"><i class="fa fa-check"></i><b>38.4</b> Assumptions</a>
<ul>
<li class="chapter" data-level="38.4.1" data-path="manova.html"><a href="manova.html#multicolinearity"><i class="fa fa-check"></i><b>38.4.1</b> Multicolinearity</a></li>
</ul></li>
<li class="chapter" data-level="38.5" data-path="manova.html"><a href="manova.html#calculating-variation"><i class="fa fa-check"></i><b>38.5</b> Calculating variation</a>
<ul>
<li class="chapter" data-level="38.5.1" data-path="manova.html"><a href="manova.html#eigenvectors-and-eigenvalues"><i class="fa fa-check"></i><b>38.5.1</b> Eigenvectors and eigenvalues</a></li>
<li class="chapter" data-level="38.5.2" data-path="manova.html"><a href="manova.html#manova-test-statistics"><i class="fa fa-check"></i><b>38.5.2</b> MANOVA test statistics</a></li>
<li class="chapter" data-level="38.5.3" data-path="manova.html"><a href="manova.html#which-manova-test-statistic-is-best"><i class="fa fa-check"></i><b>38.5.3</b> Which MANOVA test statistic is best?</a></li>
</ul></li>
<li class="chapter" data-level="38.6" data-path="manova.html"><a href="manova.html#drug-clearance-example"><i class="fa fa-check"></i><b>38.6</b> Drug clearance example</a>
<ul>
<li class="chapter" data-level="38.6.1" data-path="manova.html"><a href="manova.html#rationale-for-manova"><i class="fa fa-check"></i><b>38.6.1</b> Rationale for MANOVA</a></li>
<li class="chapter" data-level="38.6.2" data-path="manova.html"><a href="manova.html#data-structure"><i class="fa fa-check"></i><b>38.6.2</b> Data structure</a></li>
<li class="chapter" data-level="38.6.3" data-path="manova.html"><a href="manova.html#manova-procedure"><i class="fa fa-check"></i><b>38.6.3</b> MANOVA procedure</a></li>
</ul></li>
<li class="chapter" data-level="38.7" data-path="manova.html"><a href="manova.html#planning-manova-experiments"><i class="fa fa-check"></i><b>38.7</b> Planning MANOVA experiments</a>
<ul>
<li class="chapter" data-level="38.7.1" data-path="manova.html"><a href="manova.html#manova-monte-carlo-power-analysis"><i class="fa fa-check"></i><b>38.7.1</b> MANOVA Monte Carlo Power Analysis</a></li>
<li class="chapter" data-level="38.7.2" data-path="manova.html"><a href="manova.html#an-manova-monte-carlo-power-function"><i class="fa fa-check"></i><b>38.7.2</b> An MANOVA Monte Carlo Power function</a></li>
</ul></li>
<li class="chapter" data-level="38.8" data-path="manova.html"><a href="manova.html#summary-13"><i class="fa fa-check"></i><b>38.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="39" data-path="correl.html"><a href="correl.html"><i class="fa fa-check"></i><b>39</b> Correlation</a>
<ul>
<li class="chapter" data-level="39.1" data-path="correl.html"><a href="correl.html#correlation-causation"><i class="fa fa-check"></i><b>39.1</b> Correlation != Causation</a></li>
<li class="chapter" data-level="39.2" data-path="correl.html"><a href="correl.html#correlation-in-multivariate-outcomes-and-paired-designs"><i class="fa fa-check"></i><b>39.2</b> Correlation in Multivariate Outcomes and Paired Designs</a></li>
<li class="chapter" data-level="39.3" data-path="correl.html"><a href="correl.html#correlation-coefficients"><i class="fa fa-check"></i><b>39.3</b> Correlation coefficients</a>
<ul>
<li class="chapter" data-level="39.3.1" data-path="correl.html"><a href="correl.html#pearsons-correlation-coefficient"><i class="fa fa-check"></i><b>39.3.1</b> Pearson’s correlation coefficient</a></li>
<li class="chapter" data-level="39.3.2" data-path="correl.html"><a href="correl.html#spearmans-rank-correlation"><i class="fa fa-check"></i><b>39.3.2</b> Spearman’s rank correlation</a></li>
<li class="chapter" data-level="39.3.3" data-path="correl.html"><a href="correl.html#kendalls-tau"><i class="fa fa-check"></i><b>39.3.3</b> Kendall’s tau</a></li>
<li class="chapter" data-level="39.3.4" data-path="correl.html"><a href="correl.html#which-correlation-method-to-use"><i class="fa fa-check"></i><b>39.3.4</b> Which correlation method to use?</a></li>
<li class="chapter" data-level="39.3.5" data-path="correl.html"><a href="correl.html#r-correlation-analysis-functions"><i class="fa fa-check"></i><b>39.3.5</b> R correlation analysis functions</a></li>
<li class="chapter" data-level="39.3.6" data-path="correl.html"><a href="correl.html#plot-the-correlations"><i class="fa fa-check"></i><b>39.3.6</b> Plot the correlations</a></li>
<li class="chapter" data-level="39.3.7" data-path="correl.html"><a href="correl.html#calculate-a-correlation-coefficient-and-posthoc-test"><i class="fa fa-check"></i><b>39.3.7</b> Calculate a correlation coefficient and posthoc test</a></li>
<li class="chapter" data-level="39.3.8" data-path="correl.html"><a href="correl.html#interpretation-of-correlation-output"><i class="fa fa-check"></i><b>39.3.8</b> Interpretation of correlation output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="40" data-path="regress.html"><a href="regress.html"><i class="fa fa-check"></i><b>40</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="40.1" data-path="regress.html"><a href="regress.html#the-linear-regression-model"><i class="fa fa-check"></i><b>40.1</b> The linear regression model</a></li>
<li class="chapter" data-level="40.2" data-path="regress.html"><a href="regress.html#least-squares-fitting"><i class="fa fa-check"></i><b>40.2</b> Least squares fitting</a></li>
<li class="chapter" data-level="40.3" data-path="regress.html"><a href="regress.html#the-practical-importance-of-linear-model-parameters"><i class="fa fa-check"></i><b>40.3</b> The practical importance of linear model parameters</a></li>
<li class="chapter" data-level="40.4" data-path="regress.html"><a href="regress.html#linear-model-standard-errors"><i class="fa fa-check"></i><b>40.4</b> Linear model standard errors</a></li>
<li class="chapter" data-level="40.5" data-path="regress.html"><a href="regress.html#linear-regression-in-r"><i class="fa fa-check"></i><b>40.5</b> Linear regression in R</a></li>
<li class="chapter" data-level="40.6" data-path="regress.html"><a href="regress.html#intepretation"><i class="fa fa-check"></i><b>40.6</b> Intepretation</a>
<ul>
<li class="chapter" data-level="40.6.1" data-path="regress.html"><a href="regress.html#residuals"><i class="fa fa-check"></i><b>40.6.1</b> Residuals</a></li>
<li class="chapter" data-level="40.6.2" data-path="regress.html"><a href="regress.html#coefficients"><i class="fa fa-check"></i><b>40.6.2</b> Coefficients</a></li>
<li class="chapter" data-level="40.6.3" data-path="regress.html"><a href="regress.html#degrees-of-freedom-1"><i class="fa fa-check"></i><b>40.6.3</b> Degrees of freedom</a></li>
<li class="chapter" data-level="40.6.4" data-path="regress.html"><a href="regress.html#r-squared"><i class="fa fa-check"></i><b>40.6.4</b> R-squared</a></li>
<li class="chapter" data-level="40.6.5" data-path="regress.html"><a href="regress.html#f-statistic"><i class="fa fa-check"></i><b>40.6.5</b> F-statistic</a></li>
<li class="chapter" data-level="40.6.6" data-path="regress.html"><a href="regress.html#plotting-regression-results"><i class="fa fa-check"></i><b>40.6.6</b> Plotting regression results</a></li>
<li class="chapter" data-level="40.6.7" data-path="regress.html"><a href="regress.html#visualizing-residuals"><i class="fa fa-check"></i><b>40.6.7</b> Visualizing residuals</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="41" data-path="nonlinearintro.html"><a href="nonlinearintro.html"><i class="fa fa-check"></i><b>41</b> Non-linear regression introduction</a>
<ul>
<li class="chapter" data-level="41.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#uses-for-nonlinear-regression"><i class="fa fa-check"></i><b>41.1</b> Uses for nonlinear regression</a></li>
<li class="chapter" data-level="41.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#example-of-the-process"><i class="fa fa-check"></i><b>41.2</b> Example of the process</a>
<ul>
<li class="chapter" data-level="41.2.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#it-starts-with-nonlinear-data"><i class="fa fa-check"></i><b>41.2.1</b> It starts with nonlinear data</a></li>
<li class="chapter" data-level="41.2.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#which-nonlinear-model-describes-that-data"><i class="fa fa-check"></i><b>41.2.2</b> Which nonlinear model describes that data?</a></li>
<li class="chapter" data-level="41.2.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#fitting-a-model-to-data"><i class="fa fa-check"></i><b>41.2.3</b> Fitting a model to data</a></li>
<li class="chapter" data-level="41.2.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#interpreting-regression-model-output"><i class="fa fa-check"></i><b>41.2.4</b> Interpreting regression model output</a></li>
<li class="chapter" data-level="41.2.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#check-the-fit-graphically"><i class="fa fa-check"></i><b>41.2.5</b> Check the fit graphically</a></li>
<li class="chapter" data-level="41.2.6" data-path="nonlinearintro.html"><a href="nonlinearintro.html#convergence"><i class="fa fa-check"></i><b>41.2.6</b> Convergence?</a></li>
<li class="chapter" data-level="41.2.7" data-path="nonlinearintro.html"><a href="nonlinearintro.html#convergence-is-not-idiot-proof"><i class="fa fa-check"></i><b>41.2.7</b> Convergence is not idiot proof</a></li>
</ul></li>
<li class="chapter" data-level="41.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-in-general"><i class="fa fa-check"></i><b>41.3</b> Nonlinear models in general</a></li>
<li class="chapter" data-level="41.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nlfitr"><i class="fa fa-check"></i><b>41.4</b> nlfitr</a></li>
<li class="chapter" data-level="41.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-regression-functions-in-r"><i class="fa fa-check"></i><b>41.5</b> Nonlinear regression functions in R</a>
<ul>
<li class="chapter" data-level="41.5.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#the-fitting-algorithms"><i class="fa fa-check"></i><b>41.5.1</b> The fitting algorithms</a></li>
<li class="chapter" data-level="41.5.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#formula"><i class="fa fa-check"></i><b>41.5.2</b> formula</a></li>
<li class="chapter" data-level="41.5.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#data-1"><i class="fa fa-check"></i><b>41.5.3</b> data</a></li>
<li class="chapter" data-level="41.5.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#start"><i class="fa fa-check"></i><b>41.5.4</b> start</a></li>
<li class="chapter" data-level="41.5.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#weighting"><i class="fa fa-check"></i><b>41.5.5</b> weighting</a></li>
</ul></li>
<li class="chapter" data-level="41.6" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-and-parameters"><i class="fa fa-check"></i><b>41.6</b> Nonlinear models and parameters</a>
<ul>
<li class="chapter" data-level="41.6.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#hyperbolic-stimulus-response-functions"><i class="fa fa-check"></i><b>41.6.1</b> Hyperbolic stimulus response functions</a></li>
<li class="chapter" data-level="41.6.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#visualizing-nonlinear-data-and-log-scaling"><i class="fa fa-check"></i><b>41.6.2</b> Visualizing nonlinear data and log scaling</a></li>
<li class="chapter" data-level="41.6.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#selecting-the-right-model"><i class="fa fa-check"></i><b>41.6.3</b> Selecting the right model</a></li>
<li class="chapter" data-level="41.6.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#from-models-to-formulas"><i class="fa fa-check"></i><b>41.6.4</b> From models to formulas</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="42" data-path="nestedregress.html"><a href="nestedregress.html"><i class="fa fa-check"></i><b>42</b> Nested-model nonlinear regression</a>
<ul>
<li class="chapter" data-level="42.1" data-path="nestedregress.html"><a href="nestedregress.html#read-and-plot-the-data"><i class="fa fa-check"></i><b>42.1</b> Read and plot the data</a></li>
<li class="chapter" data-level="42.2" data-path="nestedregress.html"><a href="nestedregress.html#which-model-is-a-better-fit"><i class="fa fa-check"></i><b>42.2</b> Which model is a better fit?</a>
<ul>
<li class="chapter" data-level="42.2.1" data-path="nestedregress.html"><a href="nestedregress.html#nonlinear-models"><i class="fa fa-check"></i><b>42.2.1</b> Nonlinear models</a></li>
<li class="chapter" data-level="42.2.2" data-path="nestedregress.html"><a href="nestedregress.html#the-nlfitr-alternative"><i class="fa fa-check"></i><b>42.2.2</b> The nlfitr alternative</a></li>
<li class="chapter" data-level="42.2.3" data-path="nestedregress.html"><a href="nestedregress.html#troubleshooting-the-regression"><i class="fa fa-check"></i><b>42.2.3</b> Troubleshooting the regression</a></li>
<li class="chapter" data-level="42.2.4" data-path="nestedregress.html"><a href="nestedregress.html#interpreting-the-parameters"><i class="fa fa-check"></i><b>42.2.4</b> Interpreting the parameters</a></li>
<li class="chapter" data-level="42.2.5" data-path="nestedregress.html"><a href="nestedregress.html#residual-plots-to-compare-fits"><i class="fa fa-check"></i><b>42.2.5</b> Residual plots to compare fits</a></li>
<li class="chapter" data-level="42.2.6" data-path="nestedregress.html"><a href="nestedregress.html#compare-aic"><i class="fa fa-check"></i><b>42.2.6</b> Compare AIC</a></li>
<li class="chapter" data-level="42.2.7" data-path="nestedregress.html"><a href="nestedregress.html#other-ways-to-compare-models"><i class="fa fa-check"></i><b>42.2.7</b> Other ways to compare models</a></li>
<li class="chapter" data-level="42.2.8" data-path="nestedregress.html"><a href="nestedregress.html#interpretation-of-this-one-replicate"><i class="fa fa-check"></i><b>42.2.8</b> Interpretation of this one replicate</a></li>
<li class="chapter" data-level="42.2.9" data-path="nestedregress.html"><a href="nestedregress.html#alternate-analysis"><i class="fa fa-check"></i><b>42.2.9</b> Alternate analysis</a></li>
</ul></li>
<li class="chapter" data-level="42.3" data-path="nestedregress.html"><a href="nestedregress.html#summary-14"><i class="fa fa-check"></i><b>42.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="43" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html"><i class="fa fa-check"></i><b>43</b> Nonlinear regression of independent replicates</a>
<ul>
<li class="chapter" data-level="43.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-problem"><i class="fa fa-check"></i><b>43.1</b> The problem</a></li>
<li class="chapter" data-level="43.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-dataset"><i class="fa fa-check"></i><b>43.2</b> The dataset</a></li>
<li class="chapter" data-level="43.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#munging-for-regression-analysis"><i class="fa fa-check"></i><b>43.3</b> Munging for regression analysis</a>
<ul>
<li class="chapter" data-level="43.3.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#average-the-technical-replicates"><i class="fa fa-check"></i><b>43.3.1</b> Average the technical replicates</a></li>
<li class="chapter" data-level="43.3.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#create-one-table"><i class="fa fa-check"></i><b>43.3.2</b> Create one table</a></li>
<li class="chapter" data-level="43.3.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#plot-the-data"><i class="fa fa-check"></i><b>43.3.3</b> Plot the data</a></li>
<li class="chapter" data-level="43.3.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#run-the-regressions"><i class="fa fa-check"></i><b>43.3.4</b> Run the regressions</a></li>
<li class="chapter" data-level="43.3.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#clean-up-regression-results"><i class="fa fa-check"></i><b>43.3.5</b> Clean up regression results</a></li>
</ul></li>
<li class="chapter" data-level="43.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#t-test-on-half-lives"><i class="fa fa-check"></i><b>43.4</b> T-test on half-lives</a></li>
<li class="chapter" data-level="43.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#conclusion"><i class="fa fa-check"></i><b>43.5</b> Conclusion</a></li>
<li class="chapter" data-level="43.6" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#summary-figure"><i class="fa fa-check"></i><b>43.6</b> Summary figure</a></li>
</ul></li>
<li class="chapter" data-level="44" data-path="multregress.html"><a href="multregress.html"><i class="fa fa-check"></i><b>44</b> Multiple regression</a>
<ul>
<li class="chapter" data-level="44.1" data-path="multregress.html"><a href="multregress.html#the-linear-regression-model-1"><i class="fa fa-check"></i><b>44.1</b> The linear regression model</a></li>
<li class="chapter" data-level="44.2" data-path="multregress.html"><a href="multregress.html#multiple-regression-models"><i class="fa fa-check"></i><b>44.2</b> Multiple regression models</a></li>
<li class="chapter" data-level="44.3" data-path="multregress.html"><a href="multregress.html#experimental-multiple-regression"><i class="fa fa-check"></i><b>44.3</b> Experimental multiple regression</a>
<ul>
<li class="chapter" data-level="44.3.1" data-path="multregress.html"><a href="multregress.html#a-one-factor-experiment-with-3-groups"><i class="fa fa-check"></i><b>44.3.1</b> A one factor experiment with 3 groups</a></li>
<li class="chapter" data-level="44.3.2" data-path="multregress.html"><a href="multregress.html#a-two-factor-experiment-with-6-groups"><i class="fa fa-check"></i><b>44.3.2</b> A two factor experiment with 6 groups</a></li>
</ul></li>
<li class="chapter" data-level="44.4" data-path="multregress.html"><a href="multregress.html#multiple-linear-mixed-models"><i class="fa fa-check"></i><b>44.4</b> Multiple linear mixed models</a></li>
<li class="chapter" data-level="44.5" data-path="multregress.html"><a href="multregress.html#multiple-regression-of-observational-data"><i class="fa fa-check"></i><b>44.5</b> Multiple regression of observational data</a></li>
</ul></li>
<li class="chapter" data-level="45" data-path="logregress.html"><a href="logregress.html"><i class="fa fa-check"></i><b>45</b> Logistic regression</a>
<ul>
<li class="chapter" data-level="45.1" data-path="logregress.html"><a href="logregress.html#uses-of-logistic-regression"><i class="fa fa-check"></i><b>45.1</b> Uses of logistic regression</a>
<ul>
<li class="chapter" data-level="45.1.1" data-path="logregress.html"><a href="logregress.html#sidebar-doing-logistic-regression-is-machine-learning"><i class="fa fa-check"></i><b>45.1.1</b> Sidebar: Doing logistic regression is machine learning</a></li>
</ul></li>
<li class="chapter" data-level="45.2" data-path="logregress.html"><a href="logregress.html#derivation-of-the-logistic-regression-model"><i class="fa fa-check"></i><b>45.2</b> Derivation of the logistic regression model</a>
<ul>
<li class="chapter" data-level="45.2.1" data-path="logregress.html"><a href="logregress.html#relationship-of-logit-to-odds-to-the-model-coefficients-and-probability"><i class="fa fa-check"></i><b>45.2.1</b> Relationship of logit to odds to the model coefficients and probability</a></li>
<li class="chapter" data-level="45.2.2" data-path="logregress.html"><a href="logregress.html#additional-types-of-logistic-regression-models"><i class="fa fa-check"></i><b>45.2.2</b> Additional types of logistic regression models</a></li>
</ul></li>
<li class="chapter" data-level="45.3" data-path="logregress.html"><a href="logregress.html#stress-and-survival"><i class="fa fa-check"></i><b>45.3</b> Stress and survival</a>
<ul>
<li class="chapter" data-level="45.3.1" data-path="logregress.html"><a href="logregress.html#interpretation-of-output"><i class="fa fa-check"></i><b>45.3.1</b> Interpretation of output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="46" data-path="mixedlogistic.html"><a href="mixedlogistic.html"><i class="fa fa-check"></i><b>46</b> Mixed model logistic regression</a>
<ul>
<li class="chapter" data-level="46.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#mixed-models-fixed-and-random-effects"><i class="fa fa-check"></i><b>46.1</b> Mixed models, fixed and random effects</a>
<ul>
<li class="chapter" data-level="46.1.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#nfat-localization-within-smooth-muscle-cells"><i class="fa fa-check"></i><b>46.1.1</b> NFAT localization within smooth muscle cells</a></li>
</ul></li>
<li class="chapter" data-level="46.2" data-path="mixedlogistic.html"><a href="mixedlogistic.html#data-2"><i class="fa fa-check"></i><b>46.2</b> Data</a>
<ul>
<li class="chapter" data-level="46.2.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#inference-2"><i class="fa fa-check"></i><b>46.2.1</b> Inference</a></li>
</ul></li>
<li class="chapter" data-level="46.3" data-path="mixedlogistic.html"><a href="mixedlogistic.html#alternative-analysis"><i class="fa fa-check"></i><b>46.3</b> Alternative analysis</a></li>
</ul></li>
<li class="chapter" data-level="47" data-path="poissonreg.html"><a href="poissonreg.html"><i class="fa fa-check"></i><b>47</b> Poisson regression</a>
<ul>
<li class="chapter" data-level="47.1" data-path="poissonreg.html"><a href="poissonreg.html#why-not-anova"><i class="fa fa-check"></i><b>47.1</b> Why not ANOVA?</a>
<ul>
<li class="chapter" data-level="47.1.1" data-path="poissonreg.html"><a href="poissonreg.html#counting-markers"><i class="fa fa-check"></i><b>47.1.1</b> Counting markers</a></li>
<li class="chapter" data-level="47.1.2" data-path="poissonreg.html"><a href="poissonreg.html#counting-depolarizations"><i class="fa fa-check"></i><b>47.1.2</b> Counting depolarizations</a></li>
<li class="chapter" data-level="47.1.3" data-path="poissonreg.html"><a href="poissonreg.html#lever-presses"><i class="fa fa-check"></i><b>47.1.3</b> Lever presses</a></li>
</ul></li>
<li class="chapter" data-level="47.2" data-path="poissonreg.html"><a href="poissonreg.html#the-poisson-generalized-linear-model"><i class="fa fa-check"></i><b>47.2</b> The Poisson generalized linear model</a></li>
<li class="chapter" data-level="47.3" data-path="poissonreg.html"><a href="poissonreg.html#length-of-hospital-stay"><i class="fa fa-check"></i><b>47.3</b> Length of hospital stay</a></li>
<li class="chapter" data-level="47.4" data-path="poissonreg.html"><a href="poissonreg.html#output-interpretation"><i class="fa fa-check"></i><b>47.4</b> Output interpretation</a></li>
</ul></li>
<li class="chapter" data-level="48" data-path="surv.html"><a href="surv.html"><i class="fa fa-check"></i><b>48</b> Survival Analysis</a>
<ul>
<li class="chapter" data-level="48.1" data-path="surv.html"><a href="surv.html#data-records"><i class="fa fa-check"></i><b>48.1</b> Data records</a>
<ul>
<li class="chapter" data-level="48.1.1" data-path="surv.html"><a href="surv.html#deriving-times-from-recorded-dates"><i class="fa fa-check"></i><b>48.1.1</b> Deriving times from recorded dates</a></li>
<li class="chapter" data-level="48.1.2" data-path="surv.html"><a href="surv.html#censor"><i class="fa fa-check"></i><b>48.1.2</b> Censor</a></li>
</ul></li>
<li class="chapter" data-level="48.2" data-path="surv.html"><a href="surv.html#kaplan-meier-estimator"><i class="fa fa-check"></i><b>48.2</b> Kaplan-Meier estimator</a></li>
<li class="chapter" data-level="48.3" data-path="surv.html"><a href="surv.html#glioma-data"><i class="fa fa-check"></i><b>48.3</b> Glioma data</a></li>
<li class="chapter" data-level="48.4" data-path="surv.html"><a href="surv.html#kaplan-meier-plots"><i class="fa fa-check"></i><b>48.4</b> Kaplan-Meier plots</a></li>
<li class="chapter" data-level="48.5" data-path="surv.html"><a href="surv.html#log-rank-tests"><i class="fa fa-check"></i><b>48.5</b> Log rank tests</a>
<ul>
<li class="chapter" data-level="48.5.1" data-path="surv.html"><a href="surv.html#write-up-of-log-rank-test"><i class="fa fa-check"></i><b>48.5.1</b> Write up of log rank test</a></li>
<li class="chapter" data-level="48.5.2" data-path="surv.html"><a href="surv.html#comparing-more-than-two-survival-curves"><i class="fa fa-check"></i><b>48.5.2</b> Comparing more than two survival curves</a></li>
</ul></li>
<li class="chapter" data-level="48.6" data-path="surv.html"><a href="surv.html#cox-proportional-hazards-analysis"><i class="fa fa-check"></i><b>48.6</b> Cox proportional hazards analysis</a>
<ul>
<li class="chapter" data-level="48.6.1" data-path="surv.html"><a href="surv.html#cox-proportional-hazards-regression-of-glioma"><i class="fa fa-check"></i><b>48.6.1</b> Cox proportional hazards regression of glioma</a></li>
<li class="chapter" data-level="48.6.2" data-path="surv.html"><a href="surv.html#interpretation-of-cox-regression-output"><i class="fa fa-check"></i><b>48.6.2</b> Interpretation of Cox regression output</a></li>
<li class="chapter" data-level="48.6.3" data-path="surv.html"><a href="surv.html#write-up-18"><i class="fa fa-check"></i><b>48.6.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="48.7" data-path="surv.html"><a href="surv.html#summary-15"><i class="fa fa-check"></i><b>48.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="49" data-path="classification.html"><a href="classification.html"><i class="fa fa-check"></i><b>49</b> Classification</a>
<ul>
<li class="chapter" data-level="49.1" data-path="classification.html"><a href="classification.html#distances"><i class="fa fa-check"></i><b>49.1</b> Distances</a></li>
<li class="chapter" data-level="49.2" data-path="classification.html"><a href="classification.html#clustering-methods"><i class="fa fa-check"></i><b>49.2</b> Clustering methods</a></li>
<li class="chapter" data-level="49.3" data-path="classification.html"><a href="classification.html#principal-component-analysis"><i class="fa fa-check"></i><b>49.3</b> Principal component analysis</a></li>
<li class="chapter" data-level="49.4" data-path="classification.html"><a href="classification.html#synthesize-some-data"><i class="fa fa-check"></i><b>49.4</b> Synthesize some data</a></li>
<li class="chapter" data-level="49.5" data-path="classification.html"><a href="classification.html#visualize-the-data-set"><i class="fa fa-check"></i><b>49.5</b> Visualize the data set</a></li>
<li class="chapter" data-level="49.6" data-path="classification.html"><a href="classification.html#rescaling-data"><i class="fa fa-check"></i><b>49.6</b> Rescaling data</a></li>
<li class="chapter" data-level="49.7" data-path="classification.html"><a href="classification.html#are-there-clusters"><i class="fa fa-check"></i><b>49.7</b> Are there clusters?</a></li>
<li class="chapter" data-level="49.8" data-path="classification.html"><a href="classification.html#how-many-clusters"><i class="fa fa-check"></i><b>49.8</b> How many clusters?</a></li>
<li class="chapter" data-level="49.9" data-path="classification.html"><a href="classification.html#clustering"><i class="fa fa-check"></i><b>49.9</b> Clustering</a></li>
<li class="chapter" data-level="49.10" data-path="classification.html"><a href="classification.html#pam-and-clara"><i class="fa fa-check"></i><b>49.10</b> PAM and CLARA</a></li>
<li class="chapter" data-level="49.11" data-path="classification.html"><a href="classification.html#hierarchical-clustering"><i class="fa fa-check"></i><b>49.11</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="49.12" data-path="classification.html"><a href="classification.html#heatmaps"><i class="fa fa-check"></i><b>49.12</b> Heatmaps</a></li>
<li class="chapter" data-level="49.13" data-path="classification.html"><a href="classification.html#summary-16"><i class="fa fa-check"></i><b>49.13</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="50" data-path="rnaseq.html"><a href="rnaseq.html"><i class="fa fa-check"></i><b>50</b> RNA-seq with R</a>
<ul>
<li class="chapter" data-level="50.1" data-path="rnaseq.html"><a href="rnaseq.html#overview"><i class="fa fa-check"></i><b>50.1</b> Overview</a></li>
<li class="chapter" data-level="50.2" data-path="rnaseq.html"><a href="rnaseq.html#install-bioconductor"><i class="fa fa-check"></i><b>50.2</b> Install Bioconductor</a></li>
<li class="chapter" data-level="50.3" data-path="rnaseq.html"><a href="rnaseq.html#import-raw-count-data"><i class="fa fa-check"></i><b>50.3</b> Import raw count data</a></li>
<li class="chapter" data-level="50.4" data-path="rnaseq.html"><a href="rnaseq.html#munge-to-simpler-table"><i class="fa fa-check"></i><b>50.4</b> Munge to simpler table</a></li>
<li class="chapter" data-level="50.5" data-path="rnaseq.html"><a href="rnaseq.html#filtering"><i class="fa fa-check"></i><b>50.5</b> Filtering</a></li>
<li class="chapter" data-level="50.6" data-path="rnaseq.html"><a href="rnaseq.html#dgelist"><i class="fa fa-check"></i><b>50.6</b> DGEList</a></li>
<li class="chapter" data-level="50.7" data-path="rnaseq.html"><a href="rnaseq.html#visualization"><i class="fa fa-check"></i><b>50.7</b> Visualization</a></li>
<li class="chapter" data-level="50.8" data-path="rnaseq.html"><a href="rnaseq.html#classification-1"><i class="fa fa-check"></i><b>50.8</b> Classification</a></li>
<li class="chapter" data-level="50.9" data-path="rnaseq.html"><a href="rnaseq.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>50.9</b> Hierarchical clustering</a>
<ul>
<li class="chapter" data-level="50.9.1" data-path="rnaseq.html"><a href="rnaseq.html#statistical-inference-on-clustering"><i class="fa fa-check"></i><b>50.9.1</b> Statistical inference on clustering</a></li>
</ul></li>
<li class="chapter" data-level="50.10" data-path="rnaseq.html"><a href="rnaseq.html#summary-17"><i class="fa fa-check"></i><b>50.10</b> Summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">JABSTB: Statistical Design and Analysis of Experiments with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> Data Classification</h1>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="data.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(datapasta)</span>
<span id="cb58-2"><a href="data.html#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb58-3"><a href="data.html#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code></pre></div>
<p>The starting point in any statistical design is to understand the types of data that are involved.</p>
<p>First ask whether the variables are discrete or continuous.</p>
<p>Then ask if they measured, ordered or sorted?</p>
<p>Then ask which variables are controlled by the researcher and which are observed?</p>
<p>The answers will point in the proper analytic direction.</p>
<p>I cannot emphasize enough the importance of understanding data classification for mastering a statistical framework. It is the foundation. Therefore, this material covers perhaps the most important learning objectives in this course:</p>
<p>Given an experimental data set, be able to</p>
<ul>
<li>Describe the variables that are dependent or independent.</li>
<li>Describe each variable as either continuous (measured) or discrete (ordered or sorted).</li>
</ul>
<p>A second focus of this chapter is on introducing some of the more common data handling procedures we’ll do. The several scripts provide examples for how to import, inspect, subset, transform and visualize different types of data variables.</p>
<div id="dependent-and-independent-variables" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Dependent and independent variables</h2>
<p>The experimental researcher has two basic types of variables.</p>
<p>An independent variable is the predictor or explanatory variable controlled by the researcher. Independent variables are chosen, not collected. Nevertheless, they have values just like the values of collected data.</p>
<p>For example,</p>
<ul>
<li>In a blood glucose drug study, the independent variable “Treatment” would come in two levels, “Placebo” and “Drug.” The values of the “Treatment”variable are “Placebo” and “Drug”</li>
<li>In a study on how a gene influences behavior, the independent variable is “Genotype,” with three values: “wild-type,” “heterozygous” and “knockout”</li>
<li>In an enzyme activity assay, the independent variable is substrate concentration, with a dozen or so values ranging over two orders of magnitude.</li>
</ul>
<p>Conventionally, when graphing the independent variable is plotted on the abscissa, or x-axis. But YMMV.</p>
<p>A dependent variable is the response or outcome variable collected in an experiment. The values that dependent variables can take on are determined by, or are dependent upon, the level of the independent variables.</p>
<p>For example,</p>
<ul>
<li><p>The variable “blood_glucose,” collected from blood samples, has units of glucose concentration, with values over a large continuous range.</p></li>
<li><p>In a Morris Water Maze test, memory is represented by a “time” variable, which is the time taken to find a remembered location, with unit values in seconds, over a wide range.</p></li>
<li><p>Levels of product from an enzyme reaction are measured with an assay the emits fluorescence, in relative units. The dependent variable “RFU/mg” represents activity controlled by the amount of enzyme added and takes on many values over a continuous range.</p></li>
</ul>
<p>Most of the time the dependent variable is plotted on the ordinate, or y-axis, of some graph.</p>
<div id="statistical-notation" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Statistical notation</h3>
<p>Variables are abstracted in statistical modeling.</p>
<p>In the convention I’ll use, the dependent variable is usually depicted by the uppercase symbol <span class="math inline">\(Y\)</span>, to represent a variable name. Subscripts <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> are indexes for variables.</p>
<p>Here <span class="math inline">\(j\)</span> represents the number of dependent variables and can have values from 1 to <span class="math inline">\(k\)</span>. Sometimes experiments (eg, anything with an -omics) generate multiple dependent variables. <span class="math inline">\(Y_j\)</span> is the <span class="math inline">\(j^{th}\)</span> dependent variable.</p>
<p>The values a dependent variable can assume are symbolically represented as lowercase symbol <span class="math inline">\(y_i\)</span>. By convention <span class="math inline">\(y_i\)</span> is the value for the <span class="math inline">\(i^th\)</span> replicate. The values of <span class="math inline">\(i\)</span> range from 1 to <span class="math inline">\(n\)</span> independent replicates.</p>
<p>Univariate experiments will generate <span class="math inline">\(y_i\)</span> measurements from <span class="math inline">\(1 \times n\)</span> independent replicates, whereas multivariate experiments will generate <span class="math inline">\(y_{ij}\)</span> measurements from <span class="math inline">\(k \times n\)</span> independent replicates.</p>
<p>Similarly, the independent variable is usually depicted by uppercase <span class="math inline">\(X\)</span> (or some other letter) whereas the values are represented by lowercase <span class="math inline">\(x_i\)</span>.</p>
<p>I’m going to use that convention but with a twist. Independent variables denoted using <span class="math inline">\(X\)</span> will represent continuous scaled variables, whereas independent variables denoted using <span class="math inline">\(A\)</span> or <span class="math inline">\(B\)</span>, or <span class="math inline">\(C\)</span>, will represent discrete, factorial variables. Factorial variables will take on values denoted by lowercase, eg, <span class="math inline">\(a_i\)</span>, <span class="math inline">\(b_i\)</span>, <span class="math inline">\(c_i\)</span>).</p>
<p>Multivariable experiments have more than one independent variables. For example, <span class="math inline">\(X_j\)</span> represents the name of the <span class="math inline">\(j^{th}\)</span> independent variable, where <span class="math inline">\(j = 2\ to\ k\)</span>.</p>
<p>To illustrate dependent and independent variables think about a linear relationship between two continuous variables, <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span>. This relationship can be expressed using the model <span class="math inline">\(Y=\beta_0 + \beta_1 X\)</span>.</p>
<p><span class="math inline">\(X\)</span> is the variable the researcher manipulates, for example, time or the concentration of a substance. <span class="math inline">\(Y\)</span> would be a variable that the researcher measures, such as absorption or binding or fluorescence. Each has multiple values.</p>
<p>The parameters <span class="math inline">\(\beta_0\)</span> and <span class="math inline">\(\beta_1\)</span> are fixed population constants (thus the Greek symbols) that define the linear relationship between the two variables. Here they represent the y-intercept and slope, respectively, of a straight regression line.</p>
<p>Thus, <span class="math inline">\(Y\)</span> takes on different values as the researcher manipulates the levels of <span class="math inline">\(X\)</span>. Which explains why <span class="math inline">\(Y\)</span> depends on <span class="math inline">\(X\)</span>.</p>
<p>For example, here’s how the data for a protein standard curve experiment would be depicted. In the R script below the variable <span class="math inline">\(X\)</span> represents known concentrations of an immunoglobulin protein standard in <span class="math inline">\(\mu g/ml\)</span>. The researcher builds this dilution series from a stock of known concentration, thus it is the independent variable.</p>
<p>The variable <span class="math inline">\(Y\)</span> represents <span class="math inline">\(A_{595}\)</span>, light absorption in a spectrophotometer for each of the values of the standard protein. The <span class="math inline">\(A_{595}\)</span> values depend upon the immunoglobulin concentration. Estimates for <span class="math inline">\(\beta_0\)</span> and <span class="math inline">\(\beta_1\)</span> are derived from running a linear regression on the data with the <code>lm(Y~X)</code> script.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="data.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Protein assay data, X units ug/ml, Y units A595.</span></span>
<span id="cb59-2"><a href="data.html#cb59-2" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">1.25</span>, <span class="fl">2.5</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">15</span>, <span class="dv">20</span>, <span class="dv">25</span>)</span>
<span id="cb59-3"><a href="data.html#cb59-3" aria-hidden="true" tabindex="-1"></a>Y <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.000</span>, <span class="fl">0.029</span>, <span class="fl">0.060</span>, <span class="fl">0.129</span>, <span class="fl">0.250</span>, <span class="fl">0.371</span>, <span class="fl">0.491</span>, <span class="fl">0.630</span>)</span>
<span id="cb59-4"><a href="data.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="co">#derive the slope and intercept by linear regression</span></span>
<span id="cb59-5"><a href="data.html#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(Y<span class="sc">~</span>X)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = Y ~ X)
## 
## Coefficients:
## (Intercept)            X  
##  -0.0008033    0.0249705</code></pre>
<p>The output indicates the regression line intercepts the y-axis at a value very close to zero, and that every one ug/ml increment in the value of the standard protein, <span class="math inline">\(X\)</span>, predicts a 0.0249705 increment in the absorbance at 595 nm value, <span class="math inline">\(Y\)</span>.</p>
<p>Fortunately, R functions rarely forces us to abstract by the mathematical statistics convention, X and Y. We can name variables in more comfortable terms.</p>
<p>This helps make the data we are working with a bit more clear.</p>
<p>For example,</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="data.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Protein assay data, standard units ug/ml, absorption units A595.</span></span>
<span id="cb61-2"><a href="data.html#cb61-2" aria-hidden="true" tabindex="-1"></a>standard <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">1.25</span>, <span class="fl">2.5</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">15</span>, <span class="dv">20</span>, <span class="dv">25</span>)</span>
<span id="cb61-3"><a href="data.html#cb61-3" aria-hidden="true" tabindex="-1"></a>A595 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.000</span>, <span class="fl">0.029</span>, <span class="fl">0.060</span>, <span class="fl">0.129</span>, <span class="fl">0.250</span>, <span class="fl">0.371</span>, <span class="fl">0.491</span>, <span class="fl">0.630</span>)</span>
<span id="cb61-4"><a href="data.html#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="co">#derive the slope and intercept by linear regression</span></span>
<span id="cb61-5"><a href="data.html#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(A595 <span class="sc">~</span> standard)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = A595 ~ standard)
## 
## Coefficients:
## (Intercept)     standard  
##  -0.0008033    0.0249705</code></pre>
</div>
<div id="when-there-is-no-independent-variable" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> When there is no independent variable</h3>
<p>The problem of drawing causal inference from studies in which all of the variables are observed, as is common in public health and other social sciences, is beyond the scope of this course. Pearl offers an excellent primer on considerations that must be applied to extract causality from observational data <a href="http://ftp.cs.ucla.edu/pub/stat_ser/r350.pdf">here</a>.</p>
</div>
</div>
<div id="discrete-or-continuous-variables" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Discrete or continuous variables</h2>
<p>No matter if they are dependent or independent variables, all variables can be subclassified further into two categories. They are either discrete or continuous.</p>
<p>Discrete variables can only take on discrete values, while continuous variables can take on values over a continuous range. This distinction is discussed further below.</p>
<p>Variables can be subclassified further as either <strong>measured</strong>, <strong>ordered</strong>, or <strong>sorted</strong>. This subdivision fulfills a few purposes.</p>
<p>First, it’s alliterative so hopefully easier to remember. It reminds me of Waffle House hash browns, which can be either scattered, smothered or covered.</p>
<p>Different authors/software give these three types of variables different names, which creates some confusion. SPSS users must click buttons to classify data as either scalar, ordinal, or nominal, which correspond to measured, ordered and sorted. Another fairly common descriptive for the three types is interval, integer, and categorical. Conceptually, they mean the same things.</p>
<table>
<caption><span id="tab:unnamed-chunk-38">Table 9.1: </span>Common synonyms for types of variables</caption>
<thead>
<tr class="header">
<th align="left">measured</th>
<th align="left">ordered</th>
<th align="left">sorted</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">scalar</td>
<td align="left">ordinal</td>
<td align="left">nominal</td>
</tr>
<tr class="even">
<td align="left">interval</td>
<td align="left">integer</td>
<td align="left">categorical</td>
</tr>
<tr class="odd">
<td align="left">numeric</td>
<td align="left">factor/character/string/numeric</td>
<td align="left">factor/character/string</td>
</tr>
<tr class="even">
<td align="left">continuous</td>
<td align="left">discrete</td>
<td align="left">discrete</td>
</tr>
</tbody>
</table>
<p>Even if we can’t agree on names, everybody seems to agree that all variables can be reduced to three fundamental subtypes.</p>
<p>Second, this taxonomy forms the basis for drawing a pragmatic statistical modeling heuristic to help you keep the relationship between variable type and experimental design (and analysis) straight.</p>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-39"></span>
<img src="images/testing_heuristic.jpg" alt="The type of data dictates how the experiment should be modeled and analyzed." width="480" />
<p class="caption">
Figure 9.1: The type of data dictates how the experiment should be modeled and analyzed.
</p>
</div>
<p>Third, the “measured, ordered, sorted” scheme classifies variables on the basis of their information density, where measured &gt; ordered &gt; sorted. You’ll see below how discrete variables lack the kind of information density of continuous variables.</p>
<p><strong>More pragmatically, measured, ordered and sorted variables behave differently because they represent different kinds of information. They tend to have very different probability distributions. As a result statisticians have devised statistical models that are more appropriate for one or the other.</strong></p>
<div id="measured" class="section level3" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Measured variables</h3>
<p>Measured variables are fairly easy to spot. Any derivative of one of the <a href="https://www.nist.gov/pml/weights-and-measures/metric-si/si-units">seven base SI units</a> will be a measured variable.</p>
<p>The gang of seven are the ampere, second, mole, kilogram, candela, Kelvin, and meter. When in doubt, a working hack is that these variables tend to need instruments to take measurements. So if you’re using some instrument to measure something, and it is not counting events, it is a measured variable.</p>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-40"></span>
<img src="images/si_units.jpg" alt="The seven SI units" width="480" />
<p class="caption">
Figure 9.2: The seven SI units
</p>
</div>
<p>Take mass as an example. The masses of physical objects can be measured on a continuous scale of sizes ranging from super-galaxian to subatomic. Variables that are in units of mass take on a smooth continuum of values over this entire range because mass scales are infinitesimally divisible.</p>
<p>Here’s a thought experiment for what that means. Take an object that weighs a kilogram, cut it in half and measure what’s left. You have two objects that are each one half a kilogram. Now repeat that process again and again.</p>
<p>After each split something with mass always remains. Even when you arrive at the point where only a single proton remains it can be smashed into even smaller pieces in a supercollider, yielding trails of subatomic particles….most of which have observable masses.</p>
<p><strong>But here’s what’s most different about continuous variables: That continuity between gradations means that continuous variables carry more information than other types of variables.</strong></p>
<p>On a scale of micrograms, objects weighing one kilogram would have one billion subdivisions. If you have an instrument that can weigh the mass of kilogram-sized objects at microgram precision, you would say that one kilogram is comprised of a billion bits of information.</p>
<p>That’s more information than sorting an object by naming it “heavy” or “light.”</p>
<div id="visualizing-measured-variables" class="section level4" number="9.2.1.1">
<h4><span class="header-section-number">9.2.1.1</span> Visualizing measured variables</h4>
<p>In the precourse survey biostats students self-reported their height, in cm, which is a continuous scale. Height is a measured variable.</p>
<p>Viewed in scatter plots, it is easy to see how the height variable takes on many different values on the y-scale. (The points have been colored by sex, mostly to illustrate how cool it is to segment one variable (height) using another (sex)).</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="data.html#cb63-1" aria-hidden="true" tabindex="-1"></a>pcd <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;datasets/precourse.csv&quot;</span>)</span>
<span id="cb63-2"><a href="data.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="co"># omit heights &lt;125 cm and &gt;250 cm as unreliable entries</span></span>
<span id="cb63-3"><a href="data.html#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="co"># nobody that short or tall has ever taken the course!</span></span>
<span id="cb63-4"><a href="data.html#cb63-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> pcd <span class="sc">%&gt;%</span> </span>
<span id="cb63-5"><a href="data.html#cb63-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">filter</span>(height <span class="sc">&gt;</span> <span class="dv">125</span> <span class="sc">&amp;</span> height <span class="sc">&lt;</span> <span class="dv">250</span>)) <span class="sc">+</span></span>
<span id="cb63-6"><a href="data.html#cb63-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(<span class="dv">0</span>), <span class="at">y =</span> height, <span class="at">color =</span> sex), </span>
<span id="cb63-7"><a href="data.html#cb63-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">height =</span> <span class="dv">0</span>, <span class="at">width =</span> <span class="fl">0.4</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb63-8"><a href="data.html#cb63-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">begin=</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="fl">0.6</span>)<span class="sc">+</span></span>
<span id="cb63-9"><a href="data.html#cb63-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">140</span>, <span class="dv">200</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb63-10"><a href="data.html#cb63-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;biostats students&quot;</span>) <span class="sc">+</span></span>
<span id="cb63-11"><a href="data.html#cb63-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;height(cm)&quot;</span>) <span class="sc">+</span></span>
<span id="cb63-12"><a href="data.html#cb63-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb63-13"><a href="data.html#cb63-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb63-14"><a href="data.html#cb63-14" aria-hidden="true" tabindex="-1"></a>        )</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-41"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-41-1.png" alt="Scatterplot of biostat student heights segmented by sex" width="480" />
<p class="caption">
Figure 9.3: Scatterplot of biostat student heights segmented by sex
</p>
</div>
<p>It’s always important to do a quick histogram of measured data. Histograms categorize height values into bins of a given number or width. The <code>geom_histogram</code> then counts the number of values in each bin. Histograms therefore show the frequency distribution of a variable.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="data.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>pcd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(height <span class="sc">&gt;</span> <span class="dv">125</span> <span class="sc">&amp;</span> height <span class="sc">&lt;</span> <span class="dv">250</span>)) <span class="sc">+</span></span>
<span id="cb64-2"><a href="data.html#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>height), <span class="at">binwidth =</span> <span class="fl">2.5</span>, <span class="at">color =</span> <span class="st">&quot;#f2a900&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#012169&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-42"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-42-1.png" alt="Histogram of biostat student heights, the most frequent height value is 170 +/- 1.25 cm." width="480" />
<p class="caption">
Figure 9.4: Histogram of biostat student heights, the most frequent height value is 170 +/- 1.25 cm.
</p>
</div>
</div>
</div>
<div id="discrete-categorical-and-ordinal-variables" class="section level3" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Discrete categorical and ordinal variables</h3>
<p>Discrete variables are discontinuous. The units of discrete variables are indivisible. Unlike continuous variables, discrete variables have no information between their unit values.</p>
<p>There are two types of discrete variables, either <strong>ordered</strong> or <strong>sorted</strong>.</p>
<div id="sorted-data" class="section level4" number="9.2.2.1">
<h4><span class="header-section-number">9.2.2.1</span> Sorted data</h4>
<p>These variables represent objects somehow scored as belonging to one nominal category of attributes or some other. In other words, the objects are sorted into categories. You can also think of this as being sorted into buckets or bins based upon some common feature(s). The scoring process can be based upon either objective or subjective criteria.</p>
<p><em>Statisticians generally frown at researchers who sort replicates on the basis of perfectly good measured data. For example, sorting people into low, mid and high blood pressure on the basis of a cuff blood pressure measurement. These instruments generate continuous variables on inherently more informative scales. Statisticians hold that sorting after measurement discards valuable information. And cut offs can be arbitrary.</em></p>
<p>In a survey biostats students identify their sex as either male or female. The name of the variable is <code>sex</code>. The values that <code>sex</code>can have are either <code>male</code> or <code>female</code>. The variable has only two nominal levels (or two categories): male and female.</p>
<p>Perhaps most important, sorted variables have no intrinsic numeric value. Male is neither higher nor lesser than female; the values of the sex variable are just different. <em>Wanting</em> more of one or the other shouldn’t be conflated with a variables inherent properties.</p>
<p>Sorted variables are also called <code>factors</code>. That’s the dominant jargon in R. We’ll find ourselves converting variables to factors at times.</p>
<p><strong>In stats jargon, sorted, nominal, categorical, and factor are words used to represent the same kind of variable. Sorry</strong></p>
<p>Try not to let the jargon confuse you. What’s most important to understand is that sorted variables lack numeric values. So we count how many are sorted into one category or the other. That changes everything about the statistical methods we apply to experiments involving sorted variables.</p>
<p>A count is an integer value. The count cannot take on any values other than integer values. For example, we don’t have cases where there is a partial student of one sex or the other. These counts exist as discrete, discontinuous values.</p>
<p>Of course, the categorization of sex is sometimes ambiguous. When important to accommodate more than two categories, we would add additional categories to account for all possible outcomes. For example, the <code>sex</code> variable could be set to take on values of <code>male</code>, <code>female</code>, and <code>other</code>.</p>
</div>
<div id="visualizing-sorted-data" class="section level4" number="9.2.2.2">
<h4><span class="header-section-number">9.2.2.2</span> Visualizing sorted data</h4>
<p>Let’s look at the precourse survey data frame we imported above. Each row represents the data for one biostats student. Notice how the values for the sex variable are characters. There are no numbers to plot!!</p>
<p>I could assign them numbers if I wished, for example <span class="math inline">\(Male = 0\)</span> and <span class="math inline">\(Female = 1\)</span>, but that is just arbitrary numeric coding, using numbers rather than characters. The numeric values are meaningless. I could just as easily code one <span class="math inline">\(1.2345\)</span> and the other <span class="math inline">\(0.009876\)</span> and accomplish the same thing. The point being the values of sorted variables are arbitrary.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="data.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pcd)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 6
##    term enthusiasm height friends sex    fav_song
##   &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;
## 1  2014          6   170      685 Female       NA
## 2  2014          7   158.     154 Female       NA
## 3  2014          7   163.    1333 Female       NA
## 4  2014         10   168        0 Female       NA
## 5  2014          7   168.     250 Female       NA
## 6  2014          5   157      476 Female       NA</code></pre>
<p>We’ll plot the sex variable data using a function, <code>geom_bar</code>. This geom not only makes bars, it also sums up the number of <code>Male</code> and <code>Female</code> instances in the variable sex. Note there are no error bars. Counts are just a sum, there is no variation.</p>
<p>In this course, we’ll avoid bar plots for everything other than visualizing variables that summarize as counts. We could have also used pie charts or stacked bars, but those are harder to read. The chosen method is a better way to illustrate the magnitude of the counts and the difference between the two levels within the sex variable.</p>
<p>Note how this bar plot is, effectively, a histogram of the sex variable. Here the bins are defined by the two values of the variable.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="data.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pcd)<span class="sc">+</span></span>
<span id="cb67-2"><a href="data.html#cb67-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>sex), <span class="at">width =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-44"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-44-1.png" alt="Use bar plots to display the relative counts of sorted variable values." width="480" />
<p class="caption">
Figure 9.5: Use bar plots to display the relative counts of sorted variable values.
</p>
</div>
<p>Many biomedical experiments generate discrete sorted data, too.</p>
<p>Imagine the following:</p>
<ul>
<li>Neurons are poked with an electrode. Counts are recorded of the number of times they depolarize over a certain time period. Counts are compared between various treatments, such activating an upstream neuron at different levels.<br />
</li>
<li>Cells are stained for expression of a marker protein. The number of cells in which the protein is detected are counted. Counts are compared between various treatments, such as a knockdown and its control.</li>
<li>By some criteria, cells are judged to be either alive or dead and counted as such. The number of alive cells are counted after manipulating expression of a tumor suppressor gene, and compared to a control.</li>
<li>By some criteria, mice are judged to either show a disease phenotype or not, and counted as such. Disease incidence is counted in response to levels of a therapeutic agent, or a background genotype, or in response to some stressor.</li>
</ul>
<p>There are an infinite number of examples for experiments that can be performed in which a dependent variable is sorted. For each replicate the researcher decides what nominal category it should be sorted into.</p>
<p>Sometimes sorting occurs after some fairly sophisticated biological manipulation, instrumentation or biochemical analysis, where in the end all the researcher has are a count of the number of times one thing happened or some other.</p>
</div>
<div id="ordered-data" class="section level4" number="9.2.2.3">
<h4><span class="header-section-number">9.2.2.3</span> Ordered data</h4>
<p>Ordered data is, in one sense, a hybrid cross of sorted and measured data. Ordinal is another common jargon term used to describe such variables.</p>
<p>If you’ve ever taken a poll in which you’ve been asked to evaluate something on a scale ranging from something akin to “below low” to “super very high”…then you’ve experienced ordinal scaling (such scales are called Likert scales).</p>
<p>The ordered variable’s values are, in one sense, categorical: “low,” “med,” and “high” differ qualitatively. Each lacks a precise quantitative value. In that sense, they are like sorted variables.</p>
<p>However, there is a quantitative, if inexact, relationship <em>between</em> the values. “High” implies more than “med” which implies more than “low.” An underlying gradation exists in the variable’s scale. This is not just the absence or presence of an attribute. Rather, the scale allows for some amount of the attribute relative to other possible amounts of the attribute.</p>
<p>Ordinal variables are discrete, because no values exist between the categories on the scale.</p>
<p>The precourse survey is chock full of questions that generate data on an ordered scale. For example, one asks students about their <code>enthusiasm</code> for taking the course. They can select answers ranging from 1 (“Truly none”) to 10 (“Absolutely giddy with excitement”).</p>
<p>Even though numbers are used as categories, their values are arbitrary, not quantitative, except to imply a 2 means more enthusiasm than a 1, and so on. The variable values could just as easily be: “Truly none,” “Very little,” “Slightly more than very little,” … and so on.</p>
<p>In a survey, the participant selects the level for the variable. When we do an experiment, replicates are evaluated and then categorized to one of the values of the ordinal scale.</p>
<p>Disability status scales are classic ordinal scales. These are used to assess neurological abnormalities, for example, those associated with experimental multiple sclerosis. Each replicate in a study is evaluated by trained researchers and assigned the most appropriate value given its condition: 0 for no disease, 1 for limp tail, 2 for mild paraparesis, 3 for moderate paraparesis, 4 for complete hind limb paralysis, and 5 for moribund.</p>
<p>Obviously, in this ordinal scale, as the numeric value increases so to does the severity of the subject’s condition.</p>
<p>Here’s what a very small set of ordinal data might look like. Imagine doing a small pilot study to determine whether ND4 gene deletion causes neurological impairment. Here’s how the data from a lab notebook would be transcribed into an R script to make a summary table:</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="data.html#cb68-1" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;wt&quot;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&quot;ND4&quot;</span>, <span class="dv">3</span>))</span>
<span id="cb68-2"><a href="data.html#cb68-2" aria-hidden="true" tabindex="-1"></a>DSS_score <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">5</span>))</span>
<span id="cb68-3"><a href="data.html#cb68-3" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">tibble</span>(genotype, DSS_score); results</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   genotype DSS_score
##   &lt;chr&gt;        &lt;int&gt;
## 1 wt               0
## 2 wt               1
## 3 wt               1
## 4 ND4              5
## 5 ND4              3
## 6 ND4              5</code></pre>
<p>Let’s make sure we understand the variables.</p>
<p>The independent variable is <code>genotype</code>, a factorial variable that comes in two nominal levels, <code>wt</code> or <code>ND4</code>. <code>DSS_score</code> is a dependent variable that comes in 6 levels (integer values ranging from 0 to 5). I’m forcing R to read <code>DSS_score</code> for what it is, an integer rather than as a numeric.</p>
<p>However, the values 0 to 5 are nominally arbitrary. This only differs from a sorted variable because a value of 1 means more than zero, 2 means more than 1, and so on.</p>
</div>
<div id="visualizing-ordered-variables" class="section level4" number="9.2.2.4">
<h4><span class="header-section-number">9.2.2.4</span> Visualizing ordered variables</h4>
<p>Let’s visualize the enthusiasm biostats students have for taking a course.</p>
<p>Here’s a bar plot. Here again, <code>geom_bar</code> not only draws bars, it also counts the total number of biostat students who voted for each of the 10 levels. A bar plot of counted data is just a poor man’s histogram.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="data.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pcd) <span class="sc">+</span></span>
<span id="cb70-2"><a href="data.html#cb70-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x=</span>enthusiasm))<span class="sc">+</span></span>
<span id="cb70-3"><a href="data.html#cb70-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">1</span>))</span></code></pre></div>
<p><img src="jabstb_files/figure-html/unnamed-chunk-46-1.png" width="672" /></p>
<p>Viewed using the histogram function, we see the same thing, if we dial the binwidth to the unit of the ordinal scale or the number of bins to the length of the ordinal scale.</p>
<p>It is easy to see the data are skewed a bit to the right of the scale center. I mention this because ordered data are frequently skewed. Variables of counts should not be expected to obey a symmetrical normal distribution.</p>
<p>Although it is tempting to compute descriptives like means and standard deviations and even run t-tests and ANOVA on such data, slow down and have a thought about that. The values are arbitrary.</p>
<p>If instead of 1, 2, 3,… the variable values were “Truly none,” “Very little,” “Slightly more than very little,” …, how exactly would you compute a mean and standard deviation?</p>
<p>You could not.</p>
<p>A cognitive bias is equating numeric discrete yet arbitrary scale with a continuous, equal interval scale.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="data.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pcd) <span class="sc">+</span></span>
<span id="cb71-2"><a href="data.html#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>enthusiasm), </span>
<span id="cb71-3"><a href="data.html#cb71-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;#f2a900&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#012169&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-4"><a href="data.html#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">1</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-47"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-47-1.png" alt="An ordered variable plotted as histogram." width="480" />
<p class="caption">
Figure 9.6: An ordered variable plotted as histogram.
</p>
</div>
<p>Viewing scatter plots of discrete data really brings forth their discreteness. They show how there is obviously no information between levels of the ordered scale used in this survey instrument.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="data.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the height argument in geom_jitter is crucial to seeing this pattern</span></span>
<span id="cb72-2"><a href="data.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="co"># adjust the height value to see what happens</span></span>
<span id="cb72-3"><a href="data.html#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="data.html#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pcd) <span class="sc">+</span></span>
<span id="cb72-5"><a href="data.html#cb72-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(<span class="dv">1</span>), <span class="at">y=</span>enthusiasm), </span>
<span id="cb72-6"><a href="data.html#cb72-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">height=</span><span class="dv">0</span>, <span class="at">width =</span> <span class="fl">0.4</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb72-7"><a href="data.html#cb72-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb72-8"><a href="data.html#cb72-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;biostats students&quot;</span>) <span class="sc">+</span></span>
<span id="cb72-9"><a href="data.html#cb72-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb72-10"><a href="data.html#cb72-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb72-11"><a href="data.html#cb72-11" aria-hidden="true" tabindex="-1"></a>        ) </span></code></pre></div>
<pre><code>## Warning: Continuous limits supplied to discrete scale.
## Did you mean `limits = factor(...)` or `scale_*_continuous()`?</code></pre>
<pre><code>## Warning in (function (kind = NULL, normal.kind = NULL, sample.kind = NULL) :
## non-uniform &#39;Rounding&#39; sampler used</code></pre>
<p><img src="jabstb_files/figure-html/unnamed-chunk-48-1.png" width="480" /></p>
<p>Although you can calculate an mean for a variable such as this, whether you should report it to one significant digit or beyond is fraught with portent.</p>
</div>
</div>
<div id="rescaling-variables" class="section level3" number="9.2.3">
<h3><span class="header-section-number">9.2.3</span> Rescaling variables</h3>
<p>Commonly researchers transform discrete variables onto a continuous scale or continuous variables onto a discrete scale.</p>
<p>Rescaling requires some judgment. We should always be concerned that rescaling might be counterproductive.</p>
<p>The most frequent discrete to continuous transformation is probably sorted count data to percents. This usually has good descriptive utility. The problems usually arise when the transformed variable is used for statistical testing.</p>
<p>For example, the two percent values for the sex variable in the table below are useless for asking whether there is a statistical difference from, say, 50%. The transformation converted plenty of useful count data (386 bits of information) into 2 bits of information.</p>
<p>By keeping the variable values as numeric counts we can readily test whether the proportion of 144/242 differs from 50/50, by performing a proportion test (covered in Chapter <a href="categorical.html#categorical">19</a>).</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="data.html#cb75-1" aria-hidden="true" tabindex="-1"></a>pcd <span class="sc">%&gt;%</span> </span>
<span id="cb75-2"><a href="data.html#cb75-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(sex) <span class="sc">%&gt;%</span> </span>
<span id="cb75-3"><a href="data.html#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total=</span><span class="fu">sum</span>(n), <span class="at">percent=</span><span class="dv">100</span><span class="sc">*</span>n<span class="sc">/</span>total)</span></code></pre></div>
<pre><code>## # A tibble: 2 x 4
##   sex        n total percent
##   &lt;chr&gt;  &lt;int&gt; &lt;int&gt;   &lt;dbl&gt;
## 1 Female   294   464    63.4
## 2 Male     170   464    36.6</code></pre>
<p>Another common rescaling is treating ordinal variables as if they are, in fact, on a measured equal interval scale.</p>
<p>In addition to the fact that ordinal scales are often arbitrary, there are a couple of other problems that should be thought through before rescaling.</p>
<p>First, an ordinal variable like <code>enthusiasm</code> is bounded. Values below 1 and above 10 are not possible. Continuous sampling distributions, such as the Normal, are premised upon variables that have no bound.</p>
<p>Another major question that arises with ordinal scales is whether it is safe to assume the intervals between the units are equal. Is the difference between a 1 and a 2 score on the <code>enthusiasm</code> variable the same as the difference between a 9 and a 10 score?</p>
<p>When these issues cannot be validated, the nonparametric tests provide perfectly good inferential tools for ordinal data (covered in Chapter <a href="nonparametrics.html#nonparametrics">21</a>).</p>
</div>
</div>
<div id="summary-2" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Summary</h2>
<p>We’re going to talk about data a lot more the rest of the semester. So far we’ve covered these fundamentals.</p>
<ul>
<li>An experimental researchers collects measures for dependent variables after imposing independent variables on a system.</li>
<li>Both dependent and independent variables can be either discrete or continuous.</li>
<li>Discrete variables can be either ordered or sorted.</li>
<li>Continuous variables are measured.</li>
<li>Statistical treatment depends upon the type of variables involved in the experiment.</li>
<li>Working with data with software demands that we understand data structure</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rlearn1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dispersion.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["jabstb.pdf", "jabstb.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
