<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 25 Introduction to ANOVA | JABSTB: Statistical Design and Analysis of Experiments with R</title>
  <meta name="description" content="Experimental biostatistics using R." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 25 Introduction to ANOVA | JABSTB: Statistical Design and Analysis of Experiments with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Experimental biostatistics using R." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 25 Introduction to ANOVA | JABSTB: Statistical Design and Analysis of Experiments with R" />
  
  <meta name="twitter:description" content="Experimental biostatistics using R." />
  

<meta name="author" content="TJ Murphy PhD, Department of Pharmacology and Chemical Biology, School of Medicine, Emory University, Atlanta, GA biostats538@gmail.com" />


<meta name="date" content="2021-01-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="simcorrelation.html"/>
<link rel="next" href="fdistr.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">JABSTB</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="author.html"><a href="author.html"><i class="fa fa-check"></i><b>1</b> About the author and book</a></li>
<li class="chapter" data-level="2" data-path="history.html"><a href="history.html"><i class="fa fa-check"></i><b>2</b> A Brief History of Experimental Design</a></li>
<li class="chapter" data-level="3" data-path="software.html"><a href="software.html"><i class="fa fa-check"></i><b>3</b> Installing and Understanding the software</a><ul>
<li class="chapter" data-level="3.1" data-path="software.html"><a href="software.html#tldr"><i class="fa fa-check"></i><b>3.1</b> tl;dr</a></li>
<li class="chapter" data-level="3.2" data-path="software.html"><a href="software.html#spring-2021-term-versions"><i class="fa fa-check"></i><b>3.2</b> Spring 2021 term versions</a><ul>
<li class="chapter" data-level="3.2.1" data-path="software.html"><a href="software.html#what-are-these"><i class="fa fa-check"></i><b>3.2.1</b> What are these?</a></li>
<li class="chapter" data-level="3.2.2" data-path="software.html"><a href="software.html#rule-1-install-r-first-install-rstudio-second"><i class="fa fa-check"></i><b>3.2.2</b> Rule 1: Install R first, install RStudio second</a></li>
<li class="chapter" data-level="3.2.3" data-path="software.html"><a href="software.html#rule-2-dont-pay"><i class="fa fa-check"></i><b>3.2.3</b> Rule 2: Don’t pay!</a></li>
<li class="chapter" data-level="3.2.4" data-path="software.html"><a href="software.html#rule-3-dont-compile"><i class="fa fa-check"></i><b>3.2.4</b> Rule 3: Don’t compile!</a></li>
<li class="chapter" data-level="3.2.5" data-path="software.html"><a href="software.html#rule-4-install-the-right-r-software-for-your-machine"><i class="fa fa-check"></i><b>3.2.5</b> Rule 4: Install the right R software for your machine</a></li>
<li class="chapter" data-level="3.2.6" data-path="software.html"><a href="software.html#doubts"><i class="fa fa-check"></i><b>3.2.6</b> Doubts?</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="software.html"><a href="software.html#after-installation"><i class="fa fa-check"></i><b>3.3</b> After installation</a></li>
<li class="chapter" data-level="3.4" data-path="software.html"><a href="software.html#what-is-r"><i class="fa fa-check"></i><b>3.4</b> What is R?</a></li>
<li class="chapter" data-level="3.5" data-path="software.html"><a href="software.html#more-on-packages"><i class="fa fa-check"></i><b>3.5</b> More on packages</a></li>
<li class="chapter" data-level="3.6" data-path="software.html"><a href="software.html#how-to-install-and-use-packages"><i class="fa fa-check"></i><b>3.6</b> How to install and use packages</a></li>
<li class="chapter" data-level="3.7" data-path="software.html"><a href="software.html#packages-well-need"><i class="fa fa-check"></i><b>3.7</b> Packages we’ll need</a></li>
<li class="chapter" data-level="3.8" data-path="software.html"><a href="software.html#super-immportant-always-avoid-compiling"><i class="fa fa-check"></i><b>3.8</b> Super immportant: Always avoid compiling</a><ul>
<li class="chapter" data-level="3.8.1" data-path="software.html"><a href="software.html#reasons-not-to-compile"><i class="fa fa-check"></i><b>3.8.1</b> Reasons not to compile</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="software.html"><a href="software.html#summary"><i class="fa fa-check"></i><b>3.9</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>4</b> Start Learning R</a><ul>
<li class="chapter" data-level="4.1" data-path="getting-started.html"><a href="getting-started.html#digging-in-with-rrstudio"><i class="fa fa-check"></i><b>4.1</b> Digging in with R/RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started.html"><a href="getting-started.html#the-tidyverse-package"><i class="fa fa-check"></i><b>4.2</b> The tidyverse package</a><ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started.html"><a href="getting-started.html#do-swirlstats"><i class="fa fa-check"></i><b>4.2.1</b> Do swirlstats</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started.html"><a href="getting-started.html#other-resources"><i class="fa fa-check"></i><b>4.3</b> Other resources</a><ul>
<li class="chapter" data-level="4.3.1" data-path="getting-started.html"><a href="getting-started.html#internet-search"><i class="fa fa-check"></i><b>4.3.1</b> Internet search</a></li>
<li class="chapter" data-level="4.3.2" data-path="getting-started.html"><a href="getting-started.html#help-documents"><i class="fa fa-check"></i><b>4.3.2</b> Help Documents</a></li>
<li class="chapter" data-level="4.3.3" data-path="getting-started.html"><a href="getting-started.html#package-vignettes"><i class="fa fa-check"></i><b>4.3.3</b> Package Vignettes</a></li>
<li class="chapter" data-level="4.3.4" data-path="getting-started.html"><a href="getting-started.html#people"><i class="fa fa-check"></i><b>4.3.4</b> People</a></li>
<li class="chapter" data-level="4.3.5" data-path="getting-started.html"><a href="getting-started.html#typos"><i class="fa fa-check"></i><b>4.3.5</b> typos</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="philos.html"><a href="philos.html"><i class="fa fa-check"></i><b>5</b> What is your philosophy?</a><ul>
<li class="chapter" data-level="5.1" data-path="philos.html"><a href="philos.html#exploration-and-uncertainty"><i class="fa fa-check"></i><b>5.1</b> Exploration and uncertainty</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bigpic.html"><a href="bigpic.html"><i class="fa fa-check"></i><b>6</b> Stats: The Big Picture</a><ul>
<li class="chapter" data-level="6.1" data-path="bigpic.html"><a href="bigpic.html#what-are-experimental-statistics"><i class="fa fa-check"></i><b>6.1</b> What are experimental statistics?</a><ul>
<li class="chapter" data-level="6.1.1" data-path="bigpic.html"><a href="bigpic.html#descriptive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Descriptive modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="bigpic.html"><a href="bigpic.html#statistical-inference"><i class="fa fa-check"></i><b>6.1.2</b> Statistical inference</a></li>
<li class="chapter" data-level="6.1.3" data-path="bigpic.html"><a href="bigpic.html#experimental-design"><i class="fa fa-check"></i><b>6.1.3</b> Experimental design</a></li>
<li class="chapter" data-level="6.1.4" data-path="bigpic.html"><a href="bigpic.html#statistics-as-an-anti-bias-framework"><i class="fa fa-check"></i><b>6.1.4</b> Statistics as an anti-bias framework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>7</b> Statistical Sampling</a><ul>
<li class="chapter" data-level="7.1" data-path="sampling.html"><a href="sampling.html#experimental-units"><i class="fa fa-check"></i><b>7.1</b> Experimental units</a><ul>
<li class="chapter" data-level="7.1.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-to-define-the-experimental-unit"><i class="fa fa-check"></i><b>7.1.1</b> A simple test to define the experimental unit</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="sampling.html"><a href="sampling.html#independent-replicates"><i class="fa fa-check"></i><b>7.2</b> Independent Replicates</a><ul>
<li class="chapter" data-level="7.2.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-for-independence"><i class="fa fa-check"></i><b>7.2.1</b> A simple test for independence</a></li>
<li class="chapter" data-level="7.2.2" data-path="sampling.html"><a href="sampling.html#some-replication-examples"><i class="fa fa-check"></i><b>7.2.2</b> Some replication examples</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="sampling.html"><a href="sampling.html#random-process"><i class="fa fa-check"></i><b>7.3</b> Random process</a></li>
<li class="chapter" data-level="7.4" data-path="sampling.html"><a href="sampling.html#statistically-valid-samples"><i class="fa fa-check"></i><b>7.4</b> Statistically valid samples</a><ul>
<li class="chapter" data-level="7.4.1" data-path="sampling.html"><a href="sampling.html#select-random-subjects"><i class="fa fa-check"></i><b>7.4.1</b> Select random subjects</a></li>
<li class="chapter" data-level="7.4.2" data-path="sampling.html"><a href="sampling.html#randomize-to-sequence"><i class="fa fa-check"></i><b>7.4.2</b> Randomize to sequence</a></li>
<li class="chapter" data-level="7.4.3" data-path="sampling.html"><a href="sampling.html#randomize-to-location"><i class="fa fa-check"></i><b>7.4.3</b> Randomize to location</a></li>
<li class="chapter" data-level="7.4.4" data-path="sampling.html"><a href="sampling.html#randomize-to-block"><i class="fa fa-check"></i><b>7.4.4</b> Randomize to block</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="sampling.html"><a href="sampling.html#independence-of-replicates"><i class="fa fa-check"></i><b>7.5</b> Independence of replicates</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>8</b> Data Classification</a><ul>
<li class="chapter" data-level="8.1" data-path="data.html"><a href="data.html#dependent-and-independent-variables"><i class="fa fa-check"></i><b>8.1</b> Dependent and independent variables</a><ul>
<li class="chapter" data-level="8.1.1" data-path="data.html"><a href="data.html#statistical-notation"><i class="fa fa-check"></i><b>8.1.1</b> Statistical notation</a></li>
<li class="chapter" data-level="8.1.2" data-path="data.html"><a href="data.html#when-there-is-no-independent-variable"><i class="fa fa-check"></i><b>8.1.2</b> When there is no independent variable</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="data.html"><a href="data.html#discrete-or-continuous-variables"><i class="fa fa-check"></i><b>8.2</b> Discrete or continuous variables</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data.html"><a href="data.html#measured-variables"><i class="fa fa-check"></i><b>8.2.1</b> Measured variables</a></li>
<li class="chapter" data-level="8.2.2" data-path="data.html"><a href="data.html#discrete-categorical-and-ordinal-variables"><i class="fa fa-check"></i><b>8.2.2</b> Discrete categorical and ordinal variables</a></li>
<li class="chapter" data-level="8.2.3" data-path="data.html"><a href="data.html#rescaling-variables"><i class="fa fa-check"></i><b>8.2.3</b> Rescaling variables</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data.html"><a href="data.html#working-with-data"><i class="fa fa-check"></i><b>8.3</b> Working with data</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data.html"><a href="data.html#entering-data"><i class="fa fa-check"></i><b>8.3.1</b> Entering data</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data.html"><a href="data.html#summary-1"><i class="fa fa-check"></i><b>8.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dispersion.html"><a href="dispersion.html"><i class="fa fa-check"></i><b>9</b> Variability, Accuracy and Precision</a><ul>
<li class="chapter" data-level="9.0.1" data-path="dispersion.html"><a href="dispersion.html#data-are-messy"><i class="fa fa-check"></i><b>9.0.1</b> Data are messy</a></li>
<li class="chapter" data-level="9.1" data-path="dispersion.html"><a href="dispersion.html#illustration-of-a-model-with-error"><i class="fa fa-check"></i><b>9.1</b> Illustration of a model with error</a></li>
<li class="chapter" data-level="9.2" data-path="dispersion.html"><a href="dispersion.html#variance-quantifying-variation-by-least-squares"><i class="fa fa-check"></i><b>9.2</b> Variance: Quantifying variation by least squares</a><ul>
<li class="chapter" data-level="9.2.1" data-path="dispersion.html"><a href="dispersion.html#definition-of-variance"><i class="fa fa-check"></i><b>9.2.1</b> Definition of variance</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="dispersion.html"><a href="dispersion.html#standard-deviation"><i class="fa fa-check"></i><b>9.3</b> Standard deviation</a><ul>
<li class="chapter" data-level="9.3.1" data-path="dispersion.html"><a href="dispersion.html#what-does-the-standard-deviation-tell-us"><i class="fa fa-check"></i><b>9.3.1</b> What does the standard deviation tell us</a></li>
<li class="chapter" data-level="9.3.2" data-path="dispersion.html"><a href="dispersion.html#ways-of-describing-sample-variability"><i class="fa fa-check"></i><b>9.3.2</b> Ways of describing sample variability</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="dispersion.html"><a href="dispersion.html#precision-and-accuracy"><i class="fa fa-check"></i><b>9.4</b> Precision and Accuracy</a></li>
<li class="chapter" data-level="9.5" data-path="dispersion.html"><a href="dispersion.html#standard-error"><i class="fa fa-check"></i><b>9.5</b> Standard error</a><ul>
<li class="chapter" data-level="9.5.1" data-path="dispersion.html"><a href="dispersion.html#what-exactly-does-the-standard-error-represent"><i class="fa fa-check"></i><b>9.5.1</b> What exactly does the standard error represent?</a></li>
<li class="chapter" data-level="9.5.2" data-path="dispersion.html"><a href="dispersion.html#should-i-use-sd-or-sem"><i class="fa fa-check"></i><b>9.5.2</b> “Should I use SD or SEM?”</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dispersion.html"><a href="dispersion.html#confidence-intervals"><i class="fa fa-check"></i><b>9.6</b> Confidence intervals</a><ul>
<li class="chapter" data-level="9.6.1" data-path="dispersion.html"><a href="dispersion.html#ci-definition"><i class="fa fa-check"></i><b>9.6.1</b> CI Definition</a></li>
<li class="chapter" data-level="9.6.2" data-path="dispersion.html"><a href="dispersion.html#evaluating-confidence-intervals"><i class="fa fa-check"></i><b>9.6.2</b> Evaluating confidence intervals</a></li>
<li class="chapter" data-level="9.6.3" data-path="dispersion.html"><a href="dispersion.html#ci-simulation-tools"><i class="fa fa-check"></i><b>9.6.3</b> CI Simulation tools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="dispersion.html"><a href="dispersion.html#key-take-aways"><i class="fa fa-check"></i><b>9.7</b> Key take aways</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="hypotheses.html"><a href="hypotheses.html"><i class="fa fa-check"></i><b>10</b> Framing statistical hypotheses</a><ul>
<li class="chapter" data-level="10.1" data-path="hypotheses.html"><a href="hypotheses.html#the-decision-process"><i class="fa fa-check"></i><b>10.1</b> The decision process</a></li>
<li class="chapter" data-level="10.2" data-path="hypotheses.html"><a href="hypotheses.html#popper-and-falsification"><i class="fa fa-check"></i><b>10.2</b> Popper and falsification</a></li>
<li class="chapter" data-level="10.3" data-path="hypotheses.html"><a href="hypotheses.html#statistical-hypothesis-rubric"><i class="fa fa-check"></i><b>10.3</b> Statistical hypothesis rubric</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="error.html"><a href="error.html"><i class="fa fa-check"></i><b>11</b> Error</a><ul>
<li class="chapter" data-level="11.1" data-path="error.html"><a href="error.html#setting-type-1-and-type-2-error-thresholds"><i class="fa fa-check"></i><b>11.1</b> Setting type 1 and type 2 error thresholds</a><ul>
<li class="chapter" data-level="11.1.1" data-path="error.html"><a href="error.html#setting-alpha-the-type-1-error"><i class="fa fa-check"></i><b>11.1.1</b> Setting alpha-the type 1 error</a></li>
<li class="chapter" data-level="11.1.2" data-path="error.html"><a href="error.html#power-setting-beta-the-type-2-error"><i class="fa fa-check"></i><b>11.1.2</b> Power: Setting beta-the type 2 error</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="error.html"><a href="error.html#striking-the-right-balance"><i class="fa fa-check"></i><b>11.2</b> Striking the right balance</a></li>
<li class="chapter" data-level="11.3" data-path="error.html"><a href="error.html#false-discovery-rate"><i class="fa fa-check"></i><b>11.3</b> False discovery rate</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="pvalues.html"><a href="pvalues.html"><i class="fa fa-check"></i><b>12</b> P Values</a><ul>
<li class="chapter" data-level="12.1" data-path="pvalues.html"><a href="pvalues.html#definition"><i class="fa fa-check"></i><b>12.1</b> Definition</a></li>
<li class="chapter" data-level="12.2" data-path="pvalues.html"><a href="pvalues.html#null-test-statistic-distributions"><i class="fa fa-check"></i><b>12.2</b> Null test statistic distributions</a></li>
<li class="chapter" data-level="12.3" data-path="pvalues.html"><a href="pvalues.html#computing-p-values"><i class="fa fa-check"></i><b>12.3</b> Computing p-values</a></li>
<li class="chapter" data-level="12.4" data-path="pvalues.html"><a href="pvalues.html#p-values-from-null-tests-are-uniform"><i class="fa fa-check"></i><b>12.4</b> P-values from null tests are uniform</a></li>
<li class="chapter" data-level="12.5" data-path="pvalues.html"><a href="pvalues.html#how-p-values-become-error-probabilities"><i class="fa fa-check"></i><b>12.5</b> How P-values become error probabilities</a></li>
<li class="chapter" data-level="12.6" data-path="pvalues.html"><a href="pvalues.html#p-values-as-dichotomous-switches"><i class="fa fa-check"></i><b>12.6</b> P-values as dichotomous switches</a></li>
<li class="chapter" data-level="12.7" data-path="pvalues.html"><a href="pvalues.html#p-values-in-true-effects"><i class="fa fa-check"></i><b>12.7</b> P-values in true effects</a></li>
<li class="chapter" data-level="12.8" data-path="pvalues.html"><a href="pvalues.html#criticisms-of-p-values"><i class="fa fa-check"></i><b>12.8</b> Criticisms of p-values</a></li>
<li class="chapter" data-level="12.9" data-path="pvalues.html"><a href="pvalues.html#summary-how-to-use-p-values"><i class="fa fa-check"></i><b>12.9</b> Summary: How to use p-values</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="jaxwest7.html"><a href="jaxwest7.html"><i class="fa fa-check"></i><b>13</b> Reproducible Data Munging in R</a><ul>
<li class="chapter" data-level="13.1" data-path="jaxwest7.html"><a href="jaxwest7.html#jaxwest7-data"><i class="fa fa-check"></i><b>13.1</b> Jaxwest7 data</a><ul>
<li class="chapter" data-level="13.1.1" data-path="jaxwest7.html"><a href="jaxwest7.html#inspect-the-source-data"><i class="fa fa-check"></i><b>13.1.1</b> Inspect the source data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-glucose-data-into-r"><i class="fa fa-check"></i><b>13.2</b> Munge the glucose data into R</a></li>
<li class="chapter" data-level="13.3" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-bodyweight-data"><i class="fa fa-check"></i><b>13.3</b> Munge the bodyweight data</a></li>
<li class="chapter" data-level="13.4" data-path="jaxwest7.html"><a href="jaxwest7.html#merge-and-explore-the-glucose-and-bw-data"><i class="fa fa-check"></i><b>13.4</b> Merge and explore the glucose and bw data</a></li>
<li class="chapter" data-level="13.5" data-path="jaxwest7.html"><a href="jaxwest7.html#summary-2"><i class="fa fa-check"></i><b>13.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="binomial.html"><a href="binomial.html"><i class="fa fa-check"></i><b>14</b> The Binomial Distribution</a><ul>
<li class="chapter" data-level="14.1" data-path="binomial.html"><a href="binomial.html#dbinom"><i class="fa fa-check"></i><b>14.1</b> dbinom</a></li>
<li class="chapter" data-level="14.2" data-path="binomial.html"><a href="binomial.html#pbinom"><i class="fa fa-check"></i><b>14.2</b> pbinom</a></li>
<li class="chapter" data-level="14.3" data-path="binomial.html"><a href="binomial.html#qbinom"><i class="fa fa-check"></i><b>14.3</b> qbinom</a></li>
<li class="chapter" data-level="14.4" data-path="binomial.html"><a href="binomial.html#rbinom"><i class="fa fa-check"></i><b>14.4</b> rbinom</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="poisson.html"><a href="poisson.html"><i class="fa fa-check"></i><b>15</b> The Poisson Distribution</a><ul>
<li class="chapter" data-level="15.1" data-path="poisson.html"><a href="poisson.html#poisson-events"><i class="fa fa-check"></i><b>15.1</b> Poisson Events</a></li>
<li class="chapter" data-level="15.2" data-path="poisson.html"><a href="poisson.html#dpois"><i class="fa fa-check"></i><b>15.2</b> dpois</a></li>
<li class="chapter" data-level="15.3" data-path="poisson.html"><a href="poisson.html#ppois"><i class="fa fa-check"></i><b>15.3</b> ppois</a></li>
<li class="chapter" data-level="15.4" data-path="poisson.html"><a href="poisson.html#rpois"><i class="fa fa-check"></i><b>15.4</b> rpois</a></li>
<li class="chapter" data-level="15.5" data-path="poisson.html"><a href="poisson.html#overdispersion"><i class="fa fa-check"></i><b>15.5</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="normal.html"><a href="normal.html"><i class="fa fa-check"></i><b>16</b> The Normal Distribution</a><ul>
<li class="chapter" data-level="16.0.1" data-path="normal.html"><a href="normal.html#the-standard-normal"><i class="fa fa-check"></i><b>16.0.1</b> The Standard Normal</a></li>
<li class="chapter" data-level="16.1" data-path="normal.html"><a href="normal.html#dnorm"><i class="fa fa-check"></i><b>16.1</b> dnorm</a></li>
<li class="chapter" data-level="16.2" data-path="normal.html"><a href="normal.html#pnorm"><i class="fa fa-check"></i><b>16.2</b> pnorm</a><ul>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-p-values-using-pnorm"><i class="fa fa-check"></i>Calculating “p-values”" using pnorm</a></li>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-percentiles-using-pnorm"><i class="fa fa-check"></i>Calculating percentiles using pnorm</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="normal.html"><a href="normal.html#qnorm"><i class="fa fa-check"></i><b>16.3</b> qnorm</a></li>
<li class="chapter" data-level="16.4" data-path="normal.html"><a href="normal.html#rnorm"><i class="fa fa-check"></i><b>16.4</b> rnorm</a><ul>
<li class="chapter" data-level="16.4.1" data-path="normal.html"><a href="normal.html#plotting-histograms-of-some-rnorm-samples"><i class="fa fa-check"></i><b>16.4.1</b> Plotting histograms of some rnorm samples</a></li>
<li class="chapter" data-level="16.4.2" data-path="normal.html"><a href="normal.html#bins-and-binwidth"><i class="fa fa-check"></i><b>16.4.2</b> Bins and Binwidth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="categorical.html"><a href="categorical.html"><i class="fa fa-check"></i><b>17</b> Statistics for Categorical Data</a><ul>
<li class="chapter" data-level="17.1" data-path="categorical.html"><a href="categorical.html#types-of-categorical-data"><i class="fa fa-check"></i><b>17.1</b> Types of categorical data</a><ul>
<li class="chapter" data-level="17.1.1" data-path="categorical.html"><a href="categorical.html#proportions"><i class="fa fa-check"></i><b>17.1.1</b> Proportions</a></li>
<li class="chapter" data-level="17.1.2" data-path="categorical.html"><a href="categorical.html#frequencies"><i class="fa fa-check"></i><b>17.1.2</b> Frequencies</a></li>
<li class="chapter" data-level="17.1.3" data-path="categorical.html"><a href="categorical.html#associations"><i class="fa fa-check"></i><b>17.1.3</b> Associations</a></li>
<li class="chapter" data-level="17.1.4" data-path="categorical.html"><a href="categorical.html#statistics-covered-here"><i class="fa fa-check"></i><b>17.1.4</b> Statistics Covered Here</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="categorical.html"><a href="categorical.html#exact-v-asymptotic-calculations-of-p-values"><i class="fa fa-check"></i><b>17.2</b> Exact v Asymptotic Calculations of p-values</a><ul>
<li class="chapter" data-level="17.2.1" data-path="categorical.html"><a href="categorical.html#choosing-exact-or-asymptotic"><i class="fa fa-check"></i><b>17.2.1</b> Choosing exact or asymptotic</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="categorical.html"><a href="categorical.html#overview-of-the-types-of-hypothesis-testing"><i class="fa fa-check"></i><b>17.3</b> Overview of the types of hypothesis testing</a><ul>
<li class="chapter" data-level="17.3.1" data-path="categorical.html"><a href="categorical.html#proportion-analysis"><i class="fa fa-check"></i><b>17.3.1</b> Proportion analysis</a></li>
<li class="chapter" data-level="17.3.2" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-testing"><i class="fa fa-check"></i><b>17.3.2</b> Goodness of fit testing</a></li>
<li class="chapter" data-level="17.3.3" data-path="categorical.html"><a href="categorical.html#contingency-analysis"><i class="fa fa-check"></i><b>17.3.3</b> Contingency Analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="categorical.html"><a href="categorical.html#comparing-proportions"><i class="fa fa-check"></i><b>17.4</b> Comparing proportions</a><ul>
<li class="chapter" data-level="17.4.1" data-path="categorical.html"><a href="categorical.html#a-mouse-t-cell-pilot-experiment-the-cytokine-inducible-antigen-gradstudin"><i class="fa fa-check"></i><b>17.4.1</b> A Mouse T Cell Pilot Experiment: The Cytokine-inducible antigen gradstudin</a></li>
<li class="chapter" data-level="17.4.2" data-path="categorical.html"><a href="categorical.html#calculating-proportions"><i class="fa fa-check"></i><b>17.4.2</b> Calculating Proportions</a></li>
<li class="chapter" data-level="17.4.3" data-path="categorical.html"><a href="categorical.html#what-a-proportion-estimates"><i class="fa fa-check"></i><b>17.4.3</b> What A Proportion Estimates</a></li>
<li class="chapter" data-level="17.4.4" data-path="categorical.html"><a href="categorical.html#confidence-intervals-of-proportions"><i class="fa fa-check"></i><b>17.4.4</b> Confidence Intervals of Proportions</a></li>
<li class="chapter" data-level="17.4.5" data-path="categorical.html"><a href="categorical.html#a-one-sample-proportion-test"><i class="fa fa-check"></i><b>17.4.5</b> A One-Sample Proportion Test</a></li>
<li class="chapter" data-level="17.4.6" data-path="categorical.html"><a href="categorical.html#an-exact-test-for-one-proportion"><i class="fa fa-check"></i><b>17.4.6</b> An exact test for one proportion</a></li>
<li class="chapter" data-level="17.4.7" data-path="categorical.html"><a href="categorical.html#comparing-two-proportions"><i class="fa fa-check"></i><b>17.4.7</b> Comparing Two Proportions</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="categorical.html"><a href="categorical.html#exact-tests-for-two-proportions"><i class="fa fa-check"></i><b>17.5</b> Exact tests for two proportions</a></li>
<li class="chapter" data-level="17.6" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-tests"><i class="fa fa-check"></i><b>17.6</b> Goodness of fit Tests</a><ul>
<li class="chapter" data-level="17.6.1" data-path="categorical.html"><a href="categorical.html#an-exact-goodness-of-fit-test"><i class="fa fa-check"></i><b>17.6.1</b> An Exact Goodness of Fit test</a></li>
<li class="chapter" data-level="17.6.2" data-path="categorical.html"><a href="categorical.html#an-approximate-goodness-of-fit-test"><i class="fa fa-check"></i><b>17.6.2</b> An Approximate Goodness of Fit test</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="categorical.html"><a href="categorical.html#contingency-testing"><i class="fa fa-check"></i><b>17.7</b> Contingency Testing</a><ul>
<li class="chapter" data-level="17.7.1" data-path="categorical.html"><a href="categorical.html#interpretation-of-contingency-results"><i class="fa fa-check"></i><b>17.7.1</b> Interpretation of Contingency Results</a></li>
<li class="chapter" data-level="17.7.2" data-path="categorical.html"><a href="categorical.html#write-up-4"><i class="fa fa-check"></i><b>17.7.2</b> Write Up</a></li>
<li class="chapter" data-level="17.7.3" data-path="categorical.html"><a href="categorical.html#interpretation-of-chi-square-output"><i class="fa fa-check"></i><b>17.7.3</b> Interpretation of chi-square output</a></li>
<li class="chapter" data-level="17.7.4" data-path="categorical.html"><a href="categorical.html#write-up-5"><i class="fa fa-check"></i><b>17.7.4</b> Write Up</a></li>
<li class="chapter" data-level="17.7.5" data-path="categorical.html"><a href="categorical.html#which-contingency-test-is-best"><i class="fa fa-check"></i><b>17.7.5</b> Which contingency test is best?</a></li>
<li class="chapter" data-level="17.7.6" data-path="categorical.html"><a href="categorical.html#higher-dimension-contingency-analysis"><i class="fa fa-check"></i><b>17.7.6</b> Higher dimension contingency analysis</a></li>
<li class="chapter" data-level="17.7.7" data-path="categorical.html"><a href="categorical.html#other-experimental-designs-involving-categorical-data"><i class="fa fa-check"></i><b>17.7.7</b> Other experimental designs involving categorical data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="categorical.html"><a href="categorical.html#power-analysis-for-proportion-tests"><i class="fa fa-check"></i><b>17.8</b> Power analysis for proportion tests</a></li>
<li class="chapter" data-level="17.9" data-path="categorical.html"><a href="categorical.html#power-analysis-functions-for-proportion-tests"><i class="fa fa-check"></i><b>17.9</b> Power analysis functions for proportion tests</a></li>
<li class="chapter" data-level="17.10" data-path="categorical.html"><a href="categorical.html#graphing-proportions"><i class="fa fa-check"></i><b>17.10</b> Graphing Proportions</a></li>
<li class="chapter" data-level="17.11" data-path="categorical.html"><a href="categorical.html#key-takeaways"><i class="fa fa-check"></i><b>17.11</b> Key Takeaways</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="chisquare.html"><a href="chisquare.html"><i class="fa fa-check"></i><b>18</b> The Chi-square Distribution</a><ul>
<li class="chapter" data-level="18.1" data-path="chisquare.html"><a href="chisquare.html#background"><i class="fa fa-check"></i><b>18.1</b> Background</a></li>
<li class="chapter" data-level="18.2" data-path="chisquare.html"><a href="chisquare.html#dchisq"><i class="fa fa-check"></i><b>18.2</b> dchisq</a></li>
<li class="chapter" data-level="18.3" data-path="chisquare.html"><a href="chisquare.html#pchisq"><i class="fa fa-check"></i><b>18.3</b> pchisq</a><ul>
<li class="chapter" data-level="18.3.1" data-path="chisquare.html"><a href="chisquare.html#calculating-p-values-from-pchisq"><i class="fa fa-check"></i><b>18.3.1</b> Calculating p-values from pchisq</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="chisquare.html"><a href="chisquare.html#qchisq"><i class="fa fa-check"></i><b>18.4</b> qchisq</a></li>
<li class="chapter" data-level="18.5" data-path="chisquare.html"><a href="chisquare.html#rchisq"><i class="fa fa-check"></i><b>18.5</b> rchisq</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="nonparametrics.html"><a href="nonparametrics.html"><i class="fa fa-check"></i><b>19</b> Nonparametric Statistical Tests</a><ul>
<li class="chapter" data-level="19.1" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-sampling-distributions"><i class="fa fa-check"></i><b>19.1</b> Nonparametric sampling distributions</a></li>
<li class="chapter" data-level="19.2" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-discrete-data"><i class="fa fa-check"></i><b>19.2</b> Experiments involving discrete data</a><ul>
<li class="chapter" data-level="19.2.1" data-path="nonparametrics.html"><a href="nonparametrics.html#ordered-data-1"><i class="fa fa-check"></i><b>19.2.1</b> Ordered data</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-deviant-data"><i class="fa fa-check"></i><b>19.3</b> Experiments involving deviant Data</a></li>
<li class="chapter" data-level="19.4" data-path="nonparametrics.html"><a href="nonparametrics.html#sign-test"><i class="fa fa-check"></i><b>19.4</b> Sign Test</a><ul>
<li class="chapter" data-level="19.4.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation"><i class="fa fa-check"></i><b>19.4.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.4.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-6"><i class="fa fa-check"></i><b>19.4.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-one-group"><i class="fa fa-check"></i><b>19.5</b> Wilcoxon Sign Rank Test for One Group</a><ul>
<li class="chapter" data-level="19.5.1" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-experimental-designs"><i class="fa fa-check"></i><b>19.5.1</b> Wilcoxon Sign Rank Experimental Designs</a></li>
<li class="chapter" data-level="19.5.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-1"><i class="fa fa-check"></i><b>19.5.2</b> Interpretation</a></li>
<li class="chapter" data-level="19.5.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-7"><i class="fa fa-check"></i><b>19.5.3</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-mann-whitney-rank-sum-test-for-2-independent-groups"><i class="fa fa-check"></i><b>19.6</b> Wilcoxon Mann Whitney Rank Sum Test for 2 independent groups</a><ul>
<li class="chapter" data-level="19.6.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-2"><i class="fa fa-check"></i><b>19.6.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.6.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-8"><i class="fa fa-check"></i><b>19.6.2</b> Write Up</a></li>
<li class="chapter" data-level="19.6.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot"><i class="fa fa-check"></i><b>19.6.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-paired-groups"><i class="fa fa-check"></i><b>19.7</b> Wilcoxon Sign Rank Test for paired groups</a><ul>
<li class="chapter" data-level="19.7.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-3"><i class="fa fa-check"></i><b>19.7.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.7.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-9"><i class="fa fa-check"></i><b>19.7.2</b> Write up</a></li>
<li class="chapter" data-level="19.7.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot-1"><i class="fa fa-check"></i><b>19.7.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="nonparametrics.html"><a href="nonparametrics.html#kruskal-wallis"><i class="fa fa-check"></i><b>19.8</b> Kruskal-Wallis</a><ul>
<li class="chapter" data-level="19.8.1" data-path="nonparametrics.html"><a href="nonparametrics.html#the-experimental-design"><i class="fa fa-check"></i><b>19.8.1</b> The experimental design</a></li>
<li class="chapter" data-level="19.8.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-10"><i class="fa fa-check"></i><b>19.8.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.9" data-path="nonparametrics.html"><a href="nonparametrics.html#friedman-test"><i class="fa fa-check"></i><b>19.9</b> Friedman test</a><ul>
<li class="chapter" data-level="19.9.1" data-path="nonparametrics.html"><a href="nonparametrics.html#blocked-experimental-design"><i class="fa fa-check"></i><b>19.9.1</b> Blocked experimental design</a></li>
<li class="chapter" data-level="19.9.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-4"><i class="fa fa-check"></i><b>19.9.2</b> Interpretation</a></li>
<li class="chapter" data-level="19.9.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-11"><i class="fa fa-check"></i><b>19.9.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="19.10" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-power-calculations"><i class="fa fa-check"></i><b>19.10</b> Nonparametric power calculations</a><ul>
<li class="chapter" data-level="19.10.1" data-path="nonparametrics.html"><a href="nonparametrics.html#how-it-works"><i class="fa fa-check"></i><b>19.10.1</b> How it works</a></li>
<li class="chapter" data-level="19.10.2" data-path="nonparametrics.html"><a href="nonparametrics.html#simulating-response-data"><i class="fa fa-check"></i><b>19.10.2</b> Simulating response data</a></li>
<li class="chapter" data-level="19.10.3" data-path="nonparametrics.html"><a href="nonparametrics.html#an-example"><i class="fa fa-check"></i><b>19.10.3</b> An example</a></li>
<li class="chapter" data-level="19.10.4" data-path="nonparametrics.html"><a href="nonparametrics.html#nonpara.pwr"><i class="fa fa-check"></i><b>19.10.4</b> nonpara.pwr</a></li>
</ul></li>
<li class="chapter" data-level="19.11" data-path="nonparametrics.html"><a href="nonparametrics.html#summary-3"><i class="fa fa-check"></i><b>19.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="signrank.html"><a href="signrank.html"><i class="fa fa-check"></i><b>20</b> Signed Rank Distribution</a><ul>
<li class="chapter" data-level="20.1" data-path="signrank.html"><a href="signrank.html#transformation-of-data-into-sign-ranks"><i class="fa fa-check"></i><b>20.1</b> Transformation of data into sign ranks</a><ul>
<li class="chapter" data-level="20.1.1" data-path="signrank.html"><a href="signrank.html#for-a-one-group-sample"><i class="fa fa-check"></i><b>20.1.1</b> For a one group sample</a></li>
<li class="chapter" data-level="20.1.2" data-path="signrank.html"><a href="signrank.html#for-a-paired-sample"><i class="fa fa-check"></i><b>20.1.2</b> For a paired sample</a></li>
<li class="chapter" data-level="20.1.3" data-path="signrank.html"><a href="signrank.html#the-sign-rank-test-statistic-in-r"><i class="fa fa-check"></i><b>20.1.3</b> The sign rank test statistic in R</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="signrank.html"><a href="signrank.html#rs-four-sign-rank-distribution-functions"><i class="fa fa-check"></i><b>20.2</b> R’s Four Sign Rank Distribution Functions</a><ul>
<li class="chapter" data-level="20.2.1" data-path="signrank.html"><a href="signrank.html#dsignrank"><i class="fa fa-check"></i><b>20.2.1</b> dsignrank</a></li>
<li class="chapter" data-level="20.2.2" data-path="signrank.html"><a href="signrank.html#psignrank"><i class="fa fa-check"></i><b>20.2.2</b> psignrank</a></li>
<li class="chapter" data-level="20.2.3" data-path="signrank.html"><a href="signrank.html#qsignrank"><i class="fa fa-check"></i><b>20.2.3</b> qsignrank</a></li>
<li class="chapter" data-level="20.2.4" data-path="signrank.html"><a href="signrank.html#rsignrank"><i class="fa fa-check"></i><b>20.2.4</b> rsignrank</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="ranksum.html"><a href="ranksum.html"><i class="fa fa-check"></i><b>21</b> Rank Sum Distribution</a><ul>
<li class="chapter" data-level="21.0.1" data-path="ranksum.html"><a href="ranksum.html#transformation-of-data-into-rank-summs"><i class="fa fa-check"></i><b>21.0.1</b> Transformation of data into rank summs</a></li>
<li class="chapter" data-level="21.0.2" data-path="ranksum.html"><a href="ranksum.html#the-sign-rank-test-statistic-in-r-1"><i class="fa fa-check"></i><b>21.0.2</b> The sign rank test statistic in R</a></li>
<li class="chapter" data-level="21.1" data-path="ranksum.html"><a href="ranksum.html#rs-four-sign-rank-distribution-functions-1"><i class="fa fa-check"></i><b>21.1</b> R’s Four Sign Rank Distribution Functions</a><ul>
<li class="chapter" data-level="21.1.1" data-path="ranksum.html"><a href="ranksum.html#dwilcox"><i class="fa fa-check"></i><b>21.1.1</b> dwilcox</a></li>
<li class="chapter" data-level="21.1.2" data-path="ranksum.html"><a href="ranksum.html#pwilcox"><i class="fa fa-check"></i><b>21.1.2</b> pwilcox</a></li>
<li class="chapter" data-level="21.1.3" data-path="ranksum.html"><a href="ranksum.html#qwilcox"><i class="fa fa-check"></i><b>21.1.3</b> qwilcox</a></li>
<li class="chapter" data-level="21.1.4" data-path="ranksum.html"><a href="ranksum.html#rwilcox"><i class="fa fa-check"></i><b>21.1.4</b> rwilcox</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="ttests.html"><a href="ttests.html"><i class="fa fa-check"></i><b>22</b> The t-tests</a><ul>
<li class="chapter" data-level="22.1" data-path="ttests.html"><a href="ttests.html#assumptions-for-t-tests"><i class="fa fa-check"></i><b>22.1</b> Assumptions for t-tests</a><ul>
<li class="chapter" data-level="22.1.1" data-path="ttests.html"><a href="ttests.html#strong-assumptions-t-tests-are-invalid-if-these-not-met"><i class="fa fa-check"></i><b>22.1.1</b> Strong assumptions, t-tests are invalid if these not met:</a></li>
<li class="chapter" data-level="22.1.2" data-path="ttests.html"><a href="ttests.html#weak-assumptions-t-tests-are-less-robust-when-these-are-not-met"><i class="fa fa-check"></i><b>22.1.2</b> Weak assumptions: t-tests are less robust when these are not met:</a></li>
<li class="chapter" data-level="22.1.3" data-path="ttests.html"><a href="ttests.html#on-random-residual-error"><i class="fa fa-check"></i><b>22.1.3</b> On random residual error</a></li>
<li class="chapter" data-level="22.1.4" data-path="ttests.html"><a href="ttests.html#on-the-normality-assumption"><i class="fa fa-check"></i><b>22.1.4</b> On the normality assumption</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="ttests.html"><a href="ttests.html#the-t-statistic"><i class="fa fa-check"></i><b>22.2</b> The t Statistic</a></li>
<li class="chapter" data-level="22.3" data-path="ttests.html"><a href="ttests.html#one-sample-t-tests"><i class="fa fa-check"></i><b>22.3</b> One sample t tests</a><ul>
<li class="chapter" data-level="22.3.1" data-path="ttests.html"><a href="ttests.html#example"><i class="fa fa-check"></i><b>22.3.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="ttests.html"><a href="ttests.html#unpaired-t-tests"><i class="fa fa-check"></i><b>22.4</b> Unpaired t tests</a><ul>
<li class="chapter" data-level="22.4.1" data-path="ttests.html"><a href="ttests.html#students-t-test"><i class="fa fa-check"></i><b>22.4.1</b> Student’s t-test</a></li>
<li class="chapter" data-level="22.4.2" data-path="ttests.html"><a href="ttests.html#welchs-t-test"><i class="fa fa-check"></i><b>22.4.2</b> Welch’s t-test</a></li>
<li class="chapter" data-level="22.4.3" data-path="ttests.html"><a href="ttests.html#example-1"><i class="fa fa-check"></i><b>22.4.3</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="22.5" data-path="ttests.html"><a href="ttests.html#paired-t-tests"><i class="fa fa-check"></i><b>22.5</b> Paired t tests</a><ul>
<li class="chapter" data-level="22.5.1" data-path="ttests.html"><a href="ttests.html#example-2"><i class="fa fa-check"></i><b>22.5.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="ttests.html"><a href="ttests.html#t-test-hypotheses"><i class="fa fa-check"></i><b>22.6</b> t Test Hypotheses</a></li>
<li class="chapter" data-level="22.7" data-path="ttests.html"><a href="ttests.html#one-sided-or-two-sided"><i class="fa fa-check"></i><b>22.7</b> One-sided or two-sided??</a><ul>
<li class="chapter" data-level="22.7.1" data-path="ttests.html"><a href="ttests.html#one-sample-hypotheses"><i class="fa fa-check"></i><b>22.7.1</b> One sample hypotheses</a></li>
<li class="chapter" data-level="22.7.2" data-path="ttests.html"><a href="ttests.html#unpaired-hypotheses"><i class="fa fa-check"></i><b>22.7.2</b> Unpaired hypotheses</a></li>
<li class="chapter" data-level="22.7.3" data-path="ttests.html"><a href="ttests.html#paired-hypotheses"><i class="fa fa-check"></i><b>22.7.3</b> Paired hypotheses</a></li>
</ul></li>
<li class="chapter" data-level="22.8" data-path="ttests.html"><a href="ttests.html#confidence-intervals-1"><i class="fa fa-check"></i><b>22.8</b> Confidence Intervals</a></li>
<li class="chapter" data-level="22.9" data-path="ttests.html"><a href="ttests.html#reliability-of-small-samples"><i class="fa fa-check"></i><b>22.9</b> Reliability of small samples</a><ul>
<li class="chapter" data-level="22.9.1" data-path="ttests.html"><a href="ttests.html#reporting-sd-or-sem"><i class="fa fa-check"></i><b>22.9.1</b> Reporting SD or SEM?</a></li>
</ul></li>
<li class="chapter" data-level="22.10" data-path="ttests.html"><a href="ttests.html#running-the-analysis-in-r"><i class="fa fa-check"></i><b>22.10</b> Running the analysis in R</a><ul>
<li class="chapter" data-level="22.10.1" data-path="ttests.html"><a href="ttests.html#one-sample-t-test"><i class="fa fa-check"></i><b>22.10.1</b> One sample t test</a></li>
<li class="chapter" data-level="22.10.2" data-path="ttests.html"><a href="ttests.html#unpaired-t-test"><i class="fa fa-check"></i><b>22.10.2</b> Unpaired t test</a></li>
<li class="chapter" data-level="22.10.3" data-path="ttests.html"><a href="ttests.html#paired-t-test"><i class="fa fa-check"></i><b>22.10.3</b> Paired t Test</a></li>
</ul></li>
<li class="chapter" data-level="22.11" data-path="ttests.html"><a href="ttests.html#plotting-t-tests"><i class="fa fa-check"></i><b>22.11</b> Plotting t Tests</a><ul>
<li class="chapter" data-level="22.11.1" data-path="ttests.html"><a href="ttests.html#one-sample"><i class="fa fa-check"></i><b>22.11.1</b> One-sample</a></li>
<li class="chapter" data-level="22.11.2" data-path="ttests.html"><a href="ttests.html#unpaired"><i class="fa fa-check"></i><b>22.11.2</b> Unpaired</a></li>
<li class="chapter" data-level="22.11.3" data-path="ttests.html"><a href="ttests.html#paired"><i class="fa fa-check"></i><b>22.11.3</b> Paired</a></li>
</ul></li>
<li class="chapter" data-level="22.12" data-path="ttests.html"><a href="ttests.html#t-test-power"><i class="fa fa-check"></i><b>22.12</b> t Test Power</a><ul>
<li class="chapter" data-level="22.12.1" data-path="ttests.html"><a href="ttests.html#pwr.t.test"><i class="fa fa-check"></i><b>22.12.1</b> pwr.t.test</a></li>
<li class="chapter" data-level="22.12.2" data-path="ttests.html"><a href="ttests.html#t-pwr-a-monte-carlo-method"><i class="fa fa-check"></i><b>22.12.2</b> t-pwr: A Monte Carlo method</a></li>
</ul></li>
<li class="chapter" data-level="22.13" data-path="ttests.html"><a href="ttests.html#summary-4"><i class="fa fa-check"></i><b>22.13</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="ttestmc.html"><a href="ttestmc.html"><i class="fa fa-check"></i><b>23</b> Statistical design of t-tests</a><ul>
<li class="chapter" data-level="23.1" data-path="ttestmc.html"><a href="ttestmc.html#about-this-chapter"><i class="fa fa-check"></i><b>23.1</b> About this chapter</a></li>
<li class="chapter" data-level="23.2" data-path="ttestmc.html"><a href="ttestmc.html#overview-of-the-monte-carlo"><i class="fa fa-check"></i><b>23.2</b> Overview of the Monte Carlo</a><ul>
<li class="chapter" data-level="23.2.1" data-path="ttestmc.html"><a href="ttestmc.html#data-simulation"><i class="fa fa-check"></i><b>23.2.1</b> Data simulation</a></li>
<li class="chapter" data-level="23.2.2" data-path="ttestmc.html"><a href="ttestmc.html#the-test"><i class="fa fa-check"></i><b>23.2.2</b> The test</a></li>
</ul></li>
<li class="chapter" data-level="23.3" data-path="ttestmc.html"><a href="ttestmc.html#changing-sample-size"><i class="fa fa-check"></i><b>23.3</b> Changing Sample Size</a></li>
<li class="chapter" data-level="23.4" data-path="ttestmc.html"><a href="ttestmc.html#diabetes-drug-scenario"><i class="fa fa-check"></i><b>23.4</b> Diabetes drug scenario</a><ul>
<li class="chapter" data-level="23.4.1" data-path="ttestmc.html"><a href="ttestmc.html#what-would-be-a-scientifically-meaningful-response"><i class="fa fa-check"></i><b>23.4.1</b> What would be a scientifically meaningful response?</a></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="ttestmc.html"><a href="ttestmc.html#one-sample-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.5</b> One sample t-test Monte Carlo</a><ul>
<li class="chapter" data-level="23.5.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-2"><i class="fa fa-check"></i><b>23.5.1</b> Step 1</a></li>
<li class="chapter" data-level="23.5.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-2"><i class="fa fa-check"></i><b>23.5.2</b> Step 2</a></li>
<li class="chapter" data-level="23.5.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-2"><i class="fa fa-check"></i><b>23.5.3</b> Step 3</a></li>
<li class="chapter" data-level="23.5.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-2"><i class="fa fa-check"></i><b>23.5.4</b> Step 4</a></li>
<li class="chapter" data-level="23.5.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-2"><i class="fa fa-check"></i><b>23.5.5</b> Step 5</a></li>
<li class="chapter" data-level="23.5.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-2"><i class="fa fa-check"></i><b>23.5.6</b> Step 6</a></li>
<li class="chapter" data-level="23.5.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-1"><i class="fa fa-check"></i><b>23.5.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.6" data-path="ttestmc.html"><a href="ttestmc.html#unpaired-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.6</b> Unpaired t-test Monte Carlo</a><ul>
<li class="chapter" data-level="23.6.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-3"><i class="fa fa-check"></i><b>23.6.1</b> Step 1</a></li>
<li class="chapter" data-level="23.6.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-3"><i class="fa fa-check"></i><b>23.6.2</b> Step 2</a></li>
<li class="chapter" data-level="23.6.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-3"><i class="fa fa-check"></i><b>23.6.3</b> Step 3</a></li>
<li class="chapter" data-level="23.6.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-3"><i class="fa fa-check"></i><b>23.6.4</b> Step 4</a></li>
<li class="chapter" data-level="23.6.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-3"><i class="fa fa-check"></i><b>23.6.5</b> Step 5</a></li>
<li class="chapter" data-level="23.6.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-3"><i class="fa fa-check"></i><b>23.6.6</b> Step 6</a></li>
<li class="chapter" data-level="23.6.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-2"><i class="fa fa-check"></i><b>23.6.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="ttestmc.html"><a href="ttestmc.html#paired-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.7</b> Paired t-test Monte Carlo</a><ul>
<li class="chapter" data-level="23.7.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-4"><i class="fa fa-check"></i><b>23.7.1</b> Step 1</a></li>
<li class="chapter" data-level="23.7.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-4"><i class="fa fa-check"></i><b>23.7.2</b> Step 2</a></li>
<li class="chapter" data-level="23.7.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-4"><i class="fa fa-check"></i><b>23.7.3</b> Step 3</a></li>
</ul></li>
<li class="chapter" data-level="23.8" data-path="ttestmc.html"><a href="ttestmc.html#step-4-4"><i class="fa fa-check"></i><b>23.8</b> Step 4</a><ul>
<li class="chapter" data-level="23.8.1" data-path="ttestmc.html"><a href="ttestmc.html#step-5-4"><i class="fa fa-check"></i><b>23.8.1</b> Step 5</a></li>
<li class="chapter" data-level="23.8.2" data-path="ttestmc.html"><a href="ttestmc.html#step-6-4"><i class="fa fa-check"></i><b>23.8.2</b> Step 6</a></li>
<li class="chapter" data-level="23.8.3" data-path="ttestmc.html"><a href="ttestmc.html#step-7-3"><i class="fa fa-check"></i><b>23.8.3</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.9" data-path="ttestmc.html"><a href="ttestmc.html#comparison-to-pwr-tests"><i class="fa fa-check"></i><b>23.9</b> Comparison to pwr tests</a></li>
<li class="chapter" data-level="23.10" data-path="ttestmc.html"><a href="ttestmc.html#dt"><i class="fa fa-check"></i><b>23.10</b> dt</a></li>
<li class="chapter" data-level="23.11" data-path="ttestmc.html"><a href="ttestmc.html#pt"><i class="fa fa-check"></i><b>23.11</b> pt</a></li>
<li class="chapter" data-level="23.12" data-path="ttestmc.html"><a href="ttestmc.html#qt"><i class="fa fa-check"></i><b>23.12</b> qt</a></li>
<li class="chapter" data-level="23.13" data-path="ttestmc.html"><a href="ttestmc.html#rt"><i class="fa fa-check"></i><b>23.13</b> rt</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="simcorrelation.html"><a href="simcorrelation.html"><i class="fa fa-check"></i><b>24</b> Simulating correlated variables</a><ul>
<li class="chapter" data-level="24.1" data-path="simcorrelation.html"><a href="simcorrelation.html#estimating-correlation-between-two-variables"><i class="fa fa-check"></i><b>24.1</b> Estimating correlation between two variables</a></li>
<li class="chapter" data-level="24.2" data-path="simcorrelation.html"><a href="simcorrelation.html#simulating-correlated-variables"><i class="fa fa-check"></i><b>24.2</b> Simulating correlated variables</a></li>
<li class="chapter" data-level="24.3" data-path="simcorrelation.html"><a href="simcorrelation.html#monte-carlo-simulation"><i class="fa fa-check"></i><b>24.3</b> Monte Carlo simulation</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="introanova.html"><a href="introanova.html"><i class="fa fa-check"></i><b>25</b> Introduction to ANOVA</a><ul>
<li class="chapter" data-level="25.1" data-path="introanova.html"><a href="introanova.html#assumptions"><i class="fa fa-check"></i><b>25.1</b> Assumptions</a></li>
<li class="chapter" data-level="25.2" data-path="introanova.html"><a href="introanova.html#types-of-designs"><i class="fa fa-check"></i><b>25.2</b> Types of designs</a></li>
<li class="chapter" data-level="25.3" data-path="introanova.html"><a href="introanova.html#jargon-factors-and-levels"><i class="fa fa-check"></i><b>25.3</b> Jargon: Factors and levels</a><ul>
<li class="chapter" data-level="25.3.1" data-path="introanova.html"><a href="introanova.html#woring-with-factors-in-r"><i class="fa fa-check"></i><b>25.3.1</b> Woring with factors in R</a></li>
</ul></li>
<li class="chapter" data-level="25.4" data-path="introanova.html"><a href="introanova.html#anova-models-one--two--and-three-way"><i class="fa fa-check"></i><b>25.4</b> ANOVA models: One-, Two-, and Three-way</a></li>
<li class="chapter" data-level="25.5" data-path="introanova.html"><a href="introanova.html#inference"><i class="fa fa-check"></i><b>25.5</b> Inference</a><ul>
<li class="chapter" data-level="25.5.1" data-path="introanova.html"><a href="introanova.html#a-quick-peek-at-posthoc-testing"><i class="fa fa-check"></i><b>25.5.1</b> A quick peek at posthoc testing</a></li>
</ul></li>
<li class="chapter" data-level="25.6" data-path="introanova.html"><a href="introanova.html#anova-calculations"><i class="fa fa-check"></i><b>25.6</b> ANOVA calculations</a><ul>
<li class="chapter" data-level="25.6.1" data-path="introanova.html"><a href="introanova.html#sums-of-squares-partitioning"><i class="fa fa-check"></i><b>25.6.1</b> Sums of Squares partitioning</a></li>
<li class="chapter" data-level="25.6.2" data-path="introanova.html"><a href="introanova.html#the-anova-table"><i class="fa fa-check"></i><b>25.6.2</b> The ANOVA table</a></li>
<li class="chapter" data-level="25.6.3" data-path="introanova.html"><a href="introanova.html#post-hoc-group-comparisons"><i class="fa fa-check"></i><b>25.6.3</b> Post-hoc group comparisons</a></li>
</ul></li>
<li class="chapter" data-level="25.7" data-path="introanova.html"><a href="introanova.html#completely-randomized-or-related-measures"><i class="fa fa-check"></i><b>25.7</b> Completely randomized or related measures</a><ul>
<li class="chapter" data-level="25.7.1" data-path="introanova.html"><a href="introanova.html#the-problem-of-lost-data-in-related-measures-designs"><i class="fa fa-check"></i><b>25.7.1</b> The problem of lost data in related measures designs</a></li>
</ul></li>
<li class="chapter" data-level="25.8" data-path="introanova.html"><a href="introanova.html#two-way-anova"><i class="fa fa-check"></i><b>25.8</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="25.9" data-path="introanova.html"><a href="introanova.html#other-anova-models"><i class="fa fa-check"></i><b>25.9</b> Other ANOVA models</a><ul>
<li class="chapter" data-level="25.9.1" data-path="introanova.html"><a href="introanova.html#r-and-anova"><i class="fa fa-check"></i><b>25.9.1</b> R and ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="25.10" data-path="introanova.html"><a href="introanova.html#alternatives-to-anova"><i class="fa fa-check"></i><b>25.10</b> Alternatives to ANOVA</a><ul>
<li class="chapter" data-level="25.10.1" data-path="introanova.html"><a href="introanova.html#screw-anova-just-tell-me-how-to-t-test-everything"><i class="fa fa-check"></i><b>25.10.1</b> Screw ANOVA, Just Tell Me How to t-Test Everything</a></li>
</ul></li>
<li class="chapter" data-level="25.11" data-path="introanova.html"><a href="introanova.html#summary-5"><i class="fa fa-check"></i><b>25.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="fdistr.html"><a href="fdistr.html"><i class="fa fa-check"></i><b>26</b> The F distribution</a><ul>
<li class="chapter" data-level="26.1" data-path="fdistr.html"><a href="fdistr.html#background-1"><i class="fa fa-check"></i><b>26.1</b> Background</a><ul>
<li class="chapter" data-level="26.1.1" data-path="fdistr.html"><a href="fdistr.html#sample-variance-and-fs-pdf"><i class="fa fa-check"></i><b>26.1.1</b> Sample Variance and F’s PDF</a></li>
</ul></li>
<li class="chapter" data-level="26.2" data-path="fdistr.html"><a href="fdistr.html#df"><i class="fa fa-check"></i><b>26.2</b> df</a></li>
<li class="chapter" data-level="26.3" data-path="fdistr.html"><a href="fdistr.html#pf"><i class="fa fa-check"></i><b>26.3</b> pf</a></li>
<li class="chapter" data-level="26.4" data-path="fdistr.html"><a href="fdistr.html#qf"><i class="fa fa-check"></i><b>26.4</b> qf</a></li>
<li class="chapter" data-level="26.5" data-path="fdistr.html"><a href="fdistr.html#rf"><i class="fa fa-check"></i><b>26.5</b> rf</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="onewayanova.html"><a href="onewayanova.html"><i class="fa fa-check"></i><b>27</b> One-way ANOVA Completely Randomized</a><ul>
<li class="chapter" data-level="27.1" data-path="onewayanova.html"><a href="onewayanova.html#using-ezanova"><i class="fa fa-check"></i><b>27.1</b> Using <code>ezANOVA</code></a></li>
<li class="chapter" data-level="27.2" data-path="onewayanova.html"><a href="onewayanova.html#the-chickwt-data-set"><i class="fa fa-check"></i><b>27.2</b> The chickwt data set</a><ul>
<li class="chapter" data-level="27.2.1" data-path="onewayanova.html"><a href="onewayanova.html#inspect-the-data"><i class="fa fa-check"></i><b>27.2.1</b> Inspect the data</a></li>
</ul></li>
<li class="chapter" data-level="27.3" data-path="onewayanova.html"><a href="onewayanova.html#run-the-anova"><i class="fa fa-check"></i><b>27.3</b> Run the ANOVA</a><ul>
<li class="chapter" data-level="27.3.1" data-path="onewayanova.html"><a href="onewayanova.html#run-the-chickwts-one-way-anova"><i class="fa fa-check"></i><b>27.3.1</b> Run the chickwts One Way ANOVA</a></li>
<li class="chapter" data-level="27.3.2" data-path="onewayanova.html"><a href="onewayanova.html#interpreting-the-one-way-cr-anova-output"><i class="fa fa-check"></i><b>27.3.2</b> Interpreting the One-Way CR ANOVA Output</a></li>
</ul></li>
<li class="chapter" data-level="27.4" data-path="onewayanova.html"><a href="onewayanova.html#post-hoc-pairwise-comparisons"><i class="fa fa-check"></i><b>27.4</b> Post hoc pairwise comparisons</a><ul>
<li class="chapter" data-level="27.4.1" data-path="onewayanova.html"><a href="onewayanova.html#range-tests"><i class="fa fa-check"></i><b>27.4.1</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="27.5" data-path="onewayanova.html"><a href="onewayanova.html#summary-6"><i class="fa fa-check"></i><b>27.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="onewayRM.html"><a href="onewayRM.html"><i class="fa fa-check"></i><b>28</b> One-way ANOVA Related Measures</a><ul>
<li class="chapter" data-level="28.1" data-path="onewayRM.html"><a href="onewayRM.html#data-prep"><i class="fa fa-check"></i><b>28.1</b> Data prep</a></li>
<li class="chapter" data-level="28.2" data-path="onewayRM.html"><a href="onewayRM.html#data-visualization"><i class="fa fa-check"></i><b>28.2</b> Data visualization</a><ul>
<li class="chapter" data-level="28.2.1" data-path="onewayRM.html"><a href="onewayRM.html#visualize-the-statistical-design"><i class="fa fa-check"></i><b>28.2.1</b> Visualize the statistical design</a></li>
</ul></li>
<li class="chapter" data-level="28.3" data-path="onewayRM.html"><a href="onewayRM.html#the-anova"><i class="fa fa-check"></i><b>28.3</b> The ANOVA</a><ul>
<li class="chapter" data-level="28.3.1" data-path="onewayRM.html"><a href="onewayRM.html#running-ezanova"><i class="fa fa-check"></i><b>28.3.1</b> Running ezANOVA</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="onewayRM.html"><a href="onewayRM.html#interpretation-5"><i class="fa fa-check"></i><b>28.4</b> Interpretation</a></li>
<li class="chapter" data-level="28.5" data-path="onewayRM.html"><a href="onewayRM.html#post-hoc-analysis"><i class="fa fa-check"></i><b>28.5</b> Post-hoc analysis</a></li>
<li class="chapter" data-level="28.6" data-path="onewayRM.html"><a href="onewayRM.html#write-up-12"><i class="fa fa-check"></i><b>28.6</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="twowayCR.html"><a href="twowayCR.html"><i class="fa fa-check"></i><b>29</b> Two-way ANOVA Completely Randomized</a><ul>
<li class="chapter" data-level="29.1" data-path="twowayCR.html"><a href="twowayCR.html#effect-of-strain-and-diet-on-liver"><i class="fa fa-check"></i><b>29.1</b> Effect of Strain and Diet on Liver</a></li>
<li class="chapter" data-level="29.2" data-path="twowayCR.html"><a href="twowayCR.html#the-test-1"><i class="fa fa-check"></i><b>29.2</b> The test</a></li>
<li class="chapter" data-level="29.3" data-path="twowayCR.html"><a href="twowayCR.html#interpretation-of-2-way-cr-anova-output"><i class="fa fa-check"></i><b>29.3</b> Interpretation of 2 Way CR ANOVA Output</a><ul>
<li class="chapter" data-level="29.3.1" data-path="twowayCR.html"><a href="twowayCR.html#levenes"><i class="fa fa-check"></i><b>29.3.1</b> Levene’s</a></li>
<li class="chapter" data-level="29.3.2" data-path="twowayCR.html"><a href="twowayCR.html#anova-table"><i class="fa fa-check"></i><b>29.3.2</b> ANOVA Table</a></li>
</ul></li>
<li class="chapter" data-level="29.4" data-path="twowayCR.html"><a href="twowayCR.html#post-hoc-multiple-comparisons"><i class="fa fa-check"></i><b>29.4</b> Post Hoc Multiple Comparisons</a><ul>
<li class="chapter" data-level="29.4.1" data-path="twowayCR.html"><a href="twowayCR.html#pairwise.t.tests"><i class="fa fa-check"></i><b>29.4.1</b> Pairwise.t.tests</a></li>
<li class="chapter" data-level="29.4.2" data-path="twowayCR.html"><a href="twowayCR.html#write-up-13"><i class="fa fa-check"></i><b>29.4.2</b> Write Up</a></li>
<li class="chapter" data-level="29.4.3" data-path="twowayCR.html"><a href="twowayCR.html#range-tests-1"><i class="fa fa-check"></i><b>29.4.3</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="29.5" data-path="twowayCR.html"><a href="twowayCR.html#summary-7"><i class="fa fa-check"></i><b>29.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="twowayRM.html"><a href="twowayRM.html"><i class="fa fa-check"></i><b>30</b> Two-way ANOVA Related Measures</a><ul>
<li class="chapter" data-level="30.1" data-path="twowayRM.html"><a href="twowayRM.html#cell-culture"><i class="fa fa-check"></i><b>30.1</b> Cell culture</a></li>
<li class="chapter" data-level="30.2" data-path="twowayRM.html"><a href="twowayRM.html#the-test-2"><i class="fa fa-check"></i><b>30.2</b> The test</a></li>
<li class="chapter" data-level="30.3" data-path="twowayRM.html"><a href="twowayRM.html#interpretation-of-the-output"><i class="fa fa-check"></i><b>30.3</b> Interpretation of the output</a><ul>
<li class="chapter" data-level="30.3.1" data-path="twowayRM.html"><a href="twowayRM.html#anova-table-1"><i class="fa fa-check"></i><b>30.3.1</b> ANOVA Table</a></li>
<li class="chapter" data-level="30.3.2" data-path="twowayRM.html"><a href="twowayRM.html#mauchlys-sphericity-test"><i class="fa fa-check"></i><b>30.3.2</b> Mauchly’s Sphericity Test</a></li>
</ul></li>
<li class="chapter" data-level="30.4" data-path="twowayRM.html"><a href="twowayRM.html#post-hoc-multiple-comparisons-1"><i class="fa fa-check"></i><b>30.4</b> Post Hoc multiple comparisons</a></li>
<li class="chapter" data-level="30.5" data-path="twowayRM.html"><a href="twowayRM.html#write-up-14"><i class="fa fa-check"></i><b>30.5</b> Write Up</a></li>
<li class="chapter" data-level="30.6" data-path="twowayRM.html"><a href="twowayRM.html#summary-8"><i class="fa fa-check"></i><b>30.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="twowaymixed.html"><a href="twowaymixed.html"><i class="fa fa-check"></i><b>31</b> Two-way ANOVA RM/CR</a><ul>
<li class="chapter" data-level="31.1" data-path="twowaymixed.html"><a href="twowaymixed.html#chickweight-data-set"><i class="fa fa-check"></i><b>31.1</b> ChickWeight Data set</a></li>
<li class="chapter" data-level="31.2" data-path="twowaymixed.html"><a href="twowaymixed.html#munge-chickweight-data"><i class="fa fa-check"></i><b>31.2</b> Munge ChickWeight data</a></li>
<li class="chapter" data-level="31.3" data-path="twowaymixed.html"><a href="twowaymixed.html#the-test-3"><i class="fa fa-check"></i><b>31.3</b> The test</a></li>
<li class="chapter" data-level="31.4" data-path="twowaymixed.html"><a href="twowaymixed.html#interpreting-the-anova-output"><i class="fa fa-check"></i><b>31.4</b> Interpreting the ANOVA output</a><ul>
<li class="chapter" data-level="31.4.1" data-path="twowaymixed.html"><a href="twowaymixed.html#anova-the-anova-table-1"><i class="fa fa-check"></i><b>31.4.1</b> $ANOVA: The ANOVA table</a></li>
<li class="chapter" data-level="31.4.2" data-path="twowaymixed.html"><a href="twowaymixed.html#mauchlys-test-and-corrections"><i class="fa fa-check"></i><b>31.4.2</b> Mauchly’s Test and Corrections</a></li>
</ul></li>
<li class="chapter" data-level="31.5" data-path="twowaymixed.html"><a href="twowaymixed.html#post-hoc"><i class="fa fa-check"></i><b>31.5</b> Post hoc</a><ul>
<li class="chapter" data-level="31.5.1" data-path="twowaymixed.html"><a href="twowaymixed.html#pairwise.t.tests-1"><i class="fa fa-check"></i><b>31.5.1</b> Pairwise.t.tests</a></li>
<li class="chapter" data-level="31.5.2" data-path="twowaymixed.html"><a href="twowaymixed.html#tukey-test-a-range-test"><i class="fa fa-check"></i><b>31.5.2</b> Tukey test: A range test</a></li>
<li class="chapter" data-level="31.5.3" data-path="twowaymixed.html"><a href="twowaymixed.html#heres-whats-been-discovered-by-the-tukeyhsd-test"><i class="fa fa-check"></i><b>31.5.3</b> Here’s what’s been discovered by the TukeyHSD test</a></li>
<li class="chapter" data-level="31.5.4" data-path="twowaymixed.html"><a href="twowaymixed.html#why-is-the-tukeyhsd-result-so-different-than-the-pairwise.t.test"><i class="fa fa-check"></i><b>31.5.4</b> Why is the TukeyHSD result so different than the pairwise.t.test??</a></li>
</ul></li>
<li class="chapter" data-level="31.6" data-path="twowaymixed.html"><a href="twowaymixed.html#summary-9"><i class="fa fa-check"></i><b>31.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="posthoc.html"><a href="posthoc.html"><i class="fa fa-check"></i><b>32</b> ANOVA Posthoc Comparisons</a><ul>
<li class="chapter" data-level="32.1" data-path="posthoc.html"><a href="posthoc.html#overview-of-options"><i class="fa fa-check"></i><b>32.1</b> Overview of options</a><ul>
<li class="chapter" data-level="32.1.1" data-path="posthoc.html"><a href="posthoc.html#pairwise.t.test-with-p-value-adjustments"><i class="fa fa-check"></i><b>32.1.1</b> Pairwise.t.test with p-value adjustments</a></li>
<li class="chapter" data-level="32.1.2" data-path="posthoc.html"><a href="posthoc.html#range-tests-2"><i class="fa fa-check"></i><b>32.1.2</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="32.2" data-path="posthoc.html"><a href="posthoc.html#reporting-the-result-1"><i class="fa fa-check"></i><b>32.2</b> Reporting the result</a></li>
<li class="chapter" data-level="32.3" data-path="posthoc.html"><a href="posthoc.html#summary-10"><i class="fa fa-check"></i><b>32.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="jaxwest2.html"><a href="jaxwest2.html"><i class="fa fa-check"></i><b>33</b> Reproducible Data Munging Mostly with Tidyverse</a><ul>
<li class="chapter" data-level="33.1" data-path="jaxwest2.html"><a href="jaxwest2.html#look-at-the-original-data-carefully"><i class="fa fa-check"></i><b>33.1</b> Look at the original data carefully</a></li>
<li class="chapter" data-level="33.2" data-path="jaxwest2.html"><a href="jaxwest2.html#our-goal"><i class="fa fa-check"></i><b>33.2</b> Our goal</a></li>
<li class="chapter" data-level="33.3" data-path="jaxwest2.html"><a href="jaxwest2.html#step-1-read-the-data-into-r"><i class="fa fa-check"></i><b>33.3</b> Step 1: Read the data into R</a></li>
<li class="chapter" data-level="33.4" data-path="jaxwest2.html"><a href="jaxwest2.html#step-2-select-the-variables"><i class="fa fa-check"></i><b>33.4</b> Step 2: Select the variables</a></li>
<li class="chapter" data-level="33.5" data-path="jaxwest2.html"><a href="jaxwest2.html#step-3-trim-the-cases"><i class="fa fa-check"></i><b>33.5</b> Step 3: Trim the cases</a></li>
<li class="chapter" data-level="33.6" data-path="jaxwest2.html"><a href="jaxwest2.html#step-4-we-have-a-variable-problem-to-fix"><i class="fa fa-check"></i><b>33.6</b> Step 4: We have a variable problem to fix</a></li>
<li class="chapter" data-level="33.7" data-path="jaxwest2.html"><a href="jaxwest2.html#step-5-impute-or-delete"><i class="fa fa-check"></i><b>33.7</b> Step 5: Impute or Delete</a></li>
<li class="chapter" data-level="33.8" data-path="jaxwest2.html"><a href="jaxwest2.html#step-6-go-long"><i class="fa fa-check"></i><b>33.8</b> Step 6: Go long</a></li>
<li class="chapter" data-level="33.9" data-path="jaxwest2.html"><a href="jaxwest2.html#step-7-convert-other-variables-to-factor"><i class="fa fa-check"></i><b>33.9</b> Step 7: Convert other variables to factor</a></li>
<li class="chapter" data-level="33.10" data-path="jaxwest2.html"><a href="jaxwest2.html#step-8-plot"><i class="fa fa-check"></i><b>33.10</b> Step 8: Plot</a></li>
<li class="chapter" data-level="33.11" data-path="jaxwest2.html"><a href="jaxwest2.html#step-9-run-the-anova"><i class="fa fa-check"></i><b>33.11</b> Step 9: Run the ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="34" data-path="anovamc.html"><a href="anovamc.html"><i class="fa fa-check"></i><b>34</b> ANOVA power using Monte Carlo</a><ul>
<li class="chapter" data-level="34.1" data-path="anovamc.html"><a href="anovamc.html#alternatives-to-monte-carlo"><i class="fa fa-check"></i><b>34.1</b> Alternatives to Monte Carlo</a></li>
<li class="chapter" data-level="34.2" data-path="anovamc.html"><a href="anovamc.html#what-is-monte-carlo"><i class="fa fa-check"></i><b>34.2</b> What is Monte Carlo</a></li>
<li class="chapter" data-level="34.3" data-path="anovamc.html"><a href="anovamc.html#one-way-completely-randomized-anova-monte-carlo"><i class="fa fa-check"></i><b>34.3</b> One-way completely randomized ANOVA Monte Carlo</a><ul>
<li class="chapter" data-level="34.3.1" data-path="anovamc.html"><a href="anovamc.html#directions"><i class="fa fa-check"></i><b>34.3.1</b> Directions</a></li>
<li class="chapter" data-level="34.3.2" data-path="anovamc.html"><a href="anovamc.html#step-1-create-initial-values"><i class="fa fa-check"></i><b>34.3.2</b> Step 1: Create initial values</a></li>
<li class="chapter" data-level="34.3.3" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-random-sample"><i class="fa fa-check"></i><b>34.3.3</b> Step 2: Visualize one random sample</a></li>
<li class="chapter" data-level="34.3.4" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator"><i class="fa fa-check"></i><b>34.3.4</b> Step 3: Run The Power Simulator</a></li>
<li class="chapter" data-level="34.3.5" data-path="anovamc.html"><a href="anovamc.html#step-4-optimize-for-suitable-power"><i class="fa fa-check"></i><b>34.3.5</b> Step 4: Optimize for suitable power</a></li>
<li class="chapter" data-level="34.3.6" data-path="anovamc.html"><a href="anovamc.html#notes-and-considerations"><i class="fa fa-check"></i><b>34.3.6</b> Notes And Considerations</a></li>
</ul></li>
<li class="chapter" data-level="34.4" data-path="anovamc.html"><a href="anovamc.html#one-way-related-measures-anova-monte-carlo"><i class="fa fa-check"></i><b>34.4</b> One-way related measures ANOVA Monte Carlo</a></li>
<li class="chapter" data-level="34.5" data-path="anovamc.html"><a href="anovamc.html#directions-1"><i class="fa fa-check"></i><b>34.5</b> Directions</a><ul>
<li class="chapter" data-level="34.5.1" data-path="anovamc.html"><a href="anovamc.html#step-1-initial-values"><i class="fa fa-check"></i><b>34.5.1</b> Step 1: Initial values</a></li>
<li class="chapter" data-level="34.5.2" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-sample"><i class="fa fa-check"></i><b>34.5.2</b> Step 2: Visualize one sample</a></li>
<li class="chapter" data-level="34.5.3" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator-1"><i class="fa fa-check"></i><b>34.5.3</b> Step 3: Run the power simulator</a></li>
<li class="chapter" data-level="34.5.4" data-path="anovamc.html"><a href="anovamc.html#step-4-should-anything-be-changed"><i class="fa fa-check"></i><b>34.5.4</b> Step 4: Should anything be changed?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="manova.html"><a href="manova.html"><i class="fa fa-check"></i><b>35</b> Multivariate Analysis of Variance (MANOVA)</a><ul>
<li class="chapter" data-level="35.1" data-path="manova.html"><a href="manova.html#what-does-manova-test"><i class="fa fa-check"></i><b>35.1</b> What does MANOVA test?</a></li>
<li class="chapter" data-level="35.2" data-path="manova.html"><a href="manova.html#types-of-experiments"><i class="fa fa-check"></i><b>35.2</b> Types of experiments</a></li>
<li class="chapter" data-level="35.3" data-path="manova.html"><a href="manova.html#manova-is-not-many-anovas"><i class="fa fa-check"></i><b>35.3</b> MANOVA is not many ANOVAs</a><ul>
<li class="chapter" data-level="35.3.1" data-path="manova.html"><a href="manova.html#why-manova"><i class="fa fa-check"></i><b>35.3.1</b> Why MANOVA?</a></li>
</ul></li>
<li class="chapter" data-level="35.4" data-path="manova.html"><a href="manova.html#assumptions-1"><i class="fa fa-check"></i><b>35.4</b> Assumptions</a><ul>
<li class="chapter" data-level="35.4.1" data-path="manova.html"><a href="manova.html#multicolinearity"><i class="fa fa-check"></i><b>35.4.1</b> Multicolinearity</a></li>
</ul></li>
<li class="chapter" data-level="35.5" data-path="manova.html"><a href="manova.html#calculating-variation"><i class="fa fa-check"></i><b>35.5</b> Calculating variation</a><ul>
<li class="chapter" data-level="35.5.1" data-path="manova.html"><a href="manova.html#eigenvectors-and-eigenvalues"><i class="fa fa-check"></i><b>35.5.1</b> Eigenvectors and eigenvalues</a></li>
<li class="chapter" data-level="35.5.2" data-path="manova.html"><a href="manova.html#manova-test-statistics"><i class="fa fa-check"></i><b>35.5.2</b> MANOVA test statistics</a></li>
<li class="chapter" data-level="35.5.3" data-path="manova.html"><a href="manova.html#which-manova-test-statistic-is-best"><i class="fa fa-check"></i><b>35.5.3</b> Which MANOVA test statistic is best?</a></li>
</ul></li>
<li class="chapter" data-level="35.6" data-path="manova.html"><a href="manova.html#drug-clearance-example"><i class="fa fa-check"></i><b>35.6</b> Drug clearance example</a><ul>
<li class="chapter" data-level="35.6.1" data-path="manova.html"><a href="manova.html#rationale-for-manova"><i class="fa fa-check"></i><b>35.6.1</b> Rationale for MANOVA</a></li>
<li class="chapter" data-level="35.6.2" data-path="manova.html"><a href="manova.html#data-structure"><i class="fa fa-check"></i><b>35.6.2</b> Data structure</a></li>
<li class="chapter" data-level="35.6.3" data-path="manova.html"><a href="manova.html#manova-procedure"><i class="fa fa-check"></i><b>35.6.3</b> MANOVA procedure</a></li>
</ul></li>
<li class="chapter" data-level="35.7" data-path="manova.html"><a href="manova.html#planning-manova-experiments"><i class="fa fa-check"></i><b>35.7</b> Planning MANOVA experiments</a><ul>
<li class="chapter" data-level="35.7.1" data-path="manova.html"><a href="manova.html#manova-monte-carlo-power-analysis"><i class="fa fa-check"></i><b>35.7.1</b> MANOVA Monte Carlo Power Analysis</a></li>
<li class="chapter" data-level="35.7.2" data-path="manova.html"><a href="manova.html#an-manova-monte-carlo-power-function"><i class="fa fa-check"></i><b>35.7.2</b> An MANOVA Monte Carlo Power function</a></li>
</ul></li>
<li class="chapter" data-level="35.8" data-path="manova.html"><a href="manova.html#summary-11"><i class="fa fa-check"></i><b>35.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="36" data-path="correl.html"><a href="correl.html"><i class="fa fa-check"></i><b>36</b> Correlation</a><ul>
<li class="chapter" data-level="36.1" data-path="correl.html"><a href="correl.html#correlation-causation"><i class="fa fa-check"></i><b>36.1</b> Correlation != Causation</a></li>
<li class="chapter" data-level="36.2" data-path="correl.html"><a href="correl.html#correlation-in-multivariate-outcomes-and-paired-designs"><i class="fa fa-check"></i><b>36.2</b> Correlation in Multivariate Outcomes and Paired Designs</a></li>
<li class="chapter" data-level="36.3" data-path="correl.html"><a href="correl.html#correlation-coefficients"><i class="fa fa-check"></i><b>36.3</b> Correlation coefficients</a><ul>
<li class="chapter" data-level="36.3.1" data-path="correl.html"><a href="correl.html#pearsons-correlation-coefficient"><i class="fa fa-check"></i><b>36.3.1</b> Pearson’s correlation coefficient</a></li>
<li class="chapter" data-level="36.3.2" data-path="correl.html"><a href="correl.html#spearmans-rank-correlation"><i class="fa fa-check"></i><b>36.3.2</b> Spearman’s rank correlation</a></li>
<li class="chapter" data-level="36.3.3" data-path="correl.html"><a href="correl.html#kendalls-tau"><i class="fa fa-check"></i><b>36.3.3</b> Kendall’s tau</a></li>
<li class="chapter" data-level="36.3.4" data-path="correl.html"><a href="correl.html#which-correlation-method-to-use"><i class="fa fa-check"></i><b>36.3.4</b> Which correlation method to use?</a></li>
<li class="chapter" data-level="36.3.5" data-path="correl.html"><a href="correl.html#r-correlation-analysis-functions"><i class="fa fa-check"></i><b>36.3.5</b> R correlation analysis functions</a></li>
<li class="chapter" data-level="36.3.6" data-path="correl.html"><a href="correl.html#plot-the-correlations"><i class="fa fa-check"></i><b>36.3.6</b> Plot the correlations</a></li>
<li class="chapter" data-level="36.3.7" data-path="correl.html"><a href="correl.html#calculate-a-correlation-coefficient-and-posthoc-test"><i class="fa fa-check"></i><b>36.3.7</b> Calculate a correlation coefficient and posthoc test</a></li>
<li class="chapter" data-level="36.3.8" data-path="correl.html"><a href="correl.html#interpretation-of-correlation-output"><i class="fa fa-check"></i><b>36.3.8</b> Interpretation of correlation output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="regress.html"><a href="regress.html"><i class="fa fa-check"></i><b>37</b> Linear Regression</a><ul>
<li class="chapter" data-level="37.1" data-path="regress.html"><a href="regress.html#the-linear-regression-model"><i class="fa fa-check"></i><b>37.1</b> The linear regression model</a></li>
<li class="chapter" data-level="37.2" data-path="regress.html"><a href="regress.html#least-squares-fitting"><i class="fa fa-check"></i><b>37.2</b> Least squares fitting</a></li>
<li class="chapter" data-level="37.3" data-path="regress.html"><a href="regress.html#the-practical-importance-of-linear-model-parameters"><i class="fa fa-check"></i><b>37.3</b> The practical importance of linear model parameters</a></li>
<li class="chapter" data-level="37.4" data-path="regress.html"><a href="regress.html#linear-model-standard-errors"><i class="fa fa-check"></i><b>37.4</b> Linear model standard errors</a></li>
<li class="chapter" data-level="37.5" data-path="regress.html"><a href="regress.html#linear-regression-in-r"><i class="fa fa-check"></i><b>37.5</b> Linear regression in R</a></li>
<li class="chapter" data-level="37.6" data-path="regress.html"><a href="regress.html#intepretation"><i class="fa fa-check"></i><b>37.6</b> Intepretation</a><ul>
<li class="chapter" data-level="37.6.1" data-path="regress.html"><a href="regress.html#residuals"><i class="fa fa-check"></i><b>37.6.1</b> Residuals</a></li>
<li class="chapter" data-level="37.6.2" data-path="regress.html"><a href="regress.html#coefficients"><i class="fa fa-check"></i><b>37.6.2</b> Coefficients</a></li>
<li class="chapter" data-level="37.6.3" data-path="regress.html"><a href="regress.html#degrees-of-freedom-1"><i class="fa fa-check"></i><b>37.6.3</b> Degrees of freedom</a></li>
<li class="chapter" data-level="37.6.4" data-path="regress.html"><a href="regress.html#r-squared"><i class="fa fa-check"></i><b>37.6.4</b> R-squared</a></li>
<li class="chapter" data-level="37.6.5" data-path="regress.html"><a href="regress.html#f-statistic"><i class="fa fa-check"></i><b>37.6.5</b> F-statistic</a></li>
<li class="chapter" data-level="37.6.6" data-path="regress.html"><a href="regress.html#plotting-regression-results"><i class="fa fa-check"></i><b>37.6.6</b> Plotting regression results</a></li>
<li class="chapter" data-level="37.6.7" data-path="regress.html"><a href="regress.html#visualizing-residuals"><i class="fa fa-check"></i><b>37.6.7</b> Visualizing residuals</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="nonlinearintro.html"><a href="nonlinearintro.html"><i class="fa fa-check"></i><b>38</b> Non-linear regression introduction</a><ul>
<li class="chapter" data-level="38.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#uses-for-nonlinear-regression"><i class="fa fa-check"></i><b>38.1</b> Uses for nonlinear regression</a></li>
<li class="chapter" data-level="38.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#example-of-the-process"><i class="fa fa-check"></i><b>38.2</b> Example of the process</a><ul>
<li class="chapter" data-level="38.2.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#it-starts-with-nonlinear-data"><i class="fa fa-check"></i><b>38.2.1</b> It starts with nonlinear data</a></li>
<li class="chapter" data-level="38.2.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#which-nonlinear-model-describes-that-data"><i class="fa fa-check"></i><b>38.2.2</b> Which nonlinear model describes that data?</a></li>
<li class="chapter" data-level="38.2.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#fitting-a-model-to-data"><i class="fa fa-check"></i><b>38.2.3</b> Fitting a model to data</a></li>
<li class="chapter" data-level="38.2.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#interpreting-regression-model-output"><i class="fa fa-check"></i><b>38.2.4</b> Interpreting regression model output</a></li>
<li class="chapter" data-level="38.2.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#check-the-fit-graphically"><i class="fa fa-check"></i><b>38.2.5</b> Check the fit graphically</a></li>
<li class="chapter" data-level="38.2.6" data-path="nonlinearintro.html"><a href="nonlinearintro.html#convergence"><i class="fa fa-check"></i><b>38.2.6</b> Convergence?</a></li>
<li class="chapter" data-level="38.2.7" data-path="nonlinearintro.html"><a href="nonlinearintro.html#convergence-is-not-idiot-proof"><i class="fa fa-check"></i><b>38.2.7</b> Convergence is not idiot proof</a></li>
</ul></li>
<li class="chapter" data-level="38.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-in-general"><i class="fa fa-check"></i><b>38.3</b> Nonlinear models in general</a></li>
<li class="chapter" data-level="38.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nlfitr"><i class="fa fa-check"></i><b>38.4</b> nlfitr</a></li>
<li class="chapter" data-level="38.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-regression-functions-in-r"><i class="fa fa-check"></i><b>38.5</b> Nonlinear regression functions in R</a><ul>
<li class="chapter" data-level="38.5.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#the-fitting-algorithms"><i class="fa fa-check"></i><b>38.5.1</b> The fitting algorithms</a></li>
<li class="chapter" data-level="38.5.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#formula"><i class="fa fa-check"></i><b>38.5.2</b> formula</a></li>
<li class="chapter" data-level="38.5.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#data-1"><i class="fa fa-check"></i><b>38.5.3</b> data</a></li>
<li class="chapter" data-level="38.5.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#start"><i class="fa fa-check"></i><b>38.5.4</b> start</a></li>
<li class="chapter" data-level="38.5.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#weighting"><i class="fa fa-check"></i><b>38.5.5</b> weighting</a></li>
</ul></li>
<li class="chapter" data-level="38.6" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-and-parameters"><i class="fa fa-check"></i><b>38.6</b> Nonlinear models and parameters</a><ul>
<li class="chapter" data-level="38.6.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#hyperbolic-stimulus-response-functions"><i class="fa fa-check"></i><b>38.6.1</b> Hyperbolic stimulus response functions</a></li>
<li class="chapter" data-level="38.6.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#visualizing-nonlinear-data-and-log-scaling"><i class="fa fa-check"></i><b>38.6.2</b> Visualizing nonlinear data and log scaling</a></li>
<li class="chapter" data-level="38.6.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#selecting-the-right-model"><i class="fa fa-check"></i><b>38.6.3</b> Selecting the right model</a></li>
<li class="chapter" data-level="38.6.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#from-models-to-formulas"><i class="fa fa-check"></i><b>38.6.4</b> From models to formulas</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="39" data-path="nestedregress.html"><a href="nestedregress.html"><i class="fa fa-check"></i><b>39</b> Nested-model nonlinear regression</a><ul>
<li class="chapter" data-level="39.1" data-path="nestedregress.html"><a href="nestedregress.html#read-and-plot-the-data"><i class="fa fa-check"></i><b>39.1</b> Read and plot the data</a></li>
<li class="chapter" data-level="39.2" data-path="nestedregress.html"><a href="nestedregress.html#which-model-is-a-better-fit"><i class="fa fa-check"></i><b>39.2</b> Which model is a better fit?</a><ul>
<li class="chapter" data-level="39.2.1" data-path="nestedregress.html"><a href="nestedregress.html#nonlinear-models"><i class="fa fa-check"></i><b>39.2.1</b> Nonlinear models</a></li>
<li class="chapter" data-level="39.2.2" data-path="nestedregress.html"><a href="nestedregress.html#the-nlfitr-alternative"><i class="fa fa-check"></i><b>39.2.2</b> The nlfitr alternative</a></li>
<li class="chapter" data-level="39.2.3" data-path="nestedregress.html"><a href="nestedregress.html#troubleshooting-the-regression"><i class="fa fa-check"></i><b>39.2.3</b> Troubleshooting the regression</a></li>
<li class="chapter" data-level="39.2.4" data-path="nestedregress.html"><a href="nestedregress.html#interpreting-the-parameters"><i class="fa fa-check"></i><b>39.2.4</b> Interpreting the parameters</a></li>
<li class="chapter" data-level="39.2.5" data-path="nestedregress.html"><a href="nestedregress.html#residual-plots-to-compare-fits"><i class="fa fa-check"></i><b>39.2.5</b> Residual plots to compare fits</a></li>
<li class="chapter" data-level="39.2.6" data-path="nestedregress.html"><a href="nestedregress.html#compare-aic"><i class="fa fa-check"></i><b>39.2.6</b> Compare AIC</a></li>
<li class="chapter" data-level="39.2.7" data-path="nestedregress.html"><a href="nestedregress.html#other-ways-to-compare-models"><i class="fa fa-check"></i><b>39.2.7</b> Other ways to compare models</a></li>
<li class="chapter" data-level="39.2.8" data-path="nestedregress.html"><a href="nestedregress.html#interpretation-of-this-one-replicate"><i class="fa fa-check"></i><b>39.2.8</b> Interpretation of this one replicate</a></li>
<li class="chapter" data-level="39.2.9" data-path="nestedregress.html"><a href="nestedregress.html#alternate-analysis"><i class="fa fa-check"></i><b>39.2.9</b> Alternate analysis</a></li>
</ul></li>
<li class="chapter" data-level="39.3" data-path="nestedregress.html"><a href="nestedregress.html#summary-12"><i class="fa fa-check"></i><b>39.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="40" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html"><i class="fa fa-check"></i><b>40</b> Nonlinear regression of independent replicates</a><ul>
<li class="chapter" data-level="40.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-problem"><i class="fa fa-check"></i><b>40.1</b> The problem</a></li>
<li class="chapter" data-level="40.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-dataset"><i class="fa fa-check"></i><b>40.2</b> The dataset</a></li>
<li class="chapter" data-level="40.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#munging-for-regression-analysis"><i class="fa fa-check"></i><b>40.3</b> Munging for regression analysis</a><ul>
<li class="chapter" data-level="40.3.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#average-the-technical-replicates"><i class="fa fa-check"></i><b>40.3.1</b> Average the technical replicates</a></li>
<li class="chapter" data-level="40.3.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#create-one-table"><i class="fa fa-check"></i><b>40.3.2</b> Create one table</a></li>
<li class="chapter" data-level="40.3.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#plot-the-data"><i class="fa fa-check"></i><b>40.3.3</b> Plot the data</a></li>
<li class="chapter" data-level="40.3.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#run-the-regressions"><i class="fa fa-check"></i><b>40.3.4</b> Run the regressions</a></li>
<li class="chapter" data-level="40.3.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#clean-up-regression-results"><i class="fa fa-check"></i><b>40.3.5</b> Clean up regression results</a></li>
</ul></li>
<li class="chapter" data-level="40.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#t-test-on-half-lives"><i class="fa fa-check"></i><b>40.4</b> T-test on half-lives</a></li>
<li class="chapter" data-level="40.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#conclusion"><i class="fa fa-check"></i><b>40.5</b> Conclusion</a></li>
<li class="chapter" data-level="40.6" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#summary-figure"><i class="fa fa-check"></i><b>40.6</b> Summary figure</a></li>
</ul></li>
<li class="chapter" data-level="41" data-path="multregress.html"><a href="multregress.html"><i class="fa fa-check"></i><b>41</b> Multiple regression</a><ul>
<li class="chapter" data-level="41.1" data-path="multregress.html"><a href="multregress.html#the-linear-regression-model-1"><i class="fa fa-check"></i><b>41.1</b> The linear regression model</a></li>
<li class="chapter" data-level="41.2" data-path="multregress.html"><a href="multregress.html#multiple-regression-models"><i class="fa fa-check"></i><b>41.2</b> Multiple regression models</a></li>
<li class="chapter" data-level="41.3" data-path="multregress.html"><a href="multregress.html#experimental-multiple-regression"><i class="fa fa-check"></i><b>41.3</b> Experimental multiple regression</a><ul>
<li class="chapter" data-level="41.3.1" data-path="multregress.html"><a href="multregress.html#a-one-factor-experiment-with-3-groups"><i class="fa fa-check"></i><b>41.3.1</b> A one factor experiment with 3 groups</a></li>
<li class="chapter" data-level="41.3.2" data-path="multregress.html"><a href="multregress.html#a-two-factor-experiment-with-6-groups"><i class="fa fa-check"></i><b>41.3.2</b> A two factor experiment with 6 groups</a></li>
</ul></li>
<li class="chapter" data-level="41.4" data-path="multregress.html"><a href="multregress.html#multiple-linear-mixed-models"><i class="fa fa-check"></i><b>41.4</b> Multiple linear mixed models</a></li>
<li class="chapter" data-level="41.5" data-path="multregress.html"><a href="multregress.html#multiple-regression-of-observational-data"><i class="fa fa-check"></i><b>41.5</b> Multiple regression of observational data</a></li>
</ul></li>
<li class="chapter" data-level="42" data-path="logregress.html"><a href="logregress.html"><i class="fa fa-check"></i><b>42</b> Logistic regression</a><ul>
<li class="chapter" data-level="42.1" data-path="logregress.html"><a href="logregress.html#uses-of-logistic-regression"><i class="fa fa-check"></i><b>42.1</b> Uses of logistic regression</a><ul>
<li class="chapter" data-level="42.1.1" data-path="logregress.html"><a href="logregress.html#sidebar-doing-logistic-regression-is-machine-learning"><i class="fa fa-check"></i><b>42.1.1</b> Sidebar: Doing logistic regression is machine learning</a></li>
</ul></li>
<li class="chapter" data-level="42.2" data-path="logregress.html"><a href="logregress.html#derivation-of-the-logistic-regression-model"><i class="fa fa-check"></i><b>42.2</b> Derivation of the logistic regression model</a><ul>
<li class="chapter" data-level="42.2.1" data-path="logregress.html"><a href="logregress.html#relationship-of-logit-to-odds-to-the-model-coefficients-and-probability"><i class="fa fa-check"></i><b>42.2.1</b> Relationship of logit to odds to the model coefficients and probability</a></li>
<li class="chapter" data-level="42.2.2" data-path="logregress.html"><a href="logregress.html#additional-types-of-logistic-regression-models"><i class="fa fa-check"></i><b>42.2.2</b> Additional types of logistic regression models</a></li>
</ul></li>
<li class="chapter" data-level="42.3" data-path="logregress.html"><a href="logregress.html#stress-and-survival"><i class="fa fa-check"></i><b>42.3</b> Stress and survival</a><ul>
<li class="chapter" data-level="42.3.1" data-path="logregress.html"><a href="logregress.html#interpretation-of-output"><i class="fa fa-check"></i><b>42.3.1</b> Interpretation of output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="43" data-path="mixedlogistic.html"><a href="mixedlogistic.html"><i class="fa fa-check"></i><b>43</b> Mixed model logistic regression</a><ul>
<li class="chapter" data-level="43.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#mixed-models-fixed-and-random-effects"><i class="fa fa-check"></i><b>43.1</b> Mixed models, fixed and random effects</a><ul>
<li class="chapter" data-level="43.1.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#nfat-localization-within-smooth-muscle-cells"><i class="fa fa-check"></i><b>43.1.1</b> NFAT localization within smooth muscle cells</a></li>
</ul></li>
<li class="chapter" data-level="43.2" data-path="mixedlogistic.html"><a href="mixedlogistic.html#data-2"><i class="fa fa-check"></i><b>43.2</b> Data</a><ul>
<li class="chapter" data-level="43.2.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#inference-2"><i class="fa fa-check"></i><b>43.2.1</b> Inference</a></li>
</ul></li>
<li class="chapter" data-level="43.3" data-path="mixedlogistic.html"><a href="mixedlogistic.html#alternative-analysis"><i class="fa fa-check"></i><b>43.3</b> Alternative analysis</a></li>
</ul></li>
<li class="chapter" data-level="44" data-path="poissonreg.html"><a href="poissonreg.html"><i class="fa fa-check"></i><b>44</b> Poisson regression</a><ul>
<li class="chapter" data-level="44.1" data-path="poissonreg.html"><a href="poissonreg.html#why-not-anova"><i class="fa fa-check"></i><b>44.1</b> Why not ANOVA?</a><ul>
<li class="chapter" data-level="44.1.1" data-path="poissonreg.html"><a href="poissonreg.html#counting-markers"><i class="fa fa-check"></i><b>44.1.1</b> Counting markers</a></li>
<li class="chapter" data-level="44.1.2" data-path="poissonreg.html"><a href="poissonreg.html#counting-depolarizations"><i class="fa fa-check"></i><b>44.1.2</b> Counting depolarizations</a></li>
<li class="chapter" data-level="44.1.3" data-path="poissonreg.html"><a href="poissonreg.html#lever-presses"><i class="fa fa-check"></i><b>44.1.3</b> Lever presses</a></li>
</ul></li>
<li class="chapter" data-level="44.2" data-path="poissonreg.html"><a href="poissonreg.html#the-poisson-generalized-linear-model"><i class="fa fa-check"></i><b>44.2</b> The Poisson generalized linear model</a></li>
<li class="chapter" data-level="44.3" data-path="poissonreg.html"><a href="poissonreg.html#length-of-hospital-stay"><i class="fa fa-check"></i><b>44.3</b> Length of hospital stay</a></li>
<li class="chapter" data-level="44.4" data-path="poissonreg.html"><a href="poissonreg.html#output-interpretation"><i class="fa fa-check"></i><b>44.4</b> Output interpretation</a></li>
</ul></li>
<li class="chapter" data-level="45" data-path="surv.html"><a href="surv.html"><i class="fa fa-check"></i><b>45</b> Survival Analysis</a><ul>
<li class="chapter" data-level="45.1" data-path="surv.html"><a href="surv.html#data-records"><i class="fa fa-check"></i><b>45.1</b> Data records</a><ul>
<li class="chapter" data-level="45.1.1" data-path="surv.html"><a href="surv.html#deriving-times-from-recorded-dates"><i class="fa fa-check"></i><b>45.1.1</b> Deriving times from recorded dates</a></li>
<li class="chapter" data-level="45.1.2" data-path="surv.html"><a href="surv.html#censor"><i class="fa fa-check"></i><b>45.1.2</b> Censor</a></li>
</ul></li>
<li class="chapter" data-level="45.2" data-path="surv.html"><a href="surv.html#kaplan-meier-estimator"><i class="fa fa-check"></i><b>45.2</b> Kaplan-Meier estimator</a></li>
<li class="chapter" data-level="45.3" data-path="surv.html"><a href="surv.html#glioma-data"><i class="fa fa-check"></i><b>45.3</b> Glioma data</a></li>
<li class="chapter" data-level="45.4" data-path="surv.html"><a href="surv.html#kaplan-meier-plots"><i class="fa fa-check"></i><b>45.4</b> Kaplan-Meier plots</a></li>
<li class="chapter" data-level="45.5" data-path="surv.html"><a href="surv.html#log-rank-tests"><i class="fa fa-check"></i><b>45.5</b> Log rank tests</a><ul>
<li class="chapter" data-level="45.5.1" data-path="surv.html"><a href="surv.html#write-up-of-log-rank-test"><i class="fa fa-check"></i><b>45.5.1</b> Write up of log rank test</a></li>
<li class="chapter" data-level="45.5.2" data-path="surv.html"><a href="surv.html#comparing-more-than-two-survival-curves"><i class="fa fa-check"></i><b>45.5.2</b> Comparing more than two survival curves</a></li>
</ul></li>
<li class="chapter" data-level="45.6" data-path="surv.html"><a href="surv.html#cox-proportional-hazards-analysis"><i class="fa fa-check"></i><b>45.6</b> Cox proportional hazards analysis</a><ul>
<li class="chapter" data-level="45.6.1" data-path="surv.html"><a href="surv.html#cox-proportional-hazards-regression-of-glioma"><i class="fa fa-check"></i><b>45.6.1</b> Cox proportional hazards regression of glioma</a></li>
<li class="chapter" data-level="45.6.2" data-path="surv.html"><a href="surv.html#interpretation-of-cox-regression-output"><i class="fa fa-check"></i><b>45.6.2</b> Interpretation of Cox regression output</a></li>
<li class="chapter" data-level="45.6.3" data-path="surv.html"><a href="surv.html#write-up-18"><i class="fa fa-check"></i><b>45.6.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="45.7" data-path="surv.html"><a href="surv.html#summary-13"><i class="fa fa-check"></i><b>45.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="46" data-path="classification.html"><a href="classification.html"><i class="fa fa-check"></i><b>46</b> Classification</a><ul>
<li class="chapter" data-level="46.1" data-path="classification.html"><a href="classification.html#distances"><i class="fa fa-check"></i><b>46.1</b> Distances</a></li>
<li class="chapter" data-level="46.2" data-path="classification.html"><a href="classification.html#clustering-methods"><i class="fa fa-check"></i><b>46.2</b> Clustering methods</a></li>
<li class="chapter" data-level="46.3" data-path="classification.html"><a href="classification.html#principal-component-analysis"><i class="fa fa-check"></i><b>46.3</b> Principal component analysis</a></li>
<li class="chapter" data-level="46.4" data-path="classification.html"><a href="classification.html#synthesize-some-data"><i class="fa fa-check"></i><b>46.4</b> Synthesize some data</a></li>
<li class="chapter" data-level="46.5" data-path="classification.html"><a href="classification.html#visualize-the-data-set"><i class="fa fa-check"></i><b>46.5</b> Visualize the data set</a></li>
<li class="chapter" data-level="46.6" data-path="classification.html"><a href="classification.html#rescaling-data"><i class="fa fa-check"></i><b>46.6</b> Rescaling data</a></li>
<li class="chapter" data-level="46.7" data-path="classification.html"><a href="classification.html#are-there-clusters"><i class="fa fa-check"></i><b>46.7</b> Are there clusters?</a></li>
<li class="chapter" data-level="46.8" data-path="classification.html"><a href="classification.html#how-many-clusters"><i class="fa fa-check"></i><b>46.8</b> How many clusters?</a></li>
<li class="chapter" data-level="46.9" data-path="classification.html"><a href="classification.html#clustering"><i class="fa fa-check"></i><b>46.9</b> Clustering</a></li>
<li class="chapter" data-level="46.10" data-path="classification.html"><a href="classification.html#pam-and-clara"><i class="fa fa-check"></i><b>46.10</b> PAM and CLARA</a></li>
<li class="chapter" data-level="46.11" data-path="classification.html"><a href="classification.html#hierarchical-clustering"><i class="fa fa-check"></i><b>46.11</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="46.12" data-path="classification.html"><a href="classification.html#heatmaps"><i class="fa fa-check"></i><b>46.12</b> Heatmaps</a></li>
<li class="chapter" data-level="46.13" data-path="classification.html"><a href="classification.html#summary-14"><i class="fa fa-check"></i><b>46.13</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="47" data-path="rnaseq.html"><a href="rnaseq.html"><i class="fa fa-check"></i><b>47</b> RNA-seq with R</a><ul>
<li class="chapter" data-level="47.1" data-path="rnaseq.html"><a href="rnaseq.html#install-bioconductor"><i class="fa fa-check"></i><b>47.1</b> Install Bioconductor</a></li>
<li class="chapter" data-level="47.2" data-path="rnaseq.html"><a href="rnaseq.html#import-raw-count-data"><i class="fa fa-check"></i><b>47.2</b> Import raw count data</a></li>
<li class="chapter" data-level="47.3" data-path="rnaseq.html"><a href="rnaseq.html#munge-to-simpler-table"><i class="fa fa-check"></i><b>47.3</b> Munge to simpler table</a></li>
<li class="chapter" data-level="47.4" data-path="rnaseq.html"><a href="rnaseq.html#filtering"><i class="fa fa-check"></i><b>47.4</b> Filtering</a></li>
<li class="chapter" data-level="47.5" data-path="rnaseq.html"><a href="rnaseq.html#dgelist"><i class="fa fa-check"></i><b>47.5</b> DGEList</a></li>
<li class="chapter" data-level="47.6" data-path="rnaseq.html"><a href="rnaseq.html#visualization"><i class="fa fa-check"></i><b>47.6</b> Visualization</a></li>
<li class="chapter" data-level="47.7" data-path="rnaseq.html"><a href="rnaseq.html#classification-1"><i class="fa fa-check"></i><b>47.7</b> Classification</a></li>
<li class="chapter" data-level="47.8" data-path="rnaseq.html"><a href="rnaseq.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>47.8</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="47.9" data-path="rnaseq.html"><a href="rnaseq.html#summary-15"><i class="fa fa-check"></i><b>47.9</b> Summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">JABSTB: Statistical Design and Analysis of Experiments with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introanova" class="section level1">
<h1><span class="header-section-number">Chapter 25</span> Introduction to ANOVA</h1>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="introanova.html#cb552-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb552-2"><a href="introanova.html#cb552-2"></a><span class="kw">library</span>(RColorBrewer)</span></code></pre></div>
<p>The choice of any statistical design and analysis is always driven by the type of outcome and predictor variables involved. Recall that all variables are either continuous or discrete. Furthermore, it’s helpful to think of outcome variables further classified as either <a href="data.html#data">measured, ordered or sorted</a>, where measured variables are continuous, and ordered and sorted are discrete.</p>
<p>Once the dependent variable is deemed to be continuous measured, the model for the experiment is determined by the number of levels of the independent variables.</p>
<div class="figure"><span id="fig:unnamed-chunk-269"></span>
<img src="images/anova_heuristic.jpg" alt="ANOVA heuristic" width="480" />
<p class="caption">
Figure 25.1: ANOVA heuristic
</p>
</div>
<p>The analysis of variance (ANOVA) is a method to design and evaluate experiments in which the predictor variable(s) are discrete factors with three or more levels and when the outcome variable is on some continuous measured scale. ANOVA is also univariate, in so far as the analysis involves only a single outcome variable.</p>
<div id="assumptions" class="section level2">
<h2><span class="header-section-number">25.1</span> Assumptions</h2>
<p>The validity of an ANOVA depends upon fulfilling the following assumptions. These are basically the same as we discussed for t-testing.</p>
<p>Strong assumptions, do not use ANOVA when violated</p>
<ul>
<li>Every replicate is independent of all others.</li>
<li>Some random process is used when generating measurements.</li>
</ul>
<p>Weaker assumptions, ANOVA is said to be robust to violations of the following</p>
<ul>
<li>The distribution for the residuals from which the outcome variable is derived is continuous random normal.</li>
<li>The variances of the groups are approximately equal.</li>
</ul>
<p>With small samples it is usually difficult to conclude that these weaker assumptions are met. Both affirmative and negative results of tests of normality, tests for homogeneity of variance and outlier tests should be taken with a grain of salt. The smaller the sample size, the larger the grain of salt.</p>
<p>Use your judgment. Do you have any reason to believe the variable is not normally distributed? Are you not up against a zero boundary? Are you confident the variable is measured in a linear range of your assay procedure? If in doubt, analyze the data with some other procedure.</p>
<p>Data that appears markedly skewed can be transformed using log or reciprocal functions. The resulting distributions are ~ normal. ANOVA testing can be performed on those transformed values. As we discussed previously, the nonparametric analogs of ANOVA (Kruskal-Wallis and Friedman tests) are suitable options when normality is violated.</p>
<p>There are alternative analytic options as will be detailed below. In particular, there is no need to use ANOVA for discrete types of outcome data. ANOVA is not designed to analyze non-continuous data. Additionally, regression, either linear or nonlinear, is often a preferred method of analysis when the predictor variable is continuous rather than discrete.</p>
</div>
<div id="types-of-designs" class="section level2">
<h2><span class="header-section-number">25.2</span> Types of designs</h2>
<p><strong>tl;dr:</strong> ANOVA represents a family of about a dozen different statistical models. These differ by how many predictor factors are involved and whether or not the measurements are intrinsically-related. A given factor can be applied in either a completely randomized or repeated/related measures format. Experiments can involve one, two or even three factors.</p>
<p>ANOVA experiments tend to design themselves since they are fairly intuitive way of asking questions. In fact, ANOVA is probably the most widely used experimental design in the biomedical sciences. If we randomly access any article in our favorite journal, chances are it will have a figure or table depicting an ANOVA design.</p>
<p>They tend to look something like these.</p>
<pre><code>## Warning: `fun.y` is deprecated. Use `fun` instead.

## Warning: `fun.y` is deprecated. Use `fun` instead.

## Warning: `fun.y` is deprecated. Use `fun` instead.</code></pre>
<div class="figure"><span id="fig:unnamed-chunk-270"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-270-1.png" alt="Typical graphs depicting some of the different ANOVA designs." width="480" /><img src="jabstb_files/figure-html/unnamed-chunk-270-2.png" alt="Typical graphs depicting some of the different ANOVA designs." width="480" /><img src="jabstb_files/figure-html/unnamed-chunk-270-3.png" alt="Typical graphs depicting some of the different ANOVA designs." width="480" />
<p class="caption">
Figure 25.2: Typical graphs depicting some of the different ANOVA designs.
</p>
</div>
<p>Why is ANOVA so common?</p>
<p>First, ANOVA is versatile. One or more factors, each at several levels, can be tested simultaneously. We can test for the main effect of each factor, or for interactions between factors. We can also test for differences between specific individual groups within a context of many other groups, using post hoc analysis. Designs involving multiple instrinsically-related measurements from a common replicate are readily accommodated. We can have mixed designs where one factor is completely randomized and another is related measure within a single multi-factor experiments.</p>
<p>Second, ANOVA is efficient. As a general rule, fewer replicates are needed to make the same number of pairwise group comparisons than would otherwise be necessary using a t-test-based experimental design. That efficiency can improve modestly as the number of groups increases.</p>
<p>Third, we can test many hypotheses simultaneously or a grand hypothesis. Since each individual hypothesis test carries a risk of type1 error, ANOVA serves as a protocol to detect differences between many groups while ensuring that the overall type1 error, the so-called experimentwise error or the family-wise error rate (FWER), won’t inflate to intolerable levels.</p>
</div>
<div id="jargon-factors-and-levels" class="section level2">
<h2><span class="header-section-number">25.3</span> Jargon: Factors and levels</h2>
<p>In jargon, ANOVA predictor variables are classified as “factors”. ANOVA designs are said to be factorial. They are multifactorial if more than a single factor is involved.</p>
<p>In other corners, ANOVA is referred to as factorial analysis (which should not be confused with factor analysis). Where some people describe an ANOVA experiment as a “one-way ANOVA” others might describe it as “one-factor ANOVA”. Yet elsewhere it is called multiple regression. It’s all the same.</p>
<p>The factors of ANOVA represent categorical, discrete variables that are each applied at two or more levels. For example, a factor at three levels is a predictor variable that has three discrete values.</p>
<p>Imagine an experiment to explore how a particular gene influences blood glucose levels. Blood glucose levels, a continuous variable, are measured in subjects that have three different genotypes for the gene: either wild-type, or heterozygous knockouts, or homozygous knockouts. Here, genotype is a discrete predictor variable, a factor, which has three levels.</p>
<div id="woring-with-factors-in-r" class="section level3">
<h3><span class="header-section-number">25.3.1</span> Woring with factors in R</h3>
<p>R requires that ANOVA predictor variables are classified as factors in data sets.</p>
<p>The following script creates a vector object called <code>genotype</code>. The object is a representation of the genotype variable. The data class for that vector is <code>character</code> because it is comprised of character strings. But look what happens when it is packaged into a data frame called <code>my.factors</code> using the <code>data.frame</code>. R coerces <code>genotype</code> into a factor variable with 3 levels. Which is nice.</p>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="introanova.html#cb554-1"></a>genotype &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;wild-type&quot;</span>, <span class="st">&quot;heterozygote&quot;</span>, <span class="st">&quot;homozygote&quot;</span>)</span>
<span id="cb554-2"><a href="introanova.html#cb554-2"></a><span class="kw">class</span>(genotype)</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="introanova.html#cb556-1"></a>my.factors &lt;-<span class="st"> </span><span class="kw">data.frame</span>(genotype)</span>
<span id="cb556-2"><a href="introanova.html#cb556-2"></a><span class="kw">str</span>(my.factors)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    3 obs. of  1 variable:
##  $ genotype: chr  &quot;wild-type&quot; &quot;heterozygote&quot; &quot;homozygote&quot;</code></pre>
<p>But beware. The tidyverse <code>tibble</code> function is an analog of <code>data.frame</code>. Yet <code>tibble</code> does not coerce factors from character vectors. If using <code>tibble</code> or functions that create tibbles, such as <code>read_csv</code>, we’ll need to remember to convert by hand using functions like <code>as_factor</code>.</p>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="introanova.html#cb558-1"></a>genotype &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;wild-type&quot;</span>, <span class="st">&quot;heterozygote&quot;</span>, <span class="st">&quot;homozygote&quot;</span>)</span>
<span id="cb558-2"><a href="introanova.html#cb558-2"></a><span class="kw">class</span>(genotype)</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="introanova.html#cb560-1"></a>my.factors &lt;-<span class="st"> </span><span class="kw">tibble</span>(genotype)</span>
<span id="cb560-2"><a href="introanova.html#cb560-2"></a><span class="kw">str</span>(my.factors)</span></code></pre></div>
<pre><code>## tibble [3 x 1] (S3: tbl_df/tbl/data.frame)
##  $ genotype: chr [1:3] &quot;wild-type&quot; &quot;heterozygote&quot; &quot;homozygote&quot;</code></pre>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="introanova.html#cb562-1"></a><span class="kw">as_factor</span>(genotype)</span></code></pre></div>
<pre><code>## [1] wild-type    heterozygote homozygote  
## Levels: wild-type heterozygote homozygote</code></pre>
<p>This coercion won’t occur if a factor in an experiment represents a variable with numeric values. To illustrate what I mean by this, imagine adding a second factor to the genotype experiment.</p>
<p>We wish to test for the effect of an antidiabetic drug on blood glucose at 0, 10 and 30 microgram/kg. These effects would be measured at each level of the genotype factor, leading to a nifty 3 X 3 factorial design.</p>
<p>We would create the vector <code>drug</code> as an object representing the drug variable and its three levels as follows. Note however that here, R does not coerce numeric values as factors:</p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="introanova.html#cb564-1"></a>drug &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">30</span>)</span>
<span id="cb564-2"><a href="introanova.html#cb564-2"></a>my.factors &lt;-<span class="st"> </span><span class="kw">data.frame</span>(genotype, drug)</span>
<span id="cb564-3"><a href="introanova.html#cb564-3"></a><span class="kw">str</span>(my.factors)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    3 obs. of  2 variables:
##  $ genotype: chr  &quot;wild-type&quot; &quot;heterozygote&quot; &quot;homozygote&quot;
##  $ drug    : num  0 10 30</code></pre>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="introanova.html#cb566-1"></a><span class="kw">class</span>(drug)</span></code></pre></div>
<pre><code>## [1] &quot;numeric&quot;</code></pre>
<p>That’s easily fixed using the <code>as.factor</code> or <code>as_factor</code> function. Whatev.</p>
<div class="sourceCode" id="cb568"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb568-1"><a href="introanova.html#cb568-1"></a>drug &lt;-<span class="st"> </span><span class="kw">as_factor</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">30</span>))</span>
<span id="cb568-2"><a href="introanova.html#cb568-2"></a>my.factors &lt;-<span class="st"> </span><span class="kw">data.frame</span>(genotype, drug)</span>
<span id="cb568-3"><a href="introanova.html#cb568-3"></a><span class="kw">str</span>(my.factors)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    3 obs. of  2 variables:
##  $ genotype: chr  &quot;wild-type&quot; &quot;heterozygote&quot; &quot;homozygote&quot;
##  $ drug    : Factor w/ 3 levels &quot;0&quot;,&quot;10&quot;,&quot;30&quot;: 1 2 3</code></pre>
<div class="sourceCode" id="cb570"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb570-1"><a href="introanova.html#cb570-1"></a><span class="kw">class</span>(drug)</span></code></pre></div>
<pre><code>## [1] &quot;factor&quot;</code></pre>
<p>Alternately, we would enter the drug levels as character strings (tossing in some other random character value), which would be coerced into a factor at two levels when added to a data frame, like for the first example above:</p>
<div class="sourceCode" id="cb572"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb572-1"><a href="introanova.html#cb572-1"></a>drug &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;zero&quot;</span>, <span class="st">&quot;dark&quot;</span>, <span class="st">&quot;thirty&quot;</span>)</span></code></pre></div>
<p>When you forget to convert a variable to a factor, R’s anova functions will bark at you and you won’t know what the problem is because the error messages are confusing. R doesn’t know that you don’t know you should be using factors.</p>
<p><strong>You can do ANOVA with continuous predictor variables. You just have to treat them as factors.</strong></p>
<p>When a continuous predictor variable is used at many levels (eg, a time series, a dose series, etc), regression <em>per se</em> can be a better alternative to ANOVA. Regression allows for capturing additional information encoded within continuous variables (eg, rate constants, half-lives, etc).</p>
<p>But that decision is more scientific than it is statistical.</p>
</div>
</div>
<div id="anova-models-one--two--and-three-way" class="section level2">
<h2><span class="header-section-number">25.4</span> ANOVA models: One-, Two-, and Three-way</h2>
<p>If an experiment has only one factor, it is a one-way ANOVA design (alternately, “one-factor ANOVA”).</p>
<p>If an experiment has two factors, it is a two-way ANOVA or two factor design.</p>
<p>Three factors? Three-way ANOVA.</p>
<p>All of these models can also be either completely randomized, repeated/related measures, or mixed.</p>
<p>In a completely randomized structure, every level of the factor(s) is randomly assigned to replicates. Every replicate measurement is independent from all others. The two group analog is the unpaired t-test.</p>
<p>In a related measures structure, measurements within an experimental unit are intrinsically-linked. Every replicate receives <em>all</em> levels of a factor (eg, before-after, stepped dosing, or when subjects are highly homogeneous, such as cultured cells or inbred animals, or repeated recording from the same cell, etc).</p>
<p>Thus, the possible ANOVA models are quite diverse:</p>
<ul>
<li>One-way ANOVA -completely randomized</li>
<li>One-way ANOVA -related measures</li>
<li>Two-way ANOVA -completely randomized on both factors</li>
<li>Two-way ANOVA -related measures on both factors</li>
<li>Two-way ANOVA -mixed, one factor completely randomized, the other factor related measures</li>
<li>Three-way ANOVA -can be purely CR, ourely RM or mixed</li>
</ul>
<p>I should mention that ANOVA for even more than three factors is conceptually possible. However, such large, complex designs have considerable downside where, by ANOVA analysis, it is difficult to tease out which factor and level is responsible for what.</p>
<p>Three-way ANOVA, for example, allows for such a large number of hypotheses to be tested (three different main effects and four possible interaction effects, not to mention the large number of post hoc group comparisons) that it can be difficult to conclude exactly which factor at what level is responsible for any observed effects.</p>
<p>These larger designs also tend to break the efficiency rule, it’s fair to say that three way ANOVA designs tend to be over-ambitious experiments…over-designed and usually under powered for the large number of hypotheses they can test. As a general rule, I advise against doing them.</p>
<p>Designing an experiment as completely randomized or related measures is largely a scientific, not a statistical, decision. Measurements are either intrinsically-linked or they are not based upon the nature of the biological material that we have at hand. Or because of the way the experiment is being conducted. When we’ve concluded that measurements are intrinsically-linked, then the choice must be a related measures design and analysis.</p>
<p>Similarly, choosing to run experiments as one-way or two-way ANOVA designs is also scientific. In fact, intuitively, we choose a two-way design mostly to test whether two factors interact. When not interested in an interaction, it is usually best not to combine them in an experiment. The main effect questions may be more safely answered using separate one-way ANOVAs, instead.</p>
<p>We’ll discuss interaction hypotheses and effects in more detail when we arrive at our two-way ANOVA discussion in the course.</p>
</div>
<div id="inference" class="section level2">
<h2><span class="header-section-number">25.5</span> Inference</h2>
<p>ANOVA can be used inferentially as stand alone test for the main effect of a factor. More commonly, it is used as an omnibus test for permission to explore group differences. Most of the time we see it used as the latter but there is no “rule” that this must be so.</p>
<p>A positive ANOVA test result can be used to infer whether a factor, or an interaction between factors, is effective.</p>
<p>Continuing with the example, I can use a positive F-test to conclude that genotype at a given locus influences blood glucose. And just leave it at that, without demonstrating which levels of the genotype differ from each other. I would write it up as follows: <em>The genotype influences blood glucose (one-way related measures ANOVA, F(2,15)=11.48, p=0.0009).</em> Then I will point to the data, which will speak for itself revealing which groups actually differ.</p>
<p>Alternately, a positive F-test result can be used as an omnibus. Here, a positive F-test implies that at least two levels of a factor differ from each other. The positive F-test grants us access to finding exactly where those differences exist. This is done by making posthoc pairwise comparisons.</p>
<p>We would, perhaps, compare each of the responses to the three genotypes to each other; for a total of 3 t-test like comparisons. I would adjust the p-values of each of those outcomes to control the FWER for those multiple comparisons.</p>
<p>The decision to use the F-test as a stand alone or as an omnibus is driven by our scientific objectives.</p>
<div class="figure"><span id="fig:unnamed-chunk-276"></span>
<img src="images/anova_process.jpg" alt="ANOVA work flow at a 5% type1 error threshold." width="480" />
<p class="caption">
Figure 25.3: ANOVA work flow at a 5% type1 error threshold.
</p>
</div>
<div id="a-quick-peek-at-posthoc-testing" class="section level3">
<h3><span class="header-section-number">25.5.1</span> A quick peek at posthoc testing</h3>
<p>These “post hoc” comparisons are, essentially, any of several variations on the t-test that are performed simultaneously. See Chapter <a href="posthoc.html#posthoc">32</a> for more information about these.</p>
<p>These are designed in a way to limit the FWER to the predetermined type1 error threshold. Instead of all 5% error being devoted to compare two groups, the 5% is distrubuted across many comparisions. Sometimes dozens or more.</p>
<p>The choices for comparisons after the F-test can be either planned or unplanned. They should be driven by scientific, rather than statistical, reasoning.</p>
<p>Unplanned is comparing all levels of all factors to each other. We might do that if we are looking for any and all differences on a project that is more exploratory in scope, such as in a hit screen. Planned comparisons focus in on a much more limited subset of groups. We do this when the experiment is designed to test a fairly specific hypothesis or group of hypotheses.</p>
<p>What is important, statistically, is to make adjustments to the p-value threshold given all the comparisons made, so that the overall experimentwise type1 error does not exceed our declared threshold (usually 5%).</p>
<p>In an experiment whose number of groups equals <span class="math inline">\(k\)</span>, there are a total of <span class="math inline">\(m=\frac{k(k-1)}{2}\)</span> possible comparisons to make.</p>
<p>Let’s use the simplest case of a <span class="math inline">\(k=3\)</span> groups ANOVA as an example, such as the glucose genotype discussed above.</p>
<p>There are <span class="math inline">\(m=\frac{3(3-1)}{2}=3\)</span> comparisons that can be made an imagine using the Bonferroni correction (<span class="math inline">\(\alpha_{adjust}=\frac{0.05}{m}=0.01667\)</span>) the type1 error threshold is now 0.01667.</p>
<p>Thus, our adjusted threshold for declaring a statistical difference between 2 groups is now p &lt; 0.01667, rather than p &lt; 0.05.</p>
<p>If doing the 3 X 3 two-way ANOVA with genotype and drug as two factors, we now have <span class="math inline">\(k=9\)</span> groups, yielding 36 possible comparisons and the threshold <span class="math inline">\(\alpha_{adjust}=\frac{0.05}{m}=0.001389\)</span> for each comparison.</p>
</div>
</div>
<div id="anova-calculations" class="section level2">
<h2><span class="header-section-number">25.6</span> ANOVA calculations</h2>
<p>The simplest way to think about ANOVA is that it operates like a variance budgeting tool. In the final analysis, the more variance associated with the grouping factor(s) compared to the residual variance, the more likely that some group means will differ.</p>
<p>ANOVA uses the least squares method to derive and account for sources of variation within a data set.</p>
<p>Recall, our measurements of the random variable <span class="math inline">\(Y\)</span> have values of <span class="math inline">\(y_i\)</span> for <span class="math inline">\(i=1, 2,..n\)</span> and a sample mean <span class="math inline">\(\bar y\)</span>. Variance is calculated by dividing the sum of its squared deviates, <span class="math inline">\(SS\)</span>, by the sample degrees of freedom (df).</p>
<p><span class="math display">\[var(Y)=\frac{\sum_{i=1}^n(y_i-\bar y)^2}{n-1}=\frac{SS}{df}=MS\]</span></p>
<p>In ANOVA jargon the variance is also commonly referred to as the mean square <span class="math inline">\(MS\)</span>. This jargon emphasizes that variance can be thought of as an averaged deviation for sample measurements from their mean values.</p>
<p>That formula only illustrates how variance is calculated for a single sample group. What about multiple groups? As you might imagine, we have to incorporate information from all of the groups.</p>
<p>To begin to understand that, recognize that all ANOVAs, irrespective of the specific design are assumed to have two fundamental sources of variation:</p>
<ol style="list-style-type: decimal">
<li>Variation due to the experimental model, which is determined by the nature of responses to the predictor variables.</li>
<li>Residual variation, which is variation that cannot be explained by the predictor variables.</li>
</ol>
<p><strong>Law of Conservation of SS:</strong> That total variation within an experiment can be expressed as the sum of the squared deviates for these two sources. <span class="math display">\[SS_{total}=SS_{model}+SS_{residual} \]</span></p>
<p>Let’s imagine a simple one-way completely randomized ANOVA data set that looks like the graph below. There is only one factor, which has three levels. Each group has five independent replicates, for a total of 15 replicates within the entire experiment.</p>
<p>Our model seeks to account for the effect of genotype
<span class="math display">\[SS_{model}=SS_{genotype} \]</span></p>
<p>Models are perfect but data are not. Not every data point is equal to the value of the black bar it is near. The residual variation are these differences and thus <span class="math display">\[SS_{total}=SS_{genotype}+SS_{residual}\]</span></p>
<p>The graph illustrates each data point, the means for each group (black bars) and the grand mean of the sample (gold bars).</p>
<p>We can readily imagine the distances from the data points to the grand mean.</p>
<p>We can readily imagine the distances from the data points to the group means.</p>
<p>One can also appreciate and the distances from the group means to the grand mean.</p>
<p>We probably have a harder time visualizing the squares of those distances. My mind sees it Euclidean (geometrically). Larger distances, squared, lead to bigger boxes! The bigger the boxes, the greater that replicate contributes to the variance.</p>
<p>With that picture in mind, think of the variances within the experiment as follows:</p>
<ul>
<li>Total variance: average squared distances of the all the points to the grand mean</li>
<li>Model variance: weighted average squared distances of group means to the grand mean</li>
<li>Residual variance: average squared distances of the points to the group means</li>
</ul>
<div class="figure"><span id="fig:unnamed-chunk-277"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-277-1.png" alt="A completely randomized one way ANOVA, the genotype factor has three levels. Gold bar = grand mean, black bar = group means" width="672" />
<p class="caption">
Figure 25.4: A completely randomized one way ANOVA, the genotype factor has three levels. Gold bar = grand mean, black bar = group means
</p>
</div>
<div id="sums-of-squares-partitioning" class="section level3">
<h3><span class="header-section-number">25.6.1</span> Sums of Squares partitioning</h3>
<p>The first step in an ANOVA involves partitioning the variation in a data set using sums of squares.</p>
<p>In this experiment, there are <span class="math inline">\(i=1, 2,...n\)</span> independent replicates. There are also <span class="math inline">\(j=1, 2,...k\)</span> groups.</p>
<p>The total sum of squares is the sum of the squared deviation from all data points to <span class="math inline">\(\hat y\)</span>, which represents the grand mean of the sample.</p>
<p><span class="math display">\[SS_{total}=\sum_{j=1}^k\sum_{i=1}^n(y_i-\hat y)^2\]</span></p>
<p>The sum of squares for the genotype effect is sum of the weighted squared deviation between the group means, <span class="math inline">\(\bar y_j\)</span> and the grand mean. Here, <span class="math inline">\(n_j\)</span> is the sample size within the <span class="math inline">\(j^{th}\)</span> group. Thus, the group sample sizes are creating that weight.</p>
<p><span class="math display">\[SS_{genotype}=\sum_{j=1}^kn_j(\bar y_j-\hat y)^2\]</span></p>
<p>Anova software usually refers to this variation as the “treatment” sum of squares.</p>
<p>Finally, the residual sum of squares is calculated as the sum of the squared deviation between replicate values and group means,</p>
<p><span class="math display">\[SS_{residual}=\sum_{j=1}^k\sum_{i=1}^n(y_i-\bar y_j)^2\]</span></p>
<p>which, because the total variation amount of is fixed, can also be calculated as follows:</p>
<p><span class="math display">\[SS_{residual}=SS_{total}-SS_{genotype}\]</span></p>
<p>Software will tend to label this as either residual variation or as “error”. The term “error” arises from ANOVA theory, which holds that the true population means represented by these sample groups are “fixed” in the population. Thus, any deviation from the means is explained by random error.</p>
<p>Residual is jargon for the measurements of that “error”.</p>
<p>Perhaps we can intuit a few things.</p>
<p>First, the residual variation is the variation unaccounted for by the model. Meaning that whatever its causes, they are not under experimental control. If the data were perfectly explained by the model, every data point would be identical to a value of a group mean.</p>
<p>Second, if the variation around each group mean remains the same, but as the group means differ from each other more and more, the greater the fraction of the overall variation that will be associated with the model, and the less that will be associated with the residual.</p>
<p>Third, when the noise around those group means increases, less of the total variation will be associated with the model of group means, and the more with the residual.</p>
<p>In other words, big effect sizes are easier to detect, while noisy experiments tend to hide detectable differences between means, whereas clean experiments favor detecting these differences.</p>
<p>If this seems bloody obvious, and it is simple, then we should not have any problem processing how ANOVA works.</p>
<p>The following two graphs emphasize these observations. In the null graph, the group means are roughly equivalent and very nearly the same as the grand mean. There’s very little model variation relative to the grand mean. Most of the variation is in the residuals, relative to each group mean.</p>
<p>In the effective treatment graph, where the means truly differ because I coded them to differ, the residual variation is about the same as the null. The distance from each point to its group mean is about the same as in the null case.</p>
<p>But we can see there is a lot more model variation, at least compared to the null graph: The group means (black bars) really separate from the grand mean (gold line).</p>
<p><strong>ANOVA tests become very simple when we understand this. The variation associated with the model is the signal and the residual variation is the noise. F tests are simple ratios between the model variance (or elements of the model) and residual variance. F tests are simple signal-to-noise ratios, just like every other statistical test.</strong></p>
<p><img src="jabstb_files/figure-html/unnamed-chunk-278-1.png" width="288" /><img src="jabstb_files/figure-html/unnamed-chunk-278-2.png" width="288" /></p>
</div>
<div id="the-anova-table" class="section level3">
<h3><span class="header-section-number">25.6.2</span> The ANOVA table</h3>
<p>The typical ANOVA table lists the following:</p>
<ul>
<li>source of variation,</li>
<li>its <span class="math inline">\(df\)</span>,</li>
<li>its <span class="math inline">\(SS\)</span>,</li>
<li>its <span class="math inline">\(MS\)</span></li>
<li>an F-test, where appropriate</li>
<li>a p-value from the F-test</li>
</ul>
<p>ANOVA functions in R vary in their output. But here’s the ANOVA table output for the data in the last previous figure:</p>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="introanova.html#cb573-1"></a><span class="kw">anova</span>(<span class="kw">lm</span>(blood_glucose <span class="op">~</span><span class="st"> </span>genotype, data))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: blood_glucose
##           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
## genotype   2 21176.9 10588.4  22.979 7.877e-05 ***
## Residuals 12  5529.4   460.8                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div id="sums-of-squares" class="section level4">
<h4><span class="header-section-number">25.6.2.1</span> Sums of squares</h4>
<p>As described above. These are the absolute measures of variation or deviation. Their values are in squared units of the response variable. Weird. But all <span class="math inline">\(SS\)</span> sources will sum to <span class="math inline">\(SS_{total}\)</span> and that is helpful.</p>
</div>
<div id="degrees-of-freedom" class="section level4">
<h4><span class="header-section-number">25.6.2.2</span> Degrees of freedom</h4>
<p>Variance is averaged deviation. The <span class="math inline">\(SS\)</span> averaged. To calculate a variance we’ll need to divide the sum of squares by the degrees of freedom for that sum of squares. Just as sum of squares are calculated differently depending on whether it is total, or model or residual, so too are degrees of freedom.</p>
<p>The mathematical theory behind <span class="math inline">\(df\)</span> is a bit more complicated than this, but as a general rule, we lose a degree of freedom every time the calculation of a mean is involved in determination of a given <span class="math inline">\(SS\)</span>. The basic idea is this: For that mean value to be true, one of the values used in the calculation must remain fixed, while all the others are free to vary.</p>
<p>The degrees of freedom for total variance are <span class="math inline">\(df_{total}=N-1\)</span>. We use all 15 replicates, <span class="math inline">\(N\)</span>, to calculate the grand mean <span class="math inline">\(\hat y\)</span>. One degree of freedom is lost for that grand mean to be true, one of those replicate values must be fixed while the others are free to vary.</p>
<p>We have <span class="math inline">\(k\)</span> predictor levels. The degrees of freedom for the genotype model variance are <span class="math inline">\(df_{genotype}=k-1\)</span>, because the <span class="math inline">\(SS_{model}\)</span> calculation is based upon the group means, the value of one group mean must be fixed while two are free to vary.</p>
<p>The residual degrees of freedom are <span class="math inline">\(df_{residual}=N-k\)</span> because although residual values are calculated for each data point, one degree of freedom is lost to calculate each group mean.</p>
</div>
<div id="the-mean-squares" class="section level4">
<h4><span class="header-section-number">25.6.2.3</span> The mean squares</h4>
<p>The mean squares are ANOVA jargon to represent variances. Think of variance as averaged variation or averaged deviation.</p>
<p>Total variance is <span class="math display">\[MS_{total}=\frac{SS_{total}}{df_{total}} \]</span></p>
<p>The variance associated with the model is <span class="math display">\[MS_{model}=\frac{SS_{model}}{df_{model}} \]</span></p>
<p>And the residual variance is <span class="math display">\[MS_{residual}=\frac{SS_{residual}}{df_{residual}} \]</span></p>
</div>
<div id="the-f-statistic-and-p-value" class="section level4">
<h4><span class="header-section-number">25.6.2.4</span> The F statistic and p-value</h4>
<p>The null distributions of the F statistic are explained in Chapter <a href="fdistr.html#fdistr">26</a>.</p>
<p>The value of F is the ratio of two variances. For the result in the ANOVA table above <span class="math inline">\(F_{2,12}=\frac{10588.4}{460.8}=22.979\)</span>.</p>
<p>We can interpret that as saying the variance associated with the genotype model is 22.979-fold greater than the residual variance.</p>
<p>But is an F-statistic value of 22.979 with 2 and 12 degrees of freedom extreme?</p>
<p>The p-value is derived from an null F probability distribution that has 2 and 12 degrees of freedom. The p-value result in the table (<code>Pr(&gt;F)</code>) is obtained using R’s <code>pf</code> function:</p>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="introanova.html#cb575-1"></a><span class="kw">pf</span>(<span class="fl">22.979</span>, <span class="dv">2</span>, <span class="dv">12</span>, <span class="dt">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 7.87784e-05</code></pre>
<p>The example above is a simple one factor completely randomized model for which there is only one F test, because it has only one component. Experimental designs that have more components, such as two or more factors with our without related measures, will have more F tests if completely randomized. A three-way ANOVA performed a certain way can have about a dozen F tests.</p>
</div>
</div>
<div id="post-hoc-group-comparisons" class="section level3">
<h3><span class="header-section-number">25.6.3</span> Post-hoc group comparisons</h3>
<p>A positive ANOVA F test only tells you that the model (or its component) explains the response better than null.</p>
<p>The final stage of ANOVA, which is optional but done very commonly, is to run what are essentially t-test-like comparisons between groups. These posthoc tests help identify the specific factor levels that actually differ from each other.</p>
<p>The total number of comparisons that could be made are <span class="math inline">\(\frac{k(k-1)}{2}\)</span>. Sometimes we are fishing and want to do just that. Other timest, for scientific reasons, only a specific subset are of any interest. For example, we would only want to compare test groups to a negative control group.</p>
<p>There are several ways to run these post-hoc group comparisons. See Chapter <a href="posthoc.html#posthoc">32</a> for more information about these various options.</p>
<p>What is most important, however, is these be done in a way that keeps the family-wise error rate (FWER) below the pre-set type1 error threshold. That threshold is commonly 5% in biomedical research, but its value can be whatever you decide.</p>
<p>When would we change that threshold? Consider when we are conducting an exploratory “hit” screen involving large numbrers of groups. For example, we have a lot of RNAseq data. Scientifically the mistake of throwing away false negatives might be worse than the mistake of accepting false positives. In that case, raising the type1 error threshold to 10% has merit.</p>
</div>
</div>
<div id="completely-randomized-or-related-measures" class="section level2">
<h2><span class="header-section-number">25.7</span> Completely randomized or related measures</h2>
<p>Related measures (RM) ANOVA is done when the measurements are not completely independent, but instead are intrinsically-linked.</p>
<p>Examples of intrinsically-linked subjects include</p>
<ul>
<li>two measurements, one from each of a pair of identical human twins,</li>
<li>before and after on a single subject,</li>
<li>all plates and wells from a single batch or passage of a cell culture,</li>
<li>a protein preparation from a single batch,</li>
<li>a single cell in culture,</li>
<li>split tissues from one animal subject, and</li>
<li>litter mates of isogenic animal strains.</li>
</ul>
<p>There are certainly others.</p>
<p>An RM design involves taking multiple measurements from each independent replicate. Each replicate may vary randomly. That random variation between replicates can be accounted for, too. Because we can account for the variation within each replicate, that source of variation is no longer in the residual error term, but can be taken right into the model.</p>
<div id="the-problem-of-lost-data-in-related-measures-designs" class="section level3">
<h3><span class="header-section-number">25.7.1</span> The problem of lost data in related measures designs</h3>
<p>The CR vs RM design decision has a few important consequences.</p>
<p>First, when within-subject correlation is high, RM are much more efficient and less costly to produce. How much more? You can run Monte Carlo simulations to establish this for virtually any set of conditions you can imagine.</p>
<p>Second, over the course of any experiment it is possible to lose specific response values here and there. For example, a data value may be lost due to a bad lane in a replicate western blot, or we accidentally throw away a tube from a series, or we have a spill, or any of a number of such primitive errors.</p>
<p>CR ANOVA is tolerant of such losses. That leads to a missing replicate, and an unbalanced data sets, but it’s only one or a few values out of many.</p>
<p>That is not always the case with related measures designs. All of the values for every level of every factor for every replicate must be included for the analysis. If any values are missing for a given replicate, all of the remaining values for that replicate either have to be censored, or the missing values should be imputed.</p>
<p>The missing data problem becomes amplified in two way and three way related measures ANOVA. Those experiments tend to have more groups, meaning more data is at risk of being censored.</p>
<p>Researchers often ask if it is reasonable to ‘flip’ to a completely randomized analysis when they notice too many values are missing from their data set.</p>
<p>No, it is not reasonable. In fact, that’s a horrible p-value hack. The type of experimental design is scientifically-driven. Intrinsically-linked measurements are not independent, and should not be analyzed as if they are independent. To do so violates one of the two primary assumptions of the statistical analysis.</p>
<p>How to avoid this problem?</p>
<p>First, where possible, include an extra replicate or two as a hedge over what the power analysis suggests is necessary.</p>
<p>Second, don’t make a RM design too over-ambitious. Limit the number of levels to those that are scientifically important.</p>
<p>Third, be aware of the risk of lost values. Is the experimental protocol difficult? Are any protocol steps at high risk of failure? Are there any intrinsic barriers to efficiently collecting the data?</p>
</div>
</div>
<div id="two-way-anova" class="section level2">
<h2><span class="header-section-number">25.8</span> Two-way ANOVA</h2>
<p>This is a method to investigate the effects of two factors simultaneously. Thus, the variation associated with each factor can be partitioned. This also allows for assessing the variation associated with an interaction between the two factors.</p>
<p>What is an interaction? Simply, it is a response that is greater (or lesser) than the sum of the two factors combined.</p>
<p>Let’s go back to the genotype blood_glucose problem. We’ll add a factor, and simplify the study a bit. We’re interested in a gene that, when absent, raises blood glucose. We’re also interested in a drug that, when present, lowers blood glucose.</p>
<p>We have reason to hypothesize that a genotype:drug interaction might exist. For example, the gene might encode a protein that metabolizes our drug, thus impairing the drug’s ability to lower blood glucose.</p>
<p>We’ll simulate a <span class="math inline">\(2\times 2\)</span> experiment that has only the presence or absence of each of these factors.</p>
<p>Here’s the data:</p>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="introanova.html#cb577-1"></a><span class="kw">set.seed</span>(<span class="dv">12345</span>)</span>
<span id="cb577-2"><a href="introanova.html#cb577-2"></a>blood_glucose &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="kw">c</span>(<span class="kw">rnorm</span>(<span class="dv">5</span>, <span class="dv">100</span>, <span class="dv">20</span>), <span class="kw">rnorm</span>(<span class="dv">5</span>, <span class="dv">75</span>, <span class="dv">20</span>), <span class="kw">rnorm</span>(<span class="dv">5</span>, <span class="dv">200</span>, <span class="dv">20</span>), <span class="kw">rnorm</span>(<span class="dv">5</span>, <span class="dv">100</span>, <span class="dv">20</span>)), <span class="dv">1</span>)</span>
<span id="cb577-3"><a href="introanova.html#cb577-3"></a>drug &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">rep</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">30</span>),<span class="dt">each=</span><span class="dv">5</span>),<span class="dv">2</span>))</span>
<span id="cb577-4"><a href="introanova.html#cb577-4"></a>genotype &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;KO&quot;</span>), <span class="dt">each=</span><span class="dv">10</span>)</span>
<span id="cb577-5"><a href="introanova.html#cb577-5"></a></span>
<span id="cb577-6"><a href="introanova.html#cb577-6"></a>test &lt;-<span class="st"> </span><span class="kw">data.frame</span>(genotype, drug, blood_glucose)</span>
<span id="cb577-7"><a href="introanova.html#cb577-7"></a></span>
<span id="cb577-8"><a href="introanova.html#cb577-8"></a>y0 &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">subset</span>(test, genotype<span class="op">==</span><span class="st">&quot;WT&quot;</span> <span class="op">&amp;</span><span class="st"> </span>drug<span class="op">==</span><span class="dv">0</span>)<span class="op">$</span>blood_glucose)</span>
<span id="cb577-9"><a href="introanova.html#cb577-9"></a>y30 &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">subset</span>(test, genotype<span class="op">==</span><span class="st">&quot;WT&quot;</span> <span class="op">&amp;</span><span class="st"> </span>drug<span class="op">==</span><span class="dv">30</span>)<span class="op">$</span>blood_glucose)</span>
<span id="cb577-10"><a href="introanova.html#cb577-10"></a>yend0 &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">subset</span>(test, genotype<span class="op">==</span><span class="st">&quot;KO&quot;</span> <span class="op">&amp;</span><span class="st"> </span>drug<span class="op">==</span><span class="dv">0</span>)<span class="op">$</span>blood_glucose)</span>
<span id="cb577-11"><a href="introanova.html#cb577-11"></a>yend30 &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">subset</span>(test, genotype<span class="op">==</span><span class="st">&quot;KO&quot;</span> <span class="op">&amp;</span><span class="st"> </span>drug<span class="op">==</span><span class="dv">30</span>)<span class="op">$</span>blood_glucose)</span>
<span id="cb577-12"><a href="introanova.html#cb577-12"></a>  </span>
<span id="cb577-13"><a href="introanova.html#cb577-13"></a><span class="kw">ggplot</span>(test, <span class="kw">aes</span>(genotype, blood_glucose, <span class="dt">color=</span>drug))<span class="op">+</span></span>
<span id="cb577-14"><a href="introanova.html#cb577-14"></a><span class="st">  </span><span class="kw">geom_jitter</span>(<span class="dt">size=</span><span class="dv">6</span>, <span class="dt">width =</span><span class="fl">0.3</span>)<span class="op">+</span></span>
<span id="cb577-15"><a href="introanova.html#cb577-15"></a><span class="st">  </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette=</span><span class="st">&quot;Dark2&quot;</span>)<span class="op">+</span></span>
<span id="cb577-16"><a href="introanova.html#cb577-16"></a><span class="st">  </span><span class="kw">stat_summary</span>(<span class="dt">fun.y=</span>mean, <span class="dt">geom=</span><span class="st">&quot;point&quot;</span>, <span class="dt">shape =</span> <span class="dv">95</span>, <span class="dt">size=</span> <span class="dv">15</span>)<span class="op">+</span></span>
<span id="cb577-17"><a href="introanova.html#cb577-17"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;KO&quot;</span>))<span class="op">+</span></span>
<span id="cb577-18"><a href="introanova.html#cb577-18"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;blood glucose&quot;</span>)<span class="op">+</span></span>
<span id="cb577-19"><a href="introanova.html#cb577-19"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">&quot;WT&quot;</span>, <span class="dt">y=</span>y30, <span class="dt">xend=</span><span class="st">&quot;KO&quot;</span>, <span class="dt">yend=</span>yend30))<span class="op">+</span></span>
<span id="cb577-20"><a href="introanova.html#cb577-20"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">&quot;WT&quot;</span>, <span class="dt">y=</span>y0, <span class="dt">xend=</span><span class="st">&quot;KO&quot;</span>, <span class="dt">yend=</span>yend0))</span></code></pre></div>
<pre><code>## Warning: `fun.y` is deprecated. Use `fun` instead.</code></pre>
<pre><code>## Warning: Computation failed in `stat_summary()`:
## Can&#39;t convert a double vector to function</code></pre>
<p><img src="jabstb_files/figure-html/unnamed-chunk-281-1.png" width="672" /></p>
<p>An interaction effect can be represented by the differing slopes of those two lines. If the lines are not parallel, it means that the effect of the drug is not the same at both levels of genotype. Or you could say the effect of the genotype is not the same at both levels of the drug. Whatever.</p>
<p>The statistical term used to describe such phenomena is that the two factors interacted.</p>
<p>Thus, a statistical interaction occurs when the effects of two factors are not the same across all of their levels.</p>
<p>Here’s a quick ANOVA table for those data. It has an F test for each of the factors genotype and drug, and an F test for the genotype:drug interaction. All three F-tests are extreme.</p>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="introanova.html#cb580-1"></a><span class="kw">anova</span>(<span class="kw">lm</span>(blood_glucose <span class="op">~</span><span class="st"> </span>genotype <span class="op">+</span><span class="st"> </span>drug <span class="op">+</span>genotype<span class="op">*</span>drug, test))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: blood_glucose
##               Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
## genotype       1 25127.0 25127.0  94.333 4.116e-08 ***
## drug           1 26028.1 26028.1  97.716 3.225e-08 ***
## genotype:drug  1  4845.4  4845.4  18.191 0.0005923 ***
## Residuals     16  4261.8   266.4                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Since the F-test for the genotype:drug interaction is extreme, we can reject the null that there is no interaction between them. Furthermore, the existence of the interaction effect complicates the interpretation of the effects of the genotype and drug factors. Generally, an interaction effect supersedes an effect of either factor, alone.</p>
<p>Although genotype seems to have a strong effect on glucose levels, that’s really blunted in the presence of drug. And although the drug seems to reduce glucose, that effect becomes remarkably prominent when the genotype changes. When interactions occur, the effect of one factor cannot be interpreted without condition on the other factor!</p>
<p>It might be useful to see the data for the discussion that follows.</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:left;">
drug
</th>
<th style="text-align:right;">
blood_glucose
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
111.7
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
114.2
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
97.8
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
90.9
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
112.1
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
38.6
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
87.6
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
69.5
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
69.3
</td>
</tr>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
56.6
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
197.7
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
236.3
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
207.4
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
210.4
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:right;">
185.0
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
116.3
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
82.3
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
93.4
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
122.4
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:right;">
106.0
</td>
</tr>
</tbody>
</table>
<p>Four sources of variation are accounted for: the main effect of genotype, the main effect of drug, the interaction between drug and genotype, and the residual error.</p>
<p>Three F tests were performed. Each of these used <span class="math inline">\(MS_{residual}\)</span> in the denominator and the <span class="math inline">\(MS\)</span> for the respective source in the numerator.</p>
<p>How has this variation been determined?</p>
<p>The experimental model is as follows: <span class="math display">\[SS_{model}=SS_{genotype}+SS_{drug}+SS_{genotype\times drug} \]</span></p>
<p>The grand mean of all the data is computed as before: <span class="math display">\[\hat y=\frac{1}{n}\sum_{i=1}^ny_i \]</span></p>
<p>There are two factors, each at two levels. Thus, there are a total of <span class="math inline">\(j=4\)</span> experimental groups. Each group has a sample size of <span class="math inline">\(n_j=5\)</span> replicates and represent a combination of predictor variables: WT/0, WT/30, KO/0, KO/30. The mean of each group is <span class="math inline">\(\bar_j\)</span>.</p>
<p>Therefore, <span class="math display">\[SS_{model}= \sum_{j=1}^kn_j(\bar y_j-\hat y)^2 \]</span> represents the total model variation.</p>
<p>We can also artificially group these factors and levels further. Two groups correspond to the levels of the genotype factors and two correspond to the levels of the drug factor. Their means are <span class="math inline">\(\bar y_{wt}, \bar y_{ko}\)</span> and <span class="math inline">\(\bar y_0, \bar y_{30}\)</span>, respectively. Each of these groups has a sample size of <span class="math inline">\(2n_j\)</span>.</p>
<p>These contrived means are used to isolate for the variation of each of the two factors:</p>
<p><span class="math display">\[SS_{genotype}= \sum n_{wt}(\bar y_{wt}-\hat y)^2+n_{ko}(\bar y_{ko}-\hat y)^2 \]</span></p>
<p><span class="math display">\[SS_{drug}= \sum n_{0}(\bar y_{0}-\hat y)^2+n_{30}(\bar y_{30}-\hat y)^2 \]</span></p>
<p>All that remains is to account for the variation associated with the interaction effect. That can be solved for algebraically: <span class="math display">\[SS_{genotype\times drug}=SS_{model}-SS_{genotype}-SS_{drug} \]</span></p>
<p>Because two way ANOVA’s have two factors, one of the factors can be applied completely randomized, and the other can be applied as related measures. Or both factors can be completely randomized, or both can be related measures.</p>
</div>
<div id="other-anova-models" class="section level2">
<h2><span class="header-section-number">25.9</span> Other ANOVA models</h2>
<p>For all ANOVA experiments, irrespective of the design, the total amount of deviation in the data can be partitioned into model and residual terms:</p>
<p><span class="math display">\[SS_{total}=SS_{model}+ SS_{residual}\]</span></p>
<p>What’s interesting is that different ANOVA experimental designs have different models. We’re interested in two factors, factorA and factorB, and have the ability to study each at multiple levels. The interaction between factorA and factorB is <span class="math inline">\(A\times B\)</span></p>
<p>One way CR: <span class="math inline">\(SS_{model}=SS_{factorA}\)</span></p>
<p>One way RM: <span class="math inline">\(SS_{model}=SS_{factorA}+SS_{subj}\)</span></p>
<p>Two way CR: <span class="math inline">\(SS_{model}=SS_{factorA}+SS_{factorB}+SS_{A\times B}\)</span></p>
<p>Two way RM on A factor: <span class="math inline">\(SS_{model}=SS_A+SS_B+SS_{A\times B}+SS_{subj\times A}\)</span></p>
<p>Two way RM on both factors: <span class="math inline">\(SS_{model}=SS_A+SS_B+SS_{A\times B}+SS_{subj\times A}+SS_{subj\times B}+SS_{subj\times A\times B}\)</span></p>
<p>The big difference between completely randomized and related measure designs is that in the latter, we’re now accounting for the deviation associated with each replicate in the model! Otherwise, that replicate deviation would have been blended into the residuals.</p>
<p>This turns out to be a pretty big deal. When that deviation due to the subjects is pulled out of the residual, it lowers the value of the denominator of the F statistic. Thus making the F statistic larger!</p>
<div id="r-and-anova" class="section level3">
<h3><span class="header-section-number">25.9.1</span> R and ANOVA</h3>
<p>There are a handful of ways to conduct ANOVA analysis on R. These are not necessarily more right or wrong than the others. What is important to know, however, is that they do perform calculations differently under certain circumstances (eg, Type 1 v Type 2 v Type 3 SS calculations). Therefore, they produce distinct results.</p>
<p>This always confuses researchers, particularly when comparing R’s results to other software we might be more comfortable with. We wonder which output is “right”. Chances are they are all “right”. We have that output because of the way we argued the analysis.</p>
<p>This again emphasizes the need to share specific details of the analysis in our publications. In this case, specify using R, specify the package version and the R function used, and even specify the arguments. Or append an R script file as supplemental information to illustrate exactly how the analysis is performed.</p>
<p>Given data and a group of arguments we’ll call foo, R’s ANOVA function options are as follows:</p>
<p><code>anova</code> - A function in R’s base. eg, <code>anova(lm(foo))</code></p>
<p><code>aov</code> - A function in R’s base. eg, `aov(foo)</p>
<p><code>Anova</code> - A function in R’s <code>car</code> package. eg, <code>Anova(lm(foo))</code></p>
<p><code>ezAnova</code> - A function in R’s <code>ez</code> package, <code>ezAnova(foo)</code></p>
<p>There are others. For example, since ANOVA analyses are also general linear models the same basic problem can also be solved using <code>lm(foo)</code>without ANOVA. Passing an <code>lm(foo)</code> into an ANOVA function mostly just provides output in the familiar ANOVA notation. The statistical fundamentals of <code>lm(foo)</code> and <code>ezAnova(foo)</code>are identical.</p>
<p>For the ANOVA part of this course, we’ll use <code>ezANOVA</code> from the <code>ez</code> package. In particular, it is a bit more straightforward to use (and teach) in terms of arguing the experiment’s ANOVA model.</p>
<p><strong>Key Jargon to understand to do ezANOVA in R</strong></p>
<p>Specify a completely random design by defining the <strong>‘between’</strong> variable as your factor name. The between here is meant to imply comparisons between groups.</p>
<p>Specify a related measures design by defining the <strong><code>within</code></strong> variable as your factor name. The within here is meant to imply comparisons within replicates.</p>
<div id="type-of-calculation" class="section level4">
<h4><span class="header-section-number">25.9.1.1</span> Type of calculation</h4>
<p>When analyzing 2-way and 3-way ANOVA’s there are three different methods to calculate, which are referred to as type I, type II and type III.</p>
<p>When an experiment is balanced, which is to say it has equal sample sizes per group, the type of calculation is immaterial. In that case,type I, II and III yield exactly the same output.</p>
<p>Unbalanced experiments are those in which the sample sizes of groups are not the same. As long as the differences are not too large, the presence of unbalance is usually not a problem. But it will always impact the precise output of different ANOVA functions, depending upon whether they perform type I, II or III calculations.</p>
<p>This is what really causes confusion when we notice different output from different functions. This is particularly confusing when comparing the output of different ANOVA functions in R (eg, <code>Anova</code> vs <code>aov</code> vs <code>anova</code> vs <code>ezAnova</code>) and/or commercial software (eg, SAS, SPSS, Prism). The researcher scratches her head, wonders which is “correct”.</p>
<p>In one sense, they are all correct.</p>
<p>Type I, II and II sum of squares calculations are explained <a href="https://www.r-bloggers.com/anova-%E2%80%93-type-iiiiii-ss-explained/">here</a>.</p>
<p>Suffice to say this is important to not overlook. This serves to illustrate how providing good detail about the software used to analyze data is important for reproducibility.</p>
<p>The most significant point to understand is that some commercial software uses type 3 calculations by default. As a consequence, given the same data set, the results from those packages may not coincide perfectly with those of ezANOVA unless using a <code>type = 3</code> argument in the function.</p>
<p>My recommendation is to use <code>type = 2</code> when interested in only testing hypotheses about the main effects of factors, and there is no interest in an interaction if working on a two- or three-way ANOVA data set. That’s because <code>type = 2</code> is purported to yield consistently higher power for main effects.</p>
<p>Use <code>type = 3</code> when, instead, the experiment is designed to test whether an interaction occurs between factors. When an interaction occurs, the main effects are not interpretable.</p>
<p>Type I sum of squares are calculated when using the <code>anova</code> and <code>aov</code> functions of base R. This is otherwise known as “sequential” sum of squares calculation. On multifactor data with those functions, the results can differ given the order by which the factors are argued. Thus, <code>aov(lm(outcome~factorA + factorB))</code> might yield slightly different results compared to <code>aov(lm(outcome~factorB + factorA))</code>. The idea is to calculate the effect on a factor that is most interesting to you scientifically, while “controlling” for the effect of the other factor.</p>
<p>When using <code>ezANOVA</code> we can explicitly argue which calculation we wish to run, removing some ambiguity about what actually happened.</p>
</div>
</div>
</div>
<div id="alternatives-to-anova" class="section level2">
<h2><span class="header-section-number">25.10</span> Alternatives to ANOVA</h2>
<p>When the outcome variable is measured and the design is completely randomized, the data can be analyzed using the general linear model using R’s <code>lm</code> function, rather than by ANOVA. This allows for analyzing interaction effects between factors. The results will be the same as ANOVA. If the design has a related measures component, then a linear mixed effects model should be run instead. In that case, use <code>lmer</code> in the <code>lme4</code> package.</p>
<p>Alternately a nonparametric analysis can be performed using either the Kruskal-Wallis (completely randomized) for the Friedman (related measures) test. Bear in mind that no nonparametric analog for the two-way ANOVA exists. Thus, hypotheses related to interaction effects are not testable using nonparametric statistics.</p>
<p>Finally there is the generalized linear model (<code>glm</code>) for completely randomized designs or the generalized linear mixed model (<code>glmer</code>) for designs that incorporate related measures, respectively. Each of these allow for testing interactions between factors. The generalized linear models also allow for a flexible array of outcome variables. These should be used, rather than ANOVA, when the outcome variable is non-normal or is discrete. For example, these are the tools of choice when the outcome variable is binomial (logistic regression) or frequency data (Poisson regression) and there are 3 or more groups to compare. Additional families are possible.</p>
<div id="screw-anova-just-tell-me-how-to-t-test-everything" class="section level3">
<h3><span class="header-section-number">25.10.1</span> Screw ANOVA, Just Tell Me How to t-Test Everything</h3>
<p>OK, fine.</p>
<p>This is far from ideal because an ANOVA-free approach risks introducing bias through cherry-picking comparisons after unpacking the data (the HARK fallacy).</p>
<p>You don’t <em>have</em> to do ANOVA for an experiment with 3 or more groups (or anything else for that matter–you just have to be able to defend your choices). A major purpose of ANOVA is to maintain an experiment-wise type1 error of 5%. But there are other ways to accomplish this objective.</p>
<p>For example, you might skip the ANOVA step and simply run serial t-tests comparing all of the groups in an experiment. Or run t-tests to compare a <strong>pre-planned</strong> sublist of all possible comparisons. The emphasis here on pre-planning is important. Make decisions ahead of time about what is to be compared, then make only those comparisons. No more and no less. Otherwise, you’re snooping. Once you’re in snooping mode, you’re deeply biased towards opportunistic outcomes.</p>
<p>For example, you may run multiple control groups within your experiment to signal that some important aspect of the protocol is working properly, but these controls are otherwise not scientifically interesting (with respect to testing new hypotheses). You may not wish to expend any of your type1 error budget doing comparisons on these controls.</p>
<p>With those reservations noted, what follows are two ways to go about this.</p>
<p>To begin, if we have <span class="math inline">\(k\)</span> levels of predictor variables in our experiment it has a <strong>total</strong> of <span class="math inline">\(m=k(k-1)/2\)</span> possible comparisons that could be made.</p>
<p>The <code>pairwise.t.test</code> is the function to use for this purpose. Use it to make the group comparisons that interest you. choosing the <code>p.adjust.method</code> that strikes your fancy. Two of the latter are listed below.</p>
</div>
</div>
<div id="summary-5" class="section level2">
<h2><span class="header-section-number">25.11</span> Summary</h2>
<ul>
<li>We conduct ANOVA designed experiments intuitively, any time we have 3 or more groups.</li>
<li>In R make sure that variable you want to be a factor is coded as a factor.</li>
<li>Accounting for combinations of factors and whether the factor is applied randomized or related measures, We have at least 12 different ANOVA designs at our disposal.</li>
<li>F-tests compare model variance to residual variance; if F is large, the model is extreme.</li>
<li>ANOVA test results can infer effects of factors, or be permission for posthoc testing</li>
<li>Several ANOVA methods exist in R. When configured the same, they should give the same output.</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="simcorrelation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="fdistr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["jabstb.pdf", "jabstb.epub"],
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
