<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 31 Two-way ANOVA Related Measures | JABSTB: Statistical Design and Analysis of Experiments with R</title>
  <meta name="description" content="Experimental biostatistics using R." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 31 Two-way ANOVA Related Measures | JABSTB: Statistical Design and Analysis of Experiments with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Experimental biostatistics using R." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 31 Two-way ANOVA Related Measures | JABSTB: Statistical Design and Analysis of Experiments with R" />
  
  <meta name="twitter:description" content="Experimental biostatistics using R." />
  

<meta name="author" content="TJ Murphy PhD, Department of Pharmacology and Chemical Biology, School of Medicine, Emory University, Atlanta, GA biostats538@gmail.com" />


<meta name="date" content="2021-01-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="twowayCR.html"/>
<link rel="next" href="twowaymixed.html"/>
<script src="libs/header-attrs-2.5/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">JABSTB</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="author.html"><a href="author.html"><i class="fa fa-check"></i><b>1</b> About the author and book</a></li>
<li class="chapter" data-level="2" data-path="history.html"><a href="history.html"><i class="fa fa-check"></i><b>2</b> A Brief History of Experimental Design</a></li>
<li class="chapter" data-level="3" data-path="software.html"><a href="software.html"><i class="fa fa-check"></i><b>3</b> Installing and Understanding the software</a>
<ul>
<li class="chapter" data-level="3.1" data-path="software.html"><a href="software.html#tldr"><i class="fa fa-check"></i><b>3.1</b> tl;dr</a></li>
<li class="chapter" data-level="3.2" data-path="software.html"><a href="software.html#spring-2021-term-versions"><i class="fa fa-check"></i><b>3.2</b> Spring 2021 term versions</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="software.html"><a href="software.html#what-are-these"><i class="fa fa-check"></i><b>3.2.1</b> What are these?</a></li>
<li class="chapter" data-level="3.2.2" data-path="software.html"><a href="software.html#rule-1-install-r-first-install-rstudio-second"><i class="fa fa-check"></i><b>3.2.2</b> Rule 1: Install R first, install RStudio second</a></li>
<li class="chapter" data-level="3.2.3" data-path="software.html"><a href="software.html#rule-2-dont-pay"><i class="fa fa-check"></i><b>3.2.3</b> Rule 2: Don’t pay!</a></li>
<li class="chapter" data-level="3.2.4" data-path="software.html"><a href="software.html#rule-3-dont-compile"><i class="fa fa-check"></i><b>3.2.4</b> Rule 3: Don’t compile!</a></li>
<li class="chapter" data-level="3.2.5" data-path="software.html"><a href="software.html#rule-4-install-the-right-r-software-for-your-machine"><i class="fa fa-check"></i><b>3.2.5</b> Rule 4: Install the right R software for your machine</a></li>
<li class="chapter" data-level="3.2.6" data-path="software.html"><a href="software.html#doubts"><i class="fa fa-check"></i><b>3.2.6</b> Doubts?</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="software.html"><a href="software.html#after-installation"><i class="fa fa-check"></i><b>3.3</b> After installation</a></li>
<li class="chapter" data-level="3.4" data-path="software.html"><a href="software.html#what-is-r"><i class="fa fa-check"></i><b>3.4</b> What is R?</a></li>
<li class="chapter" data-level="3.5" data-path="software.html"><a href="software.html#more-on-packages"><i class="fa fa-check"></i><b>3.5</b> More on packages</a></li>
<li class="chapter" data-level="3.6" data-path="software.html"><a href="software.html#how-to-install-and-use-packages"><i class="fa fa-check"></i><b>3.6</b> How to install and use packages</a></li>
<li class="chapter" data-level="3.7" data-path="software.html"><a href="software.html#packages-well-need"><i class="fa fa-check"></i><b>3.7</b> Packages we’ll need</a></li>
<li class="chapter" data-level="3.8" data-path="software.html"><a href="software.html#super-immportant-always-avoid-compiling"><i class="fa fa-check"></i><b>3.8</b> Super immportant: Always avoid compiling</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="software.html"><a href="software.html#reasons-not-to-compile"><i class="fa fa-check"></i><b>3.8.1</b> Reasons not to compile</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="software.html"><a href="software.html#summary"><i class="fa fa-check"></i><b>3.9</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>4</b> Start Learning R</a>
<ul>
<li class="chapter" data-level="4.1" data-path="getting-started.html"><a href="getting-started.html#digging-in-with-rrstudio"><i class="fa fa-check"></i><b>4.1</b> Digging in with R/RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="getting-started.html"><a href="getting-started.html#the-tidyverse-package"><i class="fa fa-check"></i><b>4.2</b> The tidyverse package</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="getting-started.html"><a href="getting-started.html#do-swirlstats"><i class="fa fa-check"></i><b>4.2.1</b> Do swirlstats</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="getting-started.html"><a href="getting-started.html#other-resources"><i class="fa fa-check"></i><b>4.3</b> Other resources</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="getting-started.html"><a href="getting-started.html#internet-search"><i class="fa fa-check"></i><b>4.3.1</b> Internet search</a></li>
<li class="chapter" data-level="4.3.2" data-path="getting-started.html"><a href="getting-started.html#help-documents"><i class="fa fa-check"></i><b>4.3.2</b> Help Documents</a></li>
<li class="chapter" data-level="4.3.3" data-path="getting-started.html"><a href="getting-started.html#package-vignettes"><i class="fa fa-check"></i><b>4.3.3</b> Package Vignettes</a></li>
<li class="chapter" data-level="4.3.4" data-path="getting-started.html"><a href="getting-started.html#people"><i class="fa fa-check"></i><b>4.3.4</b> People</a></li>
<li class="chapter" data-level="4.3.5" data-path="getting-started.html"><a href="getting-started.html#typos"><i class="fa fa-check"></i><b>4.3.5</b> typos</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="philos.html"><a href="philos.html"><i class="fa fa-check"></i><b>5</b> What is your philosophy?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="philos.html"><a href="philos.html#exploration-and-uncertainty"><i class="fa fa-check"></i><b>5.1</b> Exploration and uncertainty</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="bigpic.html"><a href="bigpic.html"><i class="fa fa-check"></i><b>6</b> Stats: The Big Picture</a>
<ul>
<li class="chapter" data-level="6.1" data-path="bigpic.html"><a href="bigpic.html#what-are-experimental-statistics"><i class="fa fa-check"></i><b>6.1</b> What are experimental statistics?</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="bigpic.html"><a href="bigpic.html#descriptive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Descriptive modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="bigpic.html"><a href="bigpic.html#statistical-inference"><i class="fa fa-check"></i><b>6.1.2</b> Statistical inference</a></li>
<li class="chapter" data-level="6.1.3" data-path="bigpic.html"><a href="bigpic.html#experimental-design"><i class="fa fa-check"></i><b>6.1.3</b> Experimental design</a></li>
<li class="chapter" data-level="6.1.4" data-path="bigpic.html"><a href="bigpic.html#statistics-as-an-anti-bias-framework"><i class="fa fa-check"></i><b>6.1.4</b> Statistics as an anti-bias framework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>7</b> Statistical Sampling</a>
<ul>
<li class="chapter" data-level="7.1" data-path="sampling.html"><a href="sampling.html#experimental-units"><i class="fa fa-check"></i><b>7.1</b> Experimental units</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-to-define-the-experimental-unit"><i class="fa fa-check"></i><b>7.1.1</b> A simple test to define the experimental unit</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="sampling.html"><a href="sampling.html#independent-replicates"><i class="fa fa-check"></i><b>7.2</b> Independent Replicates</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="sampling.html"><a href="sampling.html#a-simple-test-for-independence"><i class="fa fa-check"></i><b>7.2.1</b> A simple test for independence</a></li>
<li class="chapter" data-level="7.2.2" data-path="sampling.html"><a href="sampling.html#some-replication-examples"><i class="fa fa-check"></i><b>7.2.2</b> Some replication examples</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="sampling.html"><a href="sampling.html#random-process"><i class="fa fa-check"></i><b>7.3</b> Random process</a></li>
<li class="chapter" data-level="7.4" data-path="sampling.html"><a href="sampling.html#statistically-valid-samples"><i class="fa fa-check"></i><b>7.4</b> Statistically valid samples</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="sampling.html"><a href="sampling.html#select-random-subjects"><i class="fa fa-check"></i><b>7.4.1</b> Select random subjects</a></li>
<li class="chapter" data-level="7.4.2" data-path="sampling.html"><a href="sampling.html#randomize-to-sequence"><i class="fa fa-check"></i><b>7.4.2</b> Randomize to sequence</a></li>
<li class="chapter" data-level="7.4.3" data-path="sampling.html"><a href="sampling.html#randomize-to-location"><i class="fa fa-check"></i><b>7.4.3</b> Randomize to location</a></li>
<li class="chapter" data-level="7.4.4" data-path="sampling.html"><a href="sampling.html#randomize-to-block"><i class="fa fa-check"></i><b>7.4.4</b> Randomize to block</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="sampling.html"><a href="sampling.html#independence-of-replicates"><i class="fa fa-check"></i><b>7.5</b> Independence of replicates</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>8</b> Data Classification</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data.html"><a href="data.html#dependent-and-independent-variables"><i class="fa fa-check"></i><b>8.1</b> Dependent and independent variables</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="data.html"><a href="data.html#statistical-notation"><i class="fa fa-check"></i><b>8.1.1</b> Statistical notation</a></li>
<li class="chapter" data-level="8.1.2" data-path="data.html"><a href="data.html#when-there-is-no-independent-variable"><i class="fa fa-check"></i><b>8.1.2</b> When there is no independent variable</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="data.html"><a href="data.html#discrete-or-continuous-variables"><i class="fa fa-check"></i><b>8.2</b> Discrete or continuous variables</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="data.html"><a href="data.html#measured-variables"><i class="fa fa-check"></i><b>8.2.1</b> Measured variables</a></li>
<li class="chapter" data-level="8.2.2" data-path="data.html"><a href="data.html#discrete-categorical-and-ordinal-variables"><i class="fa fa-check"></i><b>8.2.2</b> Discrete categorical and ordinal variables</a></li>
<li class="chapter" data-level="8.2.3" data-path="data.html"><a href="data.html#rescaling-variables"><i class="fa fa-check"></i><b>8.2.3</b> Rescaling variables</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data.html"><a href="data.html#working-with-data"><i class="fa fa-check"></i><b>8.3</b> Working with data</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="data.html"><a href="data.html#entering-data"><i class="fa fa-check"></i><b>8.3.1</b> Entering data</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data.html"><a href="data.html#summary-1"><i class="fa fa-check"></i><b>8.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dispersion.html"><a href="dispersion.html"><i class="fa fa-check"></i><b>9</b> Variability, Accuracy and Precision</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="dispersion.html"><a href="dispersion.html#data-are-messy"><i class="fa fa-check"></i><b>9.0.1</b> Data are messy</a></li>
<li class="chapter" data-level="9.1" data-path="dispersion.html"><a href="dispersion.html#illustration-of-a-model-with-error"><i class="fa fa-check"></i><b>9.1</b> Illustration of a model with error</a></li>
<li class="chapter" data-level="9.2" data-path="dispersion.html"><a href="dispersion.html#variance-quantifying-variation-by-least-squares"><i class="fa fa-check"></i><b>9.2</b> Variance: Quantifying variation by least squares</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="dispersion.html"><a href="dispersion.html#definition-of-variance"><i class="fa fa-check"></i><b>9.2.1</b> Definition of variance</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="dispersion.html"><a href="dispersion.html#standard-deviation"><i class="fa fa-check"></i><b>9.3</b> Standard deviation</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="dispersion.html"><a href="dispersion.html#what-does-the-standard-deviation-tell-us"><i class="fa fa-check"></i><b>9.3.1</b> What does the standard deviation tell us</a></li>
<li class="chapter" data-level="9.3.2" data-path="dispersion.html"><a href="dispersion.html#ways-of-describing-sample-variability"><i class="fa fa-check"></i><b>9.3.2</b> Ways of describing sample variability</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="dispersion.html"><a href="dispersion.html#precision-and-accuracy"><i class="fa fa-check"></i><b>9.4</b> Precision and Accuracy</a></li>
<li class="chapter" data-level="9.5" data-path="dispersion.html"><a href="dispersion.html#standard-error"><i class="fa fa-check"></i><b>9.5</b> Standard error</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="dispersion.html"><a href="dispersion.html#what-exactly-does-the-standard-error-represent"><i class="fa fa-check"></i><b>9.5.1</b> What exactly does the standard error represent?</a></li>
<li class="chapter" data-level="9.5.2" data-path="dispersion.html"><a href="dispersion.html#should-i-use-sd-or-sem"><i class="fa fa-check"></i><b>9.5.2</b> “Should I use SD or SEM?”</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="dispersion.html"><a href="dispersion.html#confidence-intervals"><i class="fa fa-check"></i><b>9.6</b> Confidence intervals</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="dispersion.html"><a href="dispersion.html#ci-definition"><i class="fa fa-check"></i><b>9.6.1</b> CI Definition</a></li>
<li class="chapter" data-level="9.6.2" data-path="dispersion.html"><a href="dispersion.html#evaluating-confidence-intervals"><i class="fa fa-check"></i><b>9.6.2</b> Evaluating confidence intervals</a></li>
<li class="chapter" data-level="9.6.3" data-path="dispersion.html"><a href="dispersion.html#ci-simulation-tools"><i class="fa fa-check"></i><b>9.6.3</b> CI Simulation tools</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="dispersion.html"><a href="dispersion.html#key-take-aways"><i class="fa fa-check"></i><b>9.7</b> Key take aways</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="hypotheses.html"><a href="hypotheses.html"><i class="fa fa-check"></i><b>10</b> Framing statistical hypotheses</a>
<ul>
<li class="chapter" data-level="10.1" data-path="hypotheses.html"><a href="hypotheses.html#the-decision-process"><i class="fa fa-check"></i><b>10.1</b> The decision process</a></li>
<li class="chapter" data-level="10.2" data-path="hypotheses.html"><a href="hypotheses.html#popper-and-falsification"><i class="fa fa-check"></i><b>10.2</b> Popper and falsification</a></li>
<li class="chapter" data-level="10.3" data-path="hypotheses.html"><a href="hypotheses.html#statistical-hypothesis-rubric"><i class="fa fa-check"></i><b>10.3</b> Statistical hypothesis rubric</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="error.html"><a href="error.html"><i class="fa fa-check"></i><b>11</b> Error</a>
<ul>
<li class="chapter" data-level="11.1" data-path="error.html"><a href="error.html#setting-type-1-and-type-2-error-thresholds"><i class="fa fa-check"></i><b>11.1</b> Setting type 1 and type 2 error thresholds</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="error.html"><a href="error.html#setting-alpha-the-type-1-error"><i class="fa fa-check"></i><b>11.1.1</b> Setting alpha-the type 1 error</a></li>
<li class="chapter" data-level="11.1.2" data-path="error.html"><a href="error.html#power-setting-beta-the-type-2-error"><i class="fa fa-check"></i><b>11.1.2</b> Power: Setting beta-the type 2 error</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="error.html"><a href="error.html#striking-the-right-balance"><i class="fa fa-check"></i><b>11.2</b> Striking the right balance</a></li>
<li class="chapter" data-level="11.3" data-path="error.html"><a href="error.html#false-discovery-rate"><i class="fa fa-check"></i><b>11.3</b> False discovery rate</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="pvalues.html"><a href="pvalues.html"><i class="fa fa-check"></i><b>12</b> P Values</a>
<ul>
<li class="chapter" data-level="12.1" data-path="pvalues.html"><a href="pvalues.html#definition"><i class="fa fa-check"></i><b>12.1</b> Definition</a></li>
<li class="chapter" data-level="12.2" data-path="pvalues.html"><a href="pvalues.html#null-test-statistic-distributions"><i class="fa fa-check"></i><b>12.2</b> Null test statistic distributions</a></li>
<li class="chapter" data-level="12.3" data-path="pvalues.html"><a href="pvalues.html#computing-p-values"><i class="fa fa-check"></i><b>12.3</b> Computing p-values</a></li>
<li class="chapter" data-level="12.4" data-path="pvalues.html"><a href="pvalues.html#p-values-from-null-tests-are-uniform"><i class="fa fa-check"></i><b>12.4</b> P-values from null tests are uniform</a></li>
<li class="chapter" data-level="12.5" data-path="pvalues.html"><a href="pvalues.html#how-p-values-become-error-probabilities"><i class="fa fa-check"></i><b>12.5</b> How P-values become error probabilities</a></li>
<li class="chapter" data-level="12.6" data-path="pvalues.html"><a href="pvalues.html#p-values-as-dichotomous-switches"><i class="fa fa-check"></i><b>12.6</b> P-values as dichotomous switches</a></li>
<li class="chapter" data-level="12.7" data-path="pvalues.html"><a href="pvalues.html#p-values-in-true-effects"><i class="fa fa-check"></i><b>12.7</b> P-values in true effects</a></li>
<li class="chapter" data-level="12.8" data-path="pvalues.html"><a href="pvalues.html#criticisms-of-p-values"><i class="fa fa-check"></i><b>12.8</b> Criticisms of p-values</a></li>
<li class="chapter" data-level="12.9" data-path="pvalues.html"><a href="pvalues.html#summary-how-to-use-p-values"><i class="fa fa-check"></i><b>12.9</b> Summary: How to use p-values</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="jaxwest7.html"><a href="jaxwest7.html"><i class="fa fa-check"></i><b>13</b> Reproducible Data Munging in R</a>
<ul>
<li class="chapter" data-level="13.1" data-path="jaxwest7.html"><a href="jaxwest7.html#jaxwest7-data"><i class="fa fa-check"></i><b>13.1</b> Jaxwest7 data</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="jaxwest7.html"><a href="jaxwest7.html#inspect-the-source-data"><i class="fa fa-check"></i><b>13.1.1</b> Inspect the source data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-glucose-data-into-r"><i class="fa fa-check"></i><b>13.2</b> Munge the glucose data into R</a></li>
<li class="chapter" data-level="13.3" data-path="jaxwest7.html"><a href="jaxwest7.html#munge-the-bodyweight-data"><i class="fa fa-check"></i><b>13.3</b> Munge the bodyweight data</a></li>
<li class="chapter" data-level="13.4" data-path="jaxwest7.html"><a href="jaxwest7.html#merge-and-explore-the-glucose-and-bw-data"><i class="fa fa-check"></i><b>13.4</b> Merge and explore the glucose and bw data</a></li>
<li class="chapter" data-level="13.5" data-path="jaxwest7.html"><a href="jaxwest7.html#summary-2"><i class="fa fa-check"></i><b>13.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="binomial.html"><a href="binomial.html"><i class="fa fa-check"></i><b>14</b> The Binomial Distribution</a>
<ul>
<li class="chapter" data-level="14.1" data-path="binomial.html"><a href="binomial.html#dbinom"><i class="fa fa-check"></i><b>14.1</b> dbinom</a></li>
<li class="chapter" data-level="14.2" data-path="binomial.html"><a href="binomial.html#pbinom"><i class="fa fa-check"></i><b>14.2</b> pbinom</a></li>
<li class="chapter" data-level="14.3" data-path="binomial.html"><a href="binomial.html#qbinom"><i class="fa fa-check"></i><b>14.3</b> qbinom</a></li>
<li class="chapter" data-level="14.4" data-path="binomial.html"><a href="binomial.html#rbinom"><i class="fa fa-check"></i><b>14.4</b> rbinom</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="poisson.html"><a href="poisson.html"><i class="fa fa-check"></i><b>15</b> The Poisson Distribution</a>
<ul>
<li class="chapter" data-level="15.1" data-path="poisson.html"><a href="poisson.html#poisson-events"><i class="fa fa-check"></i><b>15.1</b> Poisson Events</a></li>
<li class="chapter" data-level="15.2" data-path="poisson.html"><a href="poisson.html#dpois"><i class="fa fa-check"></i><b>15.2</b> dpois</a></li>
<li class="chapter" data-level="15.3" data-path="poisson.html"><a href="poisson.html#ppois"><i class="fa fa-check"></i><b>15.3</b> ppois</a></li>
<li class="chapter" data-level="15.4" data-path="poisson.html"><a href="poisson.html#rpois"><i class="fa fa-check"></i><b>15.4</b> rpois</a></li>
<li class="chapter" data-level="15.5" data-path="poisson.html"><a href="poisson.html#overdispersion"><i class="fa fa-check"></i><b>15.5</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="normal.html"><a href="normal.html"><i class="fa fa-check"></i><b>16</b> The Normal Distribution</a>
<ul>
<li class="chapter" data-level="16.0.1" data-path="normal.html"><a href="normal.html#the-standard-normal"><i class="fa fa-check"></i><b>16.0.1</b> The Standard Normal</a></li>
<li class="chapter" data-level="16.1" data-path="normal.html"><a href="normal.html#dnorm"><i class="fa fa-check"></i><b>16.1</b> dnorm</a></li>
<li class="chapter" data-level="16.2" data-path="normal.html"><a href="normal.html#pnorm"><i class="fa fa-check"></i><b>16.2</b> pnorm</a>
<ul>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-p-values-using-pnorm"><i class="fa fa-check"></i>Calculating “p-values”" using pnorm</a></li>
<li class="chapter" data-level="" data-path="normal.html"><a href="normal.html#calculating-percentiles-using-pnorm"><i class="fa fa-check"></i>Calculating percentiles using pnorm</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="normal.html"><a href="normal.html#qnorm"><i class="fa fa-check"></i><b>16.3</b> qnorm</a></li>
<li class="chapter" data-level="16.4" data-path="normal.html"><a href="normal.html#rnorm"><i class="fa fa-check"></i><b>16.4</b> rnorm</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="normal.html"><a href="normal.html#plotting-histograms-of-some-rnorm-samples"><i class="fa fa-check"></i><b>16.4.1</b> Plotting histograms of some rnorm samples</a></li>
<li class="chapter" data-level="16.4.2" data-path="normal.html"><a href="normal.html#bins-and-binwidth"><i class="fa fa-check"></i><b>16.4.2</b> Bins and Binwidth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="categorical.html"><a href="categorical.html"><i class="fa fa-check"></i><b>17</b> Statistics for Categorical Data</a>
<ul>
<li class="chapter" data-level="17.1" data-path="categorical.html"><a href="categorical.html#types-of-categorical-data"><i class="fa fa-check"></i><b>17.1</b> Types of categorical data</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="categorical.html"><a href="categorical.html#proportions"><i class="fa fa-check"></i><b>17.1.1</b> Proportions</a></li>
<li class="chapter" data-level="17.1.2" data-path="categorical.html"><a href="categorical.html#frequencies"><i class="fa fa-check"></i><b>17.1.2</b> Frequencies</a></li>
<li class="chapter" data-level="17.1.3" data-path="categorical.html"><a href="categorical.html#associations"><i class="fa fa-check"></i><b>17.1.3</b> Associations</a></li>
<li class="chapter" data-level="17.1.4" data-path="categorical.html"><a href="categorical.html#statistics-covered-here"><i class="fa fa-check"></i><b>17.1.4</b> Statistics Covered Here</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="categorical.html"><a href="categorical.html#exact-v-asymptotic-calculations-of-p-values"><i class="fa fa-check"></i><b>17.2</b> Exact v Asymptotic Calculations of p-values</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="categorical.html"><a href="categorical.html#choosing-exact-or-asymptotic"><i class="fa fa-check"></i><b>17.2.1</b> Choosing exact or asymptotic</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="categorical.html"><a href="categorical.html#overview-of-the-types-of-hypothesis-testing"><i class="fa fa-check"></i><b>17.3</b> Overview of the types of hypothesis testing</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="categorical.html"><a href="categorical.html#proportion-analysis"><i class="fa fa-check"></i><b>17.3.1</b> Proportion analysis</a></li>
<li class="chapter" data-level="17.3.2" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-testing"><i class="fa fa-check"></i><b>17.3.2</b> Goodness of fit testing</a></li>
<li class="chapter" data-level="17.3.3" data-path="categorical.html"><a href="categorical.html#contingency-analysis"><i class="fa fa-check"></i><b>17.3.3</b> Contingency Analysis</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="categorical.html"><a href="categorical.html#comparing-proportions"><i class="fa fa-check"></i><b>17.4</b> Comparing proportions</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="categorical.html"><a href="categorical.html#a-mouse-t-cell-pilot-experiment-the-cytokine-inducible-antigen-gradstudin"><i class="fa fa-check"></i><b>17.4.1</b> A Mouse T Cell Pilot Experiment: The Cytokine-inducible antigen gradstudin</a></li>
<li class="chapter" data-level="17.4.2" data-path="categorical.html"><a href="categorical.html#calculating-proportions"><i class="fa fa-check"></i><b>17.4.2</b> Calculating Proportions</a></li>
<li class="chapter" data-level="17.4.3" data-path="categorical.html"><a href="categorical.html#what-a-proportion-estimates"><i class="fa fa-check"></i><b>17.4.3</b> What A Proportion Estimates</a></li>
<li class="chapter" data-level="17.4.4" data-path="categorical.html"><a href="categorical.html#confidence-intervals-of-proportions"><i class="fa fa-check"></i><b>17.4.4</b> Confidence Intervals of Proportions</a></li>
<li class="chapter" data-level="17.4.5" data-path="categorical.html"><a href="categorical.html#a-one-sample-proportion-test"><i class="fa fa-check"></i><b>17.4.5</b> A One-Sample Proportion Test</a></li>
<li class="chapter" data-level="17.4.6" data-path="categorical.html"><a href="categorical.html#an-exact-test-for-one-proportion"><i class="fa fa-check"></i><b>17.4.6</b> An exact test for one proportion</a></li>
<li class="chapter" data-level="17.4.7" data-path="categorical.html"><a href="categorical.html#comparing-two-proportions"><i class="fa fa-check"></i><b>17.4.7</b> Comparing Two Proportions</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="categorical.html"><a href="categorical.html#exact-tests-for-two-proportions"><i class="fa fa-check"></i><b>17.5</b> Exact tests for two proportions</a></li>
<li class="chapter" data-level="17.6" data-path="categorical.html"><a href="categorical.html#goodness-of-fit-tests"><i class="fa fa-check"></i><b>17.6</b> Goodness of fit Tests</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="categorical.html"><a href="categorical.html#an-exact-goodness-of-fit-test"><i class="fa fa-check"></i><b>17.6.1</b> An Exact Goodness of Fit test</a></li>
<li class="chapter" data-level="17.6.2" data-path="categorical.html"><a href="categorical.html#an-approximate-goodness-of-fit-test"><i class="fa fa-check"></i><b>17.6.2</b> An Approximate Goodness of Fit test</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="categorical.html"><a href="categorical.html#contingency-testing"><i class="fa fa-check"></i><b>17.7</b> Contingency Testing</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="categorical.html"><a href="categorical.html#interpretation-of-contingency-results"><i class="fa fa-check"></i><b>17.7.1</b> Interpretation of Contingency Results</a></li>
<li class="chapter" data-level="17.7.2" data-path="categorical.html"><a href="categorical.html#write-up-4"><i class="fa fa-check"></i><b>17.7.2</b> Write Up</a></li>
<li class="chapter" data-level="17.7.3" data-path="categorical.html"><a href="categorical.html#interpretation-of-chi-square-output"><i class="fa fa-check"></i><b>17.7.3</b> Interpretation of chi-square output</a></li>
<li class="chapter" data-level="17.7.4" data-path="categorical.html"><a href="categorical.html#write-up-5"><i class="fa fa-check"></i><b>17.7.4</b> Write Up</a></li>
<li class="chapter" data-level="17.7.5" data-path="categorical.html"><a href="categorical.html#which-contingency-test-is-best"><i class="fa fa-check"></i><b>17.7.5</b> Which contingency test is best?</a></li>
<li class="chapter" data-level="17.7.6" data-path="categorical.html"><a href="categorical.html#higher-dimension-contingency-analysis"><i class="fa fa-check"></i><b>17.7.6</b> Higher dimension contingency analysis</a></li>
<li class="chapter" data-level="17.7.7" data-path="categorical.html"><a href="categorical.html#other-experimental-designs-involving-categorical-data"><i class="fa fa-check"></i><b>17.7.7</b> Other experimental designs involving categorical data</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="categorical.html"><a href="categorical.html#power-analysis-for-proportion-tests"><i class="fa fa-check"></i><b>17.8</b> Power analysis for proportion tests</a></li>
<li class="chapter" data-level="17.9" data-path="categorical.html"><a href="categorical.html#power-analysis-functions-for-proportion-tests"><i class="fa fa-check"></i><b>17.9</b> Power analysis functions for proportion tests</a></li>
<li class="chapter" data-level="17.10" data-path="categorical.html"><a href="categorical.html#graphing-proportions"><i class="fa fa-check"></i><b>17.10</b> Graphing Proportions</a></li>
<li class="chapter" data-level="17.11" data-path="categorical.html"><a href="categorical.html#key-takeaways"><i class="fa fa-check"></i><b>17.11</b> Key Takeaways</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="chisquare.html"><a href="chisquare.html"><i class="fa fa-check"></i><b>18</b> The Chi-square Distribution</a>
<ul>
<li class="chapter" data-level="18.1" data-path="chisquare.html"><a href="chisquare.html#background"><i class="fa fa-check"></i><b>18.1</b> Background</a></li>
<li class="chapter" data-level="18.2" data-path="chisquare.html"><a href="chisquare.html#dchisq"><i class="fa fa-check"></i><b>18.2</b> dchisq</a></li>
<li class="chapter" data-level="18.3" data-path="chisquare.html"><a href="chisquare.html#pchisq"><i class="fa fa-check"></i><b>18.3</b> pchisq</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="chisquare.html"><a href="chisquare.html#calculating-p-values-from-pchisq"><i class="fa fa-check"></i><b>18.3.1</b> Calculating p-values from pchisq</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="chisquare.html"><a href="chisquare.html#qchisq"><i class="fa fa-check"></i><b>18.4</b> qchisq</a></li>
<li class="chapter" data-level="18.5" data-path="chisquare.html"><a href="chisquare.html#rchisq"><i class="fa fa-check"></i><b>18.5</b> rchisq</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="nonparametrics.html"><a href="nonparametrics.html"><i class="fa fa-check"></i><b>19</b> Nonparametric Statistical Tests</a>
<ul>
<li class="chapter" data-level="19.1" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-sampling-distributions"><i class="fa fa-check"></i><b>19.1</b> Nonparametric sampling distributions</a></li>
<li class="chapter" data-level="19.2" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-discrete-data"><i class="fa fa-check"></i><b>19.2</b> Experiments involving discrete data</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="nonparametrics.html"><a href="nonparametrics.html#ordered-data-1"><i class="fa fa-check"></i><b>19.2.1</b> Ordered data</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="nonparametrics.html"><a href="nonparametrics.html#experiments-involving-deviant-data"><i class="fa fa-check"></i><b>19.3</b> Experiments involving deviant Data</a></li>
<li class="chapter" data-level="19.4" data-path="nonparametrics.html"><a href="nonparametrics.html#sign-test"><i class="fa fa-check"></i><b>19.4</b> Sign Test</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation"><i class="fa fa-check"></i><b>19.4.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.4.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-6"><i class="fa fa-check"></i><b>19.4.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-one-group"><i class="fa fa-check"></i><b>19.5</b> Wilcoxon Sign Rank Test for One Group</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-experimental-designs"><i class="fa fa-check"></i><b>19.5.1</b> Wilcoxon Sign Rank Experimental Designs</a></li>
<li class="chapter" data-level="19.5.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-1"><i class="fa fa-check"></i><b>19.5.2</b> Interpretation</a></li>
<li class="chapter" data-level="19.5.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-7"><i class="fa fa-check"></i><b>19.5.3</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-mann-whitney-rank-sum-test-for-2-independent-groups"><i class="fa fa-check"></i><b>19.6</b> Wilcoxon Mann Whitney Rank Sum Test for 2 independent groups</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-2"><i class="fa fa-check"></i><b>19.6.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.6.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-8"><i class="fa fa-check"></i><b>19.6.2</b> Write Up</a></li>
<li class="chapter" data-level="19.6.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot"><i class="fa fa-check"></i><b>19.6.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="nonparametrics.html"><a href="nonparametrics.html#wilcoxon-sign-rank-test-for-paired-groups"><i class="fa fa-check"></i><b>19.7</b> Wilcoxon Sign Rank Test for paired groups</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-3"><i class="fa fa-check"></i><b>19.7.1</b> Interpretation</a></li>
<li class="chapter" data-level="19.7.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-9"><i class="fa fa-check"></i><b>19.7.2</b> Write up</a></li>
<li class="chapter" data-level="19.7.3" data-path="nonparametrics.html"><a href="nonparametrics.html#plot-1"><i class="fa fa-check"></i><b>19.7.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="19.8" data-path="nonparametrics.html"><a href="nonparametrics.html#kruskal-wallis"><i class="fa fa-check"></i><b>19.8</b> Kruskal-Wallis</a>
<ul>
<li class="chapter" data-level="19.8.1" data-path="nonparametrics.html"><a href="nonparametrics.html#the-experimental-design"><i class="fa fa-check"></i><b>19.8.1</b> The experimental design</a></li>
<li class="chapter" data-level="19.8.2" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-10"><i class="fa fa-check"></i><b>19.8.2</b> Write Up</a></li>
</ul></li>
<li class="chapter" data-level="19.9" data-path="nonparametrics.html"><a href="nonparametrics.html#friedman-test"><i class="fa fa-check"></i><b>19.9</b> Friedman test</a>
<ul>
<li class="chapter" data-level="19.9.1" data-path="nonparametrics.html"><a href="nonparametrics.html#blocked-experimental-design"><i class="fa fa-check"></i><b>19.9.1</b> Blocked experimental design</a></li>
<li class="chapter" data-level="19.9.2" data-path="nonparametrics.html"><a href="nonparametrics.html#interpretation-4"><i class="fa fa-check"></i><b>19.9.2</b> Interpretation</a></li>
<li class="chapter" data-level="19.9.3" data-path="nonparametrics.html"><a href="nonparametrics.html#write-up-11"><i class="fa fa-check"></i><b>19.9.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="19.10" data-path="nonparametrics.html"><a href="nonparametrics.html#nonparametric-power-calculations"><i class="fa fa-check"></i><b>19.10</b> Nonparametric power calculations</a>
<ul>
<li class="chapter" data-level="19.10.1" data-path="nonparametrics.html"><a href="nonparametrics.html#how-it-works"><i class="fa fa-check"></i><b>19.10.1</b> How it works</a></li>
<li class="chapter" data-level="19.10.2" data-path="nonparametrics.html"><a href="nonparametrics.html#simulating-response-data"><i class="fa fa-check"></i><b>19.10.2</b> Simulating response data</a></li>
<li class="chapter" data-level="19.10.3" data-path="nonparametrics.html"><a href="nonparametrics.html#an-example"><i class="fa fa-check"></i><b>19.10.3</b> An example</a></li>
<li class="chapter" data-level="19.10.4" data-path="nonparametrics.html"><a href="nonparametrics.html#nonpara.pwr"><i class="fa fa-check"></i><b>19.10.4</b> nonpara.pwr</a></li>
</ul></li>
<li class="chapter" data-level="19.11" data-path="nonparametrics.html"><a href="nonparametrics.html#summary-3"><i class="fa fa-check"></i><b>19.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="signrank.html"><a href="signrank.html"><i class="fa fa-check"></i><b>20</b> Signed Rank Distribution</a>
<ul>
<li class="chapter" data-level="20.1" data-path="signrank.html"><a href="signrank.html#transformation-of-data-into-sign-ranks"><i class="fa fa-check"></i><b>20.1</b> Transformation of data into sign ranks</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="signrank.html"><a href="signrank.html#for-a-one-group-sample"><i class="fa fa-check"></i><b>20.1.1</b> For a one group sample</a></li>
<li class="chapter" data-level="20.1.2" data-path="signrank.html"><a href="signrank.html#for-a-paired-sample"><i class="fa fa-check"></i><b>20.1.2</b> For a paired sample</a></li>
<li class="chapter" data-level="20.1.3" data-path="signrank.html"><a href="signrank.html#the-sign-rank-test-statistic-in-r"><i class="fa fa-check"></i><b>20.1.3</b> The sign rank test statistic in R</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="signrank.html"><a href="signrank.html#rs-four-sign-rank-distribution-functions"><i class="fa fa-check"></i><b>20.2</b> R’s Four Sign Rank Distribution Functions</a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="signrank.html"><a href="signrank.html#dsignrank"><i class="fa fa-check"></i><b>20.2.1</b> dsignrank</a></li>
<li class="chapter" data-level="20.2.2" data-path="signrank.html"><a href="signrank.html#psignrank"><i class="fa fa-check"></i><b>20.2.2</b> psignrank</a></li>
<li class="chapter" data-level="20.2.3" data-path="signrank.html"><a href="signrank.html#qsignrank"><i class="fa fa-check"></i><b>20.2.3</b> qsignrank</a></li>
<li class="chapter" data-level="20.2.4" data-path="signrank.html"><a href="signrank.html#rsignrank"><i class="fa fa-check"></i><b>20.2.4</b> rsignrank</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="ranksum.html"><a href="ranksum.html"><i class="fa fa-check"></i><b>21</b> Rank Sum Distribution</a>
<ul>
<li class="chapter" data-level="21.0.1" data-path="ranksum.html"><a href="ranksum.html#transformation-of-data-into-rank-summs"><i class="fa fa-check"></i><b>21.0.1</b> Transformation of data into rank summs</a></li>
<li class="chapter" data-level="21.0.2" data-path="ranksum.html"><a href="ranksum.html#the-sign-rank-test-statistic-in-r-1"><i class="fa fa-check"></i><b>21.0.2</b> The sign rank test statistic in R</a></li>
<li class="chapter" data-level="21.1" data-path="ranksum.html"><a href="ranksum.html#rs-four-sign-rank-distribution-functions-1"><i class="fa fa-check"></i><b>21.1</b> R’s Four Sign Rank Distribution Functions</a>
<ul>
<li class="chapter" data-level="21.1.1" data-path="ranksum.html"><a href="ranksum.html#dwilcox"><i class="fa fa-check"></i><b>21.1.1</b> dwilcox</a></li>
<li class="chapter" data-level="21.1.2" data-path="ranksum.html"><a href="ranksum.html#pwilcox"><i class="fa fa-check"></i><b>21.1.2</b> pwilcox</a></li>
<li class="chapter" data-level="21.1.3" data-path="ranksum.html"><a href="ranksum.html#qwilcox"><i class="fa fa-check"></i><b>21.1.3</b> qwilcox</a></li>
<li class="chapter" data-level="21.1.4" data-path="ranksum.html"><a href="ranksum.html#rwilcox"><i class="fa fa-check"></i><b>21.1.4</b> rwilcox</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="ttests.html"><a href="ttests.html"><i class="fa fa-check"></i><b>22</b> The t-tests</a>
<ul>
<li class="chapter" data-level="22.1" data-path="ttests.html"><a href="ttests.html#assumptions-for-t-tests"><i class="fa fa-check"></i><b>22.1</b> Assumptions for t-tests</a>
<ul>
<li class="chapter" data-level="22.1.1" data-path="ttests.html"><a href="ttests.html#strong-assumptions-t-tests-are-invalid-if-these-not-met"><i class="fa fa-check"></i><b>22.1.1</b> Strong assumptions, t-tests are invalid if these not met:</a></li>
<li class="chapter" data-level="22.1.2" data-path="ttests.html"><a href="ttests.html#weak-assumptions-t-tests-are-less-robust-when-these-are-not-met"><i class="fa fa-check"></i><b>22.1.2</b> Weak assumptions: t-tests are less robust when these are not met:</a></li>
<li class="chapter" data-level="22.1.3" data-path="ttests.html"><a href="ttests.html#on-random-residual-error"><i class="fa fa-check"></i><b>22.1.3</b> On random residual error</a></li>
<li class="chapter" data-level="22.1.4" data-path="ttests.html"><a href="ttests.html#on-the-normality-assumption"><i class="fa fa-check"></i><b>22.1.4</b> On the normality assumption</a></li>
</ul></li>
<li class="chapter" data-level="22.2" data-path="ttests.html"><a href="ttests.html#the-t-statistic"><i class="fa fa-check"></i><b>22.2</b> The t Statistic</a></li>
<li class="chapter" data-level="22.3" data-path="ttests.html"><a href="ttests.html#one-sample-t-tests"><i class="fa fa-check"></i><b>22.3</b> One sample t tests</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="ttests.html"><a href="ttests.html#example"><i class="fa fa-check"></i><b>22.3.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="ttests.html"><a href="ttests.html#unpaired-t-tests"><i class="fa fa-check"></i><b>22.4</b> Unpaired t tests</a>
<ul>
<li class="chapter" data-level="22.4.1" data-path="ttests.html"><a href="ttests.html#students-t-test"><i class="fa fa-check"></i><b>22.4.1</b> Student’s t-test</a></li>
<li class="chapter" data-level="22.4.2" data-path="ttests.html"><a href="ttests.html#welchs-t-test"><i class="fa fa-check"></i><b>22.4.2</b> Welch’s t-test</a></li>
<li class="chapter" data-level="22.4.3" data-path="ttests.html"><a href="ttests.html#example-1"><i class="fa fa-check"></i><b>22.4.3</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="22.5" data-path="ttests.html"><a href="ttests.html#paired-t-tests"><i class="fa fa-check"></i><b>22.5</b> Paired t tests</a>
<ul>
<li class="chapter" data-level="22.5.1" data-path="ttests.html"><a href="ttests.html#example-2"><i class="fa fa-check"></i><b>22.5.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="22.6" data-path="ttests.html"><a href="ttests.html#t-test-hypotheses"><i class="fa fa-check"></i><b>22.6</b> t Test Hypotheses</a></li>
<li class="chapter" data-level="22.7" data-path="ttests.html"><a href="ttests.html#one-sided-or-two-sided"><i class="fa fa-check"></i><b>22.7</b> One-sided or two-sided??</a>
<ul>
<li class="chapter" data-level="22.7.1" data-path="ttests.html"><a href="ttests.html#one-sample-hypotheses"><i class="fa fa-check"></i><b>22.7.1</b> One sample hypotheses</a></li>
<li class="chapter" data-level="22.7.2" data-path="ttests.html"><a href="ttests.html#unpaired-hypotheses"><i class="fa fa-check"></i><b>22.7.2</b> Unpaired hypotheses</a></li>
<li class="chapter" data-level="22.7.3" data-path="ttests.html"><a href="ttests.html#paired-hypotheses"><i class="fa fa-check"></i><b>22.7.3</b> Paired hypotheses</a></li>
</ul></li>
<li class="chapter" data-level="22.8" data-path="ttests.html"><a href="ttests.html#confidence-intervals-1"><i class="fa fa-check"></i><b>22.8</b> Confidence Intervals</a></li>
<li class="chapter" data-level="22.9" data-path="ttests.html"><a href="ttests.html#reliability-of-small-samples"><i class="fa fa-check"></i><b>22.9</b> Reliability of small samples</a>
<ul>
<li class="chapter" data-level="22.9.1" data-path="ttests.html"><a href="ttests.html#reporting-sd-or-sem"><i class="fa fa-check"></i><b>22.9.1</b> Reporting SD or SEM?</a></li>
</ul></li>
<li class="chapter" data-level="22.10" data-path="ttests.html"><a href="ttests.html#running-the-analysis-in-r"><i class="fa fa-check"></i><b>22.10</b> Running the analysis in R</a>
<ul>
<li class="chapter" data-level="22.10.1" data-path="ttests.html"><a href="ttests.html#one-sample-t-test"><i class="fa fa-check"></i><b>22.10.1</b> One sample t test</a></li>
<li class="chapter" data-level="22.10.2" data-path="ttests.html"><a href="ttests.html#unpaired-t-test"><i class="fa fa-check"></i><b>22.10.2</b> Unpaired t test</a></li>
<li class="chapter" data-level="22.10.3" data-path="ttests.html"><a href="ttests.html#paired-t-test"><i class="fa fa-check"></i><b>22.10.3</b> Paired t Test</a></li>
</ul></li>
<li class="chapter" data-level="22.11" data-path="ttests.html"><a href="ttests.html#plotting-t-tests"><i class="fa fa-check"></i><b>22.11</b> Plotting t Tests</a>
<ul>
<li class="chapter" data-level="22.11.1" data-path="ttests.html"><a href="ttests.html#one-sample"><i class="fa fa-check"></i><b>22.11.1</b> One-sample</a></li>
<li class="chapter" data-level="22.11.2" data-path="ttests.html"><a href="ttests.html#unpaired"><i class="fa fa-check"></i><b>22.11.2</b> Unpaired</a></li>
<li class="chapter" data-level="22.11.3" data-path="ttests.html"><a href="ttests.html#paired"><i class="fa fa-check"></i><b>22.11.3</b> Paired</a></li>
</ul></li>
<li class="chapter" data-level="22.12" data-path="ttests.html"><a href="ttests.html#t-test-power"><i class="fa fa-check"></i><b>22.12</b> t Test Power</a>
<ul>
<li class="chapter" data-level="22.12.1" data-path="ttests.html"><a href="ttests.html#pwr.t.test"><i class="fa fa-check"></i><b>22.12.1</b> pwr.t.test</a></li>
<li class="chapter" data-level="22.12.2" data-path="ttests.html"><a href="ttests.html#t-pwr-a-monte-carlo-method"><i class="fa fa-check"></i><b>22.12.2</b> t-pwr: A Monte Carlo method</a></li>
</ul></li>
<li class="chapter" data-level="22.13" data-path="ttests.html"><a href="ttests.html#summary-4"><i class="fa fa-check"></i><b>22.13</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="ttestmc.html"><a href="ttestmc.html"><i class="fa fa-check"></i><b>23</b> Statistical design of t-tests</a>
<ul>
<li class="chapter" data-level="23.1" data-path="ttestmc.html"><a href="ttestmc.html#about-this-chapter"><i class="fa fa-check"></i><b>23.1</b> About this chapter</a></li>
<li class="chapter" data-level="23.2" data-path="ttestmc.html"><a href="ttestmc.html#overview-of-the-monte-carlo"><i class="fa fa-check"></i><b>23.2</b> Overview of the Monte Carlo</a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="ttestmc.html"><a href="ttestmc.html#data-simulation"><i class="fa fa-check"></i><b>23.2.1</b> Data simulation</a></li>
<li class="chapter" data-level="23.2.2" data-path="ttestmc.html"><a href="ttestmc.html#the-test"><i class="fa fa-check"></i><b>23.2.2</b> The test</a></li>
</ul></li>
<li class="chapter" data-level="23.3" data-path="ttestmc.html"><a href="ttestmc.html#changing-sample-size"><i class="fa fa-check"></i><b>23.3</b> Changing Sample Size</a></li>
<li class="chapter" data-level="23.4" data-path="ttestmc.html"><a href="ttestmc.html#diabetes-drug-scenario"><i class="fa fa-check"></i><b>23.4</b> Diabetes drug scenario</a>
<ul>
<li class="chapter" data-level="23.4.1" data-path="ttestmc.html"><a href="ttestmc.html#what-would-be-a-scientifically-meaningful-response"><i class="fa fa-check"></i><b>23.4.1</b> What would be a scientifically meaningful response?</a></li>
</ul></li>
<li class="chapter" data-level="23.5" data-path="ttestmc.html"><a href="ttestmc.html#one-sample-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.5</b> One sample t-test Monte Carlo</a>
<ul>
<li class="chapter" data-level="23.5.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-2"><i class="fa fa-check"></i><b>23.5.1</b> Step 1</a></li>
<li class="chapter" data-level="23.5.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-2"><i class="fa fa-check"></i><b>23.5.2</b> Step 2</a></li>
<li class="chapter" data-level="23.5.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-2"><i class="fa fa-check"></i><b>23.5.3</b> Step 3</a></li>
<li class="chapter" data-level="23.5.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-2"><i class="fa fa-check"></i><b>23.5.4</b> Step 4</a></li>
<li class="chapter" data-level="23.5.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-2"><i class="fa fa-check"></i><b>23.5.5</b> Step 5</a></li>
<li class="chapter" data-level="23.5.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-2"><i class="fa fa-check"></i><b>23.5.6</b> Step 6</a></li>
<li class="chapter" data-level="23.5.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-1"><i class="fa fa-check"></i><b>23.5.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.6" data-path="ttestmc.html"><a href="ttestmc.html#unpaired-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.6</b> Unpaired t-test Monte Carlo</a>
<ul>
<li class="chapter" data-level="23.6.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-3"><i class="fa fa-check"></i><b>23.6.1</b> Step 1</a></li>
<li class="chapter" data-level="23.6.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-3"><i class="fa fa-check"></i><b>23.6.2</b> Step 2</a></li>
<li class="chapter" data-level="23.6.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-3"><i class="fa fa-check"></i><b>23.6.3</b> Step 3</a></li>
<li class="chapter" data-level="23.6.4" data-path="ttestmc.html"><a href="ttestmc.html#step-4-3"><i class="fa fa-check"></i><b>23.6.4</b> Step 4</a></li>
<li class="chapter" data-level="23.6.5" data-path="ttestmc.html"><a href="ttestmc.html#step-5-3"><i class="fa fa-check"></i><b>23.6.5</b> Step 5</a></li>
<li class="chapter" data-level="23.6.6" data-path="ttestmc.html"><a href="ttestmc.html#step-6-3"><i class="fa fa-check"></i><b>23.6.6</b> Step 6</a></li>
<li class="chapter" data-level="23.6.7" data-path="ttestmc.html"><a href="ttestmc.html#step-7-2"><i class="fa fa-check"></i><b>23.6.7</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.7" data-path="ttestmc.html"><a href="ttestmc.html#paired-t-test-monte-carlo"><i class="fa fa-check"></i><b>23.7</b> Paired t-test Monte Carlo</a>
<ul>
<li class="chapter" data-level="23.7.1" data-path="ttestmc.html"><a href="ttestmc.html#step-1-4"><i class="fa fa-check"></i><b>23.7.1</b> Step 1</a></li>
<li class="chapter" data-level="23.7.2" data-path="ttestmc.html"><a href="ttestmc.html#step-2-4"><i class="fa fa-check"></i><b>23.7.2</b> Step 2</a></li>
<li class="chapter" data-level="23.7.3" data-path="ttestmc.html"><a href="ttestmc.html#step-3-4"><i class="fa fa-check"></i><b>23.7.3</b> Step 3</a></li>
</ul></li>
<li class="chapter" data-level="23.8" data-path="ttestmc.html"><a href="ttestmc.html#step-4-4"><i class="fa fa-check"></i><b>23.8</b> Step 4</a>
<ul>
<li class="chapter" data-level="23.8.1" data-path="ttestmc.html"><a href="ttestmc.html#step-5-4"><i class="fa fa-check"></i><b>23.8.1</b> Step 5</a></li>
<li class="chapter" data-level="23.8.2" data-path="ttestmc.html"><a href="ttestmc.html#step-6-4"><i class="fa fa-check"></i><b>23.8.2</b> Step 6</a></li>
<li class="chapter" data-level="23.8.3" data-path="ttestmc.html"><a href="ttestmc.html#step-7-3"><i class="fa fa-check"></i><b>23.8.3</b> Step 7</a></li>
</ul></li>
<li class="chapter" data-level="23.9" data-path="ttestmc.html"><a href="ttestmc.html#comparison-to-pwr-tests"><i class="fa fa-check"></i><b>23.9</b> Comparison to pwr tests</a></li>
<li class="chapter" data-level="23.10" data-path="ttestmc.html"><a href="ttestmc.html#summary-5"><i class="fa fa-check"></i><b>23.10</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="tdist.html"><a href="tdist.html"><i class="fa fa-check"></i><b>24</b> t Distributions</a>
<ul>
<li class="chapter" data-level="24.1" data-path="tdist.html"><a href="tdist.html#dt"><i class="fa fa-check"></i><b>24.1</b> dt</a></li>
<li class="chapter" data-level="24.2" data-path="tdist.html"><a href="tdist.html#pt"><i class="fa fa-check"></i><b>24.2</b> pt</a></li>
<li class="chapter" data-level="24.3" data-path="tdist.html"><a href="tdist.html#qt"><i class="fa fa-check"></i><b>24.3</b> qt</a></li>
<li class="chapter" data-level="24.4" data-path="tdist.html"><a href="tdist.html#rt"><i class="fa fa-check"></i><b>24.4</b> rt</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="simcorrelation.html"><a href="simcorrelation.html"><i class="fa fa-check"></i><b>25</b> Simulating correlated variables</a>
<ul>
<li class="chapter" data-level="25.1" data-path="simcorrelation.html"><a href="simcorrelation.html#estimating-correlation-between-two-variables"><i class="fa fa-check"></i><b>25.1</b> Estimating correlation between two variables</a></li>
<li class="chapter" data-level="25.2" data-path="simcorrelation.html"><a href="simcorrelation.html#simulating-correlated-variables"><i class="fa fa-check"></i><b>25.2</b> Simulating correlated variables</a></li>
<li class="chapter" data-level="25.3" data-path="simcorrelation.html"><a href="simcorrelation.html#monte-carlo-simulation"><i class="fa fa-check"></i><b>25.3</b> Monte Carlo simulation</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="introanova.html"><a href="introanova.html"><i class="fa fa-check"></i><b>26</b> Introduction to ANOVA</a>
<ul>
<li class="chapter" data-level="26.1" data-path="introanova.html"><a href="introanova.html#assumptions"><i class="fa fa-check"></i><b>26.1</b> Assumptions</a></li>
<li class="chapter" data-level="26.2" data-path="introanova.html"><a href="introanova.html#types-of-designs"><i class="fa fa-check"></i><b>26.2</b> Types of designs</a></li>
<li class="chapter" data-level="26.3" data-path="introanova.html"><a href="introanova.html#jargon-factors-and-levels"><i class="fa fa-check"></i><b>26.3</b> Jargon: Factors and levels</a>
<ul>
<li class="chapter" data-level="26.3.1" data-path="introanova.html"><a href="introanova.html#woring-with-factors-in-r"><i class="fa fa-check"></i><b>26.3.1</b> Woring with factors in R</a></li>
</ul></li>
<li class="chapter" data-level="26.4" data-path="introanova.html"><a href="introanova.html#anova-models-one--two--and-three-way"><i class="fa fa-check"></i><b>26.4</b> ANOVA models: One-, Two-, and Three-way</a></li>
<li class="chapter" data-level="26.5" data-path="introanova.html"><a href="introanova.html#inference"><i class="fa fa-check"></i><b>26.5</b> Inference</a>
<ul>
<li class="chapter" data-level="26.5.1" data-path="introanova.html"><a href="introanova.html#a-quick-peek-at-posthoc-testing"><i class="fa fa-check"></i><b>26.5.1</b> A quick peek at posthoc testing</a></li>
</ul></li>
<li class="chapter" data-level="26.6" data-path="introanova.html"><a href="introanova.html#anova-calculations"><i class="fa fa-check"></i><b>26.6</b> ANOVA calculations</a>
<ul>
<li class="chapter" data-level="26.6.1" data-path="introanova.html"><a href="introanova.html#sums-of-squares-partitioning"><i class="fa fa-check"></i><b>26.6.1</b> Sums of Squares partitioning</a></li>
<li class="chapter" data-level="26.6.2" data-path="introanova.html"><a href="introanova.html#the-anova-table"><i class="fa fa-check"></i><b>26.6.2</b> The ANOVA table</a></li>
<li class="chapter" data-level="26.6.3" data-path="introanova.html"><a href="introanova.html#post-hoc-group-comparisons"><i class="fa fa-check"></i><b>26.6.3</b> Post-hoc group comparisons</a></li>
</ul></li>
<li class="chapter" data-level="26.7" data-path="introanova.html"><a href="introanova.html#completely-randomized-or-related-measures"><i class="fa fa-check"></i><b>26.7</b> Completely randomized or related measures</a>
<ul>
<li class="chapter" data-level="26.7.1" data-path="introanova.html"><a href="introanova.html#the-problem-of-lost-data-in-related-measures-designs"><i class="fa fa-check"></i><b>26.7.1</b> The problem of lost data in related measures designs</a></li>
</ul></li>
<li class="chapter" data-level="26.8" data-path="introanova.html"><a href="introanova.html#two-way-anova"><i class="fa fa-check"></i><b>26.8</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="26.9" data-path="introanova.html"><a href="introanova.html#other-anova-models"><i class="fa fa-check"></i><b>26.9</b> Other ANOVA models</a>
<ul>
<li class="chapter" data-level="26.9.1" data-path="introanova.html"><a href="introanova.html#r-and-anova"><i class="fa fa-check"></i><b>26.9.1</b> R and ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="26.10" data-path="introanova.html"><a href="introanova.html#alternatives-to-anova"><i class="fa fa-check"></i><b>26.10</b> Alternatives to ANOVA</a>
<ul>
<li class="chapter" data-level="26.10.1" data-path="introanova.html"><a href="introanova.html#screw-anova-just-tell-me-how-to-t-test-everything"><i class="fa fa-check"></i><b>26.10.1</b> Screw ANOVA, Just Tell Me How to t-Test Everything</a></li>
</ul></li>
<li class="chapter" data-level="26.11" data-path="introanova.html"><a href="introanova.html#summary-6"><i class="fa fa-check"></i><b>26.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="fdistr.html"><a href="fdistr.html"><i class="fa fa-check"></i><b>27</b> The F distribution</a>
<ul>
<li class="chapter" data-level="27.1" data-path="fdistr.html"><a href="fdistr.html#background-1"><i class="fa fa-check"></i><b>27.1</b> Background</a>
<ul>
<li class="chapter" data-level="27.1.1" data-path="fdistr.html"><a href="fdistr.html#sample-variance-and-fs-pdf"><i class="fa fa-check"></i><b>27.1.1</b> Sample Variance and F’s PDF</a></li>
</ul></li>
<li class="chapter" data-level="27.2" data-path="fdistr.html"><a href="fdistr.html#df"><i class="fa fa-check"></i><b>27.2</b> df</a></li>
<li class="chapter" data-level="27.3" data-path="fdistr.html"><a href="fdistr.html#pf"><i class="fa fa-check"></i><b>27.3</b> pf</a></li>
<li class="chapter" data-level="27.4" data-path="fdistr.html"><a href="fdistr.html#qf"><i class="fa fa-check"></i><b>27.4</b> qf</a></li>
<li class="chapter" data-level="27.5" data-path="fdistr.html"><a href="fdistr.html#rf"><i class="fa fa-check"></i><b>27.5</b> rf</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="onewayanova.html"><a href="onewayanova.html"><i class="fa fa-check"></i><b>28</b> One-way ANOVA Completely Randomized</a>
<ul>
<li class="chapter" data-level="28.1" data-path="onewayanova.html"><a href="onewayanova.html#using-ezanova"><i class="fa fa-check"></i><b>28.1</b> Using <code>ezANOVA</code></a></li>
<li class="chapter" data-level="28.2" data-path="onewayanova.html"><a href="onewayanova.html#the-chickwt-data-set"><i class="fa fa-check"></i><b>28.2</b> The chickwt data set</a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="onewayanova.html"><a href="onewayanova.html#inspect-the-data"><i class="fa fa-check"></i><b>28.2.1</b> Inspect the data</a></li>
</ul></li>
<li class="chapter" data-level="28.3" data-path="onewayanova.html"><a href="onewayanova.html#run-the-anova"><i class="fa fa-check"></i><b>28.3</b> Run the ANOVA</a>
<ul>
<li class="chapter" data-level="28.3.1" data-path="onewayanova.html"><a href="onewayanova.html#run-the-chickwts-one-way-anova"><i class="fa fa-check"></i><b>28.3.1</b> Run the chickwts One Way ANOVA</a></li>
<li class="chapter" data-level="28.3.2" data-path="onewayanova.html"><a href="onewayanova.html#interpreting-the-one-way-cr-anova-output"><i class="fa fa-check"></i><b>28.3.2</b> Interpreting the One-Way CR ANOVA Output</a></li>
</ul></li>
<li class="chapter" data-level="28.4" data-path="onewayanova.html"><a href="onewayanova.html#post-hoc-pairwise-comparisons"><i class="fa fa-check"></i><b>28.4</b> Post hoc pairwise comparisons</a>
<ul>
<li class="chapter" data-level="28.4.1" data-path="onewayanova.html"><a href="onewayanova.html#range-tests"><i class="fa fa-check"></i><b>28.4.1</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="28.5" data-path="onewayanova.html"><a href="onewayanova.html#summary-7"><i class="fa fa-check"></i><b>28.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="onewayRM.html"><a href="onewayRM.html"><i class="fa fa-check"></i><b>29</b> One-way ANOVA Related Measures</a>
<ul>
<li class="chapter" data-level="29.1" data-path="onewayRM.html"><a href="onewayRM.html#data-prep"><i class="fa fa-check"></i><b>29.1</b> Data prep</a></li>
<li class="chapter" data-level="29.2" data-path="onewayRM.html"><a href="onewayRM.html#data-visualization"><i class="fa fa-check"></i><b>29.2</b> Data visualization</a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="onewayRM.html"><a href="onewayRM.html#visualize-the-statistical-design"><i class="fa fa-check"></i><b>29.2.1</b> Visualize the statistical design</a></li>
</ul></li>
<li class="chapter" data-level="29.3" data-path="onewayRM.html"><a href="onewayRM.html#the-anova"><i class="fa fa-check"></i><b>29.3</b> The ANOVA</a>
<ul>
<li class="chapter" data-level="29.3.1" data-path="onewayRM.html"><a href="onewayRM.html#running-ezanova"><i class="fa fa-check"></i><b>29.3.1</b> Running ezANOVA</a></li>
</ul></li>
<li class="chapter" data-level="29.4" data-path="onewayRM.html"><a href="onewayRM.html#interpretation-5"><i class="fa fa-check"></i><b>29.4</b> Interpretation</a></li>
<li class="chapter" data-level="29.5" data-path="onewayRM.html"><a href="onewayRM.html#post-hoc-analysis"><i class="fa fa-check"></i><b>29.5</b> Post-hoc analysis</a></li>
<li class="chapter" data-level="29.6" data-path="onewayRM.html"><a href="onewayRM.html#write-up-12"><i class="fa fa-check"></i><b>29.6</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="twowayCR.html"><a href="twowayCR.html"><i class="fa fa-check"></i><b>30</b> Two-way ANOVA Completely Randomized</a>
<ul>
<li class="chapter" data-level="30.1" data-path="twowayCR.html"><a href="twowayCR.html#effect-of-strain-and-diet-on-liver"><i class="fa fa-check"></i><b>30.1</b> Effect of Strain and Diet on Liver</a></li>
<li class="chapter" data-level="30.2" data-path="twowayCR.html"><a href="twowayCR.html#the-test-1"><i class="fa fa-check"></i><b>30.2</b> The test</a></li>
<li class="chapter" data-level="30.3" data-path="twowayCR.html"><a href="twowayCR.html#interpretation-of-2-way-cr-anova-output"><i class="fa fa-check"></i><b>30.3</b> Interpretation of 2 Way CR ANOVA Output</a>
<ul>
<li class="chapter" data-level="30.3.1" data-path="twowayCR.html"><a href="twowayCR.html#levenes"><i class="fa fa-check"></i><b>30.3.1</b> Levene’s</a></li>
<li class="chapter" data-level="30.3.2" data-path="twowayCR.html"><a href="twowayCR.html#anova-table"><i class="fa fa-check"></i><b>30.3.2</b> ANOVA Table</a></li>
</ul></li>
<li class="chapter" data-level="30.4" data-path="twowayCR.html"><a href="twowayCR.html#post-hoc-multiple-comparisons"><i class="fa fa-check"></i><b>30.4</b> Post Hoc Multiple Comparisons</a>
<ul>
<li class="chapter" data-level="30.4.1" data-path="twowayCR.html"><a href="twowayCR.html#pairwise.t.tests"><i class="fa fa-check"></i><b>30.4.1</b> Pairwise.t.tests</a></li>
<li class="chapter" data-level="30.4.2" data-path="twowayCR.html"><a href="twowayCR.html#write-up-13"><i class="fa fa-check"></i><b>30.4.2</b> Write Up</a></li>
<li class="chapter" data-level="30.4.3" data-path="twowayCR.html"><a href="twowayCR.html#range-tests-1"><i class="fa fa-check"></i><b>30.4.3</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="30.5" data-path="twowayCR.html"><a href="twowayCR.html#summary-8"><i class="fa fa-check"></i><b>30.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="twowayRM.html"><a href="twowayRM.html"><i class="fa fa-check"></i><b>31</b> Two-way ANOVA Related Measures</a>
<ul>
<li class="chapter" data-level="31.1" data-path="twowayRM.html"><a href="twowayRM.html#cell-culture"><i class="fa fa-check"></i><b>31.1</b> Cell culture</a></li>
<li class="chapter" data-level="31.2" data-path="twowayRM.html"><a href="twowayRM.html#the-test-2"><i class="fa fa-check"></i><b>31.2</b> The test</a></li>
<li class="chapter" data-level="31.3" data-path="twowayRM.html"><a href="twowayRM.html#interpretation-of-the-output"><i class="fa fa-check"></i><b>31.3</b> Interpretation of the output</a>
<ul>
<li class="chapter" data-level="31.3.1" data-path="twowayRM.html"><a href="twowayRM.html#anova-table-1"><i class="fa fa-check"></i><b>31.3.1</b> ANOVA Table</a></li>
<li class="chapter" data-level="31.3.2" data-path="twowayRM.html"><a href="twowayRM.html#mauchlys-sphericity-test"><i class="fa fa-check"></i><b>31.3.2</b> Mauchly’s Sphericity Test</a></li>
</ul></li>
<li class="chapter" data-level="31.4" data-path="twowayRM.html"><a href="twowayRM.html#post-hoc-multiple-comparisons-1"><i class="fa fa-check"></i><b>31.4</b> Post Hoc multiple comparisons</a></li>
<li class="chapter" data-level="31.5" data-path="twowayRM.html"><a href="twowayRM.html#write-up-14"><i class="fa fa-check"></i><b>31.5</b> Write Up</a></li>
<li class="chapter" data-level="31.6" data-path="twowayRM.html"><a href="twowayRM.html#summary-9"><i class="fa fa-check"></i><b>31.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="twowaymixed.html"><a href="twowaymixed.html"><i class="fa fa-check"></i><b>32</b> Two-way ANOVA RM/CR</a>
<ul>
<li class="chapter" data-level="32.1" data-path="twowaymixed.html"><a href="twowaymixed.html#chickweight-data-set"><i class="fa fa-check"></i><b>32.1</b> ChickWeight Data set</a></li>
<li class="chapter" data-level="32.2" data-path="twowaymixed.html"><a href="twowaymixed.html#munge-chickweight-data"><i class="fa fa-check"></i><b>32.2</b> Munge ChickWeight data</a></li>
<li class="chapter" data-level="32.3" data-path="twowaymixed.html"><a href="twowaymixed.html#the-test-3"><i class="fa fa-check"></i><b>32.3</b> The test</a></li>
<li class="chapter" data-level="32.4" data-path="twowaymixed.html"><a href="twowaymixed.html#interpreting-the-anova-output"><i class="fa fa-check"></i><b>32.4</b> Interpreting the ANOVA output</a>
<ul>
<li class="chapter" data-level="32.4.1" data-path="twowaymixed.html"><a href="twowaymixed.html#anova-the-anova-table-1"><i class="fa fa-check"></i><b>32.4.1</b> $ANOVA: The ANOVA table</a></li>
<li class="chapter" data-level="32.4.2" data-path="twowaymixed.html"><a href="twowaymixed.html#mauchlys-test-and-corrections"><i class="fa fa-check"></i><b>32.4.2</b> Mauchly’s Test and Corrections</a></li>
</ul></li>
<li class="chapter" data-level="32.5" data-path="twowaymixed.html"><a href="twowaymixed.html#post-hoc"><i class="fa fa-check"></i><b>32.5</b> Post hoc</a>
<ul>
<li class="chapter" data-level="32.5.1" data-path="twowaymixed.html"><a href="twowaymixed.html#pairwise.t.tests-1"><i class="fa fa-check"></i><b>32.5.1</b> Pairwise.t.tests</a></li>
<li class="chapter" data-level="32.5.2" data-path="twowaymixed.html"><a href="twowaymixed.html#tukey-test-a-range-test"><i class="fa fa-check"></i><b>32.5.2</b> Tukey test: A range test</a></li>
<li class="chapter" data-level="32.5.3" data-path="twowaymixed.html"><a href="twowaymixed.html#heres-whats-been-discovered-by-the-tukeyhsd-test"><i class="fa fa-check"></i><b>32.5.3</b> Here’s what’s been discovered by the TukeyHSD test</a></li>
<li class="chapter" data-level="32.5.4" data-path="twowaymixed.html"><a href="twowaymixed.html#why-is-the-tukeyhsd-result-so-different-than-the-pairwise.t.test"><i class="fa fa-check"></i><b>32.5.4</b> Why is the TukeyHSD result so different than the pairwise.t.test??</a></li>
</ul></li>
<li class="chapter" data-level="32.6" data-path="twowaymixed.html"><a href="twowaymixed.html#summary-10"><i class="fa fa-check"></i><b>32.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="posthoc.html"><a href="posthoc.html"><i class="fa fa-check"></i><b>33</b> ANOVA Posthoc Comparisons</a>
<ul>
<li class="chapter" data-level="33.1" data-path="posthoc.html"><a href="posthoc.html#overview-of-options"><i class="fa fa-check"></i><b>33.1</b> Overview of options</a>
<ul>
<li class="chapter" data-level="33.1.1" data-path="posthoc.html"><a href="posthoc.html#pairwise.t.test-with-p-value-adjustments"><i class="fa fa-check"></i><b>33.1.1</b> Pairwise.t.test with p-value adjustments</a></li>
<li class="chapter" data-level="33.1.2" data-path="posthoc.html"><a href="posthoc.html#range-tests-2"><i class="fa fa-check"></i><b>33.1.2</b> Range tests</a></li>
</ul></li>
<li class="chapter" data-level="33.2" data-path="posthoc.html"><a href="posthoc.html#reporting-the-result-1"><i class="fa fa-check"></i><b>33.2</b> Reporting the result</a></li>
<li class="chapter" data-level="33.3" data-path="posthoc.html"><a href="posthoc.html#summary-11"><i class="fa fa-check"></i><b>33.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="34" data-path="jaxwest2.html"><a href="jaxwest2.html"><i class="fa fa-check"></i><b>34</b> Reproducible Data Munging Mostly with Tidyverse</a>
<ul>
<li class="chapter" data-level="34.1" data-path="jaxwest2.html"><a href="jaxwest2.html#look-at-the-original-data-carefully"><i class="fa fa-check"></i><b>34.1</b> Look at the original data carefully</a></li>
<li class="chapter" data-level="34.2" data-path="jaxwest2.html"><a href="jaxwest2.html#our-goal"><i class="fa fa-check"></i><b>34.2</b> Our goal</a></li>
<li class="chapter" data-level="34.3" data-path="jaxwest2.html"><a href="jaxwest2.html#step-1-read-the-data-into-r"><i class="fa fa-check"></i><b>34.3</b> Step 1: Read the data into R</a></li>
<li class="chapter" data-level="34.4" data-path="jaxwest2.html"><a href="jaxwest2.html#step-2-select-the-variables"><i class="fa fa-check"></i><b>34.4</b> Step 2: Select the variables</a></li>
<li class="chapter" data-level="34.5" data-path="jaxwest2.html"><a href="jaxwest2.html#step-3-trim-the-cases"><i class="fa fa-check"></i><b>34.5</b> Step 3: Trim the cases</a></li>
<li class="chapter" data-level="34.6" data-path="jaxwest2.html"><a href="jaxwest2.html#step-4-we-have-a-variable-problem-to-fix"><i class="fa fa-check"></i><b>34.6</b> Step 4: We have a variable problem to fix</a></li>
<li class="chapter" data-level="34.7" data-path="jaxwest2.html"><a href="jaxwest2.html#step-5-impute-or-delete"><i class="fa fa-check"></i><b>34.7</b> Step 5: Impute or Delete</a></li>
<li class="chapter" data-level="34.8" data-path="jaxwest2.html"><a href="jaxwest2.html#step-6-go-long"><i class="fa fa-check"></i><b>34.8</b> Step 6: Go long</a></li>
<li class="chapter" data-level="34.9" data-path="jaxwest2.html"><a href="jaxwest2.html#step-7-convert-other-variables-to-factor"><i class="fa fa-check"></i><b>34.9</b> Step 7: Convert other variables to factor</a></li>
<li class="chapter" data-level="34.10" data-path="jaxwest2.html"><a href="jaxwest2.html#step-8-plot"><i class="fa fa-check"></i><b>34.10</b> Step 8: Plot</a></li>
<li class="chapter" data-level="34.11" data-path="jaxwest2.html"><a href="jaxwest2.html#step-9-run-the-anova"><i class="fa fa-check"></i><b>34.11</b> Step 9: Run the ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="35" data-path="anovamc.html"><a href="anovamc.html"><i class="fa fa-check"></i><b>35</b> ANOVA power using Monte Carlo</a>
<ul>
<li class="chapter" data-level="35.1" data-path="anovamc.html"><a href="anovamc.html#alternatives-to-monte-carlo"><i class="fa fa-check"></i><b>35.1</b> Alternatives to Monte Carlo</a></li>
<li class="chapter" data-level="35.2" data-path="anovamc.html"><a href="anovamc.html#what-is-monte-carlo"><i class="fa fa-check"></i><b>35.2</b> What is Monte Carlo</a></li>
<li class="chapter" data-level="35.3" data-path="anovamc.html"><a href="anovamc.html#one-way-completely-randomized-anova-monte-carlo"><i class="fa fa-check"></i><b>35.3</b> One-way completely randomized ANOVA Monte Carlo</a>
<ul>
<li class="chapter" data-level="35.3.1" data-path="anovamc.html"><a href="anovamc.html#directions"><i class="fa fa-check"></i><b>35.3.1</b> Directions</a></li>
<li class="chapter" data-level="35.3.2" data-path="anovamc.html"><a href="anovamc.html#step-1-create-initial-values"><i class="fa fa-check"></i><b>35.3.2</b> Step 1: Create initial values</a></li>
<li class="chapter" data-level="35.3.3" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-random-sample"><i class="fa fa-check"></i><b>35.3.3</b> Step 2: Visualize one random sample</a></li>
<li class="chapter" data-level="35.3.4" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator"><i class="fa fa-check"></i><b>35.3.4</b> Step 3: Run The Power Simulator</a></li>
<li class="chapter" data-level="35.3.5" data-path="anovamc.html"><a href="anovamc.html#step-4-optimize-for-suitable-power"><i class="fa fa-check"></i><b>35.3.5</b> Step 4: Optimize for suitable power</a></li>
<li class="chapter" data-level="35.3.6" data-path="anovamc.html"><a href="anovamc.html#notes-and-considerations"><i class="fa fa-check"></i><b>35.3.6</b> Notes And Considerations</a></li>
</ul></li>
<li class="chapter" data-level="35.4" data-path="anovamc.html"><a href="anovamc.html#one-way-related-measures-anova-monte-carlo"><i class="fa fa-check"></i><b>35.4</b> One-way related measures ANOVA Monte Carlo</a></li>
<li class="chapter" data-level="35.5" data-path="anovamc.html"><a href="anovamc.html#directions-1"><i class="fa fa-check"></i><b>35.5</b> Directions</a>
<ul>
<li class="chapter" data-level="35.5.1" data-path="anovamc.html"><a href="anovamc.html#step-1-initial-values"><i class="fa fa-check"></i><b>35.5.1</b> Step 1: Initial values</a></li>
<li class="chapter" data-level="35.5.2" data-path="anovamc.html"><a href="anovamc.html#step-2-visualize-one-sample"><i class="fa fa-check"></i><b>35.5.2</b> Step 2: Visualize one sample</a></li>
<li class="chapter" data-level="35.5.3" data-path="anovamc.html"><a href="anovamc.html#step-3-run-the-power-simulator-1"><i class="fa fa-check"></i><b>35.5.3</b> Step 3: Run the power simulator</a></li>
<li class="chapter" data-level="35.5.4" data-path="anovamc.html"><a href="anovamc.html#step-4-should-anything-be-changed"><i class="fa fa-check"></i><b>35.5.4</b> Step 4: Should anything be changed?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="manova.html"><a href="manova.html"><i class="fa fa-check"></i><b>36</b> Multivariate Analysis of Variance (MANOVA)</a>
<ul>
<li class="chapter" data-level="36.1" data-path="manova.html"><a href="manova.html#what-does-manova-test"><i class="fa fa-check"></i><b>36.1</b> What does MANOVA test?</a></li>
<li class="chapter" data-level="36.2" data-path="manova.html"><a href="manova.html#types-of-experiments"><i class="fa fa-check"></i><b>36.2</b> Types of experiments</a></li>
<li class="chapter" data-level="36.3" data-path="manova.html"><a href="manova.html#manova-is-not-many-anovas"><i class="fa fa-check"></i><b>36.3</b> MANOVA is not many ANOVAs</a>
<ul>
<li class="chapter" data-level="36.3.1" data-path="manova.html"><a href="manova.html#why-manova"><i class="fa fa-check"></i><b>36.3.1</b> Why MANOVA?</a></li>
</ul></li>
<li class="chapter" data-level="36.4" data-path="manova.html"><a href="manova.html#assumptions-1"><i class="fa fa-check"></i><b>36.4</b> Assumptions</a>
<ul>
<li class="chapter" data-level="36.4.1" data-path="manova.html"><a href="manova.html#multicolinearity"><i class="fa fa-check"></i><b>36.4.1</b> Multicolinearity</a></li>
</ul></li>
<li class="chapter" data-level="36.5" data-path="manova.html"><a href="manova.html#calculating-variation"><i class="fa fa-check"></i><b>36.5</b> Calculating variation</a>
<ul>
<li class="chapter" data-level="36.5.1" data-path="manova.html"><a href="manova.html#eigenvectors-and-eigenvalues"><i class="fa fa-check"></i><b>36.5.1</b> Eigenvectors and eigenvalues</a></li>
<li class="chapter" data-level="36.5.2" data-path="manova.html"><a href="manova.html#manova-test-statistics"><i class="fa fa-check"></i><b>36.5.2</b> MANOVA test statistics</a></li>
<li class="chapter" data-level="36.5.3" data-path="manova.html"><a href="manova.html#which-manova-test-statistic-is-best"><i class="fa fa-check"></i><b>36.5.3</b> Which MANOVA test statistic is best?</a></li>
</ul></li>
<li class="chapter" data-level="36.6" data-path="manova.html"><a href="manova.html#drug-clearance-example"><i class="fa fa-check"></i><b>36.6</b> Drug clearance example</a>
<ul>
<li class="chapter" data-level="36.6.1" data-path="manova.html"><a href="manova.html#rationale-for-manova"><i class="fa fa-check"></i><b>36.6.1</b> Rationale for MANOVA</a></li>
<li class="chapter" data-level="36.6.2" data-path="manova.html"><a href="manova.html#data-structure"><i class="fa fa-check"></i><b>36.6.2</b> Data structure</a></li>
<li class="chapter" data-level="36.6.3" data-path="manova.html"><a href="manova.html#manova-procedure"><i class="fa fa-check"></i><b>36.6.3</b> MANOVA procedure</a></li>
</ul></li>
<li class="chapter" data-level="36.7" data-path="manova.html"><a href="manova.html#planning-manova-experiments"><i class="fa fa-check"></i><b>36.7</b> Planning MANOVA experiments</a>
<ul>
<li class="chapter" data-level="36.7.1" data-path="manova.html"><a href="manova.html#manova-monte-carlo-power-analysis"><i class="fa fa-check"></i><b>36.7.1</b> MANOVA Monte Carlo Power Analysis</a></li>
<li class="chapter" data-level="36.7.2" data-path="manova.html"><a href="manova.html#an-manova-monte-carlo-power-function"><i class="fa fa-check"></i><b>36.7.2</b> An MANOVA Monte Carlo Power function</a></li>
</ul></li>
<li class="chapter" data-level="36.8" data-path="manova.html"><a href="manova.html#summary-12"><i class="fa fa-check"></i><b>36.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="37" data-path="correl.html"><a href="correl.html"><i class="fa fa-check"></i><b>37</b> Correlation</a>
<ul>
<li class="chapter" data-level="37.1" data-path="correl.html"><a href="correl.html#correlation-causation"><i class="fa fa-check"></i><b>37.1</b> Correlation != Causation</a></li>
<li class="chapter" data-level="37.2" data-path="correl.html"><a href="correl.html#correlation-in-multivariate-outcomes-and-paired-designs"><i class="fa fa-check"></i><b>37.2</b> Correlation in Multivariate Outcomes and Paired Designs</a></li>
<li class="chapter" data-level="37.3" data-path="correl.html"><a href="correl.html#correlation-coefficients"><i class="fa fa-check"></i><b>37.3</b> Correlation coefficients</a>
<ul>
<li class="chapter" data-level="37.3.1" data-path="correl.html"><a href="correl.html#pearsons-correlation-coefficient"><i class="fa fa-check"></i><b>37.3.1</b> Pearson’s correlation coefficient</a></li>
<li class="chapter" data-level="37.3.2" data-path="correl.html"><a href="correl.html#spearmans-rank-correlation"><i class="fa fa-check"></i><b>37.3.2</b> Spearman’s rank correlation</a></li>
<li class="chapter" data-level="37.3.3" data-path="correl.html"><a href="correl.html#kendalls-tau"><i class="fa fa-check"></i><b>37.3.3</b> Kendall’s tau</a></li>
<li class="chapter" data-level="37.3.4" data-path="correl.html"><a href="correl.html#which-correlation-method-to-use"><i class="fa fa-check"></i><b>37.3.4</b> Which correlation method to use?</a></li>
<li class="chapter" data-level="37.3.5" data-path="correl.html"><a href="correl.html#r-correlation-analysis-functions"><i class="fa fa-check"></i><b>37.3.5</b> R correlation analysis functions</a></li>
<li class="chapter" data-level="37.3.6" data-path="correl.html"><a href="correl.html#plot-the-correlations"><i class="fa fa-check"></i><b>37.3.6</b> Plot the correlations</a></li>
<li class="chapter" data-level="37.3.7" data-path="correl.html"><a href="correl.html#calculate-a-correlation-coefficient-and-posthoc-test"><i class="fa fa-check"></i><b>37.3.7</b> Calculate a correlation coefficient and posthoc test</a></li>
<li class="chapter" data-level="37.3.8" data-path="correl.html"><a href="correl.html#interpretation-of-correlation-output"><i class="fa fa-check"></i><b>37.3.8</b> Interpretation of correlation output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="regress.html"><a href="regress.html"><i class="fa fa-check"></i><b>38</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="38.1" data-path="regress.html"><a href="regress.html#the-linear-regression-model"><i class="fa fa-check"></i><b>38.1</b> The linear regression model</a></li>
<li class="chapter" data-level="38.2" data-path="regress.html"><a href="regress.html#least-squares-fitting"><i class="fa fa-check"></i><b>38.2</b> Least squares fitting</a></li>
<li class="chapter" data-level="38.3" data-path="regress.html"><a href="regress.html#the-practical-importance-of-linear-model-parameters"><i class="fa fa-check"></i><b>38.3</b> The practical importance of linear model parameters</a></li>
<li class="chapter" data-level="38.4" data-path="regress.html"><a href="regress.html#linear-model-standard-errors"><i class="fa fa-check"></i><b>38.4</b> Linear model standard errors</a></li>
<li class="chapter" data-level="38.5" data-path="regress.html"><a href="regress.html#linear-regression-in-r"><i class="fa fa-check"></i><b>38.5</b> Linear regression in R</a></li>
<li class="chapter" data-level="38.6" data-path="regress.html"><a href="regress.html#intepretation"><i class="fa fa-check"></i><b>38.6</b> Intepretation</a>
<ul>
<li class="chapter" data-level="38.6.1" data-path="regress.html"><a href="regress.html#residuals"><i class="fa fa-check"></i><b>38.6.1</b> Residuals</a></li>
<li class="chapter" data-level="38.6.2" data-path="regress.html"><a href="regress.html#coefficients"><i class="fa fa-check"></i><b>38.6.2</b> Coefficients</a></li>
<li class="chapter" data-level="38.6.3" data-path="regress.html"><a href="regress.html#degrees-of-freedom-1"><i class="fa fa-check"></i><b>38.6.3</b> Degrees of freedom</a></li>
<li class="chapter" data-level="38.6.4" data-path="regress.html"><a href="regress.html#r-squared"><i class="fa fa-check"></i><b>38.6.4</b> R-squared</a></li>
<li class="chapter" data-level="38.6.5" data-path="regress.html"><a href="regress.html#f-statistic"><i class="fa fa-check"></i><b>38.6.5</b> F-statistic</a></li>
<li class="chapter" data-level="38.6.6" data-path="regress.html"><a href="regress.html#plotting-regression-results"><i class="fa fa-check"></i><b>38.6.6</b> Plotting regression results</a></li>
<li class="chapter" data-level="38.6.7" data-path="regress.html"><a href="regress.html#visualizing-residuals"><i class="fa fa-check"></i><b>38.6.7</b> Visualizing residuals</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="39" data-path="nonlinearintro.html"><a href="nonlinearintro.html"><i class="fa fa-check"></i><b>39</b> Non-linear regression introduction</a>
<ul>
<li class="chapter" data-level="39.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#uses-for-nonlinear-regression"><i class="fa fa-check"></i><b>39.1</b> Uses for nonlinear regression</a></li>
<li class="chapter" data-level="39.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#example-of-the-process"><i class="fa fa-check"></i><b>39.2</b> Example of the process</a>
<ul>
<li class="chapter" data-level="39.2.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#it-starts-with-nonlinear-data"><i class="fa fa-check"></i><b>39.2.1</b> It starts with nonlinear data</a></li>
<li class="chapter" data-level="39.2.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#which-nonlinear-model-describes-that-data"><i class="fa fa-check"></i><b>39.2.2</b> Which nonlinear model describes that data?</a></li>
<li class="chapter" data-level="39.2.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#fitting-a-model-to-data"><i class="fa fa-check"></i><b>39.2.3</b> Fitting a model to data</a></li>
<li class="chapter" data-level="39.2.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#interpreting-regression-model-output"><i class="fa fa-check"></i><b>39.2.4</b> Interpreting regression model output</a></li>
<li class="chapter" data-level="39.2.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#check-the-fit-graphically"><i class="fa fa-check"></i><b>39.2.5</b> Check the fit graphically</a></li>
<li class="chapter" data-level="39.2.6" data-path="nonlinearintro.html"><a href="nonlinearintro.html#convergence"><i class="fa fa-check"></i><b>39.2.6</b> Convergence?</a></li>
<li class="chapter" data-level="39.2.7" data-path="nonlinearintro.html"><a href="nonlinearintro.html#convergence-is-not-idiot-proof"><i class="fa fa-check"></i><b>39.2.7</b> Convergence is not idiot proof</a></li>
</ul></li>
<li class="chapter" data-level="39.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-in-general"><i class="fa fa-check"></i><b>39.3</b> Nonlinear models in general</a></li>
<li class="chapter" data-level="39.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nlfitr"><i class="fa fa-check"></i><b>39.4</b> nlfitr</a></li>
<li class="chapter" data-level="39.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-regression-functions-in-r"><i class="fa fa-check"></i><b>39.5</b> Nonlinear regression functions in R</a>
<ul>
<li class="chapter" data-level="39.5.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#the-fitting-algorithms"><i class="fa fa-check"></i><b>39.5.1</b> The fitting algorithms</a></li>
<li class="chapter" data-level="39.5.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#formula"><i class="fa fa-check"></i><b>39.5.2</b> formula</a></li>
<li class="chapter" data-level="39.5.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#data-1"><i class="fa fa-check"></i><b>39.5.3</b> data</a></li>
<li class="chapter" data-level="39.5.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#start"><i class="fa fa-check"></i><b>39.5.4</b> start</a></li>
<li class="chapter" data-level="39.5.5" data-path="nonlinearintro.html"><a href="nonlinearintro.html#weighting"><i class="fa fa-check"></i><b>39.5.5</b> weighting</a></li>
</ul></li>
<li class="chapter" data-level="39.6" data-path="nonlinearintro.html"><a href="nonlinearintro.html#nonlinear-models-and-parameters"><i class="fa fa-check"></i><b>39.6</b> Nonlinear models and parameters</a>
<ul>
<li class="chapter" data-level="39.6.1" data-path="nonlinearintro.html"><a href="nonlinearintro.html#hyperbolic-stimulus-response-functions"><i class="fa fa-check"></i><b>39.6.1</b> Hyperbolic stimulus response functions</a></li>
<li class="chapter" data-level="39.6.2" data-path="nonlinearintro.html"><a href="nonlinearintro.html#visualizing-nonlinear-data-and-log-scaling"><i class="fa fa-check"></i><b>39.6.2</b> Visualizing nonlinear data and log scaling</a></li>
<li class="chapter" data-level="39.6.3" data-path="nonlinearintro.html"><a href="nonlinearintro.html#selecting-the-right-model"><i class="fa fa-check"></i><b>39.6.3</b> Selecting the right model</a></li>
<li class="chapter" data-level="39.6.4" data-path="nonlinearintro.html"><a href="nonlinearintro.html#from-models-to-formulas"><i class="fa fa-check"></i><b>39.6.4</b> From models to formulas</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="40" data-path="nestedregress.html"><a href="nestedregress.html"><i class="fa fa-check"></i><b>40</b> Nested-model nonlinear regression</a>
<ul>
<li class="chapter" data-level="40.1" data-path="nestedregress.html"><a href="nestedregress.html#read-and-plot-the-data"><i class="fa fa-check"></i><b>40.1</b> Read and plot the data</a></li>
<li class="chapter" data-level="40.2" data-path="nestedregress.html"><a href="nestedregress.html#which-model-is-a-better-fit"><i class="fa fa-check"></i><b>40.2</b> Which model is a better fit?</a>
<ul>
<li class="chapter" data-level="40.2.1" data-path="nestedregress.html"><a href="nestedregress.html#nonlinear-models"><i class="fa fa-check"></i><b>40.2.1</b> Nonlinear models</a></li>
<li class="chapter" data-level="40.2.2" data-path="nestedregress.html"><a href="nestedregress.html#the-nlfitr-alternative"><i class="fa fa-check"></i><b>40.2.2</b> The nlfitr alternative</a></li>
<li class="chapter" data-level="40.2.3" data-path="nestedregress.html"><a href="nestedregress.html#troubleshooting-the-regression"><i class="fa fa-check"></i><b>40.2.3</b> Troubleshooting the regression</a></li>
<li class="chapter" data-level="40.2.4" data-path="nestedregress.html"><a href="nestedregress.html#interpreting-the-parameters"><i class="fa fa-check"></i><b>40.2.4</b> Interpreting the parameters</a></li>
<li class="chapter" data-level="40.2.5" data-path="nestedregress.html"><a href="nestedregress.html#residual-plots-to-compare-fits"><i class="fa fa-check"></i><b>40.2.5</b> Residual plots to compare fits</a></li>
<li class="chapter" data-level="40.2.6" data-path="nestedregress.html"><a href="nestedregress.html#compare-aic"><i class="fa fa-check"></i><b>40.2.6</b> Compare AIC</a></li>
<li class="chapter" data-level="40.2.7" data-path="nestedregress.html"><a href="nestedregress.html#other-ways-to-compare-models"><i class="fa fa-check"></i><b>40.2.7</b> Other ways to compare models</a></li>
<li class="chapter" data-level="40.2.8" data-path="nestedregress.html"><a href="nestedregress.html#interpretation-of-this-one-replicate"><i class="fa fa-check"></i><b>40.2.8</b> Interpretation of this one replicate</a></li>
<li class="chapter" data-level="40.2.9" data-path="nestedregress.html"><a href="nestedregress.html#alternate-analysis"><i class="fa fa-check"></i><b>40.2.9</b> Alternate analysis</a></li>
</ul></li>
<li class="chapter" data-level="40.3" data-path="nestedregress.html"><a href="nestedregress.html#summary-13"><i class="fa fa-check"></i><b>40.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="41" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html"><i class="fa fa-check"></i><b>41</b> Nonlinear regression of independent replicates</a>
<ul>
<li class="chapter" data-level="41.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-problem"><i class="fa fa-check"></i><b>41.1</b> The problem</a></li>
<li class="chapter" data-level="41.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#the-dataset"><i class="fa fa-check"></i><b>41.2</b> The dataset</a></li>
<li class="chapter" data-level="41.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#munging-for-regression-analysis"><i class="fa fa-check"></i><b>41.3</b> Munging for regression analysis</a>
<ul>
<li class="chapter" data-level="41.3.1" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#average-the-technical-replicates"><i class="fa fa-check"></i><b>41.3.1</b> Average the technical replicates</a></li>
<li class="chapter" data-level="41.3.2" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#create-one-table"><i class="fa fa-check"></i><b>41.3.2</b> Create one table</a></li>
<li class="chapter" data-level="41.3.3" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#plot-the-data"><i class="fa fa-check"></i><b>41.3.3</b> Plot the data</a></li>
<li class="chapter" data-level="41.3.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#run-the-regressions"><i class="fa fa-check"></i><b>41.3.4</b> Run the regressions</a></li>
<li class="chapter" data-level="41.3.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#clean-up-regression-results"><i class="fa fa-check"></i><b>41.3.5</b> Clean up regression results</a></li>
</ul></li>
<li class="chapter" data-level="41.4" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#t-test-on-half-lives"><i class="fa fa-check"></i><b>41.4</b> T-test on half-lives</a></li>
<li class="chapter" data-level="41.5" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#conclusion"><i class="fa fa-check"></i><b>41.5</b> Conclusion</a></li>
<li class="chapter" data-level="41.6" data-path="nonlinearreplicates.html"><a href="nonlinearreplicates.html#summary-figure"><i class="fa fa-check"></i><b>41.6</b> Summary figure</a></li>
</ul></li>
<li class="chapter" data-level="42" data-path="multregress.html"><a href="multregress.html"><i class="fa fa-check"></i><b>42</b> Multiple regression</a>
<ul>
<li class="chapter" data-level="42.1" data-path="multregress.html"><a href="multregress.html#the-linear-regression-model-1"><i class="fa fa-check"></i><b>42.1</b> The linear regression model</a></li>
<li class="chapter" data-level="42.2" data-path="multregress.html"><a href="multregress.html#multiple-regression-models"><i class="fa fa-check"></i><b>42.2</b> Multiple regression models</a></li>
<li class="chapter" data-level="42.3" data-path="multregress.html"><a href="multregress.html#experimental-multiple-regression"><i class="fa fa-check"></i><b>42.3</b> Experimental multiple regression</a>
<ul>
<li class="chapter" data-level="42.3.1" data-path="multregress.html"><a href="multregress.html#a-one-factor-experiment-with-3-groups"><i class="fa fa-check"></i><b>42.3.1</b> A one factor experiment with 3 groups</a></li>
<li class="chapter" data-level="42.3.2" data-path="multregress.html"><a href="multregress.html#a-two-factor-experiment-with-6-groups"><i class="fa fa-check"></i><b>42.3.2</b> A two factor experiment with 6 groups</a></li>
</ul></li>
<li class="chapter" data-level="42.4" data-path="multregress.html"><a href="multregress.html#multiple-linear-mixed-models"><i class="fa fa-check"></i><b>42.4</b> Multiple linear mixed models</a></li>
<li class="chapter" data-level="42.5" data-path="multregress.html"><a href="multregress.html#multiple-regression-of-observational-data"><i class="fa fa-check"></i><b>42.5</b> Multiple regression of observational data</a></li>
</ul></li>
<li class="chapter" data-level="43" data-path="logregress.html"><a href="logregress.html"><i class="fa fa-check"></i><b>43</b> Logistic regression</a>
<ul>
<li class="chapter" data-level="43.1" data-path="logregress.html"><a href="logregress.html#uses-of-logistic-regression"><i class="fa fa-check"></i><b>43.1</b> Uses of logistic regression</a>
<ul>
<li class="chapter" data-level="43.1.1" data-path="logregress.html"><a href="logregress.html#sidebar-doing-logistic-regression-is-machine-learning"><i class="fa fa-check"></i><b>43.1.1</b> Sidebar: Doing logistic regression is machine learning</a></li>
</ul></li>
<li class="chapter" data-level="43.2" data-path="logregress.html"><a href="logregress.html#derivation-of-the-logistic-regression-model"><i class="fa fa-check"></i><b>43.2</b> Derivation of the logistic regression model</a>
<ul>
<li class="chapter" data-level="43.2.1" data-path="logregress.html"><a href="logregress.html#relationship-of-logit-to-odds-to-the-model-coefficients-and-probability"><i class="fa fa-check"></i><b>43.2.1</b> Relationship of logit to odds to the model coefficients and probability</a></li>
<li class="chapter" data-level="43.2.2" data-path="logregress.html"><a href="logregress.html#additional-types-of-logistic-regression-models"><i class="fa fa-check"></i><b>43.2.2</b> Additional types of logistic regression models</a></li>
</ul></li>
<li class="chapter" data-level="43.3" data-path="logregress.html"><a href="logregress.html#stress-and-survival"><i class="fa fa-check"></i><b>43.3</b> Stress and survival</a>
<ul>
<li class="chapter" data-level="43.3.1" data-path="logregress.html"><a href="logregress.html#interpretation-of-output"><i class="fa fa-check"></i><b>43.3.1</b> Interpretation of output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="44" data-path="mixedlogistic.html"><a href="mixedlogistic.html"><i class="fa fa-check"></i><b>44</b> Mixed model logistic regression</a>
<ul>
<li class="chapter" data-level="44.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#mixed-models-fixed-and-random-effects"><i class="fa fa-check"></i><b>44.1</b> Mixed models, fixed and random effects</a>
<ul>
<li class="chapter" data-level="44.1.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#nfat-localization-within-smooth-muscle-cells"><i class="fa fa-check"></i><b>44.1.1</b> NFAT localization within smooth muscle cells</a></li>
</ul></li>
<li class="chapter" data-level="44.2" data-path="mixedlogistic.html"><a href="mixedlogistic.html#data-2"><i class="fa fa-check"></i><b>44.2</b> Data</a>
<ul>
<li class="chapter" data-level="44.2.1" data-path="mixedlogistic.html"><a href="mixedlogistic.html#inference-2"><i class="fa fa-check"></i><b>44.2.1</b> Inference</a></li>
</ul></li>
<li class="chapter" data-level="44.3" data-path="mixedlogistic.html"><a href="mixedlogistic.html#alternative-analysis"><i class="fa fa-check"></i><b>44.3</b> Alternative analysis</a></li>
</ul></li>
<li class="chapter" data-level="45" data-path="poissonreg.html"><a href="poissonreg.html"><i class="fa fa-check"></i><b>45</b> Poisson regression</a>
<ul>
<li class="chapter" data-level="45.1" data-path="poissonreg.html"><a href="poissonreg.html#why-not-anova"><i class="fa fa-check"></i><b>45.1</b> Why not ANOVA?</a>
<ul>
<li class="chapter" data-level="45.1.1" data-path="poissonreg.html"><a href="poissonreg.html#counting-markers"><i class="fa fa-check"></i><b>45.1.1</b> Counting markers</a></li>
<li class="chapter" data-level="45.1.2" data-path="poissonreg.html"><a href="poissonreg.html#counting-depolarizations"><i class="fa fa-check"></i><b>45.1.2</b> Counting depolarizations</a></li>
<li class="chapter" data-level="45.1.3" data-path="poissonreg.html"><a href="poissonreg.html#lever-presses"><i class="fa fa-check"></i><b>45.1.3</b> Lever presses</a></li>
</ul></li>
<li class="chapter" data-level="45.2" data-path="poissonreg.html"><a href="poissonreg.html#the-poisson-generalized-linear-model"><i class="fa fa-check"></i><b>45.2</b> The Poisson generalized linear model</a></li>
<li class="chapter" data-level="45.3" data-path="poissonreg.html"><a href="poissonreg.html#length-of-hospital-stay"><i class="fa fa-check"></i><b>45.3</b> Length of hospital stay</a></li>
<li class="chapter" data-level="45.4" data-path="poissonreg.html"><a href="poissonreg.html#output-interpretation"><i class="fa fa-check"></i><b>45.4</b> Output interpretation</a></li>
</ul></li>
<li class="chapter" data-level="46" data-path="surv.html"><a href="surv.html"><i class="fa fa-check"></i><b>46</b> Survival Analysis</a>
<ul>
<li class="chapter" data-level="46.1" data-path="surv.html"><a href="surv.html#data-records"><i class="fa fa-check"></i><b>46.1</b> Data records</a>
<ul>
<li class="chapter" data-level="46.1.1" data-path="surv.html"><a href="surv.html#deriving-times-from-recorded-dates"><i class="fa fa-check"></i><b>46.1.1</b> Deriving times from recorded dates</a></li>
<li class="chapter" data-level="46.1.2" data-path="surv.html"><a href="surv.html#censor"><i class="fa fa-check"></i><b>46.1.2</b> Censor</a></li>
</ul></li>
<li class="chapter" data-level="46.2" data-path="surv.html"><a href="surv.html#kaplan-meier-estimator"><i class="fa fa-check"></i><b>46.2</b> Kaplan-Meier estimator</a></li>
<li class="chapter" data-level="46.3" data-path="surv.html"><a href="surv.html#glioma-data"><i class="fa fa-check"></i><b>46.3</b> Glioma data</a></li>
<li class="chapter" data-level="46.4" data-path="surv.html"><a href="surv.html#kaplan-meier-plots"><i class="fa fa-check"></i><b>46.4</b> Kaplan-Meier plots</a></li>
<li class="chapter" data-level="46.5" data-path="surv.html"><a href="surv.html#log-rank-tests"><i class="fa fa-check"></i><b>46.5</b> Log rank tests</a>
<ul>
<li class="chapter" data-level="46.5.1" data-path="surv.html"><a href="surv.html#write-up-of-log-rank-test"><i class="fa fa-check"></i><b>46.5.1</b> Write up of log rank test</a></li>
<li class="chapter" data-level="46.5.2" data-path="surv.html"><a href="surv.html#comparing-more-than-two-survival-curves"><i class="fa fa-check"></i><b>46.5.2</b> Comparing more than two survival curves</a></li>
</ul></li>
<li class="chapter" data-level="46.6" data-path="surv.html"><a href="surv.html#cox-proportional-hazards-analysis"><i class="fa fa-check"></i><b>46.6</b> Cox proportional hazards analysis</a>
<ul>
<li class="chapter" data-level="46.6.1" data-path="surv.html"><a href="surv.html#cox-proportional-hazards-regression-of-glioma"><i class="fa fa-check"></i><b>46.6.1</b> Cox proportional hazards regression of glioma</a></li>
<li class="chapter" data-level="46.6.2" data-path="surv.html"><a href="surv.html#interpretation-of-cox-regression-output"><i class="fa fa-check"></i><b>46.6.2</b> Interpretation of Cox regression output</a></li>
<li class="chapter" data-level="46.6.3" data-path="surv.html"><a href="surv.html#write-up-18"><i class="fa fa-check"></i><b>46.6.3</b> Write up</a></li>
</ul></li>
<li class="chapter" data-level="46.7" data-path="surv.html"><a href="surv.html#summary-14"><i class="fa fa-check"></i><b>46.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="47" data-path="classification.html"><a href="classification.html"><i class="fa fa-check"></i><b>47</b> Classification</a>
<ul>
<li class="chapter" data-level="47.1" data-path="classification.html"><a href="classification.html#distances"><i class="fa fa-check"></i><b>47.1</b> Distances</a></li>
<li class="chapter" data-level="47.2" data-path="classification.html"><a href="classification.html#clustering-methods"><i class="fa fa-check"></i><b>47.2</b> Clustering methods</a></li>
<li class="chapter" data-level="47.3" data-path="classification.html"><a href="classification.html#principal-component-analysis"><i class="fa fa-check"></i><b>47.3</b> Principal component analysis</a></li>
<li class="chapter" data-level="47.4" data-path="classification.html"><a href="classification.html#synthesize-some-data"><i class="fa fa-check"></i><b>47.4</b> Synthesize some data</a></li>
<li class="chapter" data-level="47.5" data-path="classification.html"><a href="classification.html#visualize-the-data-set"><i class="fa fa-check"></i><b>47.5</b> Visualize the data set</a></li>
<li class="chapter" data-level="47.6" data-path="classification.html"><a href="classification.html#rescaling-data"><i class="fa fa-check"></i><b>47.6</b> Rescaling data</a></li>
<li class="chapter" data-level="47.7" data-path="classification.html"><a href="classification.html#are-there-clusters"><i class="fa fa-check"></i><b>47.7</b> Are there clusters?</a></li>
<li class="chapter" data-level="47.8" data-path="classification.html"><a href="classification.html#how-many-clusters"><i class="fa fa-check"></i><b>47.8</b> How many clusters?</a></li>
<li class="chapter" data-level="47.9" data-path="classification.html"><a href="classification.html#clustering"><i class="fa fa-check"></i><b>47.9</b> Clustering</a></li>
<li class="chapter" data-level="47.10" data-path="classification.html"><a href="classification.html#pam-and-clara"><i class="fa fa-check"></i><b>47.10</b> PAM and CLARA</a></li>
<li class="chapter" data-level="47.11" data-path="classification.html"><a href="classification.html#hierarchical-clustering"><i class="fa fa-check"></i><b>47.11</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="47.12" data-path="classification.html"><a href="classification.html#heatmaps"><i class="fa fa-check"></i><b>47.12</b> Heatmaps</a></li>
<li class="chapter" data-level="47.13" data-path="classification.html"><a href="classification.html#summary-15"><i class="fa fa-check"></i><b>47.13</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="48" data-path="rnaseq.html"><a href="rnaseq.html"><i class="fa fa-check"></i><b>48</b> RNA-seq with R</a>
<ul>
<li class="chapter" data-level="48.1" data-path="rnaseq.html"><a href="rnaseq.html#install-bioconductor"><i class="fa fa-check"></i><b>48.1</b> Install Bioconductor</a></li>
<li class="chapter" data-level="48.2" data-path="rnaseq.html"><a href="rnaseq.html#import-raw-count-data"><i class="fa fa-check"></i><b>48.2</b> Import raw count data</a></li>
<li class="chapter" data-level="48.3" data-path="rnaseq.html"><a href="rnaseq.html#munge-to-simpler-table"><i class="fa fa-check"></i><b>48.3</b> Munge to simpler table</a></li>
<li class="chapter" data-level="48.4" data-path="rnaseq.html"><a href="rnaseq.html#filtering"><i class="fa fa-check"></i><b>48.4</b> Filtering</a></li>
<li class="chapter" data-level="48.5" data-path="rnaseq.html"><a href="rnaseq.html#dgelist"><i class="fa fa-check"></i><b>48.5</b> DGEList</a></li>
<li class="chapter" data-level="48.6" data-path="rnaseq.html"><a href="rnaseq.html#visualization"><i class="fa fa-check"></i><b>48.6</b> Visualization</a></li>
<li class="chapter" data-level="48.7" data-path="rnaseq.html"><a href="rnaseq.html#classification-1"><i class="fa fa-check"></i><b>48.7</b> Classification</a></li>
<li class="chapter" data-level="48.8" data-path="rnaseq.html"><a href="rnaseq.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>48.8</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="48.9" data-path="rnaseq.html"><a href="rnaseq.html#summary-16"><i class="fa fa-check"></i><b>48.9</b> Summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">JABSTB: Statistical Design and Analysis of Experiments with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="twowayRM" class="section level1" number="31">
<h1><span class="header-section-number">Chapter 31</span> Two-way ANOVA Related Measures</h1>
<div class="sourceCode" id="cb667"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb667-1"><a href="twowayRM.html#cb667-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb667-2"><a href="twowayRM.html#cb667-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ez)</span>
<span id="cb667-3"><a href="twowayRM.html#cb667-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb667-4"><a href="twowayRM.html#cb667-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb667-5"><a href="twowayRM.html#cb667-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code></pre></div>
<p>A two-way ANOVA experimental design is one that involves two predictor variables, where each predictor has 2 or more levels. There is only one outcome variable in a 2 way ANOVA and it should be continuous, measured on an equal interval scale, and ideally sampled from a normally-distributed population.</p>
<p>The predictor variables are often referred to as factors, and so ANOVA designs are synonymous with factorial designs.</p>
<p>The list of two-way ANOVA experimental designs can be as follows:</p>
<ol style="list-style-type: decimal">
<li>Completely randomized (CR) on both factors</li>
<li>Related measures (RM) on both factors</li>
<li>Mixed, CR on one factor and RM on the other</li>
</ol>
<p>In one sense, a two-way ANOVA can be thought of as two one-way ANOVA’s run simultaneously. So it is possible to assess the effects of two factors at once. The major difference, however, is the ability to also test whether an interaction exists between the two factors.</p>
<p>In this chapter we’ll focus on the data structure and analysis of a two-way ANOVA RM experimental design. These are particularly common in cell-culture and other in vitro-based experiments, owing to the homogeneity of the biological material.</p>
<p>For an experiment such as the one described below, all of the biological material assayed within a single replicate comes from a common source. Every well in every plate or every tube in every rack prepared from that source is identical, for all intents and purposes. Therefore, any measurements taken from these wells are intrinsically-linked. For this reason, the data should be analyzed as related-measures.</p>
<div class="figure"><span id="fig:unnamed-chunk-327"></span>
<img src="images/twowayRM.jpg" alt="A two-way ANOVA RM design viewed from the bench. Each combination of the two factors is tested in technical duplicate. All measurements are intrinsically-related." width="384" />
<p class="caption">
Figure 31.1: A two-way ANOVA RM design viewed from the bench. Each combination of the two factors is tested in technical duplicate. All measurements are intrinsically-related.
</p>
</div>
<p>An independent replicate occurs when the same protocol is followed, from start to finish, beginning with a new batch of plates or with a new batch of a purified protein or whatever.</p>
<p>Even so, it’s hard to argue these represent true biological replicates. The cells are likely clonal, and probably aren’t changing much from batch to batch. The statistical analysis is therefore essentially focused on establishing whether the observations occur repeatedly or not. The replicates are therefore “blocked,” usually in time. For example, an assay on the material on week1 is treated differently than one on week2 because these are two different time blocks.</p>
<div id="cell-culture" class="section level2" number="31.1">
<h2><span class="header-section-number">31.1</span> Cell culture</h2>
<p>Coronary artery atheromas are enriched in fast-growing mesenchymal-like cells, which have prominent inflammatory characteristics. They express the transcription factor NFAT, which is regulated by mitogenic receptor signaling to control inflammatory gene expression.</p>
<p>Cells from atheroma explants are easy to grow in culture, often indefinitely as for cancer cells. Eventually, a fast growing cell population dominates these monocultures when passaged serially. But these probably mimic the “bad” cells in an atheroma pretty well, so are useful to study.</p>
<p>An experiment was performed on cultured human coronary smooth muscle cells derived from an atheroma to determine if the bzip suppresser ICER attenuates NFAT and whether it is mitogen-selective.</p>
<p>One predictor variable is mitogen at 3 levels: vehicle, ADP and PDGF. A second predictor variable is an expression vector, which is either empty or encodes a cDNA to express ICER.</p>
<p>The output response, luminescence in cell extracts due to an NFAT-driven luciferase transcriptional reporter, is detected using a luminometer. Luminescence is linearly proportional to the amount of luciferase in the cells.</p>
<p>The protocol involved parallel transfections of the CASM cell line with the expression vectors and luciferase reporter, followed a few days later by treatment with the mitogens for 4 hrs before measuring luminescence. All treatments are performed in technical duplicate, from which the average value is used for the response measure.</p>
<p>Three predictions, and the corresponding null hypotheses that each tests, can be evaluated here simultaneously:</p>
<ol style="list-style-type: decimal">
<li>Icer and its control differ in how they each affect NFAT-mediated transcription. <span class="math inline">\(H0:\sigma^2_{suppressor}\le\sigma^2_{residual}\)</span></li>
<li>The mitogens differ in how they affect NFAT-mediated transcription. <span class="math inline">\(H0:\sigma^2_{mitogen}\le\sigma^2_{residual}\)</span></li>
<li>NFAT-mediated transcription is influence by a combination of suppressor and mitogen. <span class="math inline">\(H0:\sigma^2_{suppressorXmitogen}\le\sigma^2_{residual}\)</span></li>
</ol>
<p>Here’s some data:</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
replicate
</th>
<th style="text-align:left;">
vector
</th>
<th style="text-align:left;">
mitogen
</th>
<th style="text-align:right;">
Nfatluc
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
Veh
</td>
<td style="text-align:right;">
23.2
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
Veh
</td>
<td style="text-align:right;">
15.7
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
Veh
</td>
<td style="text-align:right;">
17.2
</td>
</tr>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:right;">
83.5
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:right;">
95.9
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:right;">
89.0
</td>
</tr>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
PDGF
</td>
<td style="text-align:right;">
124.6
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
PDGF
</td>
<td style="text-align:right;">
187.7
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
empty
</td>
<td style="text-align:left;">
PDGF
</td>
<td style="text-align:right;">
170.9
</td>
</tr>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
Veh
</td>
<td style="text-align:right;">
16.8
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
Veh
</td>
<td style="text-align:right;">
5.4
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
Veh
</td>
<td style="text-align:right;">
27.6
</td>
</tr>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:right;">
89.8
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:right;">
80.2
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:right;">
52.5
</td>
</tr>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
PDGF
</td>
<td style="text-align:right;">
57.0
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
PDGF
</td>
<td style="text-align:right;">
78.1
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
ICER
</td>
<td style="text-align:left;">
PDGF
</td>
<td style="text-align:right;">
67.4
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb668"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb668-1"><a href="twowayRM.html#cb668-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(hcsmc, </span>
<span id="cb668-2"><a href="twowayRM.html#cb668-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(</span>
<span id="cb668-3"><a href="twowayRM.html#cb668-3" aria-hidden="true" tabindex="-1"></a>         mitogen, </span>
<span id="cb668-4"><a href="twowayRM.html#cb668-4" aria-hidden="true" tabindex="-1"></a>         Nfatluc, </span>
<span id="cb668-5"><a href="twowayRM.html#cb668-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">group=</span>replicate)</span>
<span id="cb668-6"><a href="twowayRM.html#cb668-6" aria-hidden="true" tabindex="-1"></a>       ) <span class="sc">+</span></span>
<span id="cb668-7"><a href="twowayRM.html#cb668-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(</span>
<span id="cb668-8"><a href="twowayRM.html#cb668-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb668-9"><a href="twowayRM.html#cb668-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb668-10"><a href="twowayRM.html#cb668-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(</span>
<span id="cb668-11"><a href="twowayRM.html#cb668-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb668-12"><a href="twowayRM.html#cb668-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">color=</span>replicate), </span>
<span id="cb668-13"><a href="twowayRM.html#cb668-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">size=</span><span class="dv">4</span></span>
<span id="cb668-14"><a href="twowayRM.html#cb668-14" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb668-15"><a href="twowayRM.html#cb668-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb668-16"><a href="twowayRM.html#cb668-16" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span>vector, </span>
<span id="cb668-17"><a href="twowayRM.html#cb668-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">switch=</span><span class="st">&quot;both&quot;</span></span>
<span id="cb668-18"><a href="twowayRM.html#cb668-18" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb668-19"><a href="twowayRM.html#cb668-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(</span>
<span id="cb668-20"><a href="twowayRM.html#cb668-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">values=</span><span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">16</span>)</span>
<span id="cb668-21"><a href="twowayRM.html#cb668-21" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb668-22"><a href="twowayRM.html#cb668-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis</span>(</span>
<span id="cb668-23"><a href="twowayRM.html#cb668-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">discrete=</span>T</span>
<span id="cb668-24"><a href="twowayRM.html#cb668-24" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb668-25"><a href="twowayRM.html#cb668-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb668-26"><a href="twowayRM.html#cb668-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position=</span><span class="st">&quot;top&quot;</span></span>
<span id="cb668-27"><a href="twowayRM.html#cb668-27" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb668-28"><a href="twowayRM.html#cb668-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(</span>
<span id="cb668-29"><a href="twowayRM.html#cb668-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits=</span><span class="fu">c</span>(<span class="st">&quot;Veh&quot;</span>,<span class="st">&quot;ADP&quot;</span>,<span class="st">&quot;PDGF&quot;</span>)</span>
<span id="cb668-30"><a href="twowayRM.html#cb668-30" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb668-31"><a href="twowayRM.html#cb668-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb668-32"><a href="twowayRM.html#cb668-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y=</span><span class="st">&quot;NFAT-luciferase, light units/mg&quot;</span>, </span>
<span id="cb668-33"><a href="twowayRM.html#cb668-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">x=</span><span class="st">&quot;Mitogen&quot;</span></span>
<span id="cb668-34"><a href="twowayRM.html#cb668-34" aria-hidden="true" tabindex="-1"></a>    )</span></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-329"></span>
<img src="jabstb_files/figure-html/unnamed-chunk-329-1.png" alt="A cell culture-based two-way ANOVA RM experiment" width="672" />
<p class="caption">
Figure 31.2: A cell culture-based two-way ANOVA RM experiment
</p>
</div>
</div>
<div id="the-test-2" class="section level2" number="31.2">
<h2><span class="header-section-number">31.2</span> The test</h2>
<p>Scientific judgement dictates running this test as related measures. Each replicate was performed on a different passage of cells, approximately 1 week apart. Within each passage the cells are highly homogeneous. The transfections and treatment and biochemical measurements were all conducted in a “side-by-side” block. The measurements within a replicate are intrinsically-linked. Perhaps most importantly, each measure taken on a given day is NOT independent of all other measures that day. But we can assume they differ from week to week. There really is no other way to analyze these data.</p>
<div class="sourceCode" id="cb669"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb669-1"><a href="twowayRM.html#cb669-1" aria-hidden="true" tabindex="-1"></a>out.rm <span class="ot">&lt;-</span> <span class="fu">ezANOVA</span>(<span class="at">data =</span> hcsmc</span>
<span id="cb669-2"><a href="twowayRM.html#cb669-2" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">dv =</span> Nfatluc</span>
<span id="cb669-3"><a href="twowayRM.html#cb669-3" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">wid =</span> replicate</span>
<span id="cb669-4"><a href="twowayRM.html#cb669-4" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">within =</span> <span class="fu">c</span>(vector, mitogen)</span>
<span id="cb669-5"><a href="twowayRM.html#cb669-5" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">type =</span> <span class="dv">3</span></span>
<span id="cb669-6"><a href="twowayRM.html#cb669-6" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">return_aov =</span> T</span>
<span id="cb669-7"><a href="twowayRM.html#cb669-7" aria-hidden="true" tabindex="-1"></a>                   , <span class="at">detailed =</span> T</span>
<span id="cb669-8"><a href="twowayRM.html#cb669-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb669-9"><a href="twowayRM.html#cb669-9" aria-hidden="true" tabindex="-1"></a>); out.rm</span></code></pre></div>
<pre><code>## Warning: Converting &quot;replicate&quot; to factor for ANOVA.</code></pre>
<pre><code>## Warning: Converting &quot;vector&quot; to factor for ANOVA.</code></pre>
<pre><code>## Warning: Converting &quot;mitogen&quot; to factor for ANOVA.</code></pre>
<pre><code>## $ANOVA
##           Effect DFn DFd       SSn       SSd         F           p p&lt;.05
## 1    (Intercept)   1   2 91378.125  388.5700 470.33031 0.002119408     *
## 2         vector   1   2  6156.801  471.0011  26.14347 0.036186976     *
## 3        mitogen   2   4 29018.893 1981.2567  29.29342 0.004084641     *
## 4 vector:mitogen   2   4  7333.031  623.0722  23.53830 0.006133042     *
##         ges
## 1 0.9634772
## 2 0.6399535
## 3 0.8933620
## 4 0.6791774
## 
## $`Mauchly&#39;s Test for Sphericity`
##           Effect         W         p p&lt;.05
## 3        mitogen 0.6795372 0.8243404      
## 4 vector:mitogen 0.4963405 0.7045144      
## 
## $`Sphericity Corrections`
##           Effect       GGe      p[GG] p[GG]&lt;.05      HFe       p[HF] p[HF]&lt;.05
## 3        mitogen 0.7573102 0.01101064         * 2.620487 0.004084641         *
## 4 vector:mitogen 0.6650442 0.02123339         * 1.485468 0.006133042         *
## 
## $aov
## 
## Call:
## aov(formula = formula(aov_formula), data = data)
## 
## Grand Mean: 71.25
## 
## Stratum 1: replicate
## 
## Terms:
##                 Residuals
## Sum of Squares     388.57
## Deg. of Freedom         2
## 
## Residual standard error: 13.93862
## 
## Stratum 2: replicate:vector
## 
## Terms:
##                   vector Residuals
## Sum of Squares  6156.801   471.001
## Deg. of Freedom        1         2
## 
## Residual standard error: 15.34603
## 2 out of 3 effects not estimable
## Estimated effects are balanced
## 
## Stratum 3: replicate:mitogen
## 
## Terms:
##                   mitogen Residuals
## Sum of Squares  29018.893  1981.257
## Deg. of Freedom         2         4
## 
## Residual standard error: 22.25565
## 2 out of 4 effects not estimable
## Estimated effects may be unbalanced
## 
## Stratum 4: replicate:vector:mitogen
## 
## Terms:
##                 vector:mitogen Residuals
## Sum of Squares        7333.031   623.072
## Deg. of Freedom              2         4
## 
## Residual standard error: 12.48071
## Estimated effects may be unbalanced</code></pre>
</div>
<div id="interpretation-of-the-output" class="section level2" number="31.3">
<h2><span class="header-section-number">31.3</span> Interpretation of the output</h2>
<p>Note here that we are NOT capturing an aov object. The only use of the aov object we might have is for posthoc testing with range tests. But we should not use range tests on a pure related measures design, where the means of groups are irrelevant.</p>
<div id="anova-table-1" class="section level3" number="31.3.1">
<h3><span class="header-section-number">31.3.1</span> ANOVA Table</h3>
<p>Note just as for the 2-way ANOVA CR analysis, there are 3 F tests for the model (the intercept F test is inconsequential for now). The F value for the suppress:mitogen interaction is beyond the critical limit for a null F distribution with 2 and 4 degrees of freedom. The result is extreme (p = 0.00613). The interaction takes precedence, since the main effects are difficult to interpret if an interaction occurs.</p>
<p>It is safe to reject the interaction null and conclude that variance for its effect exceeds that for its residual. Scientifically, this means that levels of NFAT-mediated transcription are influenced by both the suppressor and the mitogen stimuli, as designed in this experiment. About 68% of the variance in the data can be explained by this interaction effect.</p>
</div>
<div id="mauchlys-sphericity-test" class="section level3" number="31.3.2">
<h3><span class="header-section-number">31.3.2</span> Mauchly’s Sphericity Test</h3>
<p>Sphericity is defined as uniform variance of the differences. Think of it as the RM analog to the CR homogeneity of variance assumption. Sphericity, as for homogeneity of variance, is an assumption ideally met in RM ANOVA for validity of the result.</p>
<p>Mauchly’s tests the null hypothesis that the variances among the differences are equal. If the test statistic were extreme, the null would be rejected, meaning these variances are unequal and sphericity is violated. If that were the case, we’d conduct inference on the basis of a corrected p-value for the ANOVA..the p[GG], which is the Geisser-Greenhouse corrected p-value.</p>
<p>In our analysis, the Mauchly test is not extreme. We have no reason to believe the sphericity assumption is not satisfied. We can use the p-value in the ANOVA table to guide our inference, without using the sphericity correction.</p>
</div>
</div>
<div id="post-hoc-multiple-comparisons-1" class="section level2" number="31.4">
<h2><span class="header-section-number">31.4</span> Post Hoc multiple comparisons</h2>
<p>The true scientific objective here is to know whether the suppressor selectively attenuate NFAT-mediated transcription by either of the mitogens. In other words, scientifically, it would be interesting to learn whether an ICER-suppressible factor participates in the pathway of one of the mitogens, but not the other.</p>
<p>Given that objective, there are some fairly specific planned comparisons worth making. If we compare ICER to Empty for each level of mitogen, we’ll have an answer to the question posed above. In other words, we only care about 3 of the 15 possible comparisons that could be made.</p>
<p>Therefore, we’ll run a <code>pairwise.t.test</code> without creating any adjusted p-values. After that, we’ll create a small vector of p-values for the 3 comparisons we wish to make and use the <code>p.adjust</code> function to correct them for multiple comparisons.</p>
<p>Step1: Run the pairwise.t.test but don’t adjust p-values</p>
<p>We use a <code>paired=T</code> argument given the intrinsically-linked relationship of the measures in the samples. Not all of the measures are independent so the unpaired t-test is inappropriate.</p>
<div class="sourceCode" id="cb674"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb674-1"><a href="twowayRM.html#cb674-1" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">pairwise.t.test</span>(hcsmc<span class="sc">$</span>Nfatluc, <span class="fu">interaction</span>(hcsmc<span class="sc">$</span>vector, hcsmc<span class="sc">$</span>mitogen), <span class="at">paired=</span>T, <span class="at">p.adj =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb674-2"><a href="twowayRM.html#cb674-2" aria-hidden="true" tabindex="-1"></a>m</span></code></pre></div>
<pre><code>## 
##  Pairwise comparisons using paired t tests 
## 
## data:  hcsmc$Nfatluc and interaction(hcsmc$vector, hcsmc$mitogen) 
## 
##            empty.ADP ICER.ADP empty.PDGF ICER.PDGF empty.Veh
## ICER.ADP   0.3413    -        -          -         -        
## empty.PDGF 0.0439    0.0803   -          -         -        
## ICER.PDGF  0.0129    0.6800   0.0191     -         -        
## empty.Veh  0.0066    0.0316   0.0214     0.0276    -        
## ICER.Veh   0.0147    0.0720   0.0215     0.0429    0.7723   
## 
## P value adjustment method: none</code></pre>
<p>Step2: Collect the p-values only for the comparisons of scientific interest</p>
<div class="sourceCode" id="cb676"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb676-1"><a href="twowayRM.html#cb676-1" aria-hidden="true" tabindex="-1"></a>pv <span class="ot">&lt;-</span> <span class="fu">c</span>(m<span class="sc">$</span>p.value[<span class="dv">1</span>,<span class="dv">1</span>], m<span class="sc">$</span>p.value[<span class="dv">3</span>,<span class="dv">3</span>], m<span class="sc">$</span>p.value[<span class="dv">5</span>,<span class="dv">5</span>]) </span>
<span id="cb676-2"><a href="twowayRM.html#cb676-2" aria-hidden="true" tabindex="-1"></a>pv</span></code></pre></div>
<pre><code>## [1] 0.34127356 0.01905007 0.77231675</code></pre>
<p>Thus, the three comparisons and their unadjusted p-values are:</p>
<ul>
<li>ICER.ADP to empty.ADP p=0.3413</li>
<li>ICER.PDGF to empty.PDGF p=0.0191</li>
<li>ICER.Veh to empty.Veh p=0.7723</li>
</ul>
<p>Step3: Adjust those p-values for multiple comparisons</p>
<div class="sourceCode" id="cb678"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb678-1"><a href="twowayRM.html#cb678-1" aria-hidden="true" tabindex="-1"></a><span class="fu">p.adjust</span>(<span class="at">p=</span>pv, <span class="at">method=</span><span class="st">&quot;bonf&quot;</span>, <span class="at">n=</span><span class="fu">length</span>(pv))</span></code></pre></div>
<pre><code>## [1] 1.00000000 0.05715022 1.00000000</code></pre>
<p>Each of these adjusted p-values is greater than the type1 error threshold of 5%. On this basis, we cannot reject the null hypothesis that these response are the same.</p>
<p>This illustrates the occasional case where a positive result at the level of the omnibus test is coupled to a negative finding in post hoc testing. In fact, that shouldn’t be a surprise. The F test sniffed out some decent sized effects within the collection of responses that are not of scientific interest.</p>
</div>
<div id="write-up-14" class="section level2" number="31.5">
<h2><span class="header-section-number">31.5</span> Write Up</h2>
<p><em>Statistical analysis of the luciferase data indicates there is an interaction between levels of the suppressor and the mitogen variables (2 way RM ANOVA, p=0.0061, n=3). However, post hoc comparisons do not support the hypothesis that ICER selectively suppresses either of the mitogens (pairwise two-sided paired t-tests, Holm adjusted p-values for multiple comparisons). The experiment may have been underpowered.</em></p>
</div>
<div id="summary-9" class="section level2" number="31.6">
<h2><span class="header-section-number">31.6</span> Summary</h2>
<p><em>We use two-way repeated measures ANOVA (both arms RM) when all measurements within each replicate are intrinsically-linked.
</em>When the experiment involves in vitro biological material there is a very good chance this ANOVA is the right choice.
<em>Inference can be done on only the main effect(s). When the interaction is positive, the main effects are difficult to interpret.
</em>Follow-up with pairwise.t.tests (paired=T), not with range tests.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="twowayCR.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="twowaymixed.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["jabstb.pdf", "jabstb.epub"],
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
